From 139edbc9a6f39f635bed05a10fcd5658a18a18c6 Mon Sep 17 00:00:00 2001
From: Marko Lindqvist <cazfi74@gmail.com>
Date: Mon, 26 Jun 2023 17:11:07 +0300
Subject: [PATCH 18/18] Set diplstate max_state correctly for teamed players

See osdn #48295

Signed-off-by: Marko Lindqvist <cazfi74@gmail.com>
---
 server/srv_main.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/server/srv_main.c b/server/srv_main.c
index fb57b336e0..520d938860 100644
--- a/server/srv_main.c
+++ b/server/srv_main.c
@@ -3363,7 +3363,9 @@ static void srv_ready(void)
       players_iterate(pdest) {
         if (players_on_same_team(pplayer, pdest)
             && player_number(pplayer) != player_number(pdest)) {
-          player_diplstate_get(pplayer, pdest)->type = DS_TEAM;
+          set_diplstate_type(player_diplstate_get(pplayer, pdest),
+                             player_diplstate_get(pdest, pplayer),
+                             DS_TEAM);
           give_shared_vision(pplayer, pdest);
           BV_SET(pplayer->real_embassy, player_index(pdest));
         }
-- 
2.40.1

