From bf089a3ef88b30c2cf525a6e7889179402107a39 Mon Sep 17 00:00:00 2001
From: Alain BKR <alain.baeckeroot@laposte.net>
Date: Fri, 5 May 2023 09:05:19 +0200
Subject: [PATCH 22/22] Fix out of bounds on cargo iter

See osdn #47982
---
 common/unit.c | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/common/unit.c b/common/unit.c
index 57a286494a..f957e01777 100644
--- a/common/unit.c
+++ b/common/unit.c
@@ -2478,11 +2478,11 @@ static void *cargo_iter_get(const struct iterator *it)
 static void cargo_iter_next(struct iterator *it)
 {
   struct cargo_iter *iter = CARGO_ITER(it);
-  const struct unit_list_link *piter = iter->links[iter->depth - 1];
+  const struct unit_list_link *piter;
   const struct unit_list_link *pnext;
 
   /* Variant 1: unit has cargo. */
-  pnext = unit_list_head(unit_transport_cargo(unit_list_link_data(piter)));
+  pnext = unit_list_head(unit_transport_cargo(cargo_iter_get(it)));
   if (NULL != pnext) {
     fc_assert(iter->depth < ARRAY_SIZE(iter->links));
     iter->links[iter->depth++] = pnext;
@@ -2492,6 +2492,8 @@ static void cargo_iter_next(struct iterator *it)
   fc_assert(iter->depth > 0);
 
   while (iter->depth > 0) {
+    piter = iter->links[iter->depth - 1];
+
     /* Variant 2: there are other cargo units at same level. */
     pnext = unit_list_link_next(piter);
     if (NULL != pnext) {
@@ -2500,7 +2502,7 @@ static void cargo_iter_next(struct iterator *it)
     }
 
     /* Variant 3: return to previous level, and do same tests. */
-    piter = iter->links[iter->depth-- - 2];
+    iter->depth--;
   }
 }
 
-- 
2.39.2

