From ed24740775fa169065fd0f65694066f8b87ded35 Mon Sep 17 00:00:00 2001
From: Marko Lindqvist <cazfi74@gmail.com>
Date: Fri, 22 Dec 2023 14:58:27 +0200
Subject: [PATCH 31/31] Meson: Enable implicit-fallthrough compiler warnings

See RM #76

Signed-off-by: Marko Lindqvist <cazfi74@gmail.com>
---
 meson.build | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index 1234216ec4..a631e45db0 100644
--- a/meson.build
+++ b/meson.build
@@ -32,7 +32,8 @@ c_cpp_args = [
   '-Wshadow',
   '-Wold-style-declaration',
   '-Wold-style-definition',
-  '-Wtype-limits'
+  '-Wtype-limits',
+  '-Wimplicit-fallthrough'
   ]
 
 foreach arg : c_args
-- 
2.43.0

