From 465d4df9fd46f5188b2719361e9b503247d18f7b Mon Sep 17 00:00:00 2001
From: Alain BKR <alain.baeckeroot@laposte.net>
Date: Sun, 9 Apr 2023 19:56:04 +0200
Subject: [PATCH 32/40] Mapgenerator: Check if lake exist before regenerate

See osdn #47825
---
 server/generator/mapgen.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/server/generator/mapgen.c b/server/generator/mapgen.c
index abab956768..c110a2efce 100644
--- a/server/generator/mapgen.c
+++ b/server/generator/mapgen.c
@@ -1364,7 +1364,10 @@ bool map_fractal_generate(bool autosize, struct unit_type *initial_unit)
     assign_continent_numbers();
 
     /* Turn small oceans into lakes. */
-    regenerate_lakes();
+    if (wld.map.num_oceans > 0) {
+      regenerate_lakes();
+    }
+
   } else {
     assign_continent_numbers();
   }
-- 
2.39.2

