From c4e6c2315cdd080ac8505308a29bb5f7b478068c Mon Sep 17 00:00:00 2001
From: Marko Lindqvist <cazfi74@gmail.com>
Date: Wed, 30 Nov 2022 05:01:29 +0200
Subject: [PATCH 34/34] Increase MAX_LEN_CITYNAME to 120

See osdn #46096

Signed-off-by: Marko Lindqvist <cazfi74@gmail.com>
---
 common/fc_types.h | 14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

diff --git a/common/fc_types.h b/common/fc_types.h
index 38de637c6c..8e5834c4d2 100644
--- a/common/fc_types.h
+++ b/common/fc_types.h
@@ -67,15 +67,15 @@ extern "C" {
 #define MAX_NUM_TECH_CLASSES   16     /* Used in the network protocol. */
 
 /* Changing these will probably break network compatibility. */
-#define MAX_LEN_NAME       48
-#define MAX_LEN_CITYNAME   80
-#define MAX_LEN_MAP_LABEL  64
-#define MAX_LEN_DEMOGRAPHY 16
-#define MAX_LEN_ALLOW_TAKE 16
+#define MAX_LEN_NAME        48
+#define MAX_LEN_CITYNAME   120
+#define MAX_LEN_MAP_LABEL   64
+#define MAX_LEN_DEMOGRAPHY  16
+#define MAX_LEN_ALLOW_TAKE  16
 #define MAX_LEN_GAME_IDENTIFIER 33
-#define MAX_GRANARY_INIS 24
+#define MAX_GRANARY_INIS    24
 #define MAX_LEN_STARTUNIT (20+1)
-#define MAX_LEN_ENUM     64
+#define MAX_LEN_ENUM        64
 
 /* Line breaks after this number of characters; be careful and use only 70 */
 #define LINE_BREAK 70
-- 
2.35.1

