From ac3c3d7f434ddbfd03076d771c2d99b65dfa8927 Mon Sep 17 00:00:00 2001
From: Marko Lindqvist <cazfi74@gmail.com>
Date: Thu, 15 Dec 2022 19:48:30 +0200
Subject: [PATCH 37/37] Inform client when the server side CMA gets enabled

See osdn #46279

Signed-off-by: Marko Lindqvist <cazfi74@gmail.com>
---
 server/cityhand.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/server/cityhand.c b/server/cityhand.c
index 96915b410d..1f1463ff74 100644
--- a/server/cityhand.c
+++ b/server/cityhand.c
@@ -552,6 +552,7 @@ void handle_web_cma_set(struct player *pplayer, int id,
     }
 
     cm_copy_parameter(pcity->cm_parameter, param);
+    pcity->server.synced = FALSE;
 
     auto_arrange_workers(pcity);
 
-- 
2.35.1

