From a69ccb1e30ae0ed15300146ef035b87bfe524554 Mon Sep 17 00:00:00 2001
From: Marko Lindqvist <cazfi74@gmail.com>
Date: Thu, 8 Feb 2024 21:53:31 +0200
Subject: [PATCH 46/46] Fix auto_arrange_workers() garbage cmp.minimal_surplus
 value

See RM #245

Signed-off-by: Marko Lindqvist <cazfi74@gmail.com>
---
 server/cityturn.c | 11 ++++++-----
 1 file changed, 6 insertions(+), 5 deletions(-)

diff --git a/server/cityturn.c b/server/cityturn.c
index 8b296ca914..a05034ba97 100644
--- a/server/cityturn.c
+++ b/server/cityturn.c
@@ -431,19 +431,20 @@ void auto_arrange_workers(struct city *pcity)
 
     if (!cmr->found_a_valid) {
       /* Drop surpluses and try again. */
-      cmp.minimal_surplus[O_FOOD] = 0;
-      cmp.minimal_surplus[O_SHIELD] = 0;
+      output_type_iterate(o) {
+        cmp.minimal_surplus[o] = 0;
+      } output_type_iterate_end;
       cmp.minimal_surplus[O_GOLD] = -FC_INFINITY;
       cm_query_result(pcity, pcmp, cmr, FALSE);
     }
   }
   if (!cmr->found_a_valid) {
-    /* Emergency management.  Get _some_ result.  This doesn't use
-     * cm_init_emergency_parameter so we can keep the factors from
+    /* Emergency management. Get _some_ result. This doesn't use
+     * cm_init_emergency_parameter() so we can keep the factors from
      * above. */
     output_type_iterate(o) {
       cmp.minimal_surplus[o] = MIN(cmp.minimal_surplus[o],
-				   MIN(pcity->surplus[o], 0));
+                                   MIN(pcity->surplus[o], 0));
     } output_type_iterate_end;
     cmp.require_happy = FALSE;
     cmp.allow_disorder = is_ai(city_owner(pcity)) ? FALSE : TRUE;
-- 
2.43.0

