diff -Nurd freeciv/client/control.c freeciv/client/control.c
--- freeciv/client/control.c	2023-04-22 00:28:05.984606306 +0300
+++ freeciv/client/control.c	2023-04-22 00:28:22.228750376 +0300
@@ -3617,7 +3617,7 @@
 {
   unit_list_iterate(get_units_in_focus(), punit) {
     struct extra_type *tgt = prev_extra_in_tile(unit_tile(punit),
-                                                ERM_CLEANPOLLUTION,
+                                                ERM_CLEAN,
                                                 unit_owner(punit),
                                                 punit);
 
@@ -3626,13 +3626,23 @@
       request_new_unit_activity_targeted(punit, ACTIVITY_CLEAN, tgt);
     } else {
       tgt = prev_extra_in_tile(unit_tile(punit),
-                               ERM_CLEANFALLOUT,
+                               ERM_CLEANPOLLUTION,
                                unit_owner(punit),
                                punit);
 
       if (tgt != NULL
           && can_unit_do_activity_targeted(punit, ACTIVITY_CLEAN, tgt)) {
         request_new_unit_activity_targeted(punit, ACTIVITY_CLEAN, tgt);
+      } else {
+        tgt = prev_extra_in_tile(unit_tile(punit),
+                                 ERM_CLEANFALLOUT,
+                                 unit_owner(punit),
+                                 punit);
+
+        if (tgt != NULL
+            && can_unit_do_activity_targeted(punit, ACTIVITY_CLEAN, tgt)) {
+          request_new_unit_activity_targeted(punit, ACTIVITY_CLEAN, tgt);
+        }
       }
     }
   } unit_list_iterate_end;
diff -Nurd freeciv/client/gui-gtk-3.22/citydlg.c freeciv/client/gui-gtk-3.22/citydlg.c
--- freeciv/client/gui-gtk-3.22/citydlg.c	2023-04-22 00:28:05.988606340 +0300
+++ freeciv/client/gui-gtk-3.22/citydlg.c	2023-04-22 00:28:22.228750376 +0300
@@ -3000,6 +3000,12 @@
                         G_CALLBACK(set_city_workertask),
                         GINT_TO_POINTER(ACTIVITY_TRANSFORM), FALSE, NULL);
     }
+    if (prev_extra_in_tile(ptile, ERM_CLEAN,
+                           city_owner(pcity), NULL) != NULL) {
+      choice_dialog_add(shl, _("Clean"),
+                        G_CALLBACK(set_city_workertask),
+                        GINT_TO_POINTER(ACTIVITY_CLEAN), FALSE, NULL);
+    }
     if (prev_extra_in_tile(ptile, ERM_CLEANPOLLUTION,
                            city_owner(pcity), NULL) != NULL) {
       choice_dialog_add(shl, _("Clean Pollution"),
diff -Nurd freeciv/client/gui-gtk-4.0/citydlg.c freeciv/client/gui-gtk-4.0/citydlg.c
--- freeciv/client/gui-gtk-4.0/citydlg.c	2023-04-22 00:28:05.992606377 +0300
+++ freeciv/client/gui-gtk-4.0/citydlg.c	2023-04-22 00:28:22.228750376 +0300
@@ -3135,6 +3135,12 @@
                         G_CALLBACK(set_city_workertask),
                         GINT_TO_POINTER(ACTIVITY_TRANSFORM), FALSE, NULL);
     }
+    if (prev_extra_in_tile(ptile, ERM_CLEAN,
+                           city_owner(pcity), NULL) != NULL) {
+      choice_dialog_add(shl, _("Clean"),
+                        G_CALLBACK(set_city_workertask),
+                        GINT_TO_POINTER(ACTIVITY_CLEAN), FALSE, NULL);
+    }
     if (prev_extra_in_tile(ptile, ERM_CLEANPOLLUTION,
                            city_owner(pcity), NULL) != NULL) {
       choice_dialog_add(shl, _("Clean Pollution"),
diff -Nurd freeciv/client/gui-gtk-4.0/menu.c freeciv/client/gui-gtk-4.0/menu.c
--- freeciv/client/gui-gtk-4.0/menu.c	2023-04-22 00:28:05.992606377 +0300
+++ freeciv/client/gui-gtk-4.0/menu.c	2023-04-22 00:28:22.228750376 +0300
@@ -3548,7 +3548,7 @@
 
   submenu = g_menu_new();
 
-  extra_type_by_rmcause_iterate(ERM_CLEANPOLLUTION, pextra) {
+  extra_type_by_rmcause_iterate(ERM_CLEAN, pextra) {
     char actname[256];
     GSimpleAction *act;
 
diff -Nurd freeciv/client/gui-sdl2/menu.c freeciv/client/gui-sdl2/menu.c
--- freeciv/client/gui-sdl2/menu.c	2023-04-22 00:28:06.000606447 +0300
+++ freeciv/client/gui-sdl2/menu.c	2023-04-22 00:28:22.228750376 +0300
@@ -1307,16 +1307,23 @@
         set_wflag(order_airbase_button, WF_HIDDEN);
       }
 
-      clean_target = prev_extra_in_tile(ptile, ERM_CLEANPOLLUTION,
+      clean_target = prev_extra_in_tile(ptile, ERM_CLEAN,
                                         unit_owner(punit), punit);
-      if (clean_target != NULL
+
+      pextra = prev_extra_in_tile(ptile, ERM_CLEANPOLLUTION,
+                                        unit_owner(punit), punit);
+      if (clean_target == NULL) {
+        clean_target = pextra; /* Fallback */
+      }
+
+      if (pextra != NULL
           && can_unit_do_activity_targeted(punit, ACTIVITY_POLLUTION,
-                                           clean_target)) {
+                                           pextra)) {
         time = turns_to_activity_done(ptile, ACTIVITY_POLLUTION,
-                                      clean_target, punit);
+                                      pextra, punit);
         /* TRANS: "Clean Pollution (P) 3 turns" */
         fc_snprintf(cbuf, sizeof(cbuf), _("Clean %s (%s) %d %s"),
-                    extra_name_translation(clean_target), "P", time,
+                    extra_name_translation(pextra), "P", time,
                     PL_("turn", "turns", time));
         copy_chars_to_utf8_str(order_pollution_button->info_label, cbuf);
         clear_wflag(order_pollution_button, WF_HIDDEN);
@@ -1332,6 +1339,11 @@
 
       pextra = prev_extra_in_tile(ptile, ERM_CLEANFALLOUT,
                                   unit_owner(punit), punit);
+
+      if (clean_target == NULL) {
+        clean_target = pextra; /* Fallback */
+      }
+
       if (pextra != NULL
           && can_unit_do_activity_targeted(punit, ACTIVITY_FALLOUT, pextra)) {
         time = turns_to_activity_done(ptile, ACTIVITY_FALLOUT, pextra,
@@ -1346,9 +1358,6 @@
         set_wflag(order_fallout_button, WF_HIDDEN);
       }
 
-      if (clean_target == NULL) {
-        clean_target = pextra;
-      }
       if (clean_target != NULL
           && can_unit_do_activity_targeted(punit, ACTIVITY_CLEAN,
                                            clean_target)) {
diff -Nurd freeciv/client/helpdata.c freeciv/client/helpdata.c
--- freeciv/client/helpdata.c	2023-04-22 00:28:05.988606340 +0300
+++ freeciv/client/helpdata.c	2023-04-22 00:28:22.228750376 +0300
@@ -270,6 +270,17 @@
             transform_result);
 
         if (clean_time != 0) {
+          extra_type_by_rmcause_iterate(ERM_CLEAN, pextra) {
+            int rmtime = pterrain->extra_removal_times[extra_index(pextra)];
+
+            if (rmtime != 0) {
+              if (clean_time < 0) {
+                clean_time = rmtime;
+              } else if (clean_time != rmtime) {
+                clean_time = 0; /* Give up */
+              }
+            }
+          } extra_type_by_rmcause_iterate_end;
           extra_type_by_rmcause_iterate(ERM_CLEANPOLLUTION, pextra) {
             int rmtime = pterrain->extra_removal_times[extra_index(pextra)];
 
@@ -313,6 +324,21 @@
     {
       int time = -1, factor = -1;
 
+      extra_type_by_rmcause_iterate(ERM_CLEAN, pextra) {
+        if (pextra->removal_time == 0) {
+          if (factor < 0) {
+            factor = pextra->removal_time_factor;
+          } else if (factor != pextra->removal_time_factor) {
+            factor = 0; /* Give up */
+          }
+        } else {
+          if (time < 0) {
+            time = pextra->removal_time;
+          } else if (time != pextra->removal_time) {
+            time = 0; /* Give up */
+          }
+        }
+      } extra_type_by_rmcause_iterate_end;
       extra_type_by_rmcause_iterate(ERM_CLEANPOLLUTION, pextra) {
         if (pextra->removal_time == 0) {
           if (factor < 0) {
diff -Nurd freeciv/common/actions.c freeciv/common/actions.c
--- freeciv/common/actions.c	2023-04-22 00:28:06.004606482 +0300
+++ freeciv/common/actions.c	2023-04-22 00:31:08.294224715 +0300
@@ -2299,7 +2299,8 @@
   case ACTRES_PILLAGE:
     return is_extra_removed_by(pextra, ERM_PILLAGE);
   case ACTRES_CLEAN:
-    return is_extra_removed_by(pextra, ERM_CLEANPOLLUTION)
+    return is_extra_removed_by(pextra, ERM_CLEAN)
+      || is_extra_removed_by(pextra, ERM_CLEANPOLLUTION)
       || is_extra_removed_by(pextra, ERM_CLEANFALLOUT);
   case ACTRES_CLEAN_POLLUTION:
     return is_extra_removed_by(pextra, ERM_CLEANPOLLUTION);
@@ -4321,31 +4322,37 @@
   case ACTRES_CLEAN:
     {
       const struct extra_type *pextra = NULL;
-      const struct extra_type *fextra = NULL;
 
       pterrain = tile_terrain(target->tile);
 
       if (target_extra != NULL) {
-        if (is_extra_removed_by(target_extra, ERM_CLEANPOLLUTION)
-            && tile_has_extra(target->tile, target_extra)) {
+        if (tile_has_extra(target->tile, target_extra)
+            && (is_extra_removed_by(target_extra, ERM_CLEAN)
+                || is_extra_removed_by(target_extra, ERM_CLEANPOLLUTION)
+                || is_extra_removed_by(target_extra, ERM_CLEANFALLOUT))) {
           pextra = target_extra;
         }
-        if (is_extra_removed_by(target_extra, ERM_CLEANFALLOUT)
-            && tile_has_extra(target->tile, target_extra)) {
-          fextra = target_extra;
-        }
       } else {
         /* TODO: Make sure that all callers set target so that
          * we don't need this fallback. */
 
         pextra = prev_extra_in_tile(target->tile,
-                                    ERM_CLEANPOLLUTION,
-                                    actor->player,
-                                    actor->unit);
-        fextra = prev_extra_in_tile(target->tile,
-                                    ERM_CLEANFALLOUT,
+                                    ERM_CLEAN,
                                     actor->player,
                                     actor->unit);
+        if (pextra == NULL) {
+          pextra = prev_extra_in_tile(target->tile,
+                                      ERM_CLEANPOLLUTION,
+                                      actor->player,
+                                      actor->unit);
+
+          if (pextra == NULL) {
+            pextra = prev_extra_in_tile(target->tile,
+                                        ERM_CLEANFALLOUT,
+                                        actor->player,
+                                        actor->unit);
+          }
+        }
       }
 
       if (pextra != NULL && pterrain->extra_removal_times[extra_index(pextra)] > 0
@@ -4353,11 +4360,6 @@
         return TRI_YES;
       }
 
-      if (fextra != NULL && pterrain->extra_removal_times[extra_index(fextra)] > 0
-          && can_remove_extra(fextra, actor->unit, target->tile)) {
-        return TRI_YES;
-      }
-
       return TRI_NO;
     }
 
diff -Nurd freeciv/common/clientutils.c freeciv/common/clientutils.c
--- freeciv/common/clientutils.c	2023-04-22 00:28:06.004606482 +0300
+++ freeciv/common/clientutils.c	2023-04-22 00:28:22.228750376 +0300
@@ -258,7 +258,9 @@
       case ACTIVITY_PILLAGE:
         rmcause = ERM_PILLAGE;
         break;
-      case ACTIVITY_CLEAN: /* ERM_CLEANPOLLUTION first, ERM_CLEANFALLOUT later */
+      case ACTIVITY_CLEAN:
+        rmcause = ERM_CLEAN; /* Also ERM_CLEANPOLLUTION and ERM_CLEANFALLOUT */
+        break;
       case ACTIVITY_POLLUTION:
         rmcause = ERM_CLEANPOLLUTION;
         break;
@@ -288,8 +290,25 @@
       }
 
       if (i == ACTIVITY_CLEAN) {
+        extra_type_by_rmcause_iterate(ERM_CLEANPOLLUTION, ep) {
+          /* Make sure it's not handled by earlier iteration already */
+          if (!is_extra_caused_by(ep, ERM_CLEAN)) {
+            int ei = extra_index(ep);
+
+            if (calc->rmextra_turns[ei][i] > 0) {
+              if (num_activities > 0) {
+                astr_add(&str, "/");
+              }
+              astr_add(&str, _("Clean %s(%d)"),
+                       extra_name_translation(ep), calc->rmextra_turns[ei][i]);
+              num_activities++;
+            }
+          }
+        } extra_type_by_rmcause_iterate_end;
         extra_type_by_rmcause_iterate(ERM_CLEANFALLOUT, ep) {
-          if (!is_extra_caused_by(ep, ERM_CLEANPOLLUTION)) {
+          /* Make sure it's not handled by earlier iterations already */
+          if (!is_extra_caused_by(ep, ERM_CLEAN)
+              && !is_extra_caused_by(ep, ERM_CLEANPOLLUTION)) {
             int ei = extra_index(ep);
 
             if (calc->rmextra_turns[ei][i] > 0) {
diff -Nurd freeciv/common/extras.c freeciv/common/extras.c
--- freeciv/common/extras.c	2023-04-22 00:28:06.004606482 +0300
+++ freeciv/common/extras.c	2023-04-22 00:28:22.228750376 +0300
@@ -977,7 +977,8 @@
 {
   /* Is any of the worker remove action bits set? */
   return (pextra->rmcauses
-          & (1 << ERM_CLEANPOLLUTION
+          & (1 << ERM_CLEAN
+             | 1 << ERM_CLEANPOLLUTION
              | 1 << ERM_CLEANFALLOUT
              | 1 << ERM_PILLAGE));
 }
@@ -1029,6 +1030,8 @@
 enum extra_rmcause activity_to_extra_rmcause(enum unit_activity act)
 {
   switch (act) {
+  case ACTIVITY_CLEAN:
+    return ERM_CLEAN;
   case ACTIVITY_PILLAGE:
     return ERM_PILLAGE;
   case ACTIVITY_POLLUTION:
diff -Nurd freeciv/common/fc_types.h freeciv/common/fc_types.h
--- freeciv/common/fc_types.h	2023-04-22 00:28:06.004606482 +0300
+++ freeciv/common/fc_types.h	2023-04-22 00:28:22.228750376 +0300
@@ -1083,14 +1083,16 @@
 #define SPECENUM_NAME extra_rmcause
 #define SPECENUM_VALUE0 ERM_PILLAGE
 #define SPECENUM_VALUE0NAME "Pillage"
-#define SPECENUM_VALUE1 ERM_CLEANPOLLUTION
-#define SPECENUM_VALUE1NAME "CleanPollution"
+#define SPECENUM_VALUE1 ERM_CLEAN
+#define SPECENUM_VALUE1NAME "Clean"
 #define SPECENUM_VALUE2 ERM_CLEANFALLOUT
 #define SPECENUM_VALUE2NAME "CleanFallout"
 #define SPECENUM_VALUE3 ERM_DISAPPEARANCE
 #define SPECENUM_VALUE3NAME "Disappear"
 #define SPECENUM_VALUE4 ERM_ENTER
 #define SPECENUM_VALUE4NAME "Enter"
+#define SPECENUM_VALUE5 ERM_CLEANPOLLUTION
+#define SPECENUM_VALUE5NAME "CleanPollution"
 #define SPECENUM_COUNT ERM_COUNT
 #define SPECENUM_BITVECTOR bv_rmcauses
 #include "specenum_gen.h"
diff -Nurd freeciv/data/alien/terrain.ruleset freeciv/data/alien/terrain.ruleset
--- freeciv/data/alien/terrain.ruleset	2023-04-22 00:28:06.028606696 +0300
+++ freeciv/data/alien/terrain.ruleset	2023-04-22 00:28:22.232750413 +0300
@@ -692,8 +692,8 @@
 ;                           (the last three require a corresponding
 ;                           [resource_*] / [base_*] / [road_*] section)
 ; rmcauses                = events that can remove extra type.
-;                           "CleanPollution", "CleanFallout", "Pillage",
-;                           "Disappear", or "Enter"
+;                           "Clean", "Pillage", "Disappear", or "Enter"
+;                           For now also "CleanPollution", "CleanFallout"
 ; infracost               = Number of infrapoints it costs to place this
 ;                           extra. 0, the default, means extra cannot be placed.
 ; graphic                 = tag specifying preferred graphic
@@ -847,7 +847,7 @@
 name           = _("Pollution")
 category       = "Nuisance"
 causes         = "Pollution"
-rmcauses       = "CleanPollution"
+rmcauses       = "Clean", "CleanPollution"
 graphic        = "tx.pollution"
 graphic_alt    = "-"
 activity_gfx   = "None"
@@ -934,7 +934,7 @@
 name           = _("Fallout")
 category       = "Nuisance"
 causes         = "Fallout"
-rmcauses       = "CleanFallout"
+rmcauses       = "Clean", "CleanFallout"
 graphic        = "tx.fallout"
 graphic_alt    = "-"
 activity_gfx   = "None"
diff -Nurd freeciv/data/civ1/terrain.ruleset freeciv/data/civ1/terrain.ruleset
--- freeciv/data/civ1/terrain.ruleset	2023-04-22 00:28:06.052606909 +0300
+++ freeciv/data/civ1/terrain.ruleset	2023-04-22 00:28:22.232750413 +0300
@@ -859,8 +859,8 @@
 ;                           (the last three require a corresponding
 ;                           [resource_*] / [base_*] / [road_*] section)
 ; rmcauses                = events that can remove extra type.
-;                           "CleanPollution", "CleanFallout", "Pillage",
-;                           "Disappear", or "Enter"
+;                           "Clean", "Pillage", "Disappear", or "Enter"
+;                           For now also "CleanPollution", "CleanFallout"
 ; infracost               = Number of infrapoints it costs to place this
 ;                           extra. 0, the default, means extra cannot be placed.
 ; graphic                 = tag specifying preferred graphic
@@ -1021,7 +1021,7 @@
 name           = _("Pollution")
 category       = "Nuisance"
 causes         = "Pollution", "Fallout"
-rmcauses       = "CleanPollution"
+rmcauses       = "Clean", "CleanPollution"
 graphic        = "tx.pollution"
 graphic_alt    = "-"
 activity_gfx   = "None"
diff -Nurd freeciv/data/civ2/terrain.ruleset freeciv/data/civ2/terrain.ruleset
--- freeciv/data/civ2/terrain.ruleset	2023-04-22 00:28:06.052606909 +0300
+++ freeciv/data/civ2/terrain.ruleset	2023-04-22 00:28:22.232750413 +0300
@@ -930,8 +930,8 @@
 ;                           (the last three require a corresponding
 ;                           [resource_*] / [base_*] / [road_*] section)
 ; rmcauses                = events that can remove extra type.
-;                           "CleanPollution", "CleanFallout", "Pillage",
-;                           "Disappear", or "Enter"
+;                           "Clean", "Pillage", "Disappear", or "Enter"
+;                           For now also "CleanPollution", "CleanFallout"
 ; infracost               = Number of infrapoints it costs to place this
 ;                           extra. 0, the default, means extra cannot be placed.
 ; graphic                 = tag specifying preferred graphic
@@ -1093,7 +1093,7 @@
 name           = _("Pollution")
 category       = "Nuisance"
 causes         = "Pollution", "Fallout"
-rmcauses       = "CleanPollution"
+rmcauses       = "Clean", "CleanPollution"
 graphic        = "tx.pollution"
 graphic_alt    = "-"
 activity_gfx   = "None"
diff -Nurd freeciv/data/civ2civ3/terrain.ruleset freeciv/data/civ2civ3/terrain.ruleset
--- freeciv/data/civ2civ3/terrain.ruleset	2023-04-22 00:28:06.052606909 +0300
+++ freeciv/data/civ2civ3/terrain.ruleset	2023-04-22 00:28:22.232750413 +0300
@@ -1135,8 +1135,8 @@
 ;                           (the last three require a corresponding
 ;                           [resource_*] / [base_*] / [road_*] section)
 ; rmcauses                = events that can remove extra type.
-;                           "CleanPollution", "CleanFallout", "Pillage",
-;                           "Disappear", or "Enter"
+;                           "Clean", "Pillage", "Disappear", or "Enter"
+;                           For now also "CleanPollution", "CleanFallout"
 ; infracost               = Number of infrapoints it costs to place this
 ;                           extra. 0, the default, means extra cannot be placed.
 ; graphic                 = tag specifying preferred graphic
@@ -1365,7 +1365,7 @@
 name           = _("Pollution")
 category       = "Nuisance"
 causes         = "Pollution"
-rmcauses       = "CleanPollution"
+rmcauses       = "Clean", "CleanPollution"
 graphic        = "tx.pollution"
 graphic_alt    = "-"
 activity_gfx   = "None"
@@ -1493,7 +1493,7 @@
 name           = _("Fallout")
 category       = "Nuisance"
 causes         = "Fallout"
-rmcauses       = "CleanFallout"
+rmcauses       = "Clean", "CleanFallout"
 graphic        = "tx.fallout"
 graphic_alt    = "-"
 activity_gfx   = "None"
diff -Nurd freeciv/data/classic/terrain.ruleset freeciv/data/classic/terrain.ruleset
--- freeciv/data/classic/terrain.ruleset	2023-04-22 00:28:06.052606909 +0300
+++ freeciv/data/classic/terrain.ruleset	2023-04-22 00:28:22.232750413 +0300
@@ -1114,8 +1114,8 @@
 ;                           (the last three require a corresponding
 ;                           [resource_*] / [base_*] / [road_*] section)
 ; rmcauses                = events that can remove extra type.
-;                           "CleanPollution", "CleanFallout", "Pillage",
-;                           "Disappear", or "Enter"
+;                           "Clean", "Pillage", "Disappear", or "Enter"
+;                           For now also "CleanPollution", "CleanFallout"
 ; infracost               = Number of infrapoints it costs to place this
 ;                           extra. 0, the default, means extra cannot be placed.
 ; graphic                 = tag specifying preferred graphic
@@ -1308,7 +1308,7 @@
 name           = _("Pollution")
 category       = "Nuisance"
 causes         = "Pollution"
-rmcauses       = "CleanPollution"
+rmcauses       = "Clean", "CleanPollution"
 graphic        = "tx.pollution"
 graphic_alt    = "-"
 activity_gfx   = "None"
@@ -1430,7 +1430,7 @@
 name           = _("Fallout")
 category       = "Nuisance"
 causes         = "Fallout"
-rmcauses       = "CleanFallout"
+rmcauses       = "Clean", "CleanFallout"
 graphic        = "tx.fallout"
 graphic_alt    = "-"
 activity_gfx   = "None"
diff -Nurd freeciv/data/goldkeep/terrain.ruleset freeciv/data/goldkeep/terrain.ruleset
--- freeciv/data/goldkeep/terrain.ruleset	2023-04-22 00:28:06.140607689 +0300
+++ freeciv/data/goldkeep/terrain.ruleset	2023-04-22 00:28:22.232750413 +0300
@@ -1090,8 +1090,8 @@
 ;                           (the last three require a corresponding
 ;                           [resource_*] / [base_*] / [road_*] section)
 ; rmcauses                = events that can remove extra type.
-;                           "CleanPollution", "CleanFallout", "Pillage",
-;                           "Disappear", or "Enter"
+;                           "Clean", "Pillage", "Disappear", or "Enter"
+;                           For now also "CleanPollution", "CleanFallout"
 ; infracost               = Number of infrapoints it costs to place this
 ;                           extra. 0, the default, means extra cannot be placed.
 ; graphic                 = tag specifying preferred graphic
@@ -1284,7 +1284,7 @@
 name           = _("Pollution")
 category       = "Nuisance"
 causes         = "Pollution"
-rmcauses       = "CleanPollution"
+rmcauses       = "Clean", "CleanPollution"
 graphic        = "tx.pollution"
 graphic_alt    = "-"
 activity_gfx   = "None"
@@ -1406,7 +1406,7 @@
 name           = _("Fallout")
 category       = "Nuisance"
 causes         = "Fallout"
-rmcauses       = "CleanFallout"
+rmcauses       = "Clean", "CleanFallout"
 graphic        = "tx.fallout"
 graphic_alt    = "-"
 activity_gfx   = "None"
diff -Nurd freeciv/data/granularity/terrain.ruleset freeciv/data/granularity/terrain.ruleset
--- freeciv/data/granularity/terrain.ruleset	2023-04-22 00:28:06.140607689 +0300
+++ freeciv/data/granularity/terrain.ruleset	2023-04-22 00:28:22.232750413 +0300
@@ -463,8 +463,8 @@
 ;                           (the last three require a corresponding
 ;                           [resource_*] / [base_*] / [road_*] section)
 ; rmcauses                = events that can remove extra type.
-;                           "CleanPollution", "CleanFallout", "Pillage",
-;                           "Disappear", or "Enter"
+;                           "Clean", "Pillage", "Disappear", or "Enter"
+;                           For now also "CleanPollution", "CleanFallout"
 ; infracost               = Number of infrapoints it costs to place this
 ;                           extra. 0, the default, means extra cannot be placed.
 ; graphic                 = tag specifying preferred graphic
diff -Nurd freeciv/data/multiplayer/terrain.ruleset freeciv/data/multiplayer/terrain.ruleset
--- freeciv/data/multiplayer/terrain.ruleset	2023-04-22 00:28:06.280608931 +0300
+++ freeciv/data/multiplayer/terrain.ruleset	2023-04-22 00:28:22.232750413 +0300
@@ -1088,8 +1088,8 @@
 ;                           (the last three require a corresponding
 ;                           [resource_*] / [base_*] / [road_*] section)
 ; rmcauses                = events that can remove extra type.
-;                           "CleanPollution", "CleanFallout", "Pillage",
-;                           "Disappear", or "Enter"
+;                           "Clean", "Pillage", "Disappear", or "Enter"
+;                           For now also "CleanPollution", "CleanFallout"
 ; infracost               = Number of infrapoints it costs to place this
 ;                           extra. 0, the default, means extra cannot be placed.
 ; graphic                 = tag specifying preferred graphic
@@ -1282,7 +1282,7 @@
 name           = _("Pollution")
 category       = "Nuisance"
 causes         = "Pollution"
-rmcauses       = "CleanPollution"
+rmcauses       = "Clean", "CleanPollution"
 graphic        = "tx.pollution"
 graphic_alt    = "-"
 activity_gfx   = "None"
@@ -1403,7 +1403,7 @@
 name           = _("Fallout")
 category       = "Nuisance"
 causes         = "Fallout"
-rmcauses       = "CleanFallout"
+rmcauses       = "Clean", "CleanFallout"
 graphic        = "tx.fallout"
 graphic_alt    = "-"
 activity_gfx   = "None"
diff -Nurd freeciv/data/ruledit/comments-3.3.txt freeciv/data/ruledit/comments-3.3.txt
--- freeciv/data/ruledit/comments-3.3.txt	2023-04-22 00:28:06.292609037 +0300
+++ freeciv/data/ruledit/comments-3.3.txt	2023-04-22 00:28:22.232750413 +0300
@@ -747,8 +747,8 @@
 ;                           (the last three require a corresponding\n\
 ;                           [resource_*] / [base_*] / [road_*] section)\n\
 ; rmcauses                = events that can remove extra type.\n\
-;                           \"CleanPollution\", \"CleanFallout\", \"Pillage\",\n\
-;                           \"Disappear\", or \"Enter\"\n\
+;                           \"Clean\", \"Pillage\", \"Disappear\", or \"Enter\"\n\
+;                           For now also \"CleanPollution\", \"CleanFallout\"\n\
 ; infracost               = Number of infrapoints it costs to place this\n\
 ;                           extra. 0, the default, means extra cannot be placed.\n\
 ; graphic                 = tag specifying preferred graphic\n\
diff -Nurd freeciv/data/sandbox/terrain.ruleset freeciv/data/sandbox/terrain.ruleset
--- freeciv/data/sandbox/terrain.ruleset	2023-04-22 00:28:06.296609072 +0300
+++ freeciv/data/sandbox/terrain.ruleset	2023-04-22 00:28:22.232750413 +0300
@@ -1136,8 +1136,8 @@
 ;                           (the last three require a corresponding
 ;                           [resource_*] / [base_*] / [road_*] section)
 ; rmcauses                = events that can remove extra type.
-;                           "CleanPollution", "CleanFallout", "Pillage",
-;                           "Disappear", or "Enter"
+;                           "Clean", "Pillage", "Disappear", or "Enter"
+;                           For now also "CleanPollution", "CleanFallout"
 ; infracost               = Number of infrapoints it costs to place this
 ;                           extra. 0, the default, means extra cannot be placed.
 ; graphic                 = tag specifying preferred graphic
@@ -1366,7 +1366,7 @@
 name           = _("Pollution")
 category       = "Nuisance"
 causes         = "Pollution"
-rmcauses       = "CleanPollution"
+rmcauses       = "Clean", "CleanPollution"
 graphic        = "tx.pollution"
 graphic_alt    = "-"
 activity_gfx   = "None"
@@ -1523,7 +1523,7 @@
 name           = _("Fallout")
 category       = "Nuisance"
 causes         = "Fallout"
-rmcauses       = "CleanFallout"
+rmcauses       = "Clean", "CleanFallout"
 graphic        = "tx.fallout"
 graphic_alt    = "-"
 activity_gfx   = "None"
diff -Nurd freeciv/data/stub/terrain.ruleset freeciv/data/stub/terrain.ruleset
--- freeciv/data/stub/terrain.ruleset	2023-04-22 00:28:06.304609144 +0300
+++ freeciv/data/stub/terrain.ruleset	2023-04-22 00:28:22.232750413 +0300
@@ -356,8 +356,8 @@
 ;                           (the last three require a corresponding
 ;                           [resource_*] / [base_*] / [road_*] section)
 ; rmcauses                = events that can remove extra type.
-;                           "CleanPollution", "CleanFallout", "Pillage",
-;                           "Disappear", or "Enter"
+;                           "Clean", "Pillage", "Disappear", or "Enter"
+;                           For now also "CleanPollution", "CleanFallout"
 ; infracost               = Number of infrapoints it costs to place this
 ;                           extra. 0, the default, means extra cannot be placed.
 ; graphic                 = tag specifying preferred graphic
diff -Nurd freeciv/data/webperimental/terrain.ruleset freeciv/data/webperimental/terrain.ruleset
--- freeciv/data/webperimental/terrain.ruleset	2023-04-22 00:28:06.332609392 +0300
+++ freeciv/data/webperimental/terrain.ruleset	2023-04-22 00:28:22.232750413 +0300
@@ -1114,8 +1114,8 @@
 ;                           (the last three require a corresponding
 ;                           [resource_*] / [base_*] / [road_*] section)
 ; rmcauses                = events that can remove extra type.
-;                           "CleanPollution", "CleanFallout", "Pillage",
-;                           "Disappear", or "Enter"
+;                           "Clean", "Pillage", "Disappear", or "Enter"
+;                           For now also "CleanPollution", "CleanFallout"
 ; infracost               = Number of infrapoints it costs to place this
 ;                           extra. 0, the default, means extra cannot be placed.
 ; graphic                 = tag specifying preferred graphic
@@ -1308,7 +1308,7 @@
 name           = _("Pollution")
 category       = "Nuisance"
 causes         = "Pollution"
-rmcauses       = "CleanPollution"
+rmcauses       = "Clean", "CleanPollution"
 graphic        = "tx.pollution"
 graphic_alt    = "-"
 activity_gfx   = "None"
@@ -1430,7 +1430,7 @@
 name           = _("Fallout")
 category       = "Nuisance"
 causes         = "Fallout"
-rmcauses       = "CleanFallout"
+rmcauses       = "Clean", "CleanFallout"
 graphic        = "tx.fallout"
 graphic_alt    = "-"
 activity_gfx   = "None"
diff -Nurd freeciv/server/rssanity.c freeciv/server/rssanity.c
--- freeciv/server/rssanity.c	2023-04-22 00:28:06.344609499 +0300
+++ freeciv/server/rssanity.c	2023-04-22 00:28:22.232750413 +0300
@@ -1148,8 +1148,8 @@
     }
     if ((requirement_vector_size(&pextra->rmreqs) > 0)
         && !(pextra->rmcauses
-             & (ERM_ENTER | ERM_CLEANPOLLUTION
-                | ERM_CLEANFALLOUT | ERM_PILLAGE))) {
+             & (ERM_ENTER | ERM_CLEAN | ERM_PILLAGE
+                | ERM_CLEANPOLLUTION | ERM_CLEANFALLOUT))) {
       ruleset_error(logger, LOG_WARN,
                     "Requirements for extra removal defined but not "
                     "a valid remove cause!");
diff -Nurd freeciv/server/unithand.c freeciv/server/unithand.c
--- freeciv/server/unithand.c	2023-04-22 00:28:06.348609534 +0300
+++ freeciv/server/unithand.c	2023-04-22 00:28:22.232750413 +0300
@@ -4299,11 +4299,16 @@
       }
 
     } else if (activity == ACTIVITY_CLEAN) {
-      activity_target = prev_extra_in_tile(unit_tile(punit), ERM_CLEANPOLLUTION,
+      activity_target = prev_extra_in_tile(unit_tile(punit), ERM_CLEAN,
                                            pplayer, punit);
+
       if (activity_target == NULL) {
-        activity_target = prev_extra_in_tile(unit_tile(punit), ERM_CLEANFALLOUT,
+        activity_target = prev_extra_in_tile(unit_tile(punit), ERM_CLEANPOLLUTION,
                                              pplayer, punit);
+        if (activity_target == NULL) {
+          activity_target = prev_extra_in_tile(unit_tile(punit), ERM_CLEANFALLOUT,
+                                               pplayer, punit);
+        }
       }
     } else if (activity == ACTIVITY_POLLUTION) {
       activity_target = prev_extra_in_tile(unit_tile(punit), ERM_CLEANPOLLUTION,
diff -Nurd freeciv/server/unittools.c freeciv/server/unittools.c
--- freeciv/server/unittools.c	2023-04-22 00:28:06.344609499 +0300
+++ freeciv/server/unittools.c	2023-04-22 00:28:22.232750413 +0300
@@ -954,32 +954,31 @@
      *       set */
     {
       struct extra_type *pextra;
-      struct extra_type *fextra;
 
       if (punit->activity_target == NULL) {
-        pextra = prev_extra_in_tile(ptile, ERM_CLEANPOLLUTION,
+        pextra = prev_extra_in_tile(ptile, ERM_CLEAN,
                                     NULL, punit);
-
         if (pextra != NULL) {
           punit->activity_target = pextra;
-          fextra = NULL; /* fextra not needed, keep compiler happy */
         } else {
-          fextra = prev_extra_in_tile(ptile, ERM_CLEANFALLOUT,
+          pextra = prev_extra_in_tile(ptile, ERM_CLEANPOLLUTION,
                                       NULL, punit);
-          punit->activity_target = fextra;
+
+          if (pextra != NULL) {
+            punit->activity_target = pextra;
+          } else {
+            pextra = prev_extra_in_tile(ptile, ERM_CLEANFALLOUT,
+                                        NULL, punit);
+            punit->activity_target = pextra;
+          }
         }
       } else {
-        if (is_extra_removed_by(punit->activity_target, ERM_CLEANPOLLUTION)) {
+        if (is_extra_removed_by(punit->activity_target, ERM_CLEAN)
+            || is_extra_removed_by(punit->activity_target, ERM_CLEANPOLLUTION)
+            || is_extra_removed_by(punit->activity_target, ERM_CLEANFALLOUT)) {
           pextra = punit->activity_target;
-          fextra = NULL; /* fextra not needed, keep compiler happy */
         } else {
           pextra = NULL;
-
-          if (is_extra_removed_by(punit->activity_target, ERM_CLEANFALLOUT)) {
-            fextra = punit->activity_target;
-          } else {
-            fextra = NULL;
-          }
         }
       }
 
@@ -988,11 +987,6 @@
           destroy_extra(ptile, pextra);
           unit_activity_done = TRUE;
         }
-      } else if (fextra != NULL) {
-        if (total_activity_done(ptile, ACTIVITY_CLEAN, fextra)) {
-          destroy_extra(ptile, fextra);
-          unit_activity_done = TRUE;
-        }
       }
     }
     break;
