From aa69de60f28f66db0b2c4dc509fb5164faf8a7db Mon Sep 17 00:00:00 2001
From: Marko Lindqvist <cazfi74@gmail.com>
Date: Tue, 10 Dec 2019 07:39:19 +0200
Subject: [PATCH] Terminate format escapes list

See hrm Bug #851727

Signed-off-by: Marko Lindqvist <cazfi74@gmail.com>
---
 utility/shared.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/utility/shared.c b/utility/shared.c
index 38274ce79a..125259c5ce 100644
--- a/utility/shared.c
+++ b/utility/shared.c
@@ -2355,7 +2355,7 @@ static size_t extract_escapes(const char *format, char *escapes,
 {
   static const char format_escapes[] = {
     '*', 'd', 'i', 'o', 'u', 'x', 'X', 'e', 'E', 'f',
-    'F', 'g', 'G', 'a', 'A', 'c', 's', 'p', 'n',
+    'F', 'g', 'G', 'a', 'A', 'c', 's', 'p', 'n', '\0'
   };
   bool reordered = FALSE;
   size_t num = 0;
-- 
2.24.0

