diff -Nurd freeciv/client/gui-qt/helpdlg.cpp freeciv/client/gui-qt/helpdlg.cpp
--- freeciv/client/gui-qt/helpdlg.cpp	2021-04-17 03:16:31.696087433 +0300
+++ freeciv/client/gui-qt/helpdlg.cpp	2021-04-17 03:16:42.964198700 +0300
@@ -1327,8 +1327,8 @@
 
       tb = new QLabel(this);
       fc_snprintf(buffer, sizeof(buffer), PL_("%d turn", "%d turns",
-                                              pterrain->irrigation_time),
-                  pterrain->irrigation_time);
+                                              pterrain->cultivate_time),
+                  pterrain->cultivate_time);
       str = N_("Irrig. Rslt/Time:");;
       str = str + link_me(terrain_name_translation(pterrain->irrigation_result),
                           HELP_TERRAIN)
diff -Nurd freeciv/client/packhand.c freeciv/client/packhand.c
--- freeciv/client/packhand.c	2021-04-17 03:16:31.704087513 +0300
+++ freeciv/client/packhand.c	2021-04-17 03:16:42.964198700 +0300
@@ -3809,6 +3809,8 @@
 
   pterrain->base_time = p->base_time;
   pterrain->road_time = p->road_time;
+  pterrain->cultivate_time = p->cultivate_time;
+  pterrain->plant_time = p->plant_time;
   pterrain->irrigation_result = terrain_by_number(p->irrigation_result);
   pterrain->irrigation_food_incr = p->irrigation_food_incr;
   pterrain->irrigation_time = p->irrigation_time;
diff -Nurd freeciv/common/networking/packets.def freeciv/common/networking/packets.def
--- freeciv/common/networking/packets.def	2021-04-17 03:16:31.712087592 +0300
+++ freeciv/common/networking/packets.def	2021-04-17 03:16:42.964198700 +0300
@@ -1660,6 +1660,10 @@
   UINT8 base_time;
   UINT8 road_time;
 
+  UINT8 cultivate_time;
+
+  UINT8 plant_time;
+
   TERRAIN irrigation_result;
   UINT8 irrigation_food_incr;
   UINT8 irrigation_time;
diff -Nurd freeciv/common/terrain.h freeciv/common/terrain.h
--- freeciv/common/terrain.h	2021-04-17 03:16:31.708087553 +0300
+++ freeciv/common/terrain.h	2021-04-17 03:16:42.964198700 +0300
@@ -200,6 +200,10 @@
   int base_time;
   int road_time;
 
+  int cultivate_time;
+
+  int plant_time;
+
   struct terrain *irrigation_result;
   int irrigation_food_incr;
   int irrigation_time;
diff -Nurd freeciv/common/tile.c freeciv/common/tile.c
--- freeciv/common/tile.c	2021-04-17 03:16:31.708087553 +0300
+++ freeciv/common/tile.c	2021-04-17 03:16:42.964198700 +0300
@@ -442,9 +442,9 @@
   case ACTIVITY_TRANSFORM:
     return pterrain->transform_time * ACTIVITY_FACTOR;
   case ACTIVITY_CULTIVATE:
-    return pterrain->irrigation_time * ACTIVITY_FACTOR;
+    return pterrain->cultivate_time * ACTIVITY_FACTOR;
   case ACTIVITY_PLANT:
-    return pterrain->mining_time * ACTIVITY_FACTOR;
+    return pterrain->plant_time * ACTIVITY_FACTOR;
   case ACTIVITY_IRRIGATE:
   case ACTIVITY_MINE:
   case ACTIVITY_BASE:
diff -Nurd freeciv/data/alien/terrain.ruleset freeciv/data/alien/terrain.ruleset
--- freeciv/data/alien/terrain.ruleset	2021-04-17 03:16:31.712087592 +0300
+++ freeciv/data/alien/terrain.ruleset	2021-04-17 03:16:42.964198700 +0300
@@ -116,6 +116,8 @@
 ;                        Nonzero values only affect extras with build_time 0.
 ; road_time            = time to build roads; if 0, cannot build any roads.
 ;                        Nonzero values only affect extras with build_time 0.
+; cultivate_time       = time to cultivate
+; plant_time           = time to plant
 ; irrigation_result    = result of 'irrigate' activity; one of:
 ;                        "no"  -- cannot irrigate
 ;                        "yes" -- can build extra
@@ -226,6 +228,8 @@
 road_trade_incr_pct  = 0
 base_time            = 0
 road_time            = 0
+cultivate_time       = 0
+plant_time           = 0
 irrigation_result    = "no"
 irrigation_food_incr = 0
 irrigation_time      = 0
@@ -265,6 +269,8 @@
 road_trade_incr_pct  = 100
 base_time            = 1
 road_time            = 2
+cultivate_time       = 0
+plant_time           = 0
 irrigation_result    = "yes"
 irrigation_food_incr = 1
 irrigation_time      = 5
@@ -307,12 +313,14 @@
 road_trade_incr_pct  = 0
 base_time            = 1
 road_time            = 5
+cultivate_time       = 0
+plant_time           = 10
 irrigation_result    = "no"
 irrigation_food_incr = 0
 irrigation_time      = 10
 mining_result        = "Radiating Rocks"
 mining_shield_incr   = 0
-mining_time          = 10
+mining_time          = 0
 transform_result     = "no"
 transform_time       = 24
 pillage_time         = 1
@@ -352,9 +360,11 @@
 road_trade_incr_pct  = 0
 base_time            = 1
 road_time            = 7
+cultivate_time       = 10
+plant_time           = 0
 irrigation_result    = "Alien Forest"
 irrigation_food_incr = 0
-irrigation_time      = 10
+irrigation_time      = 0
 mining_result        = "no"
 mining_shield_incr   = 0
 mining_time          = 10
@@ -396,6 +406,8 @@
 road_trade_incr_pct  = 0
 base_time            = 1
 road_time            = 6
+cultivate_time       = 0
+plant_time           = 0
 irrigation_result    = "yes"
 irrigation_food_incr = 1
 irrigation_time      = 10
@@ -437,6 +449,8 @@
 road_trade_incr_pct  = 0
 base_time            = 1
 road_time            = 12
+cultivate_time       = 0
+plant_time           = 0
 irrigation_result    = "no"
 irrigation_food_incr = 0
 irrigation_time      = 0
@@ -479,6 +493,8 @@
 road_trade_incr_pct  = 0
 base_time            = 1
 road_time            = 1
+cultivate_time       = 0
+plant_time           = 0
 irrigation_result    = "no"
 irrigation_food_incr = 0
 irrigation_time      = 0
@@ -520,6 +536,8 @@
 road_trade_incr_pct  = 0
 base_time            = 1
 road_time            = 1
+cultivate_time       = 0
+plant_time           = 0
 irrigation_result    = "no"
 irrigation_food_incr = 0
 irrigation_time      = 0
diff -Nurd freeciv/data/civ1/terrain.ruleset freeciv/data/civ1/terrain.ruleset
--- freeciv/data/civ1/terrain.ruleset	2021-04-17 03:16:31.720087671 +0300
+++ freeciv/data/civ1/terrain.ruleset	2021-04-17 03:16:42.964198700 +0300
@@ -110,6 +110,8 @@
 ;                        Nonzero values only affect extras with build_time 0.
 ; road_time            = time to build roads; if 0, cannot build any roads.
 ;                        Nonzero values only affect extras with build_time 0.
+; cultivate_time       = time to cultivate
+; plant_time           = time to plant
 ; irrigation_result    = result of 'irrigate' activity; one of:
 ;                        "no"  -- cannot irrigate
 ;                        "yes" -- can build extra
@@ -219,6 +221,8 @@
 road_trade_incr_pct  = 0
 base_time            = 1
 road_time            = 0
+cultivate_time       = 0
+plant_time           = 0
 irrigation_result    = "no"
 irrigation_food_incr = 0
 irrigation_time      = 0
@@ -262,6 +266,8 @@
 road_trade_incr_pct  = 0
 base_time            = 1
 road_time            = 4
+cultivate_time       = 0
+plant_time           = 0
 irrigation_result    = "no"
 irrigation_food_incr = 0
 irrigation_time      = 0
@@ -303,6 +309,8 @@
 road_trade_incr_pct  = 100
 base_time            = 1
 road_time            = 2
+cultivate_time       = 0
+plant_time           = 0
 irrigation_result    = "yes"
 irrigation_food_incr = 1
 irrigation_time      = 5
@@ -344,9 +352,11 @@
 road_trade_incr_pct  = 0
 base_time            = 1
 road_time            = 4
+cultivate_time       = 5
+plant_time           = 0
 irrigation_result    = "Plains"
 irrigation_food_incr = 0
-irrigation_time      = 5
+irrigation_time      = 0
 mining_result        = "no"
 mining_shield_incr   = 0
 mining_time          = 0
@@ -387,12 +397,14 @@
 road_trade_incr_pct  = 100
 base_time            = 1
 road_time            = 2
+cultivate_time       = 0
+plant_time           = 10
 irrigation_result    = "yes"
 irrigation_food_incr = 1
 irrigation_time      = 5
 mining_result        = "Forest"
 mining_shield_incr   = 0
-mining_time          = 10
+mining_time          = 0
 transform_result     = "no"
 transform_time       = 0
 pillage_time         = 1
@@ -427,6 +439,8 @@
 road_trade_incr_pct  = 0
 base_time            = 1
 road_time            = 4
+cultivate_time       = 0
+plant_time           = 0
 irrigation_result    = "yes"
 irrigation_food_incr = 1
 irrigation_time      = 10
@@ -472,12 +486,14 @@
 road_trade_incr_pct  = 0
 base_time            = 1
 road_time            = 4
+cultivate_time       = 15
+plant_time           = 15
 irrigation_result    = "Grassland"
 irrigation_food_incr = 0
-irrigation_time      = 15
+irrigation_time      = 0
 mining_result        = "Forest"
 mining_shield_incr   = 0
-mining_time          = 15
+mining_time          = 0
 transform_result     = "no"
 transform_time       = 0
 pillage_time         = 1
@@ -512,6 +528,8 @@
 road_trade_incr_pct  = 0
 base_time            = 1
 road_time            = 6
+cultivate_time       = 0
+plant_time           = 0
 irrigation_result    = "no"
 irrigation_food_incr = 0
 irrigation_time      = 0
@@ -555,12 +573,14 @@
 road_trade_incr_pct  = 100
 base_time            = 1
 road_time            = 2
+cultivate_time       = 0
+plant_time           = 15
 irrigation_result    = "yes"
 irrigation_food_incr = 1
 irrigation_time      = 5
 mining_result        = "Forest"
 mining_shield_incr   = 0
-mining_time          = 15
+mining_time          = 0
 transform_result     = "no"
 transform_time       = 0
 pillage_time         = 1
@@ -596,12 +616,14 @@
 road_trade_incr_pct  = 0
 base_time            = 1
 road_time            = 4
+cultivate_time       = 15
+plant_time           = 15
 irrigation_result    = "Grassland"
 irrigation_food_incr = 0
-irrigation_time      = 15
+irrigation_time      = 0
 mining_result        = "Forest"
 mining_shield_incr   = 0
-mining_time          = 15
+mining_time          = 0
 transform_result     = "no"
 transform_time       = 0
 pillage_time         = 1
@@ -638,6 +660,8 @@
 road_trade_incr_pct  = 0
 base_time            = 1
 road_time            = 2
+cultivate_time       = 0
+plant_time           = 0
 irrigation_result    = "no"
 irrigation_food_incr = 0
 irrigation_time      = 0
diff -Nurd freeciv/data/civ2/terrain.ruleset freeciv/data/civ2/terrain.ruleset
--- freeciv/data/civ2/terrain.ruleset	2021-04-17 03:16:31.720087671 +0300
+++ freeciv/data/civ2/terrain.ruleset	2021-04-17 03:16:42.964198700 +0300
@@ -110,6 +110,8 @@
 ;                        Nonzero values only affect extras with build_time 0.
 ; road_time            = time to build roads; if 0, cannot build any roads.
 ;                        Nonzero values only affect extras with build_time 0.
+; cultivate_time       = time to cultivate
+; plant_time           = time to plant
 ; irrigation_result    = result of 'irrigate' activity; one of:
 ;                        "no"  -- cannot irrigate
 ;                        "yes" -- can build extra
@@ -219,6 +221,8 @@
 road_trade_incr_pct  = 0
 base_time            = 1
 road_time            = 0
+cultivate_time       = 0
+plant_time           = 0
 irrigation_result    = "no"
 irrigation_food_incr = 0
 irrigation_time      = 0
@@ -258,6 +262,8 @@
 road_trade_incr_pct  = 0
 base_time            = 1
 road_time            = 4
+cultivate_time       = 0
+plant_time           = 0
 irrigation_result    = "no"
 irrigation_food_incr = 0
 irrigation_time      = 0
@@ -299,6 +305,8 @@
 road_trade_incr_pct  = 100
 base_time            = 1
 road_time            = 2
+cultivate_time       = 0
+plant_time           = 0
 irrigation_result    = "yes"
 irrigation_food_incr = 1
 irrigation_time      = 5
@@ -341,9 +349,11 @@
 road_trade_incr_pct  = 0
 base_time            = 1
 road_time            = 4
+cultivate_time       = 5
+plant_time           = 0
 irrigation_result    = "Plains"
 irrigation_food_incr = 0
-irrigation_time      = 5
+irrigation_time      = 0
 mining_result        = "no"
 mining_shield_incr   = 0
 mining_time          = 0
@@ -384,12 +394,14 @@
 road_trade_incr_pct  = 100
 base_time            = 1
 road_time            = 2
+cultivate_time       = 0
+plant_time           = 10
 irrigation_result    = "yes"
 irrigation_food_incr = 1
 irrigation_time      = 5
 mining_result        = "Forest"
 mining_shield_incr   = 0
-mining_time          = 10
+mining_time          = 0
 transform_result     = "Hills"
 transform_time       = 24
 pillage_time         = 1
@@ -424,6 +436,8 @@
 road_trade_incr_pct  = 0
 base_time            = 1
 road_time            = 4
+cultivate_time       = 0
+plant_time           = 0
 irrigation_result    = "yes"
 irrigation_food_incr = 1
 irrigation_time      = 10
@@ -465,12 +479,14 @@
 road_trade_incr_pct  = 0
 base_time            = 1
 road_time            = 4
+cultivate_time       = 15
+plant_time           = 15
 irrigation_result    = "Grassland"
 irrigation_food_incr = 0
-irrigation_time      = 15
+irrigation_time      = 0
 mining_result        = "Forest"
 mining_shield_incr   = 0
-mining_time          = 15
+mining_time          = 0
 transform_result     = "Plains"
 transform_time       = 24
 pillage_time         = 1
@@ -507,6 +523,8 @@
 road_trade_incr_pct  = 0
 base_time            = 1
 road_time            = 6
+cultivate_time       = 0
+plant_time           = 0
 irrigation_result    = "no"
 irrigation_food_incr = 0
 irrigation_time      = 0
@@ -547,12 +565,14 @@
 road_trade_incr_pct  = 100
 base_time            = 1
 road_time            = 2
+cultivate_time       = 0
+plant_time           = 15
 irrigation_result    = "yes"
 irrigation_food_incr = 1
 irrigation_time      = 5
 mining_result        = "Forest"
 mining_shield_incr   = 0
-mining_time          = 15
+mining_time          = 0
 transform_result     = "Grassland"
 transform_time       = 24
 pillage_time         = 1
@@ -589,12 +609,14 @@
 road_trade_incr_pct  = 0
 base_time            = 1
 road_time            = 4
+cultivate_time       = 15
+plant_time           = 15
 irrigation_result    = "Grassland"
 irrigation_food_incr = 0
-irrigation_time      = 15
+irrigation_time      = 0
 mining_result        = "Forest"
 mining_shield_incr   = 0
-mining_time          = 15
+mining_time          = 0
 transform_result     = "Plains"
 transform_time       = 24
 pillage_time         = 1
@@ -632,6 +654,8 @@
 road_trade_incr_pct  = 0
 base_time            = 1
 road_time            = 2
+cultivate_time       = 0
+plant_time           = 0
 irrigation_result    = "yes"
 irrigation_food_incr = 1
 irrigation_time      = 5
diff -Nurd freeciv/data/civ2civ3/terrain.ruleset freeciv/data/civ2civ3/terrain.ruleset
--- freeciv/data/civ2civ3/terrain.ruleset	2021-04-17 03:16:31.724087710 +0300
+++ freeciv/data/civ2civ3/terrain.ruleset	2021-04-17 03:16:42.964198700 +0300
@@ -122,6 +122,8 @@
 ;                        Nonzero values only affect extras with build_time 0.
 ; road_time            = time to build roads; if 0, cannot build any roads.
 ;                        Nonzero values only affect extras with build_time 0.
+; cultivate_time       = time to cultivate
+; plant_time           = time to plant
 ; irrigation_result    = result of 'irrigate' activity; one of:
 ;                        "no"  -- cannot irrigate
 ;                        "yes" -- can build extra
@@ -232,6 +234,8 @@
 road_trade_incr_pct  = 0
 base_time            = 0
 road_time            = 0
+cultivate_time       = 0
+plant_time           = 0
 irrigation_result    = "no"
 irrigation_food_incr = 0
 irrigation_time      = 0
@@ -273,6 +277,8 @@
 road_trade_incr_pct  = 0
 base_time            = 1
 road_time            = 0
+cultivate_time       = 0
+plant_time           = 0
 irrigation_result    = "no"
 irrigation_food_incr = 0
 irrigation_time      = 0
@@ -316,6 +322,8 @@
 road_trade_incr_pct  = 0
 base_time            = 1
 road_time            = 0
+cultivate_time       = 0
+plant_time           = 0
 irrigation_result    = "no"
 irrigation_food_incr = 0
 irrigation_time      = 0
@@ -360,6 +368,8 @@
 road_trade_incr_pct  = 0
 base_time            = 1
 road_time            = 0
+cultivate_time       = 0
+plant_time           = 0
 irrigation_result    = "no"
 irrigation_food_incr = 0
 irrigation_time      = 0
@@ -404,6 +414,8 @@
 road_trade_incr_pct  = 0
 base_time            = 1
 road_time            = 4
+cultivate_time       = 0
+plant_time           = 0
 irrigation_result    = "no"
 irrigation_food_incr = 0
 irrigation_time      = 0
@@ -449,6 +461,8 @@
 road_trade_incr_pct  = 100
 base_time            = 1
 road_time            = 2
+cultivate_time       = 0
+plant_time           = 0
 irrigation_result    = "yes"
 irrigation_food_incr = 1
 irrigation_time      = 5
@@ -503,12 +517,14 @@
 road_trade_incr_pct  = 0
 base_time            = 1
 road_time            = 4
+cultivate_time       = 10
+plant_time           = 15
 irrigation_result    = "Plains"
 irrigation_food_incr = 0
-irrigation_time      = 10
+irrigation_time      = 0
 mining_result        = "Grassland"
 mining_shield_incr   = 0
-mining_time          = 15
+mining_time          = 0
 transform_result     = "Hills"
 transform_time       = 24
 pillage_time         = 1
@@ -550,12 +566,14 @@
 road_trade_incr_pct  = 100
 base_time            = 1
 road_time            = 2
+cultivate_time       = 0
+plant_time           = 15
 irrigation_result    = "yes"
 irrigation_food_incr = 1
 irrigation_time      = 5
 mining_result        = "Forest"
 mining_shield_incr   = 0
-mining_time          = 15
+mining_time          = 0
 transform_result     = "Lake"
 transform_time       = 36
 pillage_time         = 1
@@ -594,6 +612,8 @@
 road_trade_incr_pct  = 0
 base_time            = 1
 road_time            = 4
+cultivate_time       = 0
+plant_time           = 0
 irrigation_result    = "yes"
 irrigation_food_incr = 1
 irrigation_time      = 10
@@ -642,9 +662,11 @@
 road_trade_incr_pct  = 0
 base_time            = 1
 road_time            = 4
+cultivate_time       = 10
+plant_time           = 0
 irrigation_result    = "Swamp"
 irrigation_food_incr = 0
-irrigation_time      = 10
+irrigation_time      = 0
 mining_result        = "no"
 mining_shield_incr   = 0
 mining_time          = 0
@@ -689,6 +711,8 @@
 road_trade_incr_pct  = 0
 base_time            = 1
 road_time            = 6
+cultivate_time       = 0
+plant_time           = 0
 irrigation_result    = "no"
 irrigation_food_incr = 0
 irrigation_time      = 0
@@ -738,12 +762,14 @@
 road_trade_incr_pct  = 100
 base_time            = 1
 road_time            = 2
+cultivate_time       = 0
+plant_time           = 10
 irrigation_result    = "yes"
 irrigation_food_incr = 1
 irrigation_time      = 5
 mining_result        = "Forest"
 mining_shield_incr   = 0
-mining_time          = 10
+mining_time          = 0
 transform_result     = "Lake"
 transform_time       = 36
 pillage_time         = 1
@@ -783,6 +809,8 @@
 road_trade_incr_pct  = 0
 base_time            = 1
 road_time            = 4
+cultivate_time       = 0
+plant_time           = 0
 irrigation_result    = "yes"
 irrigation_food_incr = 1
 irrigation_time      = 5
@@ -831,6 +859,8 @@
 road_trade_incr_pct  = 100
 base_time            = 1
 road_time            = 2
+cultivate_time       = 0
+plant_time           = 0
 irrigation_result    = "no"
 irrigation_food_incr = 0
 irrigation_time      = 0
diff -Nurd freeciv/data/classic/terrain.ruleset freeciv/data/classic/terrain.ruleset
--- freeciv/data/classic/terrain.ruleset	2021-04-17 03:16:31.724087710 +0300
+++ freeciv/data/classic/terrain.ruleset	2021-04-17 03:16:42.964198700 +0300
@@ -115,6 +115,8 @@
 ;                        Nonzero values only affect extras with build_time 0.
 ; road_time            = time to build roads; if 0, cannot build any roads.
 ;                        Nonzero values only affect extras with build_time 0.
+; cultivate_time       = time to cultivate
+; plant_time           = time to plant
 ; irrigation_result    = result of 'irrigate' activity; one of:
 ;                        "no"  -- cannot irrigate
 ;                        "yes" -- can build extra
@@ -225,6 +227,8 @@
 road_trade_incr_pct  = 0
 base_time            = 0
 road_time            = 0
+cultivate_time       = 0
+plant_time           = 0
 irrigation_result    = "no"
 irrigation_food_incr = 0
 irrigation_time      = 0
@@ -266,6 +270,8 @@
 road_trade_incr_pct  = 0
 base_time            = 1
 road_time            = 0
+cultivate_time       = 0
+plant_time           = 0
 irrigation_result    = "no"
 irrigation_food_incr = 0
 irrigation_time      = 0
@@ -309,6 +315,8 @@
 road_trade_incr_pct  = 0
 base_time            = 1
 road_time            = 0
+cultivate_time       = 0
+plant_time           = 0
 irrigation_result    = "no"
 irrigation_food_incr = 0
 irrigation_time      = 0
@@ -353,6 +361,8 @@
 road_trade_incr_pct  = 0
 base_time            = 1
 road_time            = 0
+cultivate_time       = 0
+plant_time           = 0
 irrigation_result    = "no"
 irrigation_food_incr = 0
 irrigation_time      = 0
@@ -397,6 +407,8 @@
 road_trade_incr_pct  = 0
 base_time            = 1
 road_time            = 4
+cultivate_time       = 0
+plant_time           = 0
 irrigation_result    = "no"
 irrigation_food_incr = 0
 irrigation_time      = 0
@@ -444,6 +456,8 @@
 road_trade_incr_pct  = 100
 base_time            = 1
 road_time            = 2
+cultivate_time       = 0
+plant_time           = 0
 irrigation_result    = "yes"
 irrigation_food_incr = 1
 irrigation_time      = 5
@@ -492,12 +506,14 @@
 road_trade_incr_pct  = 0
 base_time            = 1
 road_time            = 4
+cultivate_time       = 5
+plant_time           = 15
 irrigation_result    = "Plains"
 irrigation_food_incr = 0
-irrigation_time      = 5
+irrigation_time      = 0
 mining_result        = "Swamp"
 mining_shield_incr   = 0
-mining_time          = 15
+mining_time          = 0
 transform_result     = "Grassland"
 transform_time       = 24
 pillage_time         = 1
@@ -539,12 +555,14 @@
 road_trade_incr_pct  = 100
 base_time            = 1
 road_time            = 2
+cultivate_time       = 0
+plant_time           = 10
 irrigation_result    = "yes"
 irrigation_food_incr = 1
 irrigation_time      = 5
 mining_result        = "Forest"
 mining_shield_incr   = 0
-mining_time          = 10
+mining_time          = 0
 transform_result     = "Hills"
 transform_time       = 24
 pillage_time         = 1
@@ -583,6 +601,8 @@
 road_trade_incr_pct  = 0
 base_time            = 1
 road_time            = 4
+cultivate_time       = 0
+plant_time           = 0
 irrigation_result    = "yes"
 irrigation_food_incr = 1
 irrigation_time      = 10
@@ -628,12 +648,14 @@
 road_trade_incr_pct  = 0
 base_time            = 1
 road_time            = 4
+cultivate_time       = 15
+plant_time           = 15
 irrigation_result    = "Grassland"
 irrigation_food_incr = 0
-irrigation_time      = 15
+irrigation_time      = 0
 mining_result        = "Forest"
 mining_shield_incr   = 0
-mining_time          = 15
+mining_time          = 0
 transform_result     = "Plains"
 transform_time       = 24
 pillage_time         = 1
@@ -674,6 +696,8 @@
 road_trade_incr_pct  = 0
 base_time            = 1
 road_time            = 6
+cultivate_time       = 0
+plant_time           = 0
 irrigation_result    = "no"
 irrigation_food_incr = 0
 irrigation_time      = 0
@@ -718,12 +742,14 @@
 road_trade_incr_pct  = 100
 base_time            = 1
 road_time            = 2
+cultivate_time       = 0
+plant_time           = 15
 irrigation_result    = "yes"
 irrigation_food_incr = 1
 irrigation_time      = 5
 mining_result        = "Forest"
 mining_shield_incr   = 0
-mining_time          = 15
+mining_time          = 0
 transform_result     = "Grassland"
 transform_time       = 24
 pillage_time         = 1
@@ -763,12 +789,14 @@
 road_trade_incr_pct  = 0
 base_time            = 1
 road_time            = 4
+cultivate_time       = 15
+plant_time           = 15
 irrigation_result    = "Grassland"
 irrigation_food_incr = 0
-irrigation_time      = 15
+irrigation_time      = 0
 mining_result        = "Forest"
 mining_shield_incr   = 0
-mining_time          = 15
+mining_time          = 0
 transform_result     = "Lake"
 transform_time       = 36
 pillage_time         = 1
@@ -810,6 +838,8 @@
 road_trade_incr_pct  = 0
 base_time            = 1
 road_time            = 2
+cultivate_time       = 0
+plant_time           = 0
 irrigation_result    = "yes"
 irrigation_food_incr = 1
 irrigation_time      = 5
diff -Nurd freeciv/data/experimental/terrain.ruleset freeciv/data/experimental/terrain.ruleset
--- freeciv/data/experimental/terrain.ruleset	2021-04-17 03:16:31.712087592 +0300
+++ freeciv/data/experimental/terrain.ruleset	2021-04-17 03:16:42.968198740 +0300
@@ -117,6 +117,8 @@
 ;                        Nonzero values only affect extras with build_time 0.
 ; road_time            = time to build roads; if 0, cannot build any roads.
 ;                        Nonzero values only affect extras with build_time 0.
+; cultivate_time       = time to cultivate
+; plant_time           = time to plant
 ; irrigation_result    = result of 'irrigate' activity; one of:
 ;                        "no"  -- cannot irrigate
 ;                        "yes" -- can build extra
@@ -227,6 +229,8 @@
 road_trade_incr_pct  = 0
 base_time            = 0
 road_time            = 0
+cultivate_time       = 0
+plant_time           = 0
 irrigation_result    = "no"
 irrigation_food_incr = 0
 irrigation_time      = 0
@@ -266,6 +270,8 @@
 road_trade_incr_pct  = 0
 base_time            = 1
 road_time            = 0
+cultivate_time       = 0
+plant_time           = 0
 irrigation_result    = "no"
 irrigation_food_incr = 0
 irrigation_time      = 0
@@ -307,6 +313,8 @@
 road_trade_incr_pct  = 0
 base_time            = 1
 road_time            = 0
+cultivate_time       = 0
+plant_time           = 0
 irrigation_result    = "no"
 irrigation_food_incr = 0
 irrigation_time      = 0
@@ -349,6 +357,8 @@
 road_trade_incr_pct  = 0
 base_time            = 1
 road_time            = 0
+cultivate_time       = 0
+plant_time           = 0
 irrigation_result    = "no"
 irrigation_food_incr = 0
 irrigation_time      = 0
@@ -391,6 +401,8 @@
 road_trade_incr_pct  = 0
 base_time            = 1
 road_time            = 4
+cultivate_time       = 0
+plant_time           = 0
 irrigation_result    = "no"
 irrigation_food_incr = 0
 irrigation_time      = 0
@@ -436,6 +448,8 @@
 road_trade_incr_pct  = 100
 base_time            = 1
 road_time            = 2
+cultivate_time       = 0
+plant_time           = 0
 irrigation_result    = "yes"
 irrigation_food_incr = 1
 irrigation_time      = 5
@@ -482,12 +496,14 @@
 road_trade_incr_pct  = 0
 base_time            = 1
 road_time            = 4
+cultivate_time       = 5
+plant_time           = 15
 irrigation_result    = "Plains"
 irrigation_food_incr = 0
-irrigation_time      = 5
+irrigation_time      = 0
 mining_result        = "Swamp"
 mining_shield_incr   = 0
-mining_time          = 15
+mining_time          = 0
 transform_result     = "Grassland"
 transform_time       = 24
 pillage_time         = 1
@@ -527,12 +543,14 @@
 road_trade_incr_pct  = 100
 base_time            = 1
 road_time            = 2
+cultivate_time       = 0
+plant_time           = 10
 irrigation_result    = "yes"
 irrigation_food_incr = 1
 irrigation_time      = 5
 mining_result        = "Forest"
 mining_shield_incr   = 0
-mining_time          = 10
+mining_time          = 0
 transform_result     = "Hills"
 transform_time       = 24
 pillage_time         = 1
@@ -569,6 +587,8 @@
 road_trade_incr_pct  = 0
 base_time            = 1
 road_time            = 4
+cultivate_time       = 0
+plant_time           = 0
 irrigation_result    = "yes"
 irrigation_food_incr = 1
 irrigation_time      = 10
@@ -612,12 +632,14 @@
 road_trade_incr_pct  = 0
 base_time            = 1
 road_time            = 4
+cultivate_time       = 15
+plant_time           = 15
 irrigation_result    = "Grassland"
 irrigation_food_incr = 0
-irrigation_time      = 15
+irrigation_time      = 0
 mining_result        = "Forest"
 mining_shield_incr   = 0
-mining_time          = 15
+mining_time          = 0
 transform_result     = "Plains"
 transform_time       = 24
 pillage_time         = 1
@@ -656,6 +678,8 @@
 road_trade_incr_pct  = 0
 base_time            = 1
 road_time            = 6
+cultivate_time       = 0
+plant_time           = 0
 irrigation_result    = "no"
 irrigation_food_incr = 0
 irrigation_time      = 0
@@ -698,12 +722,14 @@
 road_trade_incr_pct  = 100
 base_time            = 1
 road_time            = 2
+cultivate_time       = 0
+plant_time           = 15
 irrigation_result    = "yes"
 irrigation_food_incr = 1
 irrigation_time      = 5
 mining_result        = "Forest"
 mining_shield_incr   = 0
-mining_time          = 15
+mining_time          = 0
 transform_result     = "Grassland"
 transform_time       = 24
 pillage_time         = 1
@@ -741,12 +767,14 @@
 road_trade_incr_pct  = 0
 base_time            = 1
 road_time            = 4
+cultivate_time       = 15
+plant_time           = 15
 irrigation_result    = "Grassland"
 irrigation_food_incr = 0
-irrigation_time      = 15
+irrigation_time      = 0
 mining_result        = "Forest"
 mining_shield_incr   = 0
-mining_time          = 15
+mining_time          = 0
 transform_result     = "Lake"
 transform_time       = 36
 pillage_time         = 1
@@ -786,6 +814,8 @@
 road_trade_incr_pct  = 0
 base_time            = 1
 road_time            = 2
+cultivate_time       = 0
+plant_time           = 0
 irrigation_result    = "yes"
 irrigation_food_incr = 1
 irrigation_time      = 5
diff -Nurd freeciv/data/granularity/terrain.ruleset freeciv/data/granularity/terrain.ruleset
--- freeciv/data/granularity/terrain.ruleset	2021-04-17 03:16:31.796088423 +0300
+++ freeciv/data/granularity/terrain.ruleset	2021-04-17 03:16:42.968198740 +0300
@@ -113,6 +113,8 @@
 ;                        Nonzero values only affect extras with build_time 0.
 ; road_time            = time to build roads; if 0, cannot build any roads.
 ;                        Nonzero values only affect extras with build_time 0.
+; cultivate_time       = time to cultivate
+; plant_time           = time to plant
 ; irrigation_result    = result of 'irrigate' activity; one of:
 ;                        "no"  -- cannot irrigate
 ;                        "yes" -- can build extra
@@ -222,6 +224,8 @@
 road_trade_incr_pct  = 100
 base_time            = 1
 road_time            = 1
+cultivate_time       = 0
+plant_time           = 0
 irrigation_result    = "yes"
 irrigation_food_incr = 1
 irrigation_time      = 5
@@ -266,6 +270,8 @@
 road_trade_incr_pct  = 100
 base_time            = 1
 road_time            = 1
+cultivate_time       = 0
+plant_time           = 0
 irrigation_result    = "no"
 irrigation_food_incr = 0
 irrigation_time      = 0
diff -Nurd freeciv/data/multiplayer/terrain.ruleset freeciv/data/multiplayer/terrain.ruleset
--- freeciv/data/multiplayer/terrain.ruleset	2021-04-17 03:16:31.908089529 +0300
+++ freeciv/data/multiplayer/terrain.ruleset	2021-04-17 03:16:42.968198740 +0300
@@ -115,6 +115,8 @@
 ;                        Nonzero values only affect extras with build_time 0.
 ; road_time            = time to build roads; if 0, cannot build any roads.
 ;                        Nonzero values only affect extras with build_time 0.
+; cultivate_time       = time to cultivate
+; plant_time           = time to plant
 ; irrigation_result    = result of 'irrigate' activity; one of:
 ;                        "no"  -- cannot irrigate
 ;                        "yes" -- can build extra
@@ -225,6 +227,8 @@
 road_trade_incr_pct  = 0
 base_time            = 0
 road_time            = 0
+cultivate_time       = 0
+plant_time           = 0
 irrigation_result    = "no"
 irrigation_food_incr = 0
 irrigation_time      = 0
@@ -264,6 +268,8 @@
 road_trade_incr_pct  = 0
 base_time            = 1
 road_time            = 0
+cultivate_time       = 0
+plant_time           = 0
 irrigation_result    = "no"
 irrigation_food_incr = 0
 irrigation_time      = 0
@@ -305,6 +311,8 @@
 road_trade_incr_pct  = 0
 base_time            = 1
 road_time            = 0
+cultivate_time       = 0
+plant_time           = 0
 irrigation_result    = "no"
 irrigation_food_incr = 0
 irrigation_time      = 0
@@ -347,6 +355,8 @@
 road_trade_incr_pct  = 0
 base_time            = 1
 road_time            = 0
+cultivate_time       = 0
+plant_time           = 0
 irrigation_result    = "no"
 irrigation_food_incr = 0
 irrigation_time      = 0
@@ -389,6 +399,8 @@
 road_trade_incr_pct  = 0
 base_time            = 1
 road_time            = 4
+cultivate_time       = 0
+plant_time           = 0
 irrigation_result    = "no"
 irrigation_food_incr = 0
 irrigation_time      = 0
@@ -434,6 +446,8 @@
 road_trade_incr_pct  = 100
 base_time            = 1
 road_time            = 2
+cultivate_time       = 0
+plant_time           = 0
 irrigation_result    = "yes"
 irrigation_food_incr = 1
 irrigation_time      = 5
@@ -480,12 +494,14 @@
 road_trade_incr_pct  = 0
 base_time            = 1
 road_time            = 4
+cultivate_time       = 5
+plant_time           = 15
 irrigation_result    = "Plains"
 irrigation_food_incr = 0
-irrigation_time      = 5
+irrigation_time      = 0
 mining_result        = "Swamp"
 mining_shield_incr   = 0
-mining_time          = 15
+mining_time          = 0
 transform_result     = "Grassland"
 transform_time       = 24
 pillage_time         = 1
@@ -525,12 +541,14 @@
 road_trade_incr_pct  = 100
 base_time            = 1
 road_time            = 2
+cultivate_time       = 0
+plant_time           = 10
 irrigation_result    = "yes"
 irrigation_food_incr = 1
 irrigation_time      = 5
 mining_result        = "Forest"
 mining_shield_incr   = 0
-mining_time          = 10
+mining_time          = 0
 transform_result     = "Hills"
 transform_time       = 24
 pillage_time         = 1
@@ -567,6 +585,8 @@
 road_trade_incr_pct  = 0
 base_time            = 1
 road_time            = 4
+cultivate_time       = 0
+plant_time           = 0
 irrigation_result    = "yes"
 irrigation_food_incr = 1
 irrigation_time      = 10
@@ -610,12 +630,14 @@
 road_trade_incr_pct  = 0
 base_time            = 1
 road_time            = 4
+cultivate_time       = 15
+plant_time           = 15
 irrigation_result    = "Grassland"
 irrigation_food_incr = 0
-irrigation_time      = 15
+irrigation_time      = 0
 mining_result        = "Forest"
 mining_shield_incr   = 0
-mining_time          = 15
+mining_time          = 0
 transform_result     = "Plains"
 transform_time       = 24
 pillage_time         = 1
@@ -654,6 +676,8 @@
 road_trade_incr_pct  = 0
 base_time            = 1
 road_time            = 6
+cultivate_time       = 0
+plant_time           = 0
 irrigation_result    = "no"
 irrigation_food_incr = 0
 irrigation_time      = 0
@@ -696,12 +720,14 @@
 road_trade_incr_pct  = 100
 base_time            = 1
 road_time            = 2
+cultivate_time       = 0
+plant_time           = 15
 irrigation_result    = "yes"
 irrigation_food_incr = 1
 irrigation_time      = 5
 mining_result        = "Forest"
 mining_shield_incr   = 0
-mining_time          = 15
+mining_time          = 0
 transform_result     = "Grassland"
 transform_time       = 24
 pillage_time         = 1
@@ -739,12 +765,14 @@
 road_trade_incr_pct  = 0
 base_time            = 1
 road_time            = 4
+cultivate_time       = 15
+plant_time           = 15
 irrigation_result    = "Grassland"
 irrigation_food_incr = 0
-irrigation_time      = 15
+irrigation_time      = 0
 mining_result        = "Forest"
 mining_shield_incr   = 0
-mining_time          = 15
+mining_time          = 0
 transform_result     = "Lake"
 transform_time       = 36
 pillage_time         = 1
@@ -784,6 +812,8 @@
 road_trade_incr_pct  = 0
 base_time            = 1
 road_time            = 2
+cultivate_time       = 0
+plant_time           = 0
 irrigation_result    = "yes"
 irrigation_food_incr = 1
 irrigation_time      = 5
diff -Nurd freeciv/data/ruledit/comments-3.1.txt freeciv/data/ruledit/comments-3.1.txt
--- freeciv/data/ruledit/comments-3.1.txt	2021-04-17 03:16:31.920089647 +0300
+++ freeciv/data/ruledit/comments-3.1.txt	2021-04-17 03:16:42.968198740 +0300
@@ -595,6 +595,8 @@
 ;                        Nonzero values only affect extras with build_time 0.\n\
 ; road_time            = time to build roads; if 0, cannot build any roads.\n\
 ;                        Nonzero values only affect extras with build_time 0.\n\
+; cultivate_time       = time to cultivate\n\
+; plant_time           = time to plant\n\
 ; irrigation_result    = result of 'irrigate' activity; one of:\n\
 ;                        \"no\"  -- cannot irrigate\n\
 ;                        \"yes\" -- can build extra\n\
diff -Nurd freeciv/data/sandbox/terrain.ruleset freeciv/data/sandbox/terrain.ruleset
--- freeciv/data/sandbox/terrain.ruleset	2021-04-17 03:16:31.920089647 +0300
+++ freeciv/data/sandbox/terrain.ruleset	2021-04-17 03:16:42.968198740 +0300
@@ -123,6 +123,8 @@
 ;                        Nonzero values only affect extras with build_time 0.
 ; road_time            = time to build roads; if 0, cannot build any roads.
 ;                        Nonzero values only affect extras with build_time 0.
+; cultivate_time       = time to cultivate
+; plant_time           = time to plant
 ; irrigation_result    = result of 'irrigate' activity; one of:
 ;                        "no"  -- cannot irrigate
 ;                        "yes" -- can build extra
@@ -233,6 +235,8 @@
 road_trade_incr_pct  = 0
 base_time            = 0
 road_time            = 0
+cultivate_time       = 0
+plant_time           = 0
 irrigation_result    = "no"
 irrigation_food_incr = 0
 irrigation_time      = 0
@@ -274,6 +278,8 @@
 road_trade_incr_pct  = 0
 base_time            = 1
 road_time            = 0
+cultivate_time       = 0
+plant_time           = 0
 irrigation_result    = "no"
 irrigation_food_incr = 0
 irrigation_time      = 0
@@ -317,6 +323,8 @@
 road_trade_incr_pct  = 0
 base_time            = 1
 road_time            = 0
+cultivate_time       = 0
+plant_time           = 0
 irrigation_result    = "no"
 irrigation_food_incr = 0
 irrigation_time      = 0
@@ -361,6 +369,8 @@
 road_trade_incr_pct  = 0
 base_time            = 1
 road_time            = 0
+cultivate_time       = 0
+plant_time           = 0
 irrigation_result    = "no"
 irrigation_food_incr = 0
 irrigation_time      = 0
@@ -405,6 +415,8 @@
 road_trade_incr_pct  = 0
 base_time            = 1
 road_time            = 4
+cultivate_time       = 0
+plant_time           = 0
 irrigation_result    = "no"
 irrigation_food_incr = 0
 irrigation_time      = 0
@@ -450,6 +462,8 @@
 road_trade_incr_pct  = 100
 base_time            = 1
 road_time            = 2
+cultivate_time       = 0
+plant_time           = 0
 irrigation_result    = "yes"
 irrigation_food_incr = 1
 irrigation_time      = 5
@@ -504,12 +518,14 @@
 road_trade_incr_pct  = 0
 base_time            = 1
 road_time            = 4
+cultivate_time       = 10
+plant_time           = 15
 irrigation_result    = "Plains"
 irrigation_food_incr = 0
-irrigation_time      = 10
+irrigation_time      = 0
 mining_result        = "Grassland"
 mining_shield_incr   = 0
-mining_time          = 15
+mining_time          = 0
 transform_result     = "Hills"
 transform_time       = 24
 pillage_time         = 1
@@ -551,12 +567,14 @@
 road_trade_incr_pct  = 100
 base_time            = 1
 road_time            = 2
+cultivate_time       = 0
+plant_time           = 15
 irrigation_result    = "yes"
 irrigation_food_incr = 1
 irrigation_time      = 5
 mining_result        = "Forest"
 mining_shield_incr   = 0
-mining_time          = 15
+mining_time          = 0
 transform_result     = "Lake"
 transform_time       = 36
 pillage_time         = 1
@@ -595,6 +613,8 @@
 road_trade_incr_pct  = 0
 base_time            = 1
 road_time            = 4
+cultivate_time       = 0
+plant_time           = 0
 irrigation_result    = "yes"
 irrigation_food_incr = 1
 irrigation_time      = 10
@@ -643,9 +663,11 @@
 road_trade_incr_pct  = 0
 base_time            = 1
 road_time            = 4
+cultivate_time       = 10
+plant_time           = 0
 irrigation_result    = "Swamp"
 irrigation_food_incr = 0
-irrigation_time      = 10
+irrigation_time      = 0
 mining_result        = "no"
 mining_shield_incr   = 0
 mining_time          = 0
@@ -690,6 +712,8 @@
 road_trade_incr_pct  = 0
 base_time            = 1
 road_time            = 6
+cultivate_time       = 0
+plant_time           = 0
 irrigation_result    = "no"
 irrigation_food_incr = 0
 irrigation_time      = 0
@@ -739,12 +763,14 @@
 road_trade_incr_pct  = 100
 base_time            = 1
 road_time            = 2
+cultivate_time       = 0
+plant_time           = 10
 irrigation_result    = "yes"
 irrigation_food_incr = 1
 irrigation_time      = 5
 mining_result        = "Forest"
 mining_shield_incr   = 0
-mining_time          = 10
+mining_time          = 0
 transform_result     = "Lake"
 transform_time       = 36
 pillage_time         = 1
@@ -784,6 +810,8 @@
 road_trade_incr_pct  = 0
 base_time            = 1
 road_time            = 4
+cultivate_time       = 0
+plant_time           = 0
 irrigation_result    = "yes"
 irrigation_food_incr = 1
 irrigation_time      = 5
@@ -832,6 +860,8 @@
 road_trade_incr_pct  = 100
 base_time            = 1
 road_time            = 2
+cultivate_time       = 0
+plant_time           = 0
 irrigation_result    = "no"
 irrigation_food_incr = 0
 irrigation_time      = 0
diff -Nurd freeciv/data/stub/terrain.ruleset freeciv/data/stub/terrain.ruleset
--- freeciv/data/stub/terrain.ruleset	2021-04-17 03:16:31.932089766 +0300
+++ freeciv/data/stub/terrain.ruleset	2021-04-17 03:16:42.968198740 +0300
@@ -104,6 +104,8 @@
 ;                        Nonzero values only affect extras with build_time 0.
 ; road_time            = time to build roads; if 0, cannot build any roads.
 ;                        Nonzero values only affect extras with build_time 0.
+; cultivate_time       = time to cultivate
+; plant_time           = time to plant
 ; irrigation_result    = result of 'irrigate' activity; one of:
 ;                        "no"  -- cannot irrigate
 ;                        "yes" -- can build extra
@@ -213,6 +215,8 @@
 road_trade_incr_pct  = 100
 base_time            = 1
 road_time            = 1
+cultivate_time       = 0
+plant_time           = 0
 irrigation_result    = "yes"
 irrigation_food_incr = 1
 irrigation_time      = 5
@@ -257,6 +261,8 @@
 road_trade_incr_pct  = 100
 base_time            = 1
 road_time            = 1
+cultivate_time       = 0
+plant_time           = 0
 irrigation_result    = "no"
 irrigation_food_incr = 0
 irrigation_time      = 0
diff -Nurd freeciv/data/webperimental/terrain.ruleset freeciv/data/webperimental/terrain.ruleset
--- freeciv/data/webperimental/terrain.ruleset	2021-04-17 03:16:31.968090122 +0300
+++ freeciv/data/webperimental/terrain.ruleset	2021-04-17 03:16:42.968198740 +0300
@@ -115,6 +115,8 @@
 ;                        Nonzero values only affect extras with build_time 0.
 ; road_time            = time to build roads; if 0, cannot build any roads.
 ;                        Nonzero values only affect extras with build_time 0.
+; cultivate_time       = time to cultivate
+; plant_time           = time to plant
 ; irrigation_result    = result of 'irrigate' activity; one of:
 ;                        "no"  -- cannot irrigate
 ;                        "yes" -- can build extra
@@ -225,6 +227,8 @@
 road_trade_incr_pct  = 0
 base_time            = 0
 road_time            = 0
+cultivate_time       = 0
+plant_time           = 0
 irrigation_result    = "no"
 irrigation_food_incr = 0
 irrigation_time      = 0
@@ -266,6 +270,8 @@
 road_trade_incr_pct  = 0
 base_time            = 1
 road_time            = 0
+cultivate_time       = 0
+plant_time           = 0
 irrigation_result    = "no"
 irrigation_food_incr = 0
 irrigation_time      = 0
@@ -309,6 +315,8 @@
 road_trade_incr_pct  = 0
 base_time            = 1
 road_time            = 0
+cultivate_time       = 0
+plant_time           = 0
 irrigation_result    = "no"
 irrigation_food_incr = 0
 irrigation_time      = 0
@@ -353,6 +361,8 @@
 road_trade_incr_pct  = 0
 base_time            = 1
 road_time            = 0
+cultivate_time       = 0
+plant_time           = 0
 irrigation_result    = "no"
 irrigation_food_incr = 0
 irrigation_time      = 0
@@ -397,6 +407,8 @@
 road_trade_incr_pct  = 0
 base_time            = 1
 road_time            = 4
+cultivate_time       = 0
+plant_time           = 0
 irrigation_result    = "no"
 irrigation_food_incr = 0
 irrigation_time      = 0
@@ -444,6 +456,8 @@
 road_trade_incr_pct  = 100
 base_time            = 1
 road_time            = 2
+cultivate_time       = 0
+plant_time           = 0
 irrigation_result    = "yes"
 irrigation_food_incr = 1
 irrigation_time      = 5
@@ -492,12 +506,14 @@
 road_trade_incr_pct  = 0
 base_time            = 1
 road_time            = 4
+cultivate_time       = 5
+plant_time           = 15
 irrigation_result    = "Plains"
 irrigation_food_incr = 0
-irrigation_time      = 5
+irrigation_time      = 0
 mining_result        = "Swamp"
 mining_shield_incr   = 0
-mining_time          = 15
+mining_time          = 0
 transform_result     = "Grassland"
 transform_time       = 24
 pillage_time         = 1
@@ -539,12 +555,14 @@
 road_trade_incr_pct  = 100
 base_time            = 1
 road_time            = 2
+cultivate_time       = 0
+plant_time           = 10
 irrigation_result    = "yes"
 irrigation_food_incr = 1
 irrigation_time      = 5
 mining_result        = "Forest"
 mining_shield_incr   = 0
-mining_time          = 10
+mining_time          = 0
 transform_result     = "Hills"
 transform_time       = 24
 pillage_time         = 1
@@ -583,6 +601,8 @@
 road_trade_incr_pct  = 0
 base_time            = 1
 road_time            = 4
+cultivate_time       = 0
+plant_time           = 0
 irrigation_result    = "yes"
 irrigation_food_incr = 1
 irrigation_time      = 10
@@ -628,12 +648,14 @@
 road_trade_incr_pct  = 0
 base_time            = 1
 road_time            = 4
+cultivate_time       = 15
+plant_time           = 15
 irrigation_result    = "Grassland"
 irrigation_food_incr = 0
-irrigation_time      = 15
+irrigation_time      = 0
 mining_result        = "Forest"
 mining_shield_incr   = 0
-mining_time          = 15
+mining_time          = 0
 transform_result     = "Plains"
 transform_time       = 24
 pillage_time         = 1
@@ -674,6 +696,8 @@
 road_trade_incr_pct  = 0
 base_time            = 1
 road_time            = 6
+cultivate_time       = 0
+plant_time           = 0
 irrigation_result    = "no"
 irrigation_food_incr = 0
 irrigation_time      = 0
@@ -718,12 +742,14 @@
 road_trade_incr_pct  = 100
 base_time            = 1
 road_time            = 2
+cultivate_time       = 0
+plant_time           = 15
 irrigation_result    = "yes"
 irrigation_food_incr = 1
 irrigation_time      = 5
 mining_result        = "Forest"
 mining_shield_incr   = 0
-mining_time          = 15
+mining_time          = 0
 transform_result     = "Grassland"
 transform_time       = 24
 pillage_time         = 1
@@ -763,12 +789,14 @@
 road_trade_incr_pct  = 0
 base_time            = 1
 road_time            = 4
+cultivate_time       = 15
+plant_time           = 15
 irrigation_result    = "Grassland"
 irrigation_food_incr = 0
-irrigation_time      = 15
+irrigation_time      = 0
 mining_result        = "Forest"
 mining_shield_incr   = 0
-mining_time          = 15
+mining_time          = 0
 transform_result     = "Lake"
 transform_time       = 36
 pillage_time         = 1
@@ -810,6 +838,8 @@
 road_trade_incr_pct  = 0
 base_time            = 1
 road_time            = 2
+cultivate_time       = 0
+plant_time           = 0
 irrigation_result    = "yes"
 irrigation_food_incr = 1
 irrigation_time      = 5
diff -Nurd freeciv/server/ruleset.c freeciv/server/ruleset.c
--- freeciv/server/ruleset.c	2021-04-17 03:16:31.980090240 +0300
+++ freeciv/server/ruleset.c	2021-04-17 03:16:42.968198740 +0300
@@ -3007,6 +3007,37 @@
         break;
       }
 
+      if (!lookup_time(file, &pterrain->cultivate_time,
+                       tsection, "cultivate_time", filename, NULL, &ok)) {
+        if (compat->compat_mode) {
+          if (pterrain->irrigation_result != pterrain) {
+            pterrain->cultivate_time = pterrain->irrigation_time;
+            pterrain->irrigation_time = 0;
+          } else {
+            pterrain->cultivate_time = 0;
+          }
+        } else {
+          ok = FALSE;
+          break;
+        }
+      }
+
+      if (!lookup_time(file, &pterrain->plant_time,
+                       tsection, "plant_time", filename, NULL, &ok)) {
+        if (compat->compat_mode) {
+          if (pterrain->mining_result != pterrain) {
+            pterrain->plant_time = pterrain->mining_time;
+            pterrain->mining_time = 0;
+          } else {
+            pterrain->plant_time = 0;
+          }
+        } else {
+          ok = FALSE;
+          break;
+        }
+
+      }
+
       if (!lookup_unit_type(file, tsection, "animal",
                             &pterrain->animal, filename,
                             rule_name_get(&pterrain->name))) {
@@ -7545,6 +7576,10 @@
     packet.base_time = pterrain->base_time;
     packet.road_time = pterrain->road_time;
 
+    packet.cultivate_time = pterrain->cultivate_time;
+
+    packet.plant_time = pterrain->plant_time;
+
     packet.irrigation_result = (pterrain->irrigation_result
 				? terrain_number(pterrain->irrigation_result)
 				: terrain_count());
diff -Nurd freeciv/tools/civmanual.c freeciv/tools/civmanual.c
--- freeciv/tools/civmanual.c	2021-04-17 03:16:31.984090280 +0300
+++ freeciv/tools/civmanual.c	2021-04-17 03:16:42.968198740 +0300
@@ -459,8 +459,8 @@
       fprintf(doc, "<table><tr bgcolor=#9bc3d1><th colspan=2>%s</th>", _("Terrain"));
       fprintf(doc, "<th>F/P/T</th><th>%s</th>", _("Resources"));
       fprintf(doc, "<th>%s<br/>%s</th>", _("Move cost"), _("Defense bonus"));
-      fprintf(doc, "<th>%s<br/>%s<br/>%s<br/>%s<br/>(%s)</th>",
-              _("Irrigation"), _("Mining"), _("Transform"),
+      fprintf(doc, "<th>%s<br/>%s<br/>%s<br/>%s<br/>%s<br/>%s<br/>(%s)</th>",
+              _("Irrigation"), _("Cultivate"), _("Mining"), _("Plant"), _("Transform"),
               /* xgettext:no-c-format */
               _("% of Road bonus"), _("turns"));
       fprintf(doc, "<th>%s<br/>%s</th>",
@@ -509,25 +509,33 @@
                 pterrain->movement_cost, pterrain->defense_bonus);
 
         fprintf(doc, "<td><table width=\"100%%\">\n");
-        if (pterrain->irrigation_result == pterrain) {
+        if (pterrain->irrigation_time != 0) {
           fprintf(doc, "<tr><td>+%d F</td><td align=\"right\">(%d)</td></tr>\n",
                   pterrain->irrigation_food_incr, pterrain->irrigation_time);
-        } else if (pterrain->irrigation_result == T_NONE) {
-          fprintf(doc, "<tr><td>%s</td></tr>\n", _("impossible"));
         } else {
+          fprintf(doc, "<tr><td>%s</td></tr>\n", _("impossible"));
+        }
+        if (pterrain->irrigation_result != NULL
+            && pterrain->irrigation_result != pterrain) {
           fprintf(doc, "<tr><td>%s</td><td align=\"right\">(%d)</td></tr>\n",
                   terrain_name_translation(pterrain->irrigation_result),
-                  pterrain->irrigation_time);
+                  pterrain->cultivate_time);
+        } else {
+          fprintf(doc, "<tr><td>%s</td></tr>\n", _("impossible"));
         }
-        if (pterrain->mining_result == pterrain) {
+        if (pterrain->mining_time != 0) {
           fprintf(doc, "<tr><td>+%d P</td><td align=\"right\">(%d)</td></tr>\n",
                   pterrain->mining_shield_incr, pterrain->mining_time);
-        } else if (pterrain->mining_result == T_NONE) {
-          fprintf(doc, "<tr><td>%s</td></tr>\n", _("impossible"));
         } else {
+          fprintf(doc, "<tr><td>%s</td></tr>\n", _("impossible"));
+        }
+        if (pterrain->mining_result != NULL
+            && pterrain->mining_result != pterrain) {
           fprintf(doc, "<tr><td>%s</td><td align=\"right\">(%d)</td></tr>\n",
                   terrain_name_translation(pterrain->mining_result),
-                  pterrain->mining_time);
+                  pterrain->plant_time);
+        } else {
+          fprintf(doc, "<tr><td>%s</td></tr>\n", _("impossible"));
         }
 
         if (pterrain->transform_result) {
diff -Nurd freeciv/tools/ruleutil/rulesave.c freeciv/tools/ruleutil/rulesave.c
--- freeciv/tools/ruleutil/rulesave.c	2021-04-17 03:16:31.988090318 +0300
+++ freeciv/tools/ruleutil/rulesave.c	2021-04-17 03:16:42.968198740 +0300
@@ -2161,6 +2161,12 @@
     secfile_insert_int(sfile, pterr->base_time, "%s.base_time", path);
     secfile_insert_int(sfile, pterr->road_time, "%s.road_time", path);
 
+    secfile_insert_int(sfile, pterr->cultivate_time,
+                       "%s.cultivate_time", path);
+
+    secfile_insert_int(sfile, pterr->plant_time,
+                       "%s.plant_time", path);
+
     save_terrain_ref(sfile, pterr->irrigation_result, pterr, path,
                      "irrigation_result");
     secfile_insert_int(sfile, pterr->irrigation_food_incr,
