diff -Nurd freeciv/ai/default/daicity.c freeciv/ai/default/daicity.c
--- freeciv/ai/default/daicity.c	2021-05-08 15:12:45.693940932 +0300
+++ freeciv/ai/default/daicity.c	2021-05-08 15:12:57.234029881 +0300
@@ -1284,7 +1284,6 @@
   case ACTRES_CLEAN_POLLUTION:
   case ACTRES_CLEAN_FALLOUT:
   case ACTRES_FORTIFY:
-  case ACTRES_SENTRY:
   case ACTRES_ROAD:
   case ACTRES_CONVERT:
   case ACTRES_BASE:
diff -Nurd freeciv/ai/default/daidiplomacy.c freeciv/ai/default/daidiplomacy.c
--- freeciv/ai/default/daidiplomacy.c	2021-05-08 15:12:45.693940932 +0300
+++ freeciv/ai/default/daidiplomacy.c	2021-05-08 15:12:57.234029881 +0300
@@ -1994,7 +1994,6 @@
     case ACTRES_CLEAN_POLLUTION:
     case ACTRES_CLEAN_FALLOUT:
     case ACTRES_FORTIFY:
-    case ACTRES_SENTRY:
     case ACTRES_ROAD:
     case ACTRES_CONVERT:
     case ACTRES_BASE:
diff -Nurd freeciv/client/control.c freeciv/client/control.c
--- freeciv/client/control.c	2021-05-08 15:12:45.693940932 +0300
+++ freeciv/client/control.c	2021-05-08 15:12:57.234029881 +0300
@@ -1520,47 +1520,21 @@
   struct pf_path *path;
 
   if ((path = path_to_nearest_allied_city(punit))) {
-    struct tile *end_tile = pf_path_last_position(path)->tile;
     int turns = pf_path_last_position(path)->turn;
     int max_hp = unit_type_get(punit)->hp;
-    bool hurt = (punit->hp + turns *
-                 (get_unit_bonus(punit, EFT_UNIT_RECOVER)
-                  - (max_hp * unit_class_get(punit)->hp_loss_pct / 100))
-                 < max_hp);
-    struct action *paction = NULL;
-
-    if (hurt) {
-      /* Look for a sentry action the unit can perform */
-      action_by_activity_iterate(caction, act_id, ACTIVITY_SENTRY) {
-        if (action_get_actor_kind(caction) != AAK_UNIT) {
-          /* Not relevant. */
-          continue;
-        }
-
-        fc_assert_action(action_get_target_kind(caction) == ATK_SELF,
-                         continue);
-
-        if (action_prob_possible(
-              action_speculate_unit_on_self(caction->id, punit,
-                                            unit_home(punit),
-                                            end_tile,
-                                            FALSE))) {
-          /* Found one. */
-          paction = caction;
-          break;
-        }
-      } action_by_activity_iterate_end;
-    }
 
-    if (hurt && paction != NULL) {
+    if (punit->hp + turns *
+        (get_unit_bonus(punit, EFT_UNIT_RECOVER)
+         - (max_hp * unit_class_get(punit)->hp_loss_pct / 100))
+	< max_hp) {
       struct unit_order order;
 
-      order.order = ORDER_PERFORM_ACTION;
+      order.order = ORDER_ACTIVITY;
       order.dir = DIR8_ORIGIN;
-      order.activity = ACTIVITY_LAST;
-      order.target = end_tile->index;
+      order.activity = ACTIVITY_SENTRY;
+      order.target = NO_TARGET;
       order.sub_target = NO_TARGET;
-      order.action = paction->id;
+      order.action = ACTION_NONE;
       send_goto_path(punit, path, &order);
     } else {
       send_goto_path(punit, path, NULL);
diff -Nurd freeciv/client/gui-qt/dialogs.cpp freeciv/client/gui-qt/dialogs.cpp
--- freeciv/client/gui-qt/dialogs.cpp	2021-05-08 15:12:45.705941024 +0300
+++ freeciv/client/gui-qt/dialogs.cpp	2021-05-08 15:12:57.234029881 +0300
@@ -136,7 +136,6 @@
 static void disembark2(QVariant data1, QVariant data2);
 static void convert_unit(QVariant data1, QVariant data2);
 static void fortify(QVariant data1, QVariant data2);
-static void sentry(QVariant data1, QVariant data2);
 static void disband_unit(QVariant data1, QVariant data2);
 static void join_city(QVariant data1, QVariant data2);
 static void unit_home_city(QVariant data1, QVariant data2);
@@ -261,7 +260,6 @@
   /* Unit acting with no target except itself. */
   action_function[ACTION_DISBAND_UNIT] = disband_unit;
   action_function[ACTION_FORTIFY] = fortify;
-  action_function[ACTION_SENTRY] = sentry;
   action_function[ACTION_CONVERT] = convert_unit;
 
   return action_function;
@@ -2284,20 +2282,6 @@
                       target_id, 0, "");
   }
 }
-
-/***********************************************************************//**
-  Action "Sentry Unit" for choice dialog
-***************************************************************************/
-static void sentry(QVariant data1, QVariant data2)
-{
-  int actor_id = data1.toInt();
-  int target_id = data2.toInt();
-
-  if (NULL != game_unit_by_number(actor_id)) {
-    request_do_action(ACTION_SENTRY, actor_id,
-                      target_id, 0, "");
-  }
-}
 
 /***********************************************************************//**
   Action Convert Unit for choice dialog
diff -Nurd freeciv/client/packhand.c freeciv/client/packhand.c
--- freeciv/client/packhand.c	2021-05-08 15:12:45.713941085 +0300
+++ freeciv/client/packhand.c	2021-05-08 15:12:57.234029881 +0300
@@ -4743,7 +4743,6 @@
       switch ((enum gen_action)act) {
       case ACTION_DISBAND_UNIT:
       case ACTION_FORTIFY:
-      case ACTION_SENTRY:
       case ACTION_CONVERT:
       case ACTION_TRANSPORT_ALIGHT:
       case ACTION_TRANSPORT_BOARD:
diff -Nurd freeciv/common/actions.c freeciv/common/actions.c
--- freeciv/common/actions.c	2021-05-08 15:12:45.717941116 +0300
+++ freeciv/common/actions.c	2021-05-08 15:12:57.234029881 +0300
@@ -1063,11 +1063,6 @@
                       ATK_SELF, ASTK_NONE,
                       ACT_TGT_COMPL_SIMPLE, TRUE, FALSE,
                       MAK_STAYS, 0, 0, FALSE);
-  actions[ACTION_SENTRY] =
-      unit_action_new(ACTION_SENTRY, ACTRES_SENTRY,
-                      ATK_SELF, ASTK_NONE,
-                      ACT_TGT_COMPL_SIMPLE, TRUE, FALSE,
-                      MAK_STAYS, 0, 0, FALSE);
   actions[ACTION_ROAD] =
       unit_action_new(ACTION_ROAD, ACTRES_ROAD,
                       ATK_TILE, ASTK_EXTRA_NOT_THERE,
@@ -1839,8 +1834,6 @@
 
   if (action_has_result(paction, ACTRES_FORTIFY)) {
     return ACTIVITY_FORTIFYING;
-  } else if (action_has_result(paction, ACTRES_SENTRY)) {
-    return ACTIVITY_SENTRY;
   } else if (action_has_result(paction, ACTRES_BASE)) {
     return ACTIVITY_BASE;
   } else if (action_has_result(paction, ACTRES_ROAD)) {
@@ -1902,12 +1895,10 @@
     return 1;
   case ACTIVITY_CONVERT:
     return unit_type_get(actor_unit)->convert_time * ACTIVITY_FACTOR;
-  case ACTIVITY_SENTRY:
-    /* Happens instantaneously, not at turn change. */
-    return ACT_TIME_INSTANTANEOUS;
   case ACTIVITY_EXPLORE:
   case ACTIVITY_IDLE:
   case ACTIVITY_FORTIFIED:
+  case ACTIVITY_SENTRY:
   case ACTIVITY_GOTO:
   case ACTIVITY_UNKNOWN:
   case ACTIVITY_PATROL_UNUSED:
@@ -1992,7 +1983,6 @@
   case ACTRES_CLEAN_POLLUTION:
   case ACTRES_CLEAN_FALLOUT:
   case ACTRES_FORTIFY:
-  case ACTRES_SENTRY:
   case ACTRES_CONVERT:
   case ACTRES_TRANSPORT_ALIGHT:
   case ACTRES_TRANSPORT_UNLOAD:
@@ -2067,7 +2057,6 @@
   case ACTRES_CULTIVATE:
   case ACTRES_PLANT:
   case ACTRES_FORTIFY:
-  case ACTRES_SENTRY:
   case ACTRES_ROAD:
   case ACTRES_CONVERT:
   case ACTRES_BASE:
@@ -3027,7 +3016,6 @@
   case ACTRES_CLEAN_POLLUTION:
   case ACTRES_CLEAN_FALLOUT:
   case ACTRES_FORTIFY:
-  case ACTRES_SENTRY:
   case ACTRES_TRANSFORM_TERRAIN:
   case ACTRES_CULTIVATE:
   case ACTRES_PLANT:
@@ -3167,14 +3155,6 @@
     }
     break;
 
-  case ACTRES_SENTRY:
-    if (!can_unit_survive_at_tile(&(wld.map), actor_unit, actor_tile)
-        && !unit_transported(actor_unit)) {
-      /* Don't let units sentry on tiles they will die on. */
-      return TRI_NO;
-    }
-    break;
-
   case ACTRES_ESTABLISH_EMBASSY:
   case ACTRES_SPY_INVESTIGATE_CITY:
   case ACTRES_SPY_POISON:
@@ -4085,7 +4065,6 @@
   case ACTRES_STRIKE_BUILDING:
   case ACTRES_STRIKE_PRODUCTION:
   case ACTRES_FORTIFY:
-  case ACTRES_SENTRY:
   case ACTRES_NONE:
     /* No known hard coded requirements. */
     break;
@@ -5055,7 +5034,6 @@
   case ACTRES_CLEAN_POLLUTION:
   case ACTRES_CLEAN_FALLOUT:
   case ACTRES_FORTIFY:
-  case ACTRES_SENTRY:
   case ACTRES_ROAD:
   case ACTRES_CONVERT:
   case ACTRES_BASE:
@@ -6109,7 +6087,6 @@
   case ACTRES_CLEAN_POLLUTION:
   case ACTRES_CLEAN_FALLOUT:
   case ACTRES_FORTIFY:
-  case ACTRES_SENTRY:
   case ACTRES_ROAD:
   case ACTRES_CONVERT:
   case ACTRES_BASE:
@@ -6519,8 +6496,6 @@
     return "ui_name_clean_fallout";
   case ACTION_FORTIFY:
     return "ui_name_fortify";
-  case ACTION_SENTRY:
-    return "ui_name_sentry";
   case ACTION_ROAD:
     return "ui_name_road";
   case ACTION_CONVERT:
@@ -6752,9 +6727,6 @@
   case ACTION_FORTIFY:
     /* TRANS: _Fortify (100% chance of success). */
     return N_("%sFortify%s");
-  case ACTION_SENTRY:
-    /* TRANS: _Sentry (100% chance of success). */
-    return N_("%sSentry%s");
   case ACTION_ROAD:
     /* TRANS: Build _Road (100% chance of success). */
     return N_("Build %sRoad%s");
@@ -6872,7 +6844,6 @@
   case ACTION_CLEAN_POLLUTION:
   case ACTION_CLEAN_FALLOUT:
   case ACTION_FORTIFY:
-  case ACTION_SENTRY:
   case ACTION_ROAD:
   case ACTION_CONVERT:
   case ACTION_BASE:
@@ -6969,7 +6940,6 @@
   case ACTION_CLEAN_POLLUTION:
   case ACTION_CLEAN_FALLOUT:
   case ACTION_FORTIFY:
-  case ACTION_SENTRY:
   case ACTION_ROAD:
   case ACTION_CONVERT:
   case ACTION_BASE:
@@ -7063,7 +7033,6 @@
   case ACTION_CLEAN_POLLUTION:
   case ACTION_CLEAN_FALLOUT:
   case ACTION_FORTIFY:
-  case ACTION_SENTRY:
   case ACTION_ROAD:
   case ACTION_CONVERT:
   case ACTION_BASE:
@@ -7169,7 +7138,6 @@
   case ACTION_CLEAN_POLLUTION:
   case ACTION_CLEAN_FALLOUT:
   case ACTION_FORTIFY:
-  case ACTION_SENTRY:
   case ACTION_ROAD:
   case ACTION_CONVERT:
   case ACTION_BASE:
@@ -7278,7 +7246,6 @@
   case ACTION_CLEAN_POLLUTION:
   case ACTION_CLEAN_FALLOUT:
   case ACTION_FORTIFY:
-  case ACTION_SENTRY:
   case ACTION_ROAD:
   case ACTION_CONVERT:
   case ACTION_BASE:
@@ -7378,7 +7345,6 @@
   case ACTION_CLEAN_POLLUTION:
   case ACTION_CLEAN_FALLOUT:
   case ACTION_FORTIFY:
-  case ACTION_SENTRY:
   case ACTION_ROAD:
   case ACTION_CONVERT:
   case ACTION_BASE:
diff -Nurd freeciv/common/actions.h freeciv/common/actions.h
--- freeciv/common/actions.h	2021-05-08 15:12:45.717941116 +0300
+++ freeciv/common/actions.h	2021-05-08 15:12:57.234029881 +0300
@@ -210,14 +210,12 @@
 #define SPECENUM_VALUE72NAME "Spread Plague"
 #define SPECENUM_VALUE73 ACTION_SPY_ATTACK
 #define SPECENUM_VALUE73NAME "Spy Attack"
-#define SPECENUM_VALUE74 ACTION_SENTRY
-#define SPECENUM_VALUE74NAME "Sentry Unit"
-#define SPECENUM_VALUE75 ACTION_USER_ACTION1
-#define SPECENUM_VALUE75NAME "User Action 1"
-#define SPECENUM_VALUE76 ACTION_USER_ACTION2
-#define SPECENUM_VALUE76NAME "User Action 2"
-#define SPECENUM_VALUE77 ACTION_USER_ACTION3
-#define SPECENUM_VALUE77NAME "User Action 3"
+#define SPECENUM_VALUE74 ACTION_USER_ACTION1
+#define SPECENUM_VALUE74NAME "User Action 1"
+#define SPECENUM_VALUE75 ACTION_USER_ACTION2
+#define SPECENUM_VALUE75NAME "User Action 2"
+#define SPECENUM_VALUE76 ACTION_USER_ACTION3
+#define SPECENUM_VALUE76NAME "User Action 3"
 #define SPECENUM_BITVECTOR bv_actions
 #define SPECENUM_COUNT ACTION_COUNT
 #include "specenum_gen.h"
diff -Nurd freeciv/common/fc_types.h freeciv/common/fc_types.h
--- freeciv/common/fc_types.h	2021-05-08 15:12:45.717941116 +0300
+++ freeciv/common/fc_types.h	2021-05-08 15:12:57.234029881 +0300
@@ -112,7 +112,7 @@
 #define SPECENUM_VALUE6 ACTIVITY_FORTRESS
 #define SPECENUM_VALUE6NAME N_("Fortress")
 #define SPECENUM_VALUE7 ACTIVITY_SENTRY
-#define SPECENUM_VALUE7NAME N_("Sentry")
+#define SPECENUM_VALUE7NAME "Sentry"
 #define SPECENUM_VALUE8 ACTIVITY_OLD_RAILROAD
 #define SPECENUM_VALUE8NAME "Unused Railroad"
 #define SPECENUM_VALUE9 ACTIVITY_PILLAGE
@@ -260,8 +260,6 @@
 #define SPECENUM_VALUE54NAME "Spread Plague"
 #define SPECENUM_VALUE55 ACTRES_SPY_ATTACK
 #define SPECENUM_VALUE55NAME "Spy Attack"
-#define SPECENUM_VALUE56 ACTRES_SENTRY
-#define SPECENUM_VALUE56NAME "Sentry Unit"
 /* All consequences are handled as (ruleset) action data. */
 #define SPECENUM_COUNT ACTRES_NONE
 #include "specenum_gen.h"
diff -Nurd freeciv/common/unit.c freeciv/common/unit.c
--- freeciv/common/unit.c	2021-05-08 15:12:45.721941147 +0300
+++ freeciv/common/unit.c	2021-05-08 15:12:57.238029911 +0300
@@ -971,11 +971,12 @@
                                           punit, ptile, target);
 
   case ACTIVITY_SENTRY:
-    /* The call below doesn't support actor tile speculation. */
-    fc_assert_msg(unit_tile(punit) == ptile,
-                  "Please use action_speculate_unit_on_self()");
-    return is_action_enabled_unit_on_self(ACTION_SENTRY,
-                                          punit);
+    if (!can_unit_survive_at_tile(&(wld.map), punit, unit_tile(punit))
+        && !unit_transported(punit)) {
+      /* Don't let units sentry on tiles they will die on. */
+      return FALSE;
+    }
+    return TRUE;
 
   case ACTIVITY_PILLAGE:
     /* The call below doesn't support actor tile speculation. */
diff -Nurd freeciv/common/unittype.c freeciv/common/unittype.c
--- freeciv/common/unittype.c	2021-05-08 15:12:45.721941147 +0300
+++ freeciv/common/unittype.c	2021-05-08 15:12:57.238029911 +0300
@@ -353,7 +353,6 @@
   case ACTRES_CLEAN_POLLUTION:
   case ACTRES_CLEAN_FALLOUT:
   case ACTRES_FORTIFY:
-  case ACTRES_SENTRY:
   case ACTRES_ROAD:
   case ACTRES_CONVERT:
   case ACTRES_BASE:
diff -Nurd freeciv/data/alien/game.ruleset freeciv/data/alien/game.ruleset
--- freeciv/data/alien/game.ruleset	2021-05-08 15:12:45.721941147 +0300
+++ freeciv/data/alien/game.ruleset	2021-05-08 15:12:57.238029911 +0300
@@ -433,9 +433,6 @@
 ; /* TRANS: _Fortify (100% chance of success). */
 ui_name_fortify = _("%sFortify%s")
 
-; /* TRANS: _Sentry (100% chance of success). */
-ui_name_sentry = _("%sSentry%s")
-
 ; /* TRANS: Build _Road (100% chance of success). */
 ui_name_road = _("Build %sRoad%s")
 
@@ -1051,9 +1048,6 @@
       "DiplRel",      "Never met",  "Local", FALSE
     }
 
-[actionenabler_sentry]
-action        = "Sentry Unit"
-
 [borders]
 ; Base border radius from city.
 radius_sq_city = 4
diff -Nurd freeciv/data/civ1/game.ruleset freeciv/data/civ1/game.ruleset
--- freeciv/data/civ1/game.ruleset	2021-05-08 15:12:45.729941208 +0300
+++ freeciv/data/civ1/game.ruleset	2021-05-08 15:12:57.238029911 +0300
@@ -407,9 +407,6 @@
 ; /* TRANS: _Fortify (100% chance of success). */
 ui_name_fortify = _("%sFortify%s")
 
-; /* TRANS: _Sentry (100% chance of success). */
-ui_name_sentry = _("%sSentry%s")
-
 ; /* TRANS: Build _Road (100% chance of success). */
 ui_name_road = _("Build %sRoad%s")
 
@@ -891,9 +888,6 @@
       "CityTile", "Center", "Local", FALSE
     }
 
-[actionenabler_sentry]
-action        = "Sentry Unit"
-
 [borders]
 ; Base border radius from city.
 radius_sq_city = 17
diff -Nurd freeciv/data/civ2/game.ruleset freeciv/data/civ2/game.ruleset
--- freeciv/data/civ2/game.ruleset	2021-05-08 15:12:45.733941239 +0300
+++ freeciv/data/civ2/game.ruleset	2021-05-08 15:12:57.238029911 +0300
@@ -452,9 +452,6 @@
 ; /* TRANS: _Fortify (100% chance of success). */
 ui_name_fortify = _("%sFortify%s")
 
-; /* TRANS: _Sentry (100% chance of success). */
-ui_name_sentry = _("%sSentry%s")
-
 ; /* TRANS: Build _Road (100% chance of success). */
 ui_name_road = _("Build %sRoad%s")
 
@@ -1162,9 +1159,6 @@
       "CityTile", "Center", "Local", FALSE
     }
 
-[actionenabler_sentry]
-action        = "Sentry Unit"
-
 [borders]
 ; Base border radius from city.
 radius_sq_city = 17
diff -Nurd freeciv/data/civ2civ3/game.ruleset freeciv/data/civ2civ3/game.ruleset
--- freeciv/data/civ2civ3/game.ruleset	2021-05-08 15:12:45.733941239 +0300
+++ freeciv/data/civ2civ3/game.ruleset	2021-05-08 15:12:57.238029911 +0300
@@ -486,9 +486,6 @@
 ; /* TRANS: _Fortify (100% chance of success). */
 ui_name_fortify = _("%sFortify%s")
 
-; /* TRANS: _Sentry (100% chance of success). */
-ui_name_sentry = _("%sSentry%s")
-
 ; /* TRANS: Build _Road (100% chance of success). */
 ui_name_road = _("Build %sRoad%s")
 
@@ -1405,9 +1402,6 @@
       "DiplRel",      "Never met",  "Local", FALSE
     }
 
-[actionenabler_sentry]
-action        = "Sentry Unit"
-
 [borders]
 ; Base border radius from city.
 radius_sq_city = 5
diff -Nurd freeciv/data/classic/game.ruleset freeciv/data/classic/game.ruleset
--- freeciv/data/classic/game.ruleset	2021-05-08 15:12:45.733941239 +0300
+++ freeciv/data/classic/game.ruleset	2021-05-08 15:12:57.238029911 +0300
@@ -475,9 +475,6 @@
 ; /* TRANS: _Fortify (100% chance of success). */
 ui_name_fortify = _("%sFortify%s")
 
-; /* TRANS: _Sentry (100% chance of success). */
-ui_name_sentry = _("%sSentry%s")
-
 ; /* TRANS: Build _Road (100% chance of success). */
 ui_name_road = _("Build %sRoad%s")
 
@@ -1243,9 +1240,6 @@
       "DiplRel",      "Never met",  "Local", FALSE
     }
 
-[actionenabler_sentry]
-action        = "Sentry Unit"
-
 [borders]
 ; Base border radius from city.
 radius_sq_city = 17
diff -Nurd freeciv/data/experimental/game.ruleset freeciv/data/experimental/game.ruleset
--- freeciv/data/experimental/game.ruleset	2021-05-08 15:12:45.997943267 +0300
+++ freeciv/data/experimental/game.ruleset	2021-05-08 15:12:57.238029911 +0300
@@ -483,9 +483,6 @@
 ; /* TRANS: _Fortify (100% chance of success). */
 ui_name_fortify = _("%sFortify%s")
 
-; /* TRANS: _Sentry (100% chance of success). */
-ui_name_sentry = _("%sSentry%s")
-
 ; /* TRANS: Build _Road (100% chance of success). */
 ui_name_road = _("Build %sRoad%s")
 
@@ -1275,9 +1272,6 @@
       "DiplRel",      "Never met",  "Local", FALSE
     }
 
-[actionenabler_sentry]
-action        = "Sentry Unit"
-
 [borders]
 ; Base border radius from city.
 radius_sq_city = 17
diff -Nurd freeciv/data/granularity/game.ruleset freeciv/data/granularity/game.ruleset
--- freeciv/data/granularity/game.ruleset	2021-05-08 15:12:45.809941824 +0300
+++ freeciv/data/granularity/game.ruleset	2021-05-08 15:12:57.238029911 +0300
@@ -325,9 +325,6 @@
 ; /* TRANS: _Build City (100% chance of success). */
 ui_name_found_city = _("%sBuild City%s")
 
-; /* TRANS: _Sentry (100% chance of success). */
-ui_name_sentry = _("%sSentry%s")
-
 ; /* <-- avoid gettext warnings
 ;
 ; Action enablers:
@@ -486,9 +483,6 @@
       "UnitFlag",      "Settlers",     "Local", TRUE
     }
 
-[actionenabler_sentry]
-action        = "Sentry Unit"
-
 [borders]
 ; Base border radius from city.
 radius_sq_city = 2
diff -Nurd freeciv/data/multiplayer/game.ruleset freeciv/data/multiplayer/game.ruleset
--- freeciv/data/multiplayer/game.ruleset	2021-05-08 15:12:45.941942837 +0300
+++ freeciv/data/multiplayer/game.ruleset	2021-05-08 15:12:57.238029911 +0300
@@ -472,9 +472,6 @@
 ; /* TRANS: _Fortify (100% chance of success). */
 ui_name_fortify = _("%sFortify%s")
 
-; /* TRANS: _Sentry (100% chance of success). */
-ui_name_sentry = _("%sSentry%s")
-
 ; /* TRANS: Build _Road (100% chance of success). */
 ui_name_road = _("Build %sRoad%s")
 
@@ -1237,9 +1234,6 @@
       "DiplRel",      "Never met",  "Local", FALSE
     }
 
-[actionenabler_sentry]
-action        = "Sentry Unit"
-
 [borders]
 ; Base border radius from city.
 radius_sq_city = 17
diff -Nurd freeciv/data/sandbox/game.ruleset freeciv/data/sandbox/game.ruleset
--- freeciv/data/sandbox/game.ruleset	2021-05-08 15:12:45.953942929 +0300
+++ freeciv/data/sandbox/game.ruleset	2021-05-08 15:12:57.238029911 +0300
@@ -491,9 +491,6 @@
 ; /* TRANS: _Fortify (100% chance of success). */
 ui_name_fortify = _("%sFortify%s")
 
-; /* TRANS: _Sentry (100% chance of success). */
-ui_name_sentry = _("%sSentry%s")
-
 ; /* TRANS: Build _Road (100% chance of success). */
 ui_name_road = _("Build %sRoad%s")
 
@@ -1685,9 +1682,6 @@
       "DiplRel",      "Never met",  "Local", FALSE
     }
 
-[actionenabler_sentry]
-action        = "Sentry Unit"
-
 [borders]
 ; Base border radius from city.
 radius_sq_city = 5
diff -Nurd freeciv/data/webperimental/game.ruleset freeciv/data/webperimental/game.ruleset
--- freeciv/data/webperimental/game.ruleset	2021-05-08 15:12:45.997943267 +0300
+++ freeciv/data/webperimental/game.ruleset	2021-05-08 15:12:57.238029911 +0300
@@ -475,9 +475,6 @@
 ; /* TRANS: _Fortify (100% chance of success). */
 ui_name_fortify = _("%sFortify%s")
 
-; /* TRANS: _Sentry (100% chance of success). */
-ui_name_sentry = _("%sSentry%s")
-
 ; /* TRANS: Build _Road (100% chance of success). */
 ui_name_road = _("Build %sRoad%s")
 
@@ -1546,9 +1543,6 @@
       "DiplRel",      "Never met",  "Local", FALSE
     }
 
-[actionenabler_sentry]
-action        = "Sentry Unit"
-
 [borders]
 ; Base border radius from city.
 radius_sq_city = 17
diff -Nurd freeciv/doc/README.actions freeciv/doc/README.actions
--- freeciv/doc/README.actions	2021-05-08 15:12:46.001943298 +0300
+++ freeciv/doc/README.actions	2021-05-08 15:12:57.238029911 +0300
@@ -926,16 +926,6 @@
  * UI name can be set using ui_name_fortify
  * the actor unit can't already be fortified (!)
 
-"Sentry Unit" - Sentry at tile or in current transport
- * UI name can be set using ui_name_sentry
- * the actor unit must be transported or on a livable tile. If the unit
-   needs fuel or loses hit points each turn being on a livable isn't enough.
-   It is then only able to sentry in a transport or on a livable tile that
-   has one of the following:
-   - a city
-   - a native extra with the Refuel flag
-   - is next to safe terrain if the actor has the Coast unit type flag
-
 "Convert Unit" - Convert the unit to another unit type.
  * UI name can be set using ui_name_convert_unit
  * actor unit must have a type to convert to (convert_to).
diff -Nurd freeciv/doc/README.effects freeciv/doc/README.effects
--- freeciv/doc/README.effects	2021-05-08 15:12:46.001943298 +0300
+++ freeciv/doc/README.effects	2021-05-08 15:14:58.238993817 +0300
@@ -107,7 +107,7 @@
 MaxUnitsOnTile is about the number of units present on a tile.
 UnitState is "Transported", "Transporting", "OnNativeTile", "OnLivableTile",
 "InNativeExtra", "OnDomesticTile", "MovedThisTurn" or "HasHomeCity".
-Activity is "Idle", "Pollution", "Mine", "Irrigate", "Fortified", "Fortress", "Sentry",
+Activity is "Idle", "Pollution", "Mine", "Irrigate", "Fortified", "Fortress",
 "Pillage", "Goto", "Explore", "Transform", "Fortifying", "Fallout", "Base", "Road",
 "Convert", "Cultivate", or "Plant".
 MinMoveFrags is the minimum move fragments the unit must have left.
diff -Nurd freeciv/server/advisors/advdata.c freeciv/server/advisors/advdata.c
--- freeciv/server/advisors/advdata.c	2021-05-08 15:12:46.005943329 +0300
+++ freeciv/server/advisors/advdata.c	2021-05-08 15:12:57.238029911 +0300
@@ -927,7 +927,6 @@
           case ACTRES_DISBAND_UNIT:
           case ACTRES_PARADROP:
           case ACTRES_FORTIFY:
-          case ACTRES_SENTRY:
             /* Wants the ability to do this to it self. Don't want others
              * to target it. Do nothing since action_immune_government()
              * doesn't separate based on who the actor is. */
diff -Nurd freeciv/server/rscompat.c freeciv/server/rscompat.c
--- freeciv/server/rscompat.c	2021-05-08 15:12:46.009943360 +0300
+++ freeciv/server/rscompat.c	2021-05-08 15:12:57.238029911 +0300
@@ -846,10 +846,6 @@
     action_enabler_add(enabler);
 
     enabler = action_enabler_new();
-    enabler->action = ACTION_SENTRY;
-    action_enabler_add(enabler);
-
-    enabler = action_enabler_new();
     enabler->action = ACTION_ROAD;
     e_req = req_from_values(VUT_UTFLAG, REQ_RANGE_LOCAL, FALSE, TRUE, FALSE,
                             UTYF_SETTLERS);
diff -Nurd freeciv/server/savegame/savecompat.c freeciv/server/savegame/savecompat.c
--- freeciv/server/savegame/savecompat.c	2021-05-08 15:12:46.013943391 +0300
+++ freeciv/server/savegame/savecompat.c	2021-05-08 15:12:57.238029911 +0300
@@ -1576,7 +1576,6 @@
     case ACTIVITY_TRANSFORM:
     case ACTIVITY_CONVERT:
     case ACTIVITY_FORTIFYING:
-    case ACTIVITY_SENTRY:
     case ACTIVITY_BASE:
     case ACTIVITY_GEN_ROAD:
     case ACTIVITY_PILLAGE:
@@ -1590,6 +1589,9 @@
         }
       } action_iterate_end;
       break;
+    case ACTIVITY_SENTRY:
+      /* Not an action */
+      break;
     case ACTIVITY_EXPLORE:
     case ACTIVITY_IDLE:
     case ACTIVITY_GOTO:
diff -Nurd freeciv/server/unithand.c freeciv/server/unithand.c
--- freeciv/server/unithand.c	2021-05-08 15:12:46.009943360 +0300
+++ freeciv/server/unithand.c	2021-05-08 15:12:57.242029942 +0300
@@ -793,7 +793,6 @@
   case ACTRES_CLEAN_POLLUTION:
   case ACTRES_CLEAN_FALLOUT:
   case ACTRES_FORTIFY:
-  case ACTRES_SENTRY:
   case ACTRES_CONVERT:
   case ACTRES_ROAD:
   case ACTRES_BASE:
@@ -3055,10 +3054,6 @@
     ACTION_STARTED_UNIT_SELF(action_type, actor_unit,
                              do_action_activity(actor_unit, paction));
     break;
-  case ACTRES_SENTRY:
-    ACTION_STARTED_UNIT_SELF(action_type, actor_unit,
-                             do_action_activity(actor_unit, paction));
-    break;
   case ACTRES_CONVERT:
     ACTION_STARTED_UNIT_SELF(action_type, actor_unit,
                              do_action_activity(actor_unit, paction));
diff -Nurd freeciv/server/unittools.c freeciv/server/unittools.c
--- freeciv/server/unittools.c	2021-05-08 15:12:46.013943391 +0300
+++ freeciv/server/unittools.c	2021-05-08 15:12:57.242029942 +0300
@@ -3895,31 +3895,7 @@
             && !unit_has_orders(punit)
             && punit->ssa_controller == SSA_NONE
             && !can_unit_exist_at_tile(&(wld.map), punit, pdesttile)) {
-          struct action *paction = NULL;
-
-          /* Look for a sentry action the unit can perform */
-          action_by_activity_iterate(caction, act_id, ACTIVITY_SENTRY) {
-            if (action_get_actor_kind(caction) != AAK_UNIT) {
-              /* Not relevant. */
-              continue;
-            }
-
-            fc_assert_action(action_get_target_kind(caction) == ATK_SELF,
-                             continue);
-
-            if (is_action_enabled_unit_on_self(caction->id, punit)) {
-              /* Found one. */
-              paction = caction;
-              break;
-            }
-          } action_by_activity_iterate_end;
-
-          if (paction != NULL) {
-            /* A sentry action was found. */
-            unit_perform_action(pplayer, punit->id,
-                                IDENTITY_NUMBER_ZERO, NO_TARGET, NULL,
-                                paction->id, ACT_REQ_RULES);
-          }
+          set_unit_activity(punit, ACTIVITY_SENTRY);
         }
 
         send_unit_info(NULL, punit);
@@ -4690,8 +4666,15 @@
       break;
     case ORDER_ACTIVITY:
       switch (orders[i].activity) {
-      /* Replaced by action orders */
       case ACTIVITY_SENTRY:
+        if (i != length - 1) {
+          /* Only allowed as the last order. */
+          log_error("activity %d is not allowed at index %d.", orders[i].activity,
+                    i);
+          return FALSE;
+        }
+        break;
+      /* Replaced by action orders */
       case ACTIVITY_BASE:
       case ACTIVITY_GEN_ROAD:
       case ACTIVITY_FALLOUT:
@@ -4818,7 +4801,6 @@
             || !(utype_is_unmoved_by_action(paction, NULL)
                  || utype_is_moved_to_tgt_by_action(paction, NULL))
             /* or if the unit will end up standing still, */
-            || action_has_result(paction, ACTRES_SENTRY)
             || action_has_result(paction, ACTRES_FORTIFY)) {
           /* than having this action in the middle of a unit's orders is
            * probably wrong. */
