From 16f8617f4c7d5ac53cd48461e34b8785ddcad278 Mon Sep 17 00:00:00 2001
From: Marko Lindqvist <cazfi74@gmail.com>
Date: Fri, 7 Oct 2022 18:20:32 +0300
Subject: [PATCH 16/48] Make vision site able to hold maximum city name length

See osdn #45787

Signed-off-by: Marko Lindqvist <cazfi74@gmail.com>
---
 common/vision.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/common/vision.h b/common/vision.h
index 737ac63cb9..e2afa74521 100644
--- a/common/vision.h
+++ b/common/vision.h
@@ -110,7 +110,7 @@ bool vision_reveal_tiles(struct vision *vision, bool reveal_tiles);
  * so be careful with pointers!
  */
 struct vision_site {
-  char name[MAX_LEN_NAME];
+  char name[MAX_LEN_CITYNAME];
   struct tile *location; /* Cannot be NULL */
   struct player *owner;  /* May be NULL, always check! */
 
-- 
2.35.1

