From 7568da79ccde3860581643751052e08cde57ee3c Mon Sep 17 00:00:00 2001
From: Marko Lindqvist <cazfi74@gmail.com>
Date: Wed, 18 Dec 2019 04:36:46 +0200
Subject: [PATCH 23/23] Fix sell_random_unit() crash with recursive transports

See hrm Bug #852938

Signed-off-by: Marko Lindqvist <cazfi74@gmail.com>
---
 server/cityturn.c | 12 +++++++++---
 1 file changed, 9 insertions(+), 3 deletions(-)

diff --git a/server/cityturn.c b/server/cityturn.c
index 5d6c15910f..ffcf576e8b 100644
--- a/server/cityturn.c
+++ b/server/cityturn.c
@@ -2649,8 +2649,11 @@ static struct unit *sell_random_unit(struct player *pplayer,
   cargo = unit_list_new();
 
   /* Check if unit is transporting other units from punitlist,
-   * and sell one of those (recursively) instead. */
-  unit_list_iterate(unit_transport_cargo(punit), pcargo) {
+   * and sell one of those (recursively) instead.
+   * Note that in case of recursive transports we have to iterate
+   * also through those middle transports that themselves are not in
+   * punitlist. */
+  unit_cargo_iterate(punit, pcargo) {
     /* Optimization, do not iterate over punitlist
      * if we are sure that pcargo is not in it. */
     if (pcargo->server.upkeep_payed[O_GOLD] > 0) {
@@ -2660,9 +2663,12 @@ static struct unit *sell_random_unit(struct player *pplayer,
         }
       } unit_list_iterate_end;
     }
-  } unit_list_iterate_end;
+  } unit_cargo_iterate_end;
 
   if (unit_list_size(cargo) > 0) {
+    /* Recursively sell. Note that cargo list has both
+     * leaf units and middle transports in case of
+     * recursive transports. */
     struct unit *ret = sell_random_unit(pplayer, cargo);
 
     unit_list_destroy(cargo);
-- 
2.24.0

