diff --git a/common/networking/packets.def b/common/networking/packets.def
index 8bdd1a1c1e..b10d40abcf 100644
--- a/common/networking/packets.def
+++ b/common/networking/packets.def
@@ -1270,7 +1270,7 @@ end
 
 # Information about the ping times of the connections.
 PACKET_CONN_PING_INFO = 116; sc, lsend
-  UINT8 connections;
+  UINT16 connections;
   CONNECTION conn_id[MAX_NUM_CONNECTIONS:connections];
   SFLOAT10x6 ping_time[MAX_NUM_CONNECTIONS:connections];
 end
