From ef47e8480aaa5e7ff28be45de93ae6a1994c05ab Mon Sep 17 00:00:00 2001
From: Marko Lindqvist <cazfi74@gmail.com>
Date: Sat, 24 Sep 2022 16:08:24 +0300
Subject: [PATCH 38/38] Fix city_add_improvement_with_gov_notice() memory leak

See osdn #45707

Signed-off-by: Marko Lindqvist <cazfi74@gmail.com>
---
 server/citytools.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/server/citytools.c b/server/citytools.c
index 83d5bf48c8..c5f019ecf8 100644
--- a/server/citytools.c
+++ b/server/citytools.c
@@ -3551,6 +3551,8 @@ void city_add_improvement_with_gov_notice(struct city *pcity,
     players_iterate_alive(pplayer) {
       notify_new_government_options(pplayer, govs_data, buf);
     } players_iterate_alive_end;
+
+    free_current_governments_data(govs_data);
   } else {
     city_add_improvement(pcity, pimprove);
   }
-- 
2.35.1

