diff --git a/client/packhand.c b/client/packhand.c
index 9b8cd81f77..537e7d84e4 100644
--- a/client/packhand.c
+++ b/client/packhand.c
@@ -988,7 +988,9 @@ void handle_city_info(const struct packet_city_info *packet)
   use for it.
   TODO: Do not generate code calling this in the C-client.
 ****************************************************************************/
-void handle_web_city_info_addition(int id, int granary_size,
+void handle_web_city_info_addition(int id, bool cma_enabled,
+                                   const struct cm_parameter *cm_parameter,
+                                   int granary_size,
                                    int granary_turns)
 {
 }
diff --git a/common/networking/packets.def b/common/networking/packets.def
index ce49d66436..ce44d9fcfa 100644
--- a/common/networking/packets.def
+++ b/common/networking/packets.def
@@ -205,6 +205,7 @@ type WORKLIST           = worklist(struct worklist)
 # string that is URI encoded in the JSON protocol
 type ESTRING            = estring(char)
 type UNIT_ORDER         = unit_order(struct unit_order)
+type CM_PARAMETER       = cm_parameter(struct cm_parameter)
 
 # typedefs for enums
 type ACHIEVEMENT_TYPE   = uint8(enum achievement_type)
@@ -2357,6 +2358,9 @@ end
 PACKET_WEB_CITY_INFO_ADDITION = 256; sc, lsend, is-game-info, force, cancel(PACKET_CITY_SHORT_INFO)
   CITY id; key
 
+  BOOL cma_enabled;
+  CM_PARAMETER cm_parameter;
+
   UINT16 granary_size;
   TURN granary_turns;
 end
diff --git a/server/citytools.c b/server/citytools.c
index 4a96a632de..eb22322eb9 100644
--- a/server/citytools.c
+++ b/server/citytools.c
@@ -2629,6 +2629,14 @@ void package_city(struct city *pcity, struct packet_city_info *packet,
   if (web_packet != NULL) {
     web_packet->id = pcity->id;
 
+    if (pcity->cm_parameter != NULL) {
+      web_packet->cma_enabled = TRUE;
+      cm_copy_parameter(&web_packet->cm_parameter, pcity->cm_parameter);
+    } else {
+      web_packet->cma_enabled = FALSE;
+      memset(&web_packet->cm_parameter, 0, sizeof(web_packet->cm_parameter));
+    }
+
     web_packet->granary_size = city_granary_size(city_size_get(pcity));
     web_packet->granary_turns = city_turns_to_grow(pcity);
   }
