Index: freeciv/server/unithand.c
===================================================================
--- freeciv/server/unithand.c	(revision 24756)
+++ freeciv/server/unithand.c	(working copy)
@@ -224,6 +224,8 @@
 
     dsend_packet_unit_actions(pc, actor_unit_id, target_tile_id,
                               probabilities);
+    /* temporary fix to prevent segfaults in the code below */
+    return;
   }
 
   target_unit = is_other_players_unit_tile(target_tile, actor_player);
