diff --git a/configure.ac b/configure.ac
index f4be6ae..054cd07 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1397,6 +1397,8 @@ AC_CONFIG_FILES([Makefile
           data/civ2civ3/Makefile
 	  data/civ1/Makefile
 	  data/civ2/Makefile
+          data/fcweb/Makefile
+          data/webperimental/Makefile
 	  data/scenarios/Makefile
 	  data/nation/Makefile
 	  data/themes/Makefile
diff --git a/data/Makefile.am b/data/Makefile.am
index 6b198f4..5600588 100644
--- a/data/Makefile.am
+++ b/data/Makefile.am
@@ -90,7 +90,7 @@ else
 CLIENTDATADIRS = 
 endif
 
-SRV_RE_DATADIRS = default classic civ1 civ2 sandbox experimental multiplayer alien civ2civ3 nation
+SRV_RE_DATADIRS = default classic civ1 civ2 sandbox experimental multiplayer alien civ2civ3 fcweb webperimental nation
 if SERVER
 SERVERDATADIRS = $(SRV_RE_DATADIRS) scenarios
 else
