From 69b5a60f8b93f9bab63ec2999d78b1dc12737512 Mon Sep 17 00:00:00 2001
From: Sveinung Kvilhaugsvik <sveinung84@users.sourceforge.net>
Date: Sun, 26 Jul 2015 15:46:18 +0200
Subject: [PATCH 2/2] AI: Explode Nuclear if possible.

---
 ai/default/aiair.c   | 6 ++++++
 ai/default/aitools.c | 5 +++++
 2 files changed, 11 insertions(+)

diff --git a/ai/default/aiair.c b/ai/default/aiair.c
index 16aa0ec..3ed1bf3 100644
--- a/ai/default/aiair.c
+++ b/ai/default/aiair.c
@@ -405,6 +405,12 @@ void dai_manage_airunit(struct ai_type *ait, struct player *pplayer,
           handle_unit_do_action(unit_owner(punit),
                                 punit->id, tile_index(dst_tile),
                                 0, "", ACTION_BOMBARD);
+        } else if (is_action_enabled_unit_on_tile(ACTION_NUKE,
+                                                  punit, dst_tile)) {
+          /* Choose explode nuclear. */
+          handle_unit_do_action(unit_owner(punit),
+                                punit->id, tile_index(dst_tile),
+                                0, "", ACTION_NUKE);
         } else {
           (void) unit_move_handling(punit, dst_tile, TRUE, FALSE);
         }
diff --git a/ai/default/aitools.c b/ai/default/aitools.c
index 64b7c00..c549311 100644
--- a/ai/default/aitools.c
+++ b/ai/default/aitools.c
@@ -853,6 +853,11 @@ bool dai_unit_attack(struct ai_type *ait, struct unit *punit, struct tile *ptile
     /* Choose bombard. */
     handle_unit_do_action(unit_owner(punit), punit->id, tile_index(ptile),
                           0, "", ACTION_BOMBARD);
+  } if (is_action_enabled_unit_on_tile(ACTION_NUKE,
+                                       punit, ptile)) {
+    /* Choose explode nuclear. */
+    handle_unit_do_action(unit_owner(punit), punit->id, tile_index(ptile),
+                          0, "", ACTION_NUKE);
   } else {
     /* Regular attack. */
     (void) unit_move_handling(punit, ptile, FALSE, FALSE);
-- 
2.1.4

