diff -Nurd -X.diff_ignore freeciv/ai/aitraits.c freeciv/ai/aitraits.c
--- freeciv/ai/aitraits.c	2014-08-27 04:41:51.984421555 +0300
+++ freeciv/ai/aitraits.c	2014-08-27 04:42:12.760373243 +0300
@@ -59,7 +59,13 @@
 **************************************************************************/
 int ai_trait_get_value(enum trait tr, struct player *pplayer)
 {
-  int val = pplayer->ai_common.traits[tr].val + pplayer->ai_common.traits[tr].mod;
+  int val;
+
+  if (pplayer->ai_common.traits == NULL) {
+    return TRAIT_DEFAULT_VALUE;
+  }
+
+  val = pplayer->ai_common.traits[tr].val + pplayer->ai_common.traits[tr].mod;
 
   /* Clip so that value is at least 1, and maximum is
    * TRAIT_DEFAULT_VALUE as many times as TRAIT_DEFAULT value is
diff --git a/server/savegame3.c b/server/savegame3.c
index 82f3228..9fc1fcf 100644
--- a/server/savegame3.c
+++ b/server/savegame3.c
@@ -3901,10 +3901,12 @@ static void sg_save_player_main(struct savedata *saving,
     int j;
 
     for (tr = trait_begin(), j = 0; tr != trait_end(); tr = trait_next(tr), j++) {
-      secfile_insert_int(saving->file, plr->ai_common.traits[tr].val,
-                         "player%d.trait%d.val", plrno, j);
-      secfile_insert_int(saving->file, plr->ai_common.traits[tr].mod,
-                         "player%d.trait%d.mod", plrno, j);
+      if (plr->ai_common.traits != NULL) {
+        secfile_insert_int(saving->file, plr->ai_common.traits[tr].val,
+                           "player%d.trait%d.val", plrno, j);
+        secfile_insert_int(saving->file, plr->ai_common.traits[tr].mod,
+                           "player%d.trait%d.mod", plrno, j);
+      }
     }
   }
 
