diff -Nurd -X.diff_ignore freeciv/data/classic/terrain.ruleset freeciv/data/classic/terrain.ruleset
--- freeciv/data/classic/terrain.ruleset	2012-08-08 00:18:48.000000000 +0300
+++ freeciv/data/classic/terrain.ruleset	2012-08-08 00:20:16.000000000 +0300
@@ -314,7 +314,7 @@
 
 [terrain_glacier]
 name                 = _("Glacier")
-graphic              = "arctic"
+graphic              = "tundra"   ;FIXME: This is a cludge, because webclient isn't able to show arctic graphics.
 graphic_alt          = "-"
 identifier           = "a"
 movement_cost        = 2
