diff -Nurd freeciv/data/default/terrain.ruleset freeciv/data/default/terrain.ruleset
--- freeciv/data/default/terrain.ruleset	2012-05-16 18:20:18.000000000 +0300
+++ freeciv/data/default/terrain.ruleset	2012-05-16 18:20:01.000000000 +0300
@@ -293,7 +293,7 @@
 
 [terrain_glacier]
 name                 = _("Glacier")
-graphic              = "arctic"
+graphic              = "tundra"   ;FIXME: This is a cludge, because webclient isn't able to show arctic graphics.
 graphic_alt          = "-"
 identifier           = "a"
 movement_cost        = 2
