diff -Nurd -X.diff_ignore freeciv/fc_version freeciv/fc_version
--- freeciv/fc_version	2013-03-16 14:31:42.613041174 +0200
+++ freeciv/fc_version	2013-03-16 14:32:43.033039847 +0200
@@ -30,7 +30,7 @@
 #   - Avoid adding a new mandatory capability to the development branch for
 #     as long as possible.  We want to maintain network compatibility with
 #     the stable branch for as long as possible.
-NETWORK_CAPSTRING_MANDATORY="+Freeciv.Devel-2.5-2012.Nov.22"
+NETWORK_CAPSTRING_MANDATORY="+Freeciv.Web.Devel-2.5-2013.Mar.16"
 NETWORK_CAPSTRING_OPTIONAL=""
 
 FREECIV_DISTRIBUTOR=""
