diff -Nurd -X.diff_ignore freeciv/fc_version freeciv/fc_version
--- freeciv/fc_version	2013-03-31 23:26:03.918840760 +0300
+++ freeciv/fc_version	2013-03-31 23:26:47.462839804 +0300
@@ -30,7 +30,7 @@
 #   - Avoid adding a new mandatory capability to the development branch for
 #     as long as possible.  We want to maintain network compatibility with
 #     the stable branch for as long as possible.
-NETWORK_CAPSTRING_MANDATORY="+Freeciv.Devel-2.5-2013.Mar.26-2"
+NETWORK_CAPSTRING_MANDATORY="+Freeciv.Web.Devel-2.5-2013.Mar.31"
 NETWORK_CAPSTRING_OPTIONAL=""
 
 FREECIV_DISTRIBUTOR=""
