diff --git a/freeciv/common/packets.def b/freeciv/common/packets.def
index 728e4c1..246266c 100644
--- a/freeciv/common/packets.def
+++ b/freeciv/common/packets.def
@@ -791,6 +791,8 @@ PACKET_UNIT_INFO=49; sc,lsend
   BOOL has_orders;
   UINT16 orders_length, orders_index;
   BOOL orders_repeat, orders_vigilant;
+  BOOL caravan_trade;
+  BOOL caravan_wonder;
   ORDERS orders[MAX_LEN_ROUTE:orders_length];
   DIRECTION orders_dirs[MAX_LEN_ROUTE:orders_length];
   ACTIVITY orders_activities[MAX_LEN_ROUTE:orders_length];
diff --git a/freeciv/common/packets_gen.c b/freeciv/common/packets_gen.c
index caa12d1..5caa5a9 100644
--- a/freeciv/common/packets_gen.c
+++ b/freeciv/common/packets_gen.c
@@ -7259,6 +7259,8 @@ static struct packet_unit_info *receive_packet_unit_info_100(struct connection *
   }
   dio_get_bool8(&din, &real_packet->orders_repeat);
   dio_get_bool8(&din, &real_packet->orders_vigilant);
+  dio_get_bool8(&din, &real_packet->caravan_trade);
+  dio_get_bool8(&din, &real_packet->caravan_wonder);
   
   {
     int i;
@@ -7372,6 +7374,8 @@ static int send_packet_unit_info_100(struct connection *pc, const struct packet_
   dio_put_uint32(&dout, real_packet->orders_index);
   dio_put_bool8(&dout, real_packet->orders_repeat);
   dio_put_bool8(&dout, real_packet->orders_vigilant);
+  dio_put_bool8(&dout, real_packet->caravan_trade);
+  dio_put_bool8(&dout, real_packet->caravan_wonder);
 
     {
       int i;
diff --git a/freeciv/common/packets_gen.h b/freeciv/common/packets_gen.h
index 30800fe..24a2c56 100644
--- a/freeciv/common/packets_gen.h
+++ b/freeciv/common/packets_gen.h
@@ -498,6 +498,8 @@ struct packet_unit_info {
   int orders_index;
   bool orders_repeat;
   bool orders_vigilant;
+  bool caravan_trade;
+  bool caravan_wonder;
   enum unit_orders orders[MAX_LEN_ROUTE];
   enum direction8 orders_dirs[MAX_LEN_ROUTE];
   enum unit_activity orders_activities[MAX_LEN_ROUTE];
diff --git a/freeciv/server/unithand.c b/freeciv/server/unithand.c
index 5be2b1b..7db02e1 100644
--- a/freeciv/server/unithand.c
+++ b/freeciv/server/unithand.c
@@ -1698,9 +1698,9 @@ static bool base_handle_unit_establish_trade(struct player *pplayer, int unit_id
    
   }
 
-  sz_strlcpy(homecity_link, city_link(pcity_homecity));
-  sz_strlcpy(destcity_link, city_link(pcity_dest));
-  sz_strlcpy(punit_link, unit_link(punit));
+  sz_strlcpy(homecity_link, city_name(pcity_homecity));
+  sz_strlcpy(destcity_link, city_name(pcity_dest));
+  sz_strlcpy(punit_link, unit_rule_name(punit));
 
   if (!can_cities_trade(pcity_homecity, pcity_dest)) {
     notify_player(pplayer, pcity_dest->tile, E_BAD_COMMAND,
diff --git a/freeciv/server/unittools.c b/freeciv/server/unittools.c
index f305370..00b01cd 100644
--- a/freeciv/server/unittools.c
+++ b/freeciv/server/unittools.c
@@ -1892,6 +1892,15 @@ void package_unit(struct unit *punit, struct packet_unit_info *packet)
     packet->orders_repeat = packet->orders_vigilant = FALSE;
     /* No need to initialize array. */
   }
+
+  packet->caravan_trade = (unit_has_type_flag(punit, F_TRADE_ROUTE)
+	   && !unit_has_orders(punit)
+	   && unit_can_est_traderoute_here(punit));
+
+  packet->caravan_wonder = (unit_has_type_flag(punit, F_HELP_WONDER)
+	   && !unit_has_orders(punit)
+	   && unit_can_help_build_wonder_here(punit));
+
 }
 
 /**************************************************************************
