diff -Nurd -X.diff_ignore freeciv/server/citytools.c freeciv/server/citytools.c
--- freeciv/server/citytools.c	2012-09-02 02:19:14.000000000 +0300
+++ freeciv/server/citytools.c	2012-09-02 02:27:38.000000000 +0300
@@ -1727,7 +1727,7 @@
   packet->happy = pdcity->happy;
   packet->unhappy = pdcity->unhappy;
 
-  packet->improvements = pdcity->improvements;
+  //packet->improvements = pdcity->improvements;
 }
 
 /**************************************************************************
@@ -1951,6 +1951,7 @@
 {
   int i;
   int ppl = 0;
+  char impr_buf[MAX_NUM_ITEMS + 1];
 
   packet->id=pcity->id;
   packet->owner = player_number(city_owner(pcity));
@@ -2041,7 +2042,10 @@
   packet->shield_stock = pcity->shield_stock;
   packet->pollution = pcity->pollution;
   packet->illness_trade = pcity->illness_trade;
-  packet->city_options = pcity->city_options;
+  packet->granary_size = city_granary_size(city_size_get(pcity));
+  packet->granary_turns = city_turns_to_grow(pcity);
+  packet->buy_gold_cost = city_production_buy_gold_cost(pcity);
+  /* packet->city_options = pcity->city_options; */
 
   packet->production_kind = pcity->production.kind;
   packet->production_value = universal_number(&pcity->production);
@@ -2057,7 +2058,7 @@
   packet->caravan_shields=pcity->caravan_shields;
   packet->last_turns_shield_surplus = pcity->last_turns_shield_surplus;
 
-  worklist_copy(&packet->worklist, &pcity->worklist);
+  //worklist_copy(&packet->worklist, &pcity->worklist);
   packet->diplomat_investigate=dipl_invest;
 
   packet->airlift = pcity->airlift;
@@ -2068,12 +2069,13 @@
   packet->walls = city_got_citywalls(pcity);
   packet->city_image = get_city_bonus(pcity, EFT_CITY_IMAGE);
 
-  BV_CLR_ALL(packet->improvements);
   improvement_iterate(pimprove) {
-    if (city_has_building(pcity, pimprove)) {
-      BV_SET(packet->improvements, improvement_index(pimprove));
-    }
+    impr_buf[improvement_index(pimprove)] = city_has_building(pcity, pimprove)
+      ? '1' : '0';
   } improvement_iterate_end;
+  impr_buf[improvement_count()] = '\0';
+  sz_strlcpy(packet->improvements, impr_buf);
+  
 }
 
 /**************************************************************************
