diff --git a/common/networking/packets.def b/common/networking/packets.def
index 257adc5165..bc457efe10 100644
--- a/common/networking/packets.def
+++ b/common/networking/packets.def
@@ -741,6 +741,8 @@ PACKET_CITY_INFO = 31; sc, lsend, is-game-info, force, cancel(PACKET_CITY_SHORT_
 
   WORKLIST worklist;
 
+  STRING can_build_unit[MAX_LEN_MSG];
+  STRING can_build_improvement[MAX_LEN_MSG];
   BV_IMPRS improvements;
   BV_CITY_OPTIONS city_options;
   ESTRING name[MAX_LEN_CITYNAME];
diff --git a/server/citytools.c b/server/citytools.c
index 88ea703517..0768b189a9 100644
--- a/server/citytools.c
+++ b/server/citytools.c
@@ -2365,6 +2365,8 @@ void package_city(struct city *pcity, struct packet_city_info *packet,
 {
   int i;
   int ppl = 0;
+  char can_build_impr_buf[MAX_NUM_ITEMS + 1];
+  char can_build_unit_buf[MAX_NUM_ITEMS + 1];
 
   packet->id = pcity->id;
   packet->owner = player_number(city_owner(pcity));
@@ -2530,6 +2532,22 @@ void package_city(struct city *pcity, struct packet_city_info *packet,
     memset(&packet->cm_parameter, 0, sizeof(packet->cm_parameter));
   }
 
+  improvement_iterate(pimprove) {
+    can_build_impr_buf[improvement_index(pimprove)] =
+	    can_city_build_improvement_now(pcity, pimprove)
+      ? '1' : '0';
+  } improvement_iterate_end;
+  can_build_impr_buf[improvement_count()] = '\0';
+  sz_strlcpy(packet->can_build_improvement, can_build_impr_buf);
+
+  unit_type_iterate(punittype) {
+    can_build_unit_buf[utype_index(punittype)] =
+	    can_city_build_unit_now(pcity, punittype)
+      ? '1' : '0';
+  } unit_type_iterate_end;
+  can_build_unit_buf[utype_count()] = '\0';
+  sz_strlcpy(packet->can_build_unit, can_build_unit_buf);
+
   BV_CLR_ALL(packet->improvements);
   improvement_iterate(pimprove) {
     if (city_has_building(pcity, pimprove)) {
