diff -Nurd -X.diff_ignore freeciv/common/packets.def freeciv/common/packets.def
--- freeciv/common/packets.def	2012-07-16 02:51:14.000000000 +0300
+++ freeciv/common/packets.def	2012-07-16 02:51:20.000000000 +0300
@@ -543,10 +543,10 @@
   BOOL diplomat_investigate;
   BOOL walls;
   
-  WORKLIST worklist;
+  #WORKLIST worklist;
 
-  BV_IMPRS improvements;
-  BV_CITY_OPTIONS city_options;
+  STRING improvements[MAX_LEN_MSG];
+  #BV_CITY_OPTIONS city_options;
   STRING name[MAX_LEN_NAME];
 end
 
@@ -563,7 +563,7 @@
   BOOL happy;
   BOOL unhappy;
 
-  BV_IMPRS improvements;
+  STRING improvements[MAX_LEN_MSG];
   STRING name[MAX_LEN_NAME];
 end
 
diff -Nurd -X.diff_ignore freeciv/configure.ac freeciv/configure.ac
--- freeciv/configure.ac	2012-07-16 02:48:57.000000000 +0300
+++ freeciv/configure.ac	2012-07-16 02:52:19.000000000 +0300
@@ -847,7 +847,7 @@
 
 dnl Checks for additional server libraries:
 if test "x$server" = "xyes"; then
-    SERVER_LIBS="-lm ${SERVER_LIBS}"
+    SERVER_LIBS="-lm ${SERVER_LIBS} -lssl"
 
     dnl Some systems (e.g., BeOS) need this lib
     AC_CHECK_LIB(bind, gethostbyaddr, SERVER_LIBS="-lbind $SERVER_LIBS")
diff -Nurd -X.diff_ignore freeciv/server/citytools.c freeciv/server/citytools.c
--- freeciv/server/citytools.c	2012-07-16 02:48:33.000000000 +0300
+++ freeciv/server/citytools.c	2012-07-16 02:51:20.000000000 +0300
@@ -1730,7 +1730,7 @@
   packet->happy = pdcity->happy;
   packet->unhappy = pdcity->unhappy;
 
-  packet->improvements = pdcity->improvements;
+  //packet->improvements = pdcity->improvements;
 }
 
 /**************************************************************************
@@ -1948,6 +1948,7 @@
 {
   int i;
   int ppl = 0;
+  char impr_buf[MAX_NUM_ITEMS + 1];
 
   packet->id=pcity->id;
   packet->owner = player_number(city_owner(pcity));
@@ -2019,7 +2020,7 @@
   packet->shield_stock = pcity->shield_stock;
   packet->pollution = pcity->pollution;
   packet->illness_trade = pcity->illness_trade;
-  packet->city_options = pcity->city_options;
+  /* packet->city_options = pcity->city_options; */
 
   packet->production_kind = pcity->production.kind;
   packet->production_value = universal_number(&pcity->production);
@@ -2035,7 +2036,7 @@
   packet->caravan_shields=pcity->caravan_shields;
   packet->last_turns_shield_surplus = pcity->last_turns_shield_surplus;
 
-  worklist_copy(&packet->worklist, &pcity->worklist);
+  //worklist_copy(&packet->worklist, &pcity->worklist);
   packet->diplomat_investigate=dipl_invest;
 
   packet->airlift = pcity->airlift;
@@ -2045,12 +2046,13 @@
 
   packet->walls = city_got_citywalls(pcity);
 
-  BV_CLR_ALL(packet->improvements);
   improvement_iterate(pimprove) {
-    if (city_has_building(pcity, pimprove)) {
-      BV_SET(packet->improvements, improvement_index(pimprove));
-    }
+    impr_buf[improvement_index(pimprove)] = city_has_building(pcity, pimprove)
+      ? '1' : '0';
   } improvement_iterate_end;
+  impr_buf[improvement_count()] = '\0';
+  sz_strlcpy(packet->improvements, impr_buf);
+  
 }
 
 /**************************************************************************
