diff -Nurd -X.diff_ignore freeciv/configure.ac freeciv/configure.ac
--- freeciv/configure.ac	2012-07-16 02:53:09.000000000 +0300
+++ freeciv/configure.ac	2012-07-16 02:59:06.000000000 +0300
@@ -656,6 +656,11 @@
   AC_DEFINE([HAVE_THREADS], [1], [Defined if there is any thread implementation at all])
 fi
 
+AM_PATH_GLIB_2_0(2.29.0,,
+AC_MSG_ERROR(Test for Glib failed. See the file 'INSTALL' for help))
+
+LIBS="$LIBS $GLIB_LIBS"
+
 dnl Check and choose clients
 if test "x$client" != "xno"; then
 
@@ -1088,7 +1093,7 @@
      fi
    fi]],[[]])
 
-CFLAGS="$EXTRA_DEBUG_CFLAGS $CFLAGS"
+CFLAGS="$EXTRA_DEBUG_CFLAGS $CFLAGS $GLIB_CFLAGS"
 CXXFLAGS="$EXTRA_DEBUG_CXXFLAGS $CXXFLAGS"
 
 dnl Rebuild 'configure' whenever fc_version changes, if maintainer mode enabled.
diff -Nurd -X.diff_ignore freeciv/server/unithand.c freeciv/server/unithand.c
--- freeciv/server/unithand.c	2012-07-16 02:51:14.000000000 +0300
+++ freeciv/server/unithand.c	2012-07-16 03:00:22.000000000 +0300
@@ -19,9 +19,12 @@
 #include <stdlib.h>
 #include <string.h>
 
+#include <glib.h>
+
 /* utility */
 #include "astring.h"
 #include "fcintl.h"
+#include "fciconv.h"
 #include "mem.h"
 #include "rand.h"
 #include "shared.h"
@@ -705,7 +708,16 @@
   res = unit_add_or_build_city_test(punit);
 
   if (UAB_BUILD_OK == res) {
-    city_build(pplayer, punit, name);
+    /* Unescape city name, which has been escaped in Javascript. */
+    char* unescaped_text = g_uri_unescape_string(name, NULL);
+    char result_buf[MAX_LEN_NAME];
+
+    convert_string(unescaped_text,
+                   "latin1",
+                   "UTF-8",
+                   (char*)result_buf, MAX_LEN_NAME);
+    city_build(pplayer, punit, result_buf);
+    free(unescaped_text);
   } else if (UAB_ADD_OK == res) {
     city_add_unit(pplayer, punit);
   } else {
