diff -Nurd -X.diff_ignore freeciv/common/packets.def freeciv/common/packets.def
--- freeciv/common/packets.def	2012-05-16 20:56:43.000000000 +0300
+++ freeciv/common/packets.def	2012-05-16 20:59:21.000000000 +0300
@@ -576,10 +576,10 @@
   BOOL diplomat_investigate;
   BOOL walls;
   
-  WORKLIST worklist;
+  #WORKLIST worklist;
 
-  BV_IMPRS improvements;
-  BV_CITY_OPTIONS city_options;
+  STRING improvements[MAX_LEN_MSG];
+  #BV_CITY_OPTIONS city_options;
   STRING name[MAX_LEN_NAME];
 end
 
@@ -599,7 +599,7 @@
   BOOL happy;
   BOOL unhappy;
 
-  BV_IMPRS improvements;
+  STRING improvements[MAX_LEN_MSG];
   STRING name[MAX_LEN_NAME];
 end
 
diff -Nurd -X.diff_ignore freeciv/configure.ac freeciv/configure.ac
--- freeciv/configure.ac	2012-05-16 20:56:43.000000000 +0300
+++ freeciv/configure.ac	2012-05-16 20:59:19.000000000 +0300
@@ -705,7 +705,7 @@
 
     AC_CHECK_FUNC(pow)
     if test $ac_cv_func_pow = no; then
-        AC_CHECK_LIB(m, pow, SERVER_LIBS="$SERVER_LIBS -lm",
+        AC_CHECK_LIB(m, pow, SERVER_LIBS="$SERVER_LIBS -lm -lssl",
 		 AC_MSG_ERROR(Did not find math lib!))
     fi
 fi
diff -Nurd -X.diff_ignore freeciv/server/citytools.c freeciv/server/citytools.c
--- freeciv/server/citytools.c	2012-05-16 20:56:43.000000000 +0300
+++ freeciv/server/citytools.c	2012-05-16 20:59:19.000000000 +0300
@@ -1586,7 +1586,7 @@
   packet->happy = pdcity->happy;
   packet->unhappy = pdcity->unhappy;
 
-  packet->improvements = pdcity->improvements;
+  //packet->improvements = pdcity->improvements;
 }
 
 /**************************************************************************
@@ -1800,6 +1800,7 @@
 {
   int i;
   int ppl = 0;
+  char impr_buf[MAX_NUM_ITEMS + 1];
 
   packet->id=pcity->id;
   packet->owner = player_number(city_owner(pcity));
@@ -1870,7 +1871,7 @@
   packet->shield_stock=pcity->shield_stock;
   packet->pollution=pcity->pollution;
   packet->illness = pcity->illness;
-  packet->city_options = pcity->city_options;
+  //packet->city_options = pcity->city_options;
   
   packet->production_kind = pcity->production.kind;
   packet->production_value = universal_number(&pcity->production);
@@ -1886,7 +1887,7 @@
   packet->caravan_shields=pcity->caravan_shields;
   packet->last_turns_shield_surplus = pcity->last_turns_shield_surplus;
 
-  worklist_copy(&packet->worklist, &pcity->worklist);
+  //worklist_copy(&packet->worklist, &pcity->worklist);
   packet->diplomat_investigate=dipl_invest;
 
   packet->airlift = pcity->airlift;
@@ -1896,12 +1897,13 @@
 
   packet->walls = city_got_citywalls(pcity);
 
-  BV_CLR_ALL(packet->improvements);
   improvement_iterate(pimprove) {
-    if (city_has_building(pcity, pimprove)) {
-      BV_SET(packet->improvements, improvement_index(pimprove));
-    }
+    impr_buf[improvement_index(pimprove)] = city_has_building(pcity, pimprove)
+      ? '1' : '0';
   } improvement_iterate_end;
+  impr_buf[improvement_count()] = '\0';
+  sz_strlcpy(packet->improvements, impr_buf);
+  
 }
 
 /**************************************************************************
