diff --git freeciv/common/packets.def freeciv/common/packets.def
index 8b8835f..3d265ca 100644
--- freeciv/common/packets.def
+++ freeciv/common/packets.def
@@ -637,6 +637,7 @@ PACKET_CITY_INFO = 31; sc, lsend, is-game-info, force, cancel(PACKET_CITY_SHORT_
   STRING shield_output[MAX_LEN_MSG];
   STRING trade_output[MAX_LEN_MSG];
   #BV_CITY_OPTIONS city_options;
+  BOOL disbandable_city;
   STRING name[MAX_LEN_NAME];
   BOOL unhappy;
 end
@@ -2067,3 +2068,7 @@ end
 PACKET_INFO_TEXT_MESSAGE = 255; sc, dsend
   STRING message[MAX_LEN_MSG];
 end
+
+PACKET_CITY_DISBANDABLE_REQ = 256; cs, dsend
+  CITY city_id;
+end
diff --git freeciv/server/cityhand.c freeciv/server/cityhand.c
index b3df033..3662501 100644
--- freeciv/server/cityhand.c
+++ freeciv/server/cityhand.c
@@ -98,6 +98,28 @@ void handle_city_name_suggestion_req(struct player *pplayer, int unit_id)
 }
 
 /**************************************************************************
+  Handle request to toggle city disbandable state
+**************************************************************************/
+void handle_city_disbandable_req(struct player *pplayer, int city_id)
+{
+
+  struct city *pcity = player_city_by_number(pplayer, city_id);
+
+  if (!pcity) {
+    return;
+  }
+
+  if (BV_ISSET(pcity->city_options, CITYO_DISBAND)) {
+    BV_CLR(pcity->city_options, CITYO_DISBAND);
+  } else {
+    BV_SET(pcity->city_options, CITYO_DISBAND);
+  }
+
+  send_city_info(pplayer, pcity);
+
+}
+
+/**************************************************************************
   Handle request to change specialist type
 **************************************************************************/
 void handle_city_change_specialist(struct player *pplayer, int city_id,
diff --git freeciv/server/citytools.c freeciv/server/citytools.c
index b29bfd3..cd49ff2 100644
--- freeciv/server/citytools.c
+++ freeciv/server/citytools.c
@@ -2184,6 +2184,8 @@ void package_city(struct city *pcity, struct packet_city_info *packet,
   packet->buy_gold_cost = city_production_buy_gold_cost(pcity);
   /* packet->city_options = pcity->city_options; */
 
+  packet->disbandable_city = BV_ISSET(pcity->city_options, CITYO_DISBAND);
+
   packet->production_kind = pcity->production.kind;
   packet->production_value = universal_number(&pcity->production);
 
