diff --git a/common/city.c b/common/city.c
index 84d1f2c..0215b1a 100644
--- a/common/city.c
+++ b/common/city.c
@@ -2606,6 +2606,7 @@ inline void set_city_production(struct city *pcity)
   /* Add on special extra incomes: trade routes and tithes. */
   trade_routes_iterate(pcity, proute) {
     struct city *tcity = game_city_by_number(proute->partner);
+    if (tcity != NULL) {
     bool can_trade = can_cities_trade(pcity, tcity);
 
     if (!can_trade) {
@@ -2625,6 +2626,9 @@ inline void set_city_production(struct city *pcity)
     } else {
       proute->value = 0;
     }
+    } else {
+      log_debug("Error setting city trade route extra, tcity is NULL.");
+    }
   } trade_routes_iterate_end;
   pcity->prod[O_GOLD] += get_city_tithes_bonus(pcity);
 
