diff -Nurd -X.diff_ignore freeciv/server/citytools.c freeciv/server/citytools.c
--- freeciv/server/citytools.c	2012-05-16 21:02:54.000000000 +0300
+++ freeciv/server/citytools.c	2012-05-16 21:05:39.000000000 +0300
@@ -1801,5 +1801,7 @@
   int i;
   int ppl = 0;
+  char can_build_impr_buf[MAX_NUM_ITEMS + 1];
+  char can_build_unit_buf[MAX_NUM_ITEMS + 1];
 
   packet->id=pcity->id;
   packet->owner = player_number(city_owner(pcity));
@@ -1898,6 +1900,23 @@
   packet->city_image = get_city_bonus(pcity, EFT_CITY_IMAGE);
 
+  improvement_iterate(pimprove) {
+    can_build_impr_buf[improvement_index(pimprove)] = 
+	    can_city_build_improvement_now(pcity, pimprove)
+      ? '1' : '0';
+  } improvement_iterate_end;
+  can_build_impr_buf[improvement_count()] = '\0';
+  sz_strlcpy(packet->can_build_improvement, can_build_impr_buf);
+
+  unit_type_iterate(punittype) {
+    can_build_unit_buf[utype_index(punittype)] = 
+	    can_city_build_unit_now(pcity, punittype)
+      ? '1' : '0';
+  } unit_type_iterate_end;
+  can_build_unit_buf[utype_count()] = '\0';
+  sz_strlcpy(packet->can_build_unit, can_build_unit_buf);
+
+
   BV_CLR_ALL(packet->improvements);
   improvement_iterate(pimprove) {
     if (city_has_building(pcity, pimprove)) {
       BV_SET(packet->improvements, improvement_index(pimprove));
