diff -Nurd -X.diff_ignore freeciv/configure.ac freeciv/configure.ac
--- freeciv/configure.ac	2012-05-16 21:02:54.000000000 +0300
+++ freeciv/configure.ac	2012-05-16 21:08:12.000000000 +0300
@@ -523,6 +523,11 @@
      FTWL_LIBS="$FTWL_LIBS "`sdl-config --libs`" -lpng "`freetype-config --libs -lGL -lGLU`
 fi
 
+AM_PATH_GLIB_2_0(2.29.0,,
+AC_MSG_ERROR(Test for Glib failed. See the file 'INSTALL' for help))
+
+LIBS="$LIBS $GLIB_LIBS"
+
 dnl Check and choose clients
 if test "x$client" != "xno"; then
 
@@ -919,7 +924,7 @@
      fi
    fi]],[[]])
 
-CFLAGS="$EXTRA_DEBUG_CFLAGS $CFLAGS"
+CFLAGS="$EXTRA_DEBUG_CFLAGS $CFLAGS $GLIB_CFLAGS"
 CXXFLAGS="$EXTRA_DEBUG_CXXFLAGS $CXXFLAGS"
 
 dnl Rebuild 'configure' whenever version.in changes, if maintainer mode enabled.
diff -Nurd -X.diff_ignore freeciv/server/unithand.c freeciv/server/unithand.c
--- freeciv/server/unithand.c	2012-05-16 21:02:54.000000000 +0300
+++ freeciv/server/unithand.c	2012-05-16 21:08:12.000000000 +0300
@@ -20,8 +20,11 @@
 #include <stdlib.h>
 #include <string.h>
 
+#include <glib.h>
+
 /* utility */
 #include "fcintl.h"
+#include "fciconv.h"
 #include "mem.h"
 #include "rand.h"
 #include "shared.h"
@@ -716,7 +719,15 @@
   res = test_unit_add_or_build_city(punit);
 
   if (res == AB_BUILD_OK) {
-    city_build(pplayer, punit, name);
+    /* Unescape city name, which has been escaped in Javascript. */
+    char* unescaped_text = g_uri_unescape_string(name, NULL);
+    char result_buf[MAX_LEN_NAME];
+    convert_string(unescaped_text,
+		     "latin1",
+		     "UTF-8",
+		     (char*)result_buf, MAX_LEN_NAME);
+    city_build(pplayer, punit, result_buf);
+    free(unescaped_text);
   } else if (res == AB_ADD_OK) {
     city_add_unit(pplayer, punit);
   } else {
