diff --git a/freeciv/server/unithand.c b/freeciv/server/unithand.c
index 7db02e1..a4fa56d 100644
--- a/freeciv/server/unithand.c
+++ b/freeciv/server/unithand.c
@@ -20,8 +20,11 @@
 #include <stdlib.h>
 #include <string.h>
 
+#include <glib.h>
+
 /* utility */
 #include "fcintl.h"
+#include "fciconv.h"
 #include "mem.h"
 #include "rand.h"
 #include "shared.h"
@@ -718,7 +721,15 @@ void handle_unit_build_city(struct player *pplayer, int unit_id, char *name)
   res = test_unit_add_or_build_city(punit);
 
   if (res == AB_BUILD_OK) {
-    city_build(pplayer, punit, name);
+    /* Unescape city name, which has been escaped in Javascript. */
+    char* unescaped_text = g_uri_unescape_string(name, NULL);
+    char result_buf[MAX_LEN_NAME];
+    convert_string(unescaped_text,
+		     "latin1",
+		     "UTF-8",
+		     (char*)result_buf, MAX_LEN_NAME);
+    city_build(pplayer, punit, result_buf);
+    free(unescaped_text);
   } else if (res == AB_ADD_OK) {
     city_add_unit(pplayer, punit);
   } else {
diff --git a/freeciv/configure.ac b/freeciv/configure.ac
index 37cf2ac..13ac92b 100644
--- a/freeciv/configure.ac
+++ b/freeciv/configure.ac
@@ -523,6 +523,11 @@ if test "$ftwl" = opengl ; then
      FTWL_LIBS="$FTWL_LIBS "`sdl-config --libs`" -lpng "`freetype-config --libs -lGL -lGLU`
 fi
 
+AM_PATH_GLIB_2_0(2.29.0,,
+AC_MSG_ERROR(Test for Glib failed. See the file 'INSTALL' for help))
+
+LIBS="$LIBS $GLIB_LIBS"
+
 dnl Check and choose clients
 if test "x$client" != "xno"; then
 
@@ -922,7 +927,7 @@ AC_CONFIG_COMMANDS([fc_default-5],[[if test x`uname -s` = xBeOS ; then
      fi
    fi]],[[]])
 
-CFLAGS="$EXTRA_DEBUG_CFLAGS $CFLAGS"
+CFLAGS="$EXTRA_DEBUG_CFLAGS $CFLAGS $GLIB_CFLAGS"
 CXXFLAGS="$EXTRA_DEBUG_CXXFLAGS $CXXFLAGS"
 
 dnl Rebuild 'configure' whenever version.in changes, if maintainer mode enabled.
