Index: configure
===================================================================
--- configure	(revisjon 111)
+++ configure	(arbeidskopi)
@@ -36196,7 +36196,7 @@
 { $as_echo "$as_me:$LINENO: result: $ac_cv_lib_m_pow" >&5
 $as_echo "$ac_cv_lib_m_pow" >&6; }
 if test "x$ac_cv_lib_m_pow" = x""yes; then
-  SERVER_LIBS="$SERVER_LIBS -lm"
+  SERVER_LIBS="$SERVER_LIBS -lm -lssl"
 else
   { { $as_echo "$as_me:$LINENO: error: Did not find math lib!" >&5
 $as_echo "$as_me: error: Did not find math lib!" >&2;}
Index: configure.ac
===================================================================
--- configure.ac	(revisjon 111)
+++ configure.ac	(arbeidskopi)
@@ -705,7 +705,7 @@
 
     AC_CHECK_FUNC(pow)
     if test $ac_cv_func_pow = no; then
-        AC_CHECK_LIB(m, pow, SERVER_LIBS="$SERVER_LIBS -lm",
+        AC_CHECK_LIB(m, pow, SERVER_LIBS="$SERVER_LIBS -lm -lssl",
 		 AC_MSG_ERROR(Did not find math lib!))
     fi
 fi
Index: server/citytools.c
===================================================================
--- server/citytools.c	(revisjon 64)
+++ server/citytools.c	(arbeidskopi)
@@ -1586,7 +1586,7 @@
   packet->happy = pdcity->happy;
   packet->unhappy = pdcity->unhappy;
 
-  packet->improvements = pdcity->improvements;
+  //packet->improvements = pdcity->improvements;
 }
 
 /**************************************************************************
@@ -1800,6 +1800,7 @@
 {
   int i;
   int ppl = 0;
+  char impr_buf[MAX_NUM_ITEMS + 1];
 
   packet->id=pcity->id;
   packet->owner = player_number(city_owner(pcity));
@@ -1870,7 +1871,7 @@
   packet->shield_stock=pcity->shield_stock;
   packet->pollution=pcity->pollution;
   packet->illness = pcity->illness;
-  packet->city_options = pcity->city_options;
+  //packet->city_options = pcity->city_options;
   
   packet->production_kind = pcity->production.kind;
   packet->production_value = universal_number(&pcity->production);
@@ -1886,7 +1887,7 @@
   packet->caravan_shields=pcity->caravan_shields;
   packet->last_turns_shield_surplus = pcity->last_turns_shield_surplus;
 
-  worklist_copy(&packet->worklist, &pcity->worklist);
+  //worklist_copy(&packet->worklist, &pcity->worklist);
   packet->diplomat_investigate=dipl_invest;
 
   packet->airlift = pcity->airlift;
@@ -1896,12 +1897,13 @@
 
   packet->walls = city_got_citywalls(pcity);
 
-  BV_CLR_ALL(packet->improvements);
   improvement_iterate(pimprove) {
-    if (city_has_building(pcity, pimprove)) {
-      BV_SET(packet->improvements, improvement_index(pimprove));
-    }
+    impr_buf[improvement_index(pimprove)] = city_has_building(pcity, pimprove)
+      ? '1' : '0';
   } improvement_iterate_end;
+  impr_buf[improvement_count()] = '\0';
+  sz_strlcpy(packet->improvements, impr_buf);
+  
 }
 
 /**************************************************************************
Index: config.status
===================================================================
--- config.status	(revisjon 111)
+++ config.status	(arbeidskopi)
@@ -909,7 +909,7 @@
 S["GNULIB_CALLOC_POSIX"]="0"
 S["GNULIB_REALLOC_POSIX"]="0"
 S["GNULIB_MALLOC_POSIX"]="0"
-S["SERVER_LIBS"]="-lreadline   -lm -L/usr/local/lib -lcurl -lidn -lssl -lcrypto -lldap -lrt -lssl -lcrypto -ldl -lz -lz"
+S["SERVER_LIBS"]="-lreadline   -lm -lssl -L/usr/local/lib -lcurl -lidn -lssl -lcrypto -lldap -lrt -lssl -lcrypto -ldl -lz -lz"
 S["MINGW32_FALSE"]=""
 S["MINGW32_TRUE"]="#"
 S["FTWL_OPENGL_FALSE"]=""
Index: common/packets.def
===================================================================
--- common/packets.def	(revisjon 119)
+++ common/packets.def	(arbeidskopi)
@@ -577,10 +577,10 @@
   BOOL diplomat_investigate;
   BOOL walls;
   
-  WORKLIST worklist;
+  #WORKLIST worklist;
 
-  BV_IMPRS improvements;
-  BV_CITY_OPTIONS city_options;
+  STRING improvements[MAX_LEN_MSG];
+  #BV_CITY_OPTIONS city_options;
   STRING name[MAX_LEN_NAME];
 end
 
@@ -600,7 +600,7 @@
   BOOL happy;
   BOOL unhappy;
 
-  BV_IMPRS improvements;
+  STRING improvements[MAX_LEN_MSG];
   STRING name[MAX_LEN_NAME];
 end
 
Index: common/packets_gen.c
===================================================================
--- common/packets_gen.c	(revisjon 119)
+++ common/packets_gen.c	(arbeidskopi)
@@ -4316,9 +4316,7 @@
   dio_get_bool8(&din, &real_packet->was_happy);
   dio_get_bool8(&din, &real_packet->diplomat_investigate);
   dio_get_bool8(&din, &real_packet->walls);
-  dio_get_worklist(&din, &real_packet->worklist);
-  DIO_BV_GET(&din, real_packet->improvements);
-  DIO_BV_GET(&din, real_packet->city_options);
+  dio_get_string(&din, real_packet->improvements, sizeof(real_packet->improvements));
   dio_get_string(&din, real_packet->name, sizeof(real_packet->name));
 
   RECEIVE_PACKET_END(real_packet);
@@ -4460,9 +4458,7 @@
   dio_put_bool8(&dout, real_packet->was_happy);
   dio_put_bool8(&dout, real_packet->diplomat_investigate);
   dio_put_bool8(&dout, real_packet->walls);
-  dio_put_worklist(&dout, &real_packet->worklist);
-DIO_BV_PUT(&dout, packet->improvements);
-DIO_BV_PUT(&dout, packet->city_options);
+  dio_put_string(&dout, real_packet->improvements);
   dio_put_string(&dout, real_packet->name);
 
   SEND_PACKET_END;
@@ -4575,7 +4571,7 @@
   dio_get_bool8(&din, &real_packet->walls);
   dio_get_bool8(&din, &real_packet->happy);
   dio_get_bool8(&din, &real_packet->unhappy);
-  DIO_BV_GET(&din, real_packet->improvements);
+  dio_get_string(&din, real_packet->improvements, sizeof(real_packet->improvements));
   dio_get_string(&din, real_packet->name, sizeof(real_packet->name));
 
   RECEIVE_PACKET_END(real_packet);
@@ -4596,7 +4592,7 @@
   dio_put_bool8(&dout, real_packet->walls);
   dio_put_bool8(&dout, real_packet->happy);
   dio_put_bool8(&dout, real_packet->unhappy);
-DIO_BV_PUT(&dout, packet->improvements);
+  dio_put_string(&dout, real_packet->improvements);
   dio_put_string(&dout, real_packet->name);
 
   SEND_PACKET_END;
Index: common/packets_gen.h
===================================================================
--- common/packets_gen.h	(revisjon 78)
+++ common/packets_gen.h	(arbeidskopi)
@@ -298,9 +298,7 @@
   bool was_happy;
   bool diplomat_investigate;
   bool walls;
-  struct worklist worklist;
-  bv_imprs improvements;
-  bv_city_options city_options;
+  char improvements[MAX_LEN_MSG];
   char name[MAX_LEN_NAME];
 };
 
@@ -315,7 +313,7 @@
   bool walls;
   bool happy;
   bool unhappy;
-  bv_imprs improvements;
+  char improvements[MAX_LEN_MSG];
   char name[MAX_LEN_NAME];
 };
 
Index: autom4te.cache/output.1
===================================================================
--- autom4te.cache/output.1	(revisjon 111)
+++ autom4te.cache/output.1	(arbeidskopi)
@@ -36196,7 +36196,7 @@
 { $as_echo "$as_me:$LINENO: result: $ac_cv_lib_m_pow" >&5
 $as_echo "$ac_cv_lib_m_pow" >&6; }
 if test "x$ac_cv_lib_m_pow" = x""yes; then
-  SERVER_LIBS="$SERVER_LIBS -lm"
+  SERVER_LIBS="$SERVER_LIBS -lm -lssl"
 else
   { { $as_echo "$as_me:$LINENO: error: Did not find math lib!" >&5
 $as_echo "$as_me: error: Did not find math lib!" >&2;}
Index: autom4te.cache/output.2
===================================================================
--- autom4te.cache/output.2	(revisjon 111)
+++ autom4te.cache/output.2	(arbeidskopi)
@@ -36196,7 +36196,7 @@
 { $as_echo "$as_me:$LINENO: result: $ac_cv_lib_m_pow" >&5
 $as_echo "$ac_cv_lib_m_pow" >&6; }
 if test "x$ac_cv_lib_m_pow" = x""yes; then
-  SERVER_LIBS="$SERVER_LIBS -lm"
+  SERVER_LIBS="$SERVER_LIBS -lm -lssl"
 else
   { { $as_echo "$as_me:$LINENO: error: Did not find math lib!" >&5
 $as_echo "$as_me: error: Did not find math lib!" >&2;}
Index: autom4te.cache/requests
===================================================================
--- autom4te.cache/requests	(revisjon 111)
+++ autom4te.cache/requests	(arbeidskopi)
@@ -68,113 +68,113 @@
                         'configure.ac'
                       ],
                       {
+                        'AC_GGZ_CHECK' => 1,
                         'AM_ENABLE_STATIC' => 1,
-                        'AC_GGZ_CHECK' => 1,
+                        'FC_XAW_CLIENT' => 1,
                         'AC_LIBTOOL_LANG_RC_CONFIG' => 1,
-                        'FC_XAW_CLIENT' => 1,
                         '_LT_AC_SHELL_INIT' => 1,
                         'AC_CHECK_FT2' => 1,
                         'AC_DEFUN' => 1,
                         '_LT_AC_LANG_CXX_CONFIG' => 1,
                         'AC_PROG_LIBTOOL' => 1,
                         'AM_PROG_MKDIR_P' => 1,
+                        'FC_WIN32_CLIENT' => 1,
                         'AM_AUTOMAKE_VERSION' => 1,
-                        'FC_WIN32_CLIENT' => 1,
                         'AC_LIB_RPATH' => 1,
                         'AM_MISSING_PROG' => 1,
                         'AM_LIBCHARSET' => 1,
                         'AM_PATH_GTK_2_0' => 1,
+                        'FC_CHECK_X_PROTO_NARROWPROTO_WORKS' => 1,
                         'AC_LIBTOOL_PROG_LD_HARDCODE_LIBPATH' => 1,
-                        'FC_CHECK_X_PROTO_NARROWPROTO_WORKS' => 1,
                         '_LT_AC_LANG_C_CONFIG' => 1,
                         'FC_EXPAND_DIR' => 1,
+                        'FC_COMPILER_FLAGS' => 1,
                         'AM_PROG_INSTALL_STRIP' => 1,
-                        'FC_COMPILER_FLAGS' => 1,
                         'AC_GGZ_GGZDMOD' => 1,
                         '_m4_warn' => 1,
                         'AC_LIBTOOL_OBJDIR' => 1,
+                        'FC_LD_FLAGS' => 1,
                         'gl_FUNC_ARGZ' => 1,
-                        'FC_LD_FLAGS' => 1,
+                        'LTOBSOLETE_VERSION' => 1,
                         'AM_SANITY_CHECK' => 1,
-                        'LTOBSOLETE_VERSION' => 1,
                         'AC_LIBTOOL_LANG_GCJ_CONFIG' => 1,
                         'AC_LIBTOOL_PROG_COMPILER_PIC' => 1,
+                        'LT_LIB_M' => 1,
+                        'AM_PATH_SDL' => 1,
                         '_LT_AC_CHECK_DLFCN' => 1,
                         'AC_LIBTOOL_SYS_GLOBAL_SYMBOL_PIPE' => 1,
-                        'AM_PATH_SDL' => 1,
-                        'LT_LIB_M' => 1,
                         'LTSUGAR_VERSION' => 1,
                         '_LT_PROG_LTMAIN' => 1,
+                        'LT_SYS_SYMBOL_USCORE' => 1,
                         '_AM_PROG_TAR' => 1,
-                        'LT_SYS_SYMBOL_USCORE' => 1,
                         'AC_LIBTOOL_GCJ' => 1,
                         'AC_GGZ_LIBGGZ' => 1,
+                        'gl_FUNC_MKSTEMP' => 1,
+                        'LT_SYS_DLOPEN_DEPLIBS' => 1,
                         'LT_FUNC_DLSYM_USCORE' => 1,
-                        'LT_SYS_DLOPEN_DEPLIBS' => 1,
-                        'gl_FUNC_MKSTEMP' => 1,
+                        'AC_LIB_ARG_WITH' => 1,
+                        'AC_LIBTOOL_CONFIG' => 1,
                         '_LT_AC_LANG_F77' => 1,
-                        'AC_LIBTOOL_CONFIG' => 1,
-                        'AC_LIB_ARG_WITH' => 1,
+                        'FC_C_FLAGS' => 1,
+                        'AC_LTDL_DLLIB' => 1,
                         '_AM_SUBST_NOTMAKE' => 1,
-                        'AC_LTDL_DLLIB' => 1,
-                        'FC_C_FLAGS' => 1,
                         '_AM_AUTOCONF_VERSION' => 1,
                         'AM_DISABLE_SHARED' => 1,
                         '_LTDL_SETUP' => 1,
+                        'AM_PROG_LIBTOOL' => 1,
                         '_LT_AC_LANG_CXX' => 1,
-                        'AM_PROG_LIBTOOL' => 1,
+                        'AM_PROG_LD' => 1,
+                        '_LT_AC_FILE_LTDLL_C' => 1,
                         'AC_LIB_LTDL' => 1,
-                        '_LT_AC_FILE_LTDLL_C' => 1,
-                        'AM_PROG_LD' => 1,
                         'FC_DEBUG' => 1,
                         'AM_ICONV_LINK' => 1,
                         'AC_LIB_PREPARE_MULTILIB' => 1,
                         'AU_DEFUN' => 1,
+                        'AC_GGZ_GGZCORE' => 1,
                         'AC_PROG_NM' => 1,
-                        'AC_GGZ_GGZCORE' => 1,
                         'AC_LIBTOOL_DLOPEN' => 1,
+                        'AC_PROG_LD' => 1,
                         'FC_CXX_FLAGS' => 1,
-                        'AC_PROG_LD' => 1,
+                        'AC_ENABLE_FAST_INSTALL' => 1,
                         'AC_PROG_LD_GNU' => 1,
-                        'AC_ENABLE_FAST_INSTALL' => 1,
                         'FC_NO_CLIENT' => 1,
+                        'FC_XPM_PATHS' => 1,
+                        'AC_LIB_HAVE_LINKFLAGS' => 1,
                         'AC_LIBTOOL_FC' => 1,
-                        'AC_LIB_HAVE_LINKFLAGS' => 1,
-                        'FC_XPM_PATHS' => 1,
+                        'AM_ICONV_LINKFLAGS_BODY' => 1,
                         'LTDL_CONVENIENCE' => 1,
-                        'AM_ICONV_LINKFLAGS_BODY' => 1,
                         '_AM_SET_OPTION' => 1,
                         'AC_LTDL_PREOPEN' => 1,
                         '_LT_LINKER_BOILERPLATE' => 1,
+                        'FC_CHECK_X_LIB' => 1,
+                        'GLIB_RUN_PROG' => 1,
+                        'AC_LIBTOOL_LANG_CXX_CONFIG' => 1,
                         'AC_LIBTOOL_PROG_CC_C_O' => 1,
-                        'AC_LIBTOOL_LANG_CXX_CONFIG' => 1,
-                        'GLIB_RUN_PROG' => 1,
-                        'FC_CHECK_X_LIB' => 1,
                         'gl_PREREQ_ARGZ' => 1,
                         'AC_LIB_PREFIX' => 1,
+                        'gl_STDLIB_MODULE_INDICATOR' => 1,
                         'AC_C99_INITIALIZERS' => 1,
-                        'gl_STDLIB_MODULE_INDICATOR' => 1,
+                        'LT_SUPPORTED_TAG' => 1,
                         'AM_OUTPUT_DEPENDENCY_COMMANDS' => 1,
-                        'LT_SUPPORTED_TAG' => 1,
+                        'LT_PROG_RC' => 1,
                         'LT_SYS_MODULE_EXT' => 1,
-                        'LT_PROG_RC' => 1,
                         'AC_DEFUN_ONCE' => 1,
                         'FC_GTK2_CLIENT' => 1,
                         '_LT_AC_LANG_GCJ' => 1,
                         'AC_GGZ_REMOVEDUPS' => 1,
                         'AC_' => 1,
+                        'AC_LTDL_OBJDIR' => 1,
                         'FC_CHECK_AUTH' => 1,
-                        'AC_LTDL_OBJDIR' => 1,
+                        'FC_SDL_CLIENT' => 1,
                         '_LT_PATH_TOOL_PREFIX' => 1,
-                        'FC_SDL_CLIENT' => 1,
+                        'AM_ICONV' => 1,
                         'AC_LIBTOOL_RC' => 1,
-                        'AM_ICONV' => 1,
+                        'AC_DISABLE_FAST_INSTALL' => 1,
                         '_LT_AC_PROG_ECHO_BACKSLASH' => 1,
-                        'AC_DISABLE_FAST_INSTALL' => 1,
                         'FC_CHECK_GETTIMEOFDAY_RUNTIME' => 1,
+                        'include' => 1,
+                        '_LT_AC_TRY_DLOPEN_SELF' => 1,
                         '_LT_AC_SYS_LIBPATH_AIX' => 1,
-                        '_LT_AC_TRY_DLOPEN_SELF' => 1,
-                        'include' => 1,
                         'LT_AC_PROG_SED' => 1,
                         'AM_ENABLE_SHARED' => 1,
                         'AC_LIB_APPENDTOVAR' => 1,
@@ -183,93 +183,93 @@
                         'AC_ENABLE_SHARED' => 1,
                         'FC_ADD_WORDS_TO_VAR' => 1,
                         'AC_LIB_WITH_FINAL_PREFIX' => 1,
+                        'AC_ENABLE_STATIC' => 1,
                         'AC_LIBTOOL_SYS_HARD_LINK_LOCKS' => 1,
-                        'AC_ENABLE_STATIC' => 1,
                         '_LT_AC_TAGVAR' => 1,
                         'AC_LIBTOOL_LANG_F77_CONFIG' => 1,
                         'AM_CONDITIONAL' => 1,
                         'LT_LIB_DLLOAD' => 1,
+                        'AC_FUNC_VSNPRINTF' => 1,
                         'AM_LANGINFO_CODESET' => 1,
-                        'AC_FUNC_VSNPRINTF' => 1,
+                        'LTVERSION_VERSION' => 1,
                         'LTDL_INIT' => 1,
-                        'LTVERSION_VERSION' => 1,
+                        'm4_include' => 1,
                         'AM_PROG_INSTALL_SH' => 1,
-                        'm4_include' => 1,
                         'AC_PROG_EGREP' => 1,
+                        'AC_PATH_MAGIC' => 1,
                         '_AC_AM_CONFIG_HEADER_HOOK' => 1,
-                        'AC_PATH_MAGIC' => 1,
                         'AC_LTDL_SYSSEARCHPATH' => 1,
                         'FC_HAS_READLINE' => 1,
+                        'FC_CHECK_SOUND' => 1,
+                        'FC_CHECK_X_PROTO_DEFINE' => 1,
                         'AM_MAKE_INCLUDE' => 1,
-                        'FC_CHECK_X_PROTO_DEFINE' => 1,
-                        'FC_CHECK_SOUND' => 1,
+                        '_LT_AC_TAGCONFIG' => 1,
                         'LT_CMD_MAX_LEN' => 1,
-                        '_LT_AC_TAGCONFIG' => 1,
                         'AC_GGZ_SERVER' => 1,
                         'm4_pattern_forbid' => 1,
+                        'AC_LIB_PREPARE_PREFIX' => 1,
                         '_LT_LINKER_OPTION' => 1,
-                        'AC_LIB_PREPARE_PREFIX' => 1,
                         'AC_LIBTOOL_COMPILER_OPTION' => 1,
+                        'gl_PREREQ_MKSTEMP' => 1,
                         'AC_DISABLE_SHARED' => 1,
-                        'gl_PREREQ_MKSTEMP' => 1,
                         '_LT_COMPILER_BOILERPLATE' => 1,
+                        'AC_LIBTOOL_WIN32_DLL' => 1,
                         'AC_LIBTOOL_SETUP' => 1,
-                        'AC_LIBTOOL_WIN32_DLL' => 1,
+                        'AM_GLIB_GNU_GETTEXT' => 1,
                         'AC_PROG_LD_RELOAD_FLAG' => 1,
-                        'AM_GLIB_GNU_GETTEXT' => 1,
                         'AC_LTDL_DLSYM_USCORE' => 1,
+                        'AM_MISSING_HAS_RUN' => 1,
                         'LT_LANG' => 1,
-                        'AM_MISSING_HAS_RUN' => 1,
+                        'AC_GGZ_INIT' => 1,
                         'LT_SYS_DLSEARCH_PATH' => 1,
-                        'AC_GGZ_INIT' => 1,
                         'LT_CONFIG_LTDL_DIR' => 1,
+                        'FC_CHECK_READLINE_RUNTIME' => 1,
+                        'AC_LIBTOOL_DLOPEN_SELF' => 1,
                         'LT_OUTPUT' => 1,
-                        'AC_LIBTOOL_DLOPEN_SELF' => 1,
-                        'FC_CHECK_READLINE_RUNTIME' => 1,
+                        'AC_C99_STDINT_H' => 1,
                         'AC_LIB_PROG_LD_GNU' => 1,
-                        'AC_C99_STDINT_H' => 1,
                         'AC_LIBTOOL_PROG_LD_SHLIBS' => 1,
                         '_PKG_SHORT_ERRORS_SUPPORTED' => 1,
+                        'AC_WITH_LTDL' => 1,
                         'AC_LIBTOOL_LINKER_OPTION' => 1,
-                        'AC_WITH_LTDL' => 1,
                         'PKG_CHECK_EXISTS' => 1,
                         'AC_GGZ_VERSION' => 1,
+                        'LT_AC_PROG_RC' => 1,
                         'AC_LIBTOOL_CXX' => 1,
-                        'LT_AC_PROG_RC' => 1,
                         'LT_INIT' => 1,
+                        'LT_AC_PROG_GCJ' => 1,
                         'LT_SYS_DLOPEN_SELF' => 1,
-                        'LT_AC_PROG_GCJ' => 1,
+                        'AM_DISABLE_STATIC' => 1,
                         'AM_DEP_TRACK' => 1,
-                        'AM_DISABLE_STATIC' => 1,
+                        'AM_CONFIG_HEADER' => 1,
                         '_AC_PROG_LIBTOOL' => 1,
-                        'AM_CONFIG_HEADER' => 1,
                         '_AM_IF_OPTION' => 1,
                         'AC_PATH_TOOL_PREFIX' => 1,
+                        'm4_pattern_allow' => 1,
                         'AC_LIBTOOL_F77' => 1,
-                        'm4_pattern_allow' => 1,
                         'AM_SET_LEADING_DOT' => 1,
                         'LT_AC_PROG_EGREP' => 1,
                         '_AM_DEPENDENCIES' => 1,
                         'AC_LIBTOOL_LANG_C_CONFIG' => 1,
+                        'AC_GGZ_CONFIG' => 1,
                         'LTOPTIONS_VERSION' => 1,
-                        'AC_GGZ_CONFIG' => 1,
+                        'AC_LIB_LINKFLAGS' => 1,
                         '_LT_AC_SYS_COMPILER' => 1,
-                        'AC_LIB_LINKFLAGS' => 1,
                         'AM_PROG_NM' => 1,
                         'PKG_CHECK_MODULES' => 1,
                         'gl_STDLIB_H' => 1,
                         'AC_GGZ_GGZMOD' => 1,
                         'AC_LIBLTDL_CONVENIENCE' => 1,
+                        'FC_CHECK_X_PROTO_FETCH' => 1,
                         'AC_DEPLIBS_CHECK_METHOD' => 1,
-                        'FC_CHECK_X_PROTO_FETCH' => 1,
+                        'AC_LIBLTDL_INSTALLABLE' => 1,
                         'AM_SET_CURRENT_AUTOMAKE_VERSION' => 1,
-                        'AC_LIBLTDL_INSTALLABLE' => 1,
+                        'AC_C99_VARIABLE_ARRAYS' => 1,
                         'AC_LTDL_ENABLE_INSTALL' => 1,
-                        'AC_C99_VARIABLE_ARRAYS' => 1,
+                        'LT_PROG_GCJ' => 1,
                         'AC_LIBTOOL_SYS_DYNAMIC_LINKER' => 1,
-                        'LT_PROG_GCJ' => 1,
+                        'AC_C99_VARIADIC_MACROS' => 1,
                         'AM_GLIB_DEFINE_LOCALEDIR' => 1,
-                        'AC_C99_VARIADIC_MACROS' => 1,
                         'AM_INIT_AUTOMAKE' => 1,
                         'AC_GGZ_CHECK_SERVER' => 1,
                         'AC_DISABLE_STATIC' => 1,
@@ -281,33 +281,33 @@
                         'LT_SYS_MODULE_PATH' => 1,
                         'AC_GGZ_GTK' => 1,
                         'AC_GGZ_FIND_FILE' => 1,
+                        'LT_WITH_LTDL' => 1,
                         'AC_LIBTOOL_POSTDEP_PREDEP' => 1,
-                        'LT_WITH_LTDL' => 1,
+                        'gl_STDLIB_H_DEFAULTS' => 1,
                         'AC_LTDL_SHLIBPATH' => 1,
-                        'gl_STDLIB_H_DEFAULTS' => 1,
                         'AM_AUX_DIR_EXPAND' => 1,
                         'AC_LIB_LINKFLAGS_FROM_LIBS' => 1,
+                        'FC_CHECK_X_PROTO_FUNCPROTO_COMPILE' => 1,
+                        'AC_LIBTOOL_PROG_COMPILER_NO_RTTI' => 1,
                         '_LT_AC_LANG_F77_CONFIG' => 1,
-                        'AC_LIBTOOL_PROG_COMPILER_NO_RTTI' => 1,
-                        'FC_CHECK_X_PROTO_FUNCPROTO_COMPILE' => 1,
+                        '_LT_COMPILER_OPTION' => 1,
                         '_AM_SET_OPTIONS' => 1,
-                        '_LT_COMPILER_OPTION' => 1,
+                        'AM_RUN_LOG' => 1,
                         '_AM_OUTPUT_DEPENDENCY_COMMANDS' => 1,
-                        'AM_RUN_LOG' => 1,
+                        'AC_LIBTOOL_PICMODE' => 1,
+                        'AC_LTDL_SYS_DLOPEN_DEPLIBS' => 1,
                         'AC_LIBTOOL_SYS_OLD_ARCHIVE' => 1,
-                        'AC_LTDL_SYS_DLOPEN_DEPLIBS' => 1,
-                        'AC_LIBTOOL_PICMODE' => 1,
                         'AC_LIB_LINKFLAGS_BODY' => 1,
+                        'AC_CHECK_LIBM' => 1,
                         'LT_PATH_LD' => 1,
-                        'AC_CHECK_LIBM' => 1,
+                        'AC_GGZ_ERROR' => 1,
                         'AC_LIBTOOL_SYS_LIB_STRIP' => 1,
-                        'AC_GGZ_ERROR' => 1,
                         '_AM_MANGLE_OPTION' => 1,
+                        'AC_LIBTOOL_SYS_MAX_CMD_LEN' => 1,
                         'AC_LTDL_SYMBOL_USCORE' => 1,
-                        'AC_LIBTOOL_SYS_MAX_CMD_LEN' => 1,
                         'AM_SET_DEPDIR' => 1,
+                        'PKG_PROG_PKG_CONFIG' => 1,
                         '_LT_CC_BASENAME' => 1,
-                        'PKG_PROG_PKG_CONFIG' => 1,
                         '_LT_LIBOBJ' => 1,
                         'AC_LIB_PROG_LD' => 1
                       }
@@ -324,25 +324,25 @@
                         'configure.ac'
                       ],
                       {
+                        'AM_PROG_F77_C_O' => 1,
                         '_LT_AC_TAGCONFIG' => 1,
-                        'AM_PROG_F77_C_O' => 1,
+                        'm4_pattern_forbid' => 1,
                         'AC_INIT' => 1,
-                        'm4_pattern_forbid' => 1,
+                        'AC_CANONICAL_TARGET' => 1,
                         '_AM_COND_IF' => 1,
-                        'AC_CANONICAL_TARGET' => 1,
+                        'AC_CONFIG_LIBOBJ_DIR' => 1,
                         'AC_SUBST' => 1,
-                        'AC_CONFIG_LIBOBJ_DIR' => 1,
+                        'AC_CANONICAL_HOST' => 1,
                         'AC_FC_SRCEXT' => 1,
-                        'AC_CANONICAL_HOST' => 1,
                         'AC_PROG_LIBTOOL' => 1,
                         'AM_INIT_AUTOMAKE' => 1,
                         'AC_CONFIG_SUBDIRS' => 1,
                         'AM_AUTOMAKE_VERSION' => 1,
                         'LT_CONFIG_LTDL_DIR' => 1,
+                        'AC_CONFIG_LINKS' => 1,
                         'AC_REQUIRE_AUX_FILE' => 1,
-                        'AC_CONFIG_LINKS' => 1,
+                        'LT_SUPPORTED_TAG' => 1,
                         'm4_sinclude' => 1,
-                        'LT_SUPPORTED_TAG' => 1,
                         'AM_MAINTAINER_MODE' => 1,
                         'AM_GNU_GETTEXT_INTL_SUBDIR' => 1,
                         '_m4_warn' => 1,
@@ -358,13 +358,13 @@
                         'AC_CANONICAL_BUILD' => 1,
                         'AC_FC_FREEFORM' => 1,
                         'AH_OUTPUT' => 1,
+                        'AC_CONFIG_AUX_DIR' => 1,
                         '_AM_SUBST_NOTMAKE' => 1,
-                        'AC_CONFIG_AUX_DIR' => 1,
+                        'm4_pattern_allow' => 1,
+                        'AM_PROG_CC_C_O' => 1,
                         'sinclude' => 1,
-                        'AM_PROG_CC_C_O' => 1,
-                        'm4_pattern_allow' => 1,
+                        'AM_CONDITIONAL' => 1,
                         'AC_CANONICAL_SYSTEM' => 1,
-                        'AM_CONDITIONAL' => 1,
                         'AC_CONFIG_HEADERS' => 1,
                         'AC_DEFINE_TRACE_LITERAL' => 1,
                         'm4_include' => 1,
@@ -437,113 +437,113 @@
                         'configure.ac'
                       ],
                       {
+                        'AM_ENABLE_STATIC' => 1,
                         'AC_GGZ_CHECK' => 1,
-                        'AM_ENABLE_STATIC' => 1,
+                        'AC_LIBTOOL_LANG_RC_CONFIG' => 1,
                         'FC_XAW_CLIENT' => 1,
-                        'AC_LIBTOOL_LANG_RC_CONFIG' => 1,
                         '_LT_AC_SHELL_INIT' => 1,
                         'AC_CHECK_FT2' => 1,
                         'AC_DEFUN' => 1,
                         '_LT_AC_LANG_CXX_CONFIG' => 1,
                         'AC_PROG_LIBTOOL' => 1,
                         'AM_PROG_MKDIR_P' => 1,
+                        'AM_AUTOMAKE_VERSION' => 1,
                         'FC_WIN32_CLIENT' => 1,
-                        'AM_AUTOMAKE_VERSION' => 1,
                         'AC_LIB_RPATH' => 1,
                         'AM_MISSING_PROG' => 1,
                         'AM_LIBCHARSET' => 1,
                         'AM_PATH_GTK_2_0' => 1,
+                        'AC_LIBTOOL_PROG_LD_HARDCODE_LIBPATH' => 1,
                         'FC_CHECK_X_PROTO_NARROWPROTO_WORKS' => 1,
-                        'AC_LIBTOOL_PROG_LD_HARDCODE_LIBPATH' => 1,
                         '_LT_AC_LANG_C_CONFIG' => 1,
                         'FC_EXPAND_DIR' => 1,
+                        'AM_PROG_INSTALL_STRIP' => 1,
                         'FC_COMPILER_FLAGS' => 1,
-                        'AM_PROG_INSTALL_STRIP' => 1,
                         'AC_GGZ_GGZDMOD' => 1,
                         '_m4_warn' => 1,
                         'AC_LIBTOOL_OBJDIR' => 1,
+                        'gl_FUNC_ARGZ' => 1,
                         'FC_LD_FLAGS' => 1,
-                        'gl_FUNC_ARGZ' => 1,
+                        'AM_SANITY_CHECK' => 1,
                         'LTOBSOLETE_VERSION' => 1,
-                        'AM_SANITY_CHECK' => 1,
                         'AC_LIBTOOL_LANG_GCJ_CONFIG' => 1,
                         'AC_LIBTOOL_PROG_COMPILER_PIC' => 1,
+                        '_LT_AC_CHECK_DLFCN' => 1,
+                        'AC_LIBTOOL_SYS_GLOBAL_SYMBOL_PIPE' => 1,
                         'AM_PATH_SDL' => 1,
                         'LT_LIB_M' => 1,
-                        'AC_LIBTOOL_SYS_GLOBAL_SYMBOL_PIPE' => 1,
-                        '_LT_AC_CHECK_DLFCN' => 1,
                         'LTSUGAR_VERSION' => 1,
                         '_LT_PROG_LTMAIN' => 1,
+                        '_AM_PROG_TAR' => 1,
                         'LT_SYS_SYMBOL_USCORE' => 1,
-                        '_AM_PROG_TAR' => 1,
                         'AC_LIBTOOL_GCJ' => 1,
                         'AC_GGZ_LIBGGZ' => 1,
+                        'LT_FUNC_DLSYM_USCORE' => 1,
+                        'LT_SYS_DLOPEN_DEPLIBS' => 1,
                         'gl_FUNC_MKSTEMP' => 1,
-                        'LT_SYS_DLOPEN_DEPLIBS' => 1,
-                        'LT_FUNC_DLSYM_USCORE' => 1,
+                        '_LT_AC_LANG_F77' => 1,
+                        'AC_LIBTOOL_CONFIG' => 1,
                         'AC_LIB_ARG_WITH' => 1,
-                        'AC_LIBTOOL_CONFIG' => 1,
-                        '_LT_AC_LANG_F77' => 1,
+                        '_AM_SUBST_NOTMAKE' => 1,
+                        'AC_LTDL_DLLIB' => 1,
                         'FC_C_FLAGS' => 1,
-                        'AC_LTDL_DLLIB' => 1,
-                        '_AM_SUBST_NOTMAKE' => 1,
                         '_AM_AUTOCONF_VERSION' => 1,
                         'AM_DISABLE_SHARED' => 1,
                         '_LTDL_SETUP' => 1,
+                        '_LT_AC_LANG_CXX' => 1,
                         'AM_PROG_LIBTOOL' => 1,
-                        '_LT_AC_LANG_CXX' => 1,
+                        'AC_LIB_LTDL' => 1,
+                        '_LT_AC_FILE_LTDLL_C' => 1,
                         'AM_PROG_LD' => 1,
-                        '_LT_AC_FILE_LTDLL_C' => 1,
-                        'AC_LIB_LTDL' => 1,
                         'FC_DEBUG' => 1,
                         'AM_ICONV_LINK' => 1,
                         'AC_LIB_PREPARE_MULTILIB' => 1,
                         'AU_DEFUN' => 1,
+                        'AC_PROG_NM' => 1,
                         'AC_GGZ_GGZCORE' => 1,
-                        'AC_PROG_NM' => 1,
                         'AC_LIBTOOL_DLOPEN' => 1,
+                        'FC_CXX_FLAGS' => 1,
                         'AC_PROG_LD' => 1,
-                        'FC_CXX_FLAGS' => 1,
+                        'AC_PROG_LD_GNU' => 1,
                         'AC_ENABLE_FAST_INSTALL' => 1,
-                        'AC_PROG_LD_GNU' => 1,
                         'FC_NO_CLIENT' => 1,
+                        'AC_LIBTOOL_FC' => 1,
+                        'AC_LIB_HAVE_LINKFLAGS' => 1,
                         'FC_XPM_PATHS' => 1,
-                        'AC_LIB_HAVE_LINKFLAGS' => 1,
-                        'AC_LIBTOOL_FC' => 1,
+                        'LTDL_CONVENIENCE' => 1,
                         'AM_ICONV_LINKFLAGS_BODY' => 1,
-                        'LTDL_CONVENIENCE' => 1,
                         '_AM_SET_OPTION' => 1,
                         'AC_LTDL_PREOPEN' => 1,
                         '_LT_LINKER_BOILERPLATE' => 1,
+                        'AC_LIBTOOL_PROG_CC_C_O' => 1,
+                        'AC_LIBTOOL_LANG_CXX_CONFIG' => 1,
+                        'GLIB_RUN_PROG' => 1,
                         'FC_CHECK_X_LIB' => 1,
-                        'GLIB_RUN_PROG' => 1,
-                        'AC_LIBTOOL_LANG_CXX_CONFIG' => 1,
-                        'AC_LIBTOOL_PROG_CC_C_O' => 1,
                         'gl_PREREQ_ARGZ' => 1,
                         'AC_LIB_PREFIX' => 1,
+                        'AC_C99_INITIALIZERS' => 1,
                         'gl_STDLIB_MODULE_INDICATOR' => 1,
-                        'AC_C99_INITIALIZERS' => 1,
+                        'AM_OUTPUT_DEPENDENCY_COMMANDS' => 1,
                         'LT_SUPPORTED_TAG' => 1,
-                        'AM_OUTPUT_DEPENDENCY_COMMANDS' => 1,
+                        'LT_SYS_MODULE_EXT' => 1,
                         'LT_PROG_RC' => 1,
-                        'LT_SYS_MODULE_EXT' => 1,
                         'AC_DEFUN_ONCE' => 1,
                         'FC_GTK2_CLIENT' => 1,
                         '_LT_AC_LANG_GCJ' => 1,
                         'AC_GGZ_REMOVEDUPS' => 1,
                         'AC_' => 1,
+                        'FC_CHECK_AUTH' => 1,
                         'AC_LTDL_OBJDIR' => 1,
-                        'FC_CHECK_AUTH' => 1,
+                        '_LT_PATH_TOOL_PREFIX' => 1,
                         'FC_SDL_CLIENT' => 1,
-                        '_LT_PATH_TOOL_PREFIX' => 1,
+                        'AC_LIBTOOL_RC' => 1,
                         'AM_ICONV' => 1,
-                        'AC_LIBTOOL_RC' => 1,
+                        '_LT_AC_PROG_ECHO_BACKSLASH' => 1,
                         'AC_DISABLE_FAST_INSTALL' => 1,
-                        '_LT_AC_PROG_ECHO_BACKSLASH' => 1,
                         'FC_CHECK_GETTIMEOFDAY_RUNTIME' => 1,
+                        '_LT_AC_SYS_LIBPATH_AIX' => 1,
+                        '_LT_AC_TRY_DLOPEN_SELF' => 1,
                         'include' => 1,
-                        '_LT_AC_TRY_DLOPEN_SELF' => 1,
-                        '_LT_AC_SYS_LIBPATH_AIX' => 1,
                         'LT_AC_PROG_SED' => 1,
                         'AM_ENABLE_SHARED' => 1,
                         'AC_LIB_APPENDTOVAR' => 1,
@@ -552,93 +552,93 @@
                         'AC_ENABLE_SHARED' => 1,
                         'FC_ADD_WORDS_TO_VAR' => 1,
                         'AC_LIB_WITH_FINAL_PREFIX' => 1,
+                        'AC_LIBTOOL_SYS_HARD_LINK_LOCKS' => 1,
                         'AC_ENABLE_STATIC' => 1,
-                        'AC_LIBTOOL_SYS_HARD_LINK_LOCKS' => 1,
                         '_LT_AC_TAGVAR' => 1,
                         'AC_LIBTOOL_LANG_F77_CONFIG' => 1,
                         'AM_CONDITIONAL' => 1,
                         'LT_LIB_DLLOAD' => 1,
+                        'AM_LANGINFO_CODESET' => 1,
                         'AC_FUNC_VSNPRINTF' => 1,
-                        'AM_LANGINFO_CODESET' => 1,
+                        'LTDL_INIT' => 1,
                         'LTVERSION_VERSION' => 1,
-                        'LTDL_INIT' => 1,
+                        'AM_PROG_INSTALL_SH' => 1,
                         'm4_include' => 1,
-                        'AM_PROG_INSTALL_SH' => 1,
                         'AC_PROG_EGREP' => 1,
+                        '_AC_AM_CONFIG_HEADER_HOOK' => 1,
                         'AC_PATH_MAGIC' => 1,
-                        '_AC_AM_CONFIG_HEADER_HOOK' => 1,
                         'AC_LTDL_SYSSEARCHPATH' => 1,
                         'FC_HAS_READLINE' => 1,
+                        'AM_MAKE_INCLUDE' => 1,
+                        'FC_CHECK_X_PROTO_DEFINE' => 1,
                         'FC_CHECK_SOUND' => 1,
-                        'FC_CHECK_X_PROTO_DEFINE' => 1,
-                        'AM_MAKE_INCLUDE' => 1,
+                        'LT_CMD_MAX_LEN' => 1,
                         '_LT_AC_TAGCONFIG' => 1,
-                        'LT_CMD_MAX_LEN' => 1,
                         'AC_GGZ_SERVER' => 1,
                         'm4_pattern_forbid' => 1,
+                        '_LT_LINKER_OPTION' => 1,
                         'AC_LIB_PREPARE_PREFIX' => 1,
-                        '_LT_LINKER_OPTION' => 1,
                         'AC_LIBTOOL_COMPILER_OPTION' => 1,
+                        'AC_DISABLE_SHARED' => 1,
                         'gl_PREREQ_MKSTEMP' => 1,
-                        'AC_DISABLE_SHARED' => 1,
                         '_LT_COMPILER_BOILERPLATE' => 1,
+                        'AC_LIBTOOL_SETUP' => 1,
                         'AC_LIBTOOL_WIN32_DLL' => 1,
-                        'AC_LIBTOOL_SETUP' => 1,
+                        'AC_PROG_LD_RELOAD_FLAG' => 1,
                         'AM_GLIB_GNU_GETTEXT' => 1,
-                        'AC_PROG_LD_RELOAD_FLAG' => 1,
                         'AC_LTDL_DLSYM_USCORE' => 1,
+                        'LT_LANG' => 1,
                         'AM_MISSING_HAS_RUN' => 1,
-                        'LT_LANG' => 1,
+                        'LT_SYS_DLSEARCH_PATH' => 1,
                         'AC_GGZ_INIT' => 1,
-                        'LT_SYS_DLSEARCH_PATH' => 1,
                         'LT_CONFIG_LTDL_DIR' => 1,
+                        'LT_OUTPUT' => 1,
+                        'AC_LIBTOOL_DLOPEN_SELF' => 1,
                         'FC_CHECK_READLINE_RUNTIME' => 1,
-                        'AC_LIBTOOL_DLOPEN_SELF' => 1,
-                        'LT_OUTPUT' => 1,
+                        'AC_LIB_PROG_LD_GNU' => 1,
                         'AC_C99_STDINT_H' => 1,
-                        'AC_LIB_PROG_LD_GNU' => 1,
                         'AC_LIBTOOL_PROG_LD_SHLIBS' => 1,
                         '_PKG_SHORT_ERRORS_SUPPORTED' => 1,
+                        'AC_LIBTOOL_LINKER_OPTION' => 1,
                         'AC_WITH_LTDL' => 1,
-                        'AC_LIBTOOL_LINKER_OPTION' => 1,
                         'PKG_CHECK_EXISTS' => 1,
                         'AC_GGZ_VERSION' => 1,
+                        'AC_LIBTOOL_CXX' => 1,
                         'LT_AC_PROG_RC' => 1,
-                        'AC_LIBTOOL_CXX' => 1,
                         'LT_INIT' => 1,
+                        'LT_SYS_DLOPEN_SELF' => 1,
                         'LT_AC_PROG_GCJ' => 1,
-                        'LT_SYS_DLOPEN_SELF' => 1,
+                        'AM_DEP_TRACK' => 1,
                         'AM_DISABLE_STATIC' => 1,
-                        'AM_DEP_TRACK' => 1,
+                        '_AC_PROG_LIBTOOL' => 1,
                         'AM_CONFIG_HEADER' => 1,
-                        '_AC_PROG_LIBTOOL' => 1,
                         '_AM_IF_OPTION' => 1,
                         'AC_PATH_TOOL_PREFIX' => 1,
+                        'AC_LIBTOOL_F77' => 1,
                         'm4_pattern_allow' => 1,
-                        'AC_LIBTOOL_F77' => 1,
                         'AM_SET_LEADING_DOT' => 1,
                         'LT_AC_PROG_EGREP' => 1,
                         '_AM_DEPENDENCIES' => 1,
                         'AC_LIBTOOL_LANG_C_CONFIG' => 1,
+                        'LTOPTIONS_VERSION' => 1,
                         'AC_GGZ_CONFIG' => 1,
-                        'LTOPTIONS_VERSION' => 1,
+                        '_LT_AC_SYS_COMPILER' => 1,
                         'AC_LIB_LINKFLAGS' => 1,
-                        '_LT_AC_SYS_COMPILER' => 1,
                         'AM_PROG_NM' => 1,
                         'PKG_CHECK_MODULES' => 1,
                         'gl_STDLIB_H' => 1,
                         'AC_GGZ_GGZMOD' => 1,
                         'AC_LIBLTDL_CONVENIENCE' => 1,
+                        'AC_DEPLIBS_CHECK_METHOD' => 1,
                         'FC_CHECK_X_PROTO_FETCH' => 1,
-                        'AC_DEPLIBS_CHECK_METHOD' => 1,
+                        'AM_SET_CURRENT_AUTOMAKE_VERSION' => 1,
                         'AC_LIBLTDL_INSTALLABLE' => 1,
-                        'AM_SET_CURRENT_AUTOMAKE_VERSION' => 1,
+                        'AC_LTDL_ENABLE_INSTALL' => 1,
                         'AC_C99_VARIABLE_ARRAYS' => 1,
-                        'AC_LTDL_ENABLE_INSTALL' => 1,
+                        'AC_LIBTOOL_SYS_DYNAMIC_LINKER' => 1,
                         'LT_PROG_GCJ' => 1,
-                        'AC_LIBTOOL_SYS_DYNAMIC_LINKER' => 1,
+                        'AM_GLIB_DEFINE_LOCALEDIR' => 1,
                         'AC_C99_VARIADIC_MACROS' => 1,
-                        'AM_GLIB_DEFINE_LOCALEDIR' => 1,
                         'AM_INIT_AUTOMAKE' => 1,
                         'AC_GGZ_CHECK_SERVER' => 1,
                         'AC_DISABLE_STATIC' => 1,
@@ -650,33 +650,33 @@
                         'LT_SYS_MODULE_PATH' => 1,
                         'AC_GGZ_GTK' => 1,
                         'AC_GGZ_FIND_FILE' => 1,
+                        'AC_LIBTOOL_POSTDEP_PREDEP' => 1,
                         'LT_WITH_LTDL' => 1,
-                        'AC_LIBTOOL_POSTDEP_PREDEP' => 1,
+                        'AC_LTDL_SHLIBPATH' => 1,
                         'gl_STDLIB_H_DEFAULTS' => 1,
-                        'AC_LTDL_SHLIBPATH' => 1,
                         'AM_AUX_DIR_EXPAND' => 1,
                         'AC_LIB_LINKFLAGS_FROM_LIBS' => 1,
+                        '_LT_AC_LANG_F77_CONFIG' => 1,
+                        'AC_LIBTOOL_PROG_COMPILER_NO_RTTI' => 1,
                         'FC_CHECK_X_PROTO_FUNCPROTO_COMPILE' => 1,
-                        'AC_LIBTOOL_PROG_COMPILER_NO_RTTI' => 1,
-                        '_LT_AC_LANG_F77_CONFIG' => 1,
+                        '_AM_SET_OPTIONS' => 1,
                         '_LT_COMPILER_OPTION' => 1,
-                        '_AM_SET_OPTIONS' => 1,
+                        '_AM_OUTPUT_DEPENDENCY_COMMANDS' => 1,
                         'AM_RUN_LOG' => 1,
-                        '_AM_OUTPUT_DEPENDENCY_COMMANDS' => 1,
+                        'AC_LIBTOOL_SYS_OLD_ARCHIVE' => 1,
+                        'AC_LTDL_SYS_DLOPEN_DEPLIBS' => 1,
                         'AC_LIBTOOL_PICMODE' => 1,
-                        'AC_LTDL_SYS_DLOPEN_DEPLIBS' => 1,
-                        'AC_LIBTOOL_SYS_OLD_ARCHIVE' => 1,
                         'AC_LIB_LINKFLAGS_BODY' => 1,
+                        'LT_PATH_LD' => 1,
                         'AC_CHECK_LIBM' => 1,
-                        'LT_PATH_LD' => 1,
+                        'AC_LIBTOOL_SYS_LIB_STRIP' => 1,
                         'AC_GGZ_ERROR' => 1,
-                        'AC_LIBTOOL_SYS_LIB_STRIP' => 1,
                         '_AM_MANGLE_OPTION' => 1,
+                        'AC_LTDL_SYMBOL_USCORE' => 1,
                         'AC_LIBTOOL_SYS_MAX_CMD_LEN' => 1,
-                        'AC_LTDL_SYMBOL_USCORE' => 1,
                         'AM_SET_DEPDIR' => 1,
+                        '_LT_CC_BASENAME' => 1,
                         'PKG_PROG_PKG_CONFIG' => 1,
-                        '_LT_CC_BASENAME' => 1,
                         '_LT_LIBOBJ' => 1,
                         'AC_LIB_PROG_LD' => 1
                       }
