diff -Nurd -X.diff_ignore freeciv/common/packets.def freeciv/common/packets.def
--- freeciv/common/packets.def	2016-02-23 07:21:02.578745732 +0200
+++ freeciv/common/packets.def	2016-02-24 00:57:15.048056132 +0200
@@ -28,6 +28,8 @@
   Start with "type" and have the format "type <alias> = <src>". You
   can now use <alias> at every place a type is expected. Nested type
   defs are possible.
+  freeciv-web build does not like comments on the end of the line.
+  Please put comments to lines of their own.
 
  Packets:
  ========
@@ -309,7 +311,8 @@
 type GOLD               = UINT32
 type HP                 = UINT8
 type PERCENT            = UINT8
-type MOVEFRAGS          = UINT16  # units of TERRAIN_CONTROL.move_fragments
+# units of TERRAIN_CONTROL.move_fragments
+type MOVEFRAGS          = UINT16
 type PHASE              = SINT16
 type TURN               = SINT16
 type XYSIZE             = UINT16
