diff --git freeciv/server/plrhand.c freeciv/server/plrhand.c
index ad373a9..41bbf15 100644
--- freeciv/server/plrhand.c
+++ freeciv/server/plrhand.c
@@ -1042,6 +1042,7 @@ static void package_player_info(struct player *plr,
   enum plr_info_level info_level;
   enum plr_info_level highest_team_level;
   struct government *pgov = NULL;
+  struct research *presearch;
 
   if (receiver) {
     info_level = player_info_level(plr, receiver);
@@ -1127,6 +1128,13 @@ static void package_player_info(struct player *plr,
     }
   }
 
+  presearch = research_get(plr);
+  if (presearch->researching != A_UNSET) {
+    packet->current_research_cost = research_total_bulbs_required(presearch, presearch->researching, FALSE);
+  } else {
+    packet->current_research_cost = 0;
+  }
+
   /* Make absolutely sure - in case you lose your embassy! */
   if (info_level >= INFO_EMBASSY 
       || (receiver
diff --git a/server/techtools.c b/server/techtools.c
index 925d227..abe32ee 100644
--- a/server/techtools.c
+++ b/server/techtools.c
@@ -287,6 +287,7 @@ void send_research_info(const struct research *presearch,
       if (presearch == research_get(pplayer)) {
         /* Case research owner. */
         send_packet_research_info(pconn, &full_info);
+	send_player_info_c(conn_get_player(pconn), conn_get_player(pconn)->connections);
       } else {
         /* 'pplayer' may have an embassy for looking to 'presearch'. */
         research_players_iterate(presearch, powner) {
