Index: common/packets.def
===================================================================
--- common/packets.def	(revisjon 78)
+++ common/packets.def	(arbeidskopi)
@@ -922,7 +922,7 @@
 PACKET_UNIT_DIPLOMAT_ACTION=70;cs,dsend
   UNIT diplomat_id;
   UNIT target_id;   # city_id or unit_id
-  SINT16 value;
+  UINT32 value;
   DIPLOMAT_ACTION action_type;
 end
 
Index: common/packets_gen.c
===================================================================
--- common/packets_gen.c	(revisjon 78)
+++ common/packets_gen.c	(arbeidskopi)
@@ -9589,7 +9589,7 @@
   {
     int readin;
   
-    dio_get_sint16(&din, &readin);
+    dio_get_uint32(&din, &readin);
     real_packet->value = readin;
   }
   {
@@ -9609,7 +9609,7 @@
 
   dio_put_uint32(&dout, real_packet->diplomat_id);
   dio_put_uint32(&dout, real_packet->target_id);
-  dio_put_sint16(&dout, real_packet->value);
+  dio_put_uint32(&dout, real_packet->value);
   dio_put_uint8(&dout, real_packet->action_type);
 
   SEND_PACKET_END;
