diff --git a/Makefile.am b/Makefile.am
index 6c4fa0d..05eef60 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -25,7 +25,7 @@ else
 DSUBDIRS =
 endif
 
-SUBDIRS = $(DSUBDIRS) dependencies utility common tests win32 $(SSUBDIRS) $(CSUBDIRS) translations doc tools
+SUBDIRS = $(DSUBDIRS) dependencies utility common tests win32 $(SSUBDIRS) $(CSUBDIRS) doc tools
 
 ## Automake should distribute config.rpath automatically - and new
 ## versions probably do. We still support versions that do not, so
diff --git a/configure.ac b/configure.ac
index 1b07f3d..28437f7 100644
--- a/configure.ac
+++ b/configure.ac
@@ -573,9 +573,6 @@ int main()
   ]])],[AC_MSG_RESULT([yes])
   ],[
   AC_MSG_RESULT([no])
-  AC_MSG_ERROR([You are unable to create *.gmo files.
-This is development version which doesn't include them. Please update your
-gettext package. Or use --disable-nls instead.])
   ],[echo error
   ])
 fi
