diff --git a/server/generator/mapgen.c b/server/generator/mapgen.c
index 965dbdf..b13608a 100644
--- a/server/generator/mapgen.c
+++ b/server/generator/mapgen.c
@@ -2986,10 +2986,7 @@ static bool fair_map_place_island_rand(struct fair_tile *ptarget,
 
   /* Try hard placement. */
   r = fc_rand(MAP_INDEX_SIZE);
-  for (i = r + 1; i != r; i++) {
-    if (i >= MAP_INDEX_SIZE) {
-      i = 0;
-    }
+  for (i = (r + 1) % MAP_INDEX_SIZE; i != r; i = (i + 1) % MAP_INDEX_SIZE) {
     index_to_map_pos(&x, &y, i);
     if (fair_map_copy(ptarget, x, y, psource, &geometry, -1)) {
       return TRUE;
@@ -3370,10 +3367,11 @@ static struct fair_tile *fair_map_island_new(int size, int startpos_num)
       pftile = pisland + k;
       while (pftile->flags & FTF_NO_HUT) {
         pftile++;
+        if (pftile - pisland == MAP_INDEX_SIZE) {
+          pftile = pisland;
+        }
         if (pftile - pisland == k) {
           break;
-        } else if (pftile - pisland == MAP_INDEX_SIZE) {
-          pftile = pisland;
         }
       }
       if (pftile->flags & FTF_NO_HUT) {
