diff --git freeciv/server/generator/mapgen.c freeciv/server/generator/mapgen.c
index f2c23ac..b45d036 100644
--- freeciv/server/generator/mapgen.c
+++ freeciv/server/generator/mapgen.c
@@ -3607,11 +3607,10 @@ static bool map_generate_fair_islands(void)
           break;
         }
         i += k;
-        team_players_num -= k;
         j++;
       } teams_iterate_end;
 
-      fc_assert(!done || team_players_num == 0);
+      fc_assert(!done || i == team_players_num);
     }
 
     if (done) {
@@ -3622,9 +3621,8 @@ static bool map_generate_fair_islands(void)
           done = FALSE;
           break;
         }
-        single_players_num -= players_per_island;
       }
-      fc_assert(!done || single_players_num == 0);
+      fc_assert(!done || i == player_count());
     }
     fair_map_destroy(pisland);
 
