diff --git a/common/packets.def b/common/packets.def
index b0e07fc..e0ab80b 100644
--- a/common/packets.def
+++ b/common/packets.def
@@ -196,10 +196,10 @@ type UINT32             = uint32(int)
 type MEMORY             = memory(unsigned char)
 type REQUIREMENT        = requirement(struct requirement)
 type STRING             = string(char)
-type TECH_LIST          = tech_list(int)
-type UNIT_LIST          = unit_list(int)
-type BUILDING_LIST      = building_list(int)
-type WORKLIST           = worklist(struct worklist)
+# type TECH_LIST          = tech_list(int)
+# type UNIT_LIST          = unit_list(int)
+# type BUILDING_LIST      = building_list(int)
+type WORKLIST           = string(char)
 # string that is URI encoded in the JSON protocol
 type ESTRING            = estring(char)
 
@@ -293,7 +293,7 @@ type UNIT               = UINT16
 type UNIT_TYPE          = uint8(Unit_type_id)
 
 # other typedefs
-type CITY_COORD         = UINT8
+type CITY_COORD         = UINT32
 type GOLD               = UINT32
 type HP                 = UINT8
 type PERCENT            = UINT8
@@ -422,7 +422,7 @@ PACKET_TILE_INFO = 15; sc, lsend, is-game-info
 
   TERRAIN terrain;
   RESOURCE resource;
-  BV_EXTRAS extras;
+  STRING extras[MAX_EXTRA_TYPES];
   STRING spec_sprite[MAX_LEN_NAME];
   STRING label[MAX_LEN_NAME];
 end
@@ -431,6 +431,7 @@ end
 PACKET_GAME_INFO = 16; sc, is-info
   UINT8 add_to_size_limit;
   UINT8 aifill;
+  UINT32 mapsize;
   PERSISTENT_READY persistent_ready;
   AIRLIFTING_STYLE airlifting_style;
   UINT8 angrycitizen;
@@ -542,6 +543,7 @@ PACKET_GAME_INFO = 16; sc, is-info
   STRING calendar_fragment_name[MAX_CALENDAR_FRAGMENTS][MAX_LEN_NAME];
   BOOL civil_war_enabled;
   BOOL paradrop_to_transport;
+  STRING meta_message[MAX_LEN_NAME];
 end
 
 # This cannot have is-info set. Sending the same value a second time after a
@@ -627,6 +629,9 @@ PACKET_CITY_INFO = 31; sc, lsend, is-game-info, force, cancel(PACKET_CITY_SHORT_
   SINT16 citizen_base[O_LAST];
   SINT16 usage[O_LAST];
   UINT16 food_stock, shield_stock;
+  UINT16 granary_size;
+  TURN granary_turns;
+  UINT16 buy_gold_cost;
 
   UINT8 traderoute_count;
 
@@ -652,11 +657,18 @@ PACKET_CITY_INFO = 31; sc, lsend, is-game-info, force, cancel(PACKET_CITY_SHORT_
   UINT8 walls;
   SINT8 city_image;
 
-  WORKLIST worklist;
+  WORKLIST worklist[MAX_LEN_WORKLIST_WEB];
 
-  BV_IMPRS improvements;
+  STRING can_build_unit[MAX_LEN_MSG];
+  STRING can_build_improvement[MAX_LEN_MSG];
+  STRING improvements[MAX_LEN_MSG];
+
+  STRING food_output[MAX_LEN_MSG];
+  STRING shield_output[MAX_LEN_MSG];
+  STRING trade_output[MAX_LEN_MSG];
   BV_CITY_OPTIONS city_options;
   ESTRING name[MAX_LEN_NAME];
+  BOOL unhappy;
 end
 
 PACKET_CITY_SHORT_INFO = 32; sc, lsend, is-game-info, cancel(PACKET_CITY_INFO)
@@ -675,7 +688,7 @@ PACKET_CITY_SHORT_INFO = 32; sc, lsend, is-game-info, cancel(PACKET_CITY_INFO)
 
   SINT8 city_image;
 
-  BV_IMPRS improvements;
+  STRING improvements[MAX_LEN_MSG];
   ESTRING name[MAX_LEN_NAME];
 end
 
@@ -704,7 +717,7 @@ end
 
 PACKET_CITY_WORKLIST = 36; cs, dsend
   CITY city_id;
-  WORKLIST worklist;
+  #WORKLIST worklist[MAX_LEN_MSG];
 end
 
 PACKET_CITY_MAKE_SPECIALIST = 37; cs, dsend
@@ -808,12 +821,14 @@ PACKET_PLAYER_INFO = 51; sc, is-info
   BV_PLAYER gives_shared_vision;
   UINT16 culture;
   SINT16 love[MAX_NUM_PLAYER_SLOTS];
+  UINT32 current_research_cost;
 
   UINT8 color_red;
   UINT8 color_green;
   UINT8 color_blue;
 
   CITYSPE wonders[B_LAST]; diff
+  UINT32  net_income;
   
   SINT32 multiplier[MAX_NUM_MULTIPLIERS];
   SINT32 multiplier_target[MAX_NUM_MULTIPLIERS];
@@ -1089,25 +1104,25 @@ end
 
 PACKET_DIPLOMACY_CREATE_CLAUSE_REQ = 99; cs, dsend
   PLAYER counterpart, giver;
-  CLAUSE type;
+  CLAUSE clause_type;
   UINT32 value;
 end
 
 PACKET_DIPLOMACY_CREATE_CLAUSE = 100; sc, dsend, lsend
   PLAYER counterpart, giver;
-  CLAUSE type;
+  CLAUSE clause_type;
   UINT32 value;
 end
 
 PACKET_DIPLOMACY_REMOVE_CLAUSE_REQ = 101; cs, dsend
   PLAYER counterpart, giver;
-  CLAUSE type;
+  CLAUSE clause_type;
   UINT32 value;
 end
 
 PACKET_DIPLOMACY_REMOVE_CLAUSE = 102; sc, dsend, lsend
   PLAYER counterpart, giver;
-  CLAUSE type;
+  CLAUSE clause_type;
   UINT32 value;
 end
 
@@ -1331,8 +1346,9 @@ end
 PACKET_RULESET_GAME = 141; sc, lsend
   UINT8 default_specialist;
 
-  TECH_LIST global_init_techs[MAX_NUM_TECH_LIST];
-  BUILDING_LIST global_init_buildings[MAX_NUM_BUILDING_LIST];
+  # These are removed from freeciv-web
+  # TECH_LIST global_init_techs[MAX_NUM_TECH_LIST];
+  # BUILDING_LIST global_init_buildings[MAX_NUM_BUILDING_LIST];
 
   UINT8 veteran_levels;
   STRING veteran_name[MAX_VET_LEVELS:veteran_levels][MAX_LEN_NAME];
@@ -1459,9 +1475,9 @@ PACKET_RULESET_NATION = 148; sc, lsend
   UINT8 groups[MAX_NUM_NATION_GROUPS:ngroups];
 
   GOVERNMENT init_government_id;
-  TECH_LIST init_techs[MAX_NUM_TECH_LIST];
-  UNIT_LIST init_units[MAX_NUM_UNIT_LIST];
-  BUILDING_LIST init_buildings[MAX_NUM_BUILDING_LIST];
+  # TECH_LIST init_techs[MAX_NUM_TECH_LIST];
+  # UNIT_LIST init_units[MAX_NUM_UNIT_LIST];
+  # BUILDING_LIST init_buildings[MAX_NUM_BUILDING_LIST];
 end
 
 # Separate from PACKET_RULESET_NATION since it can change in pregame
@@ -2136,3 +2152,39 @@ end
 PACKET_PLAY_MUSIC = 245; sc, lsend
   STRING tag[MAX_LEN_NAME];
 end
+
+/*************** Webclient specific packets ****************/
+/* Use range 256:511 for these                             */
+# used for showing GOTO path in web client.
+PACKET_GOTO_PATH_REQ = 256; cs, dsend
+  UNIT unit_id;
+  TILE goal;
+end
+
+PACKET_GOTO_PATH = 257; sc, dsend
+  UNIT unit_id;
+  UINT16 length;
+  DIRECTION dir[MAX_LEN_ROUTE:length];
+  TILE dest;
+  UINT32 turns;
+end
+
+PACKET_INFO_TEXT_REQ = 258; cs, dsend
+  TILE loc;
+  UNIT visible_unit;
+end
+
+PACKET_INFO_TEXT_MESSAGE = 259; sc, dsend
+  STRING message[MAX_LEN_MSG];
+end
+
+PACKET_CITY_WORKLIST_ADD = 261; cs, dsend
+  CITY city_id;
+  UINT32 kind;
+  UINT32 value;
+end
+
+PACKET_CITY_WORKLIST_REMOVE = 262; cs, dsend
+  CITY city_id;
+  UINT32 index;
+end
