diff -Nurd -X.diff_ignore freeciv/common/featured_text.c freeciv/common/featured_text.c
--- freeciv/common/featured_text.c	2014-09-11 01:41:13.951730689 +0300
+++ freeciv/common/featured_text.c	2014-09-11 01:41:34.947685408 +0300
@@ -33,8 +33,8 @@
 
 #include "featured_text.h"
 
-#define SEQ_START '['
-#define SEQ_STOP ']'
+#define SEQ_START '<'
+#define SEQ_STOP '>'
 #define SEQ_END '/'
 
 #define MAX_LEN_STR 32
@@ -141,7 +141,7 @@
   case TTT_UNDERLINE:
     return "u";
   case TTT_COLOR:
-    return "c";
+    return "font";
   case TTT_LINK:
     return "l";
   };
@@ -238,7 +238,7 @@
     {
       if (!find_option(sequence, "foreground", ptag->color.foreground,
                        sizeof(ptag->color.foreground))
-          && !find_option(sequence, "fg", ptag->color.foreground,
+          && !find_option(sequence, "color", ptag->color.foreground,
                           sizeof(ptag->color.foreground))) {
         ptag->color.foreground[0] = '\0';
       }
@@ -482,7 +482,7 @@
                                text_tag_type_short_name(ptag->type));
 
       if (ptag->color.foreground[0] != '\0') {
-        ret += fc_snprintf(buf + ret, len - ret, " fg=\"%s\"",
+        ret += fc_snprintf(buf + ret, len - ret, " color=\"%s\"",
                            ptag->color.foreground);
       }
       if (ptag->color.background[0] != '\0') {
diff -Nurd -X.diff_ignore freeciv/server/ruleset.c freeciv/server/ruleset.c
--- freeciv/server/ruleset.c	2014-09-11 01:41:13.955730680 +0300
+++ freeciv/server/ruleset.c	2014-09-11 01:41:34.951685399 +0300
@@ -6272,6 +6272,8 @@
     misc_p.move_bonus[i] = vlevel->move_bonus;
   }
 
+  /* FIXME: removed for webclient. */
+#if 0
   fc_assert(sizeof(misc_p.global_init_techs)
             == sizeof(game.rgame.global_init_techs));
   fc_assert(ARRAY_SIZE(misc_p.global_init_techs)
@@ -6286,6 +6288,7 @@
     misc_p.global_init_buildings[i] =
       game.rgame.global_init_buildings[i];
   }
+#endif
 
   misc_p.default_specialist = DEFAULT_SPECIALIST;
 
