diff -Nurd freeciv/ai/default/aidiplomat.c freeciv/ai/default/aidiplomat.c
--- freeciv/ai/default/aidiplomat.c	2023-03-20 07:14:33.450796239 +0200
+++ freeciv/ai/default/aidiplomat.c	2023-03-20 07:14:46.323059631 +0200
@@ -766,6 +766,11 @@
       || unit_data->task == AIUNIT_DEFEND_HOME) {
     bool failure = FALSE;
 
+    if (punit->goto_tile == NULL) {
+      dai_unit_new_task(ait, punit, AIUNIT_NONE, NULL);
+      return;
+    }
+
     ctarget = tile_city(punit->goto_tile);
     if (pf_map_position(pfm, punit->goto_tile, &pos)
         && ctarget) {
diff -Nurd freeciv/ai/default/daiunit.c freeciv/ai/default/daiunit.c
--- freeciv/ai/default/daiunit.c	2023-03-20 07:14:33.450796239 +0200
+++ freeciv/ai/default/daiunit.c	2023-03-20 07:14:53.451206237 +0200
@@ -2287,9 +2287,16 @@
        || unit_data->task == AIUNIT_WONDER)) {
     /* we are moving to our destination */
     /* we check to see if our current goal is feasible */
-    struct city *city_dest = tile_city(punit->goto_tile);
+    struct city *city_dest = NULL;
 
-    if ((city_dest == NULL) 
+    if (punit->goto_tile != NULL) {
+      city_dest = tile_city(punit->goto_tile);
+    } else {
+      dai_unit_new_task(ait, punit, AIUNIT_NONE, NULL);
+      city_dest = NULL;
+    }
+
+    if (city_dest == NULL
         || !pplayers_allied(unit_owner(punit), city_dest->owner)
         || (unit_data->task == AIUNIT_TRADE
             && !(can_cities_trade(homecity, city_dest)
