diff -Nurd -X.diff_ignore freeciv/common/packets.def freeciv/common/packets.def
--- freeciv/common/packets.def	2016-02-24 00:36:52.946368017 +0200
+++ freeciv/common/packets.def	2016-02-24 00:38:11.130306360 +0200
@@ -200,10 +200,10 @@
 # Marking it this way is useful as documentation. The marking can also be
 # used in non vanilla generate_packets.py packet generators.
 type STRVEC             = STRING
-type TECH_LIST          = tech_list(int)
-type UNIT_LIST          = unit_list(int)
-type BUILDING_LIST      = building_list(int)
-type WORKLIST           = worklist(struct worklist)
+# type TECH_LIST          = tech_list(int)
+# type UNIT_LIST          = unit_list(int)
+# type BUILDING_LIST      = building_list(int)
+type WORKLIST           = string(char)
 # string that is URI encoded in the JSON protocol
 type ESTRING            = estring(char)
 
@@ -297,7 +297,7 @@
 type UNIT_TYPE          = uint8(Unit_type_id)
 
 # other typedefs
-type CITY_COORD         = UINT8
+type CITY_COORD         = UINT32
 type GOLD               = UINT32
 type HP                 = UINT8
 type PERCENT            = UINT8
@@ -632,6 +632,9 @@
   SINT16 citizen_base[O_LAST];
   SINT16 usage[O_LAST];
   UINT16 food_stock, shield_stock;
+  UINT16 granary_size;
+  TURN granary_turns;
+  UINT16 buy_gold_cost;
 
   UINT8 traderoute_count;
 
@@ -657,11 +660,18 @@
   UINT8 walls;
   SINT8 city_image;
 
-  WORKLIST worklist;
+  WORKLIST worklist[MAX_LEN_WORKLIST_WEB];
 
+  STRING can_build_unit[MAX_LEN_MSG];
+  STRING can_build_improvement[MAX_LEN_MSG];
   BV_IMPRS improvements;
+
+  STRING food_output[MAX_LEN_MSG];
+  STRING shield_output[MAX_LEN_MSG];
+  STRING trade_output[MAX_LEN_MSG];
   BV_CITY_OPTIONS city_options;
   ESTRING name[MAX_LEN_NAME];
+  BOOL unhappy;
 end
 
 PACKET_CITY_SHORT_INFO = 32; sc, lsend, is-game-info, cancel(PACKET_CITY_INFO)
@@ -709,7 +719,7 @@
 
 PACKET_CITY_WORKLIST = 36; cs, dsend
   CITY city_id;
-  WORKLIST worklist;
+  #WORKLIST worklist[MAX_LEN_MSG];
 end
 
 PACKET_CITY_MAKE_SPECIALIST = 37; cs, dsend
@@ -813,12 +823,14 @@
   BV_PLAYER gives_shared_vision;
   UINT16 culture;
   SINT16 love[MAX_NUM_PLAYER_SLOTS];
+  UINT32 current_research_cost;
 
   UINT8 color_red;
   UINT8 color_green;
   UINT8 color_blue;
 
   CITYSPE wonders[B_LAST]; diff
+  UINT32  net_income;
   
   SINT32 multiplier[MAX_NUM_MULTIPLIERS];
   SINT32 multiplier_target[MAX_NUM_MULTIPLIERS];
@@ -1334,8 +1346,9 @@
 PACKET_RULESET_GAME = 141; sc, lsend
   UINT8 default_specialist;
 
-  TECH_LIST global_init_techs[MAX_NUM_TECH_LIST];
-  BUILDING_LIST global_init_buildings[MAX_NUM_BUILDING_LIST];
+  # These are removed from freeciv-web
+  # TECH_LIST global_init_techs[MAX_NUM_TECH_LIST];
+  # BUILDING_LIST global_init_buildings[MAX_NUM_BUILDING_LIST];
 
   UINT8 veteran_levels;
   STRING veteran_name[MAX_VET_LEVELS:veteran_levels][MAX_LEN_NAME];
@@ -1462,9 +1475,9 @@
   UINT8 groups[MAX_NUM_NATION_GROUPS:ngroups];
 
   GOVERNMENT init_government_id;
-  TECH_LIST init_techs[MAX_NUM_TECH_LIST];
-  UNIT_LIST init_units[MAX_NUM_UNIT_LIST];
-  BUILDING_LIST init_buildings[MAX_NUM_BUILDING_LIST];
+  # TECH_LIST init_techs[MAX_NUM_TECH_LIST];
+  # UNIT_LIST init_units[MAX_NUM_UNIT_LIST];
+  # BUILDING_LIST init_buildings[MAX_NUM_BUILDING_LIST];
 end
 
 # Separate from PACKET_RULESET_NATION since it can change in pregame
@@ -2139,3 +2152,39 @@
 PACKET_PLAY_MUSIC = 245; sc, lsend
   STRING tag[MAX_LEN_NAME];
 end
+
+/*************** Webclient specific packets ****************/
+/* Use range 256:511 for these                             */
+# used for showing GOTO path in web client.
+PACKET_GOTO_PATH_REQ = 256; cs, dsend
+  UNIT unit_id;
+  TILE goal;
+end
+
+PACKET_GOTO_PATH = 257; sc, dsend
+  UNIT unit_id;
+  UINT16 length;
+  DIRECTION dir[MAX_LEN_ROUTE:length];
+  TILE dest;
+  UINT32 turns;
+end
+
+PACKET_INFO_TEXT_REQ = 258; cs, dsend
+  TILE loc;
+  UNIT visible_unit;
+end
+
+PACKET_INFO_TEXT_MESSAGE = 259; sc, dsend
+  STRING message[MAX_LEN_MSG];
+end
+
+PACKET_CITY_WORKLIST_ADD = 261; cs, dsend
+  CITY city_id;
+  UINT32 kind;
+  UINT32 value;
+end
+
+PACKET_CITY_WORKLIST_REMOVE = 262; cs, dsend
+  CITY city_id;
+  UINT32 index;
+end
