diff -Nurd -X.diff_ignore freeciv/common/connection.c freeciv/common/connection.c
--- freeciv/common/connection.c	2012-07-17 02:35:58.000000000 +0300
+++ freeciv/common/connection.c	2012-07-17 03:49:42.000000000 +0300
@@ -290,6 +290,8 @@
   }
 
   pconn->statistics.bytes_send += len;
+
+#if 0
   if (0 < pconn->send_buffer->do_buffer_sends) {
     flush_connection_send_buffer_packets(pconn);
     if (!add_connection_data(pconn, data, len)) {
@@ -298,7 +300,9 @@
       return FALSE;
     }
     flush_connection_send_buffer_packets(pconn);
-  } else {
+  } else
+#endif
+  {
     flush_connection_send_buffer_all(pconn);
     if (!add_connection_data(pconn, data, len)) {
       log_verbose("cut connection %s due to huge send buffer (2)",
diff -Nurd -X.diff_ignore freeciv/common/connection.h freeciv/common/connection.h
--- freeciv/common/connection.h	2012-07-17 02:35:58.000000000 +0300
+++ freeciv/common/connection.h	2012-07-17 03:49:42.000000000 +0300
@@ -26,7 +26,7 @@
 #include <sys/time.h>
 #endif
 
-#define USE_COMPRESSION
+//#define USE_COMPRESSION   /* webclient doesn't like compression*/
 
 /**************************************************************************
   The connection struct and related stuff.
diff -Nurd -X.diff_ignore freeciv/common/featured_text.c freeciv/common/featured_text.c
--- freeciv/common/featured_text.c	2012-07-17 03:43:50.000000000 +0300
+++ freeciv/common/featured_text.c	2012-07-17 03:49:42.000000000 +0300
@@ -33,8 +33,8 @@
 
 #include "featured_text.h"
 
-#define SEQ_START '['
-#define SEQ_STOP ']'
+#define SEQ_START '<'
+#define SEQ_STOP '>'
 #define SEQ_END '/'
 
 #define MAX_LEN_STR 32
@@ -134,7 +134,7 @@
   case TTT_UNDERLINE:
     return "u";
   case TTT_COLOR:
-    return "c";
+    return "font";
   case TTT_LINK:
     return "l";
   };
@@ -231,7 +231,7 @@
     {
       if (!find_option(sequence, "foreground", ptag->color.foreground,
                        sizeof(ptag->color.foreground))
-          && !find_option(sequence, "fg", ptag->color.foreground,
+          && !find_option(sequence, "color", ptag->color.foreground,
                           sizeof(ptag->color.foreground))) {
         ptag->color.foreground[0] = '\0';
       }
@@ -475,7 +475,7 @@
                                text_tag_type_short_name(ptag->type));
 
       if (ptag->color.foreground[0] != '\0') {
-        ret += fc_snprintf(buf + ret, len - ret, " fg=\"%s\"",
+        ret += fc_snprintf(buf + ret, len - ret, " color=\"%s\"",
                            ptag->color.foreground);
       }
       if (ptag->color.background[0] != '\0') {
diff -Nurd -X.diff_ignore freeciv/common/generate_packets.py freeciv/common/generate_packets.py
--- freeciv/common/generate_packets.py	2012-07-17 02:35:58.000000000 +0300
+++ freeciv/common/generate_packets.py	2012-07-17 03:49:42.000000000 +0300
@@ -27,7 +27,7 @@
 
 ### The following parameters CHANGE the protocol. You have been warned.
 fold_bool_into_header=1
-disable_delta=0
+disable_delta=1
 
 ################# END OF PARAMETERS ####################
 
diff -Nurd -X.diff_ignore freeciv/common/packets.def freeciv/common/packets.def
--- freeciv/common/packets.def	2012-07-17 03:49:35.000000000 +0300
+++ freeciv/common/packets.def	2012-07-17 03:50:40.000000000 +0300
@@ -171,7 +171,7 @@
 type SINT16             = sint16(int)
 type SINT32             = sint32(int)
 type UINT8              = uint8(int)
-type UINT16             = uint16(int)
+type UINT16             = uint32(int)
 type UINT32             = uint32(int)
 
 # typedefs for arrays/structs
@@ -179,7 +179,8 @@
 type MEMORY             = memory(unsigned char)
 type REQUIREMENT        = requirement(struct requirement)
 type STRING             = string(char)
-type TECH_LIST          = tech_list(int)
+# The TECH_LIST has been removed for the web client.  -Andreas
+#type TECH_LIST          = tech_list(int)
 type WORKLIST           = worklist(struct worklist)
 
 # typedefs for enums
@@ -194,7 +195,7 @@
 type DIPLOMAT_ACTION    = uint8(enum diplomat_actions)
 type DIRECTION          = uint8(enum direction8)
 type EFFECT_TYPE        = uint8(enum effect_type)
-type EVENT              = sint16(enum event_type)
+type EVENT              = sint32(enum event_type)
 type GUI_TYPE           = uint8(enum gui_type)
 type IMPR_GENUS         = uint8(enum impr_genus_id)
 type IMPR_RANGE         = uint8(enum impr_range)
@@ -206,7 +207,7 @@
 type REQ_RANGE          = uint8(enum req_range)
 type REQ_TYPE           = uint8(enum universals_n)
 type RIVER_MOVE         = uint8(enum special_river_move)
-type SPECIAL            = uint16(enum tile_special_type)
+type SPECIAL            = uint32(enum tile_special_type)
 type SSET_CLASS         = uint8(enum sset_class)
 type SSET_TYPE          = uint8(enum sset_type)
 
@@ -234,7 +235,7 @@
 type EDIT_TECH_MODE     = uint8(enum editor_tech_mode)
 type GOVERNMENT         = UINT8
 type IMPROVEMENT        = uint8(Impr_type_id)
-type NATION             = sint16(Nation_type_id)
+type NATION             = UINT32
 type PLAYER             = SINT8
 type RESOURCE           = uint8(Resource_type_id)
 type SPECIALIST         = uint8(Specialist_type_id)
@@ -1168,7 +1169,7 @@
 PACKET_RULESET_GAME = 141; sc, lsend
   UINT8 default_specialist;
 
-  TECH_LIST global_init_techs[MAX_NUM_TECH_LIST];
+  #TECH_LIST global_init_techs[MAX_NUM_TECH_LIST]; - Removed for web client. - Andreas
 
   UINT8 veteran_levels;
   STRING veteran_name[MAX_VET_LEVELS:veteran_levels][MAX_LEN_NAME];
@@ -1439,7 +1440,7 @@
   UINT16 id; key
   STRING name[MAX_LEN_NAME];
   STRING short_help[MAX_LEN_PACKET];
-  STRING extra_help[MAX_LEN_PACKET];
+  # STRING extra_help[MAX_LEN_PACKET];
   UINT8 category;
 end
 
diff -Nurd -X.diff_ignore freeciv/data/amplio/ancientcities.spec freeciv/data/amplio/ancientcities.spec
--- freeciv/data/amplio/ancientcities.spec	2012-07-17 02:35:52.000000000 +0300
+++ freeciv/data/amplio/ancientcities.spec	2012-07-17 03:49:42.000000000 +0300
@@ -25,8 +25,7 @@
     Polynesian style by CapTVK
     Celtic style by Erwan, adapted to 96x48 by CapTVK
     Roman style by CapTVK
-    City walls by Hogne Håskjold
-"
+    City walls by Hogne Håskjold"
 
 [file]
 gfx = "amplio/ancientcities"
diff -Nurd -X.diff_ignore freeciv/data/amplio/explosions.spec freeciv/data/amplio/explosions.spec
--- freeciv/data/amplio/explosions.spec	2012-07-17 02:35:52.000000000 +0300
+++ freeciv/data/amplio/explosions.spec	2012-07-17 03:49:42.000000000 +0300
@@ -10,8 +10,7 @@
     Explosions from xgalaga <http://rumsey.org/xgal.html>
     Credits:
     Joe Rumsey <ogre@atomic.com>
-    Mike McGrath <mcgrath@epx.cis.umn.edu>
-"
+    Mike McGrath <mcgrath@epx.cis.umn.edu>"
 
 [file]
 gfx = "amplio/explosions"
@@ -28,5 +27,4 @@
   1,  0, "explode.unit_1"
   2,  0, "explode.unit_2"
   3,  0, "explode.unit_3"
-  4,  0, "explode.unit_4"
-}
+  4,  0, "explode.unit_4"}
diff -Nurd -X.diff_ignore freeciv/data/amplio/fog.spec freeciv/data/amplio/fog.spec
--- freeciv/data/amplio/fog.spec	2012-07-17 02:35:52.000000000 +0300
+++ freeciv/data/amplio/fog.spec	2012-07-17 03:49:42.000000000 +0300
@@ -3,8 +3,7 @@
 options = "+Freeciv-2.3-spec"
 
 artists = "
-    Brendon Oram (Yautja)
-"
+    Brendon Oram (Yautja)"
 
 [file]
 gfx = "amplio/fog"
@@ -98,5 +97,4 @@
   8,  5, "t.fog_u_u_k_u"
   8,  6, "t.fog_u_u_u_f"
   8,  7, "t.fog_u_u_u_k"
-  8,  8, "t.fog_u_u_u_u"
-}
+  8,  8, "t.fog_u_u_u_u"}
diff -Nurd -X.diff_ignore freeciv/data/amplio/grid.spec freeciv/data/amplio/grid.spec
--- freeciv/data/amplio/grid.spec	2012-07-17 02:35:52.000000000 +0300
+++ freeciv/data/amplio/grid.spec	2012-07-17 03:49:42.000000000 +0300
@@ -9,8 +9,7 @@
 artists = "
     Jason Dorje Short <jdorje@freeciv.org>
     Hogne Håskjold    <hogne@freeciv.org>
-    Eleazar / J. W. Bjerk <www.jwbjerk.com>
-"
+    Eleazar / J. W. Bjerk <www.jwbjerk.com>"
 
 [file]
 gfx = "amplio/grid"
@@ -39,5 +38,4 @@
   2, 0, "grid.borders.n"
   2, 1, "grid.borders.s"
   2, 2, "grid.borders.w"
-  2, 3, "grid.borders.e"
-}
+  2, 3, "grid.borders.e"}
diff -Nurd -X.diff_ignore freeciv/data/amplio/icons.spec freeciv/data/amplio/icons.spec
--- freeciv/data/amplio/icons.spec	2012-07-17 02:35:52.000000000 +0300
+++ freeciv/data/amplio/icons.spec	2012-07-17 03:49:42.000000000 +0300
@@ -9,8 +9,7 @@
 artists = "
     Hogne Håskjold <hogne@freeciv.org>
     civicon: Made from the globe in the intro picture
-    cityicon: City gfx by smiley
-"
+    cityicon: City gfx by smiley"
 
 [extra]
 sprites = 
diff -Nurd -X.diff_ignore freeciv/data/amplio/medievalcities.spec freeciv/data/amplio/medievalcities.spec
--- freeciv/data/amplio/medievalcities.spec	2012-07-17 02:35:52.000000000 +0300
+++ freeciv/data/amplio/medievalcities.spec	2012-07-17 03:49:42.000000000 +0300
@@ -23,8 +23,7 @@
 artists = "
     Medieval by CapTVK
     Steam Age by Smiley, www.firstcultural.com
-    City walls by Hogne Håskjold
-"
+    City walls by Hogne Håskjold"
 
 [file]
 gfx = "amplio/medievalcities"
diff -Nurd -X.diff_ignore freeciv/data/amplio/moderncities.spec freeciv/data/amplio/moderncities.spec
--- freeciv/data/amplio/moderncities.spec	2012-07-17 02:35:52.000000000 +0300
+++ freeciv/data/amplio/moderncities.spec	2012-07-17 03:49:42.000000000 +0300
@@ -25,8 +25,7 @@
     Modern, Post Modern and Electric Age by Smiley, www.firstcultural.com
     City walls by Hogne Håskjold
     Buoy by Eleazar
-    Ruins by Vincent Croisier
-"
+    Ruins by Vincent Croisier"
 
 [file]
 gfx = "amplio/moderncities"
diff -Nurd -X.diff_ignore freeciv/data/amplio/nuke.spec freeciv/data/amplio/nuke.spec
--- freeciv/data/amplio/nuke.spec	2012-07-17 02:35:52.000000000 +0300
+++ freeciv/data/amplio/nuke.spec	2012-07-17 03:49:42.000000000 +0300
@@ -7,8 +7,7 @@
 
 artists = "
     Tatu Rissanen <tatu.rissanen@hut.fi>
-    Jeff Mallatt <jjm@codewell.com> (miscellaneous)
-"
+    Jeff Mallatt <jjm@codewell.com> (miscellaneous)"
 
 [file]
 gfx = "amplio/nuke"
@@ -22,5 +21,4 @@
 pixel_border = 0
 
 tiles = { "row", "column", "tag"
- 0, 0, "explode.nuke"
-}
+ 0, 0, "explode.nuke"}
diff -Nurd -X.diff_ignore freeciv/data/amplio/ocean.spec freeciv/data/amplio/ocean.spec
--- freeciv/data/amplio/ocean.spec	2012-07-17 02:35:52.000000000 +0300
+++ freeciv/data/amplio/ocean.spec	2012-07-17 03:49:42.000000000 +0300
@@ -10,8 +10,7 @@
     Yautja
     CapTVK
     Eleazar
-    William Allen Simpson
-"
+    William Allen Simpson"
 
 [file]
 gfx = "amplio/ocean"
@@ -107,5 +106,4 @@
   0, 3, "t.l0.cellgroup_l_l_l_s"
   0, 2, "t.l0.cellgroup_d_l_l_l"
   0, 1, "t.l0.cellgroup_s_l_l_l"
-  0, 0, "t.l0.cellgroup_l_l_l_l"
-}
+  0, 0, "t.l0.cellgroup_l_l_l_l"}
diff -Nurd -X.diff_ignore freeciv/data/amplio/select.spec freeciv/data/amplio/select.spec
--- freeciv/data/amplio/select.spec	2012-07-17 02:35:52.000000000 +0300
+++ freeciv/data/amplio/select.spec	2012-07-17 03:49:42.000000000 +0300
@@ -8,8 +8,7 @@
 artists = "
     Jason Dorje Short <jdorje@freeciv.org>
     Eleazar / J. W. Bjerk <www.jwbjerk.com>
-    qwm
-"
+    qwm"
 
 [file]
 gfx = "amplio/select-alpha"
@@ -26,5 +25,4 @@
   0, 0, "unit.select0"
   0, 1, "unit.select1"
   0, 2, "unit.select2"
-  0, 3, "unit.select3"
-}
+  0, 3, "unit.select3"}
diff -Nurd -X.diff_ignore freeciv/data/amplio/terrain1.spec freeciv/data/amplio/terrain1.spec
--- freeciv/data/amplio/terrain1.spec	2012-07-17 02:35:52.000000000 +0300
+++ freeciv/data/amplio/terrain1.spec	2012-07-17 03:49:42.000000000 +0300
@@ -10,8 +10,7 @@
     Hogne Håskjold
     Daniel Speyer
     Yautja
-    CapTVK
-"
+    CapTVK"
 
 [file]
 gfx = "amplio/terrain1"
@@ -127,5 +126,4 @@
   7,  0, "t.blend.arctic" ;ice over neighbors
  15,  3, "t.blend.coast"
  15,  4, "user.attention"
- 15,  5, "tx.fog"
-}
+ 15,  5, "tx.fog"}
diff -Nurd -X.diff_ignore freeciv/data/amplio/terrain2.spec freeciv/data/amplio/terrain2.spec
--- freeciv/data/amplio/terrain2.spec	2012-07-17 02:35:52.000000000 +0300
+++ freeciv/data/amplio/terrain2.spec	2012-07-17 03:49:42.000000000 +0300
@@ -11,8 +11,7 @@
     Tim F. Smith <yoohootim@hotmail.com>
     Yautja
     Daniel Speyer
-    Eleazar
-"
+    Eleazar"
 
 [file]
 gfx = "amplio/terrain2"
@@ -82,9 +81,7 @@
  9,  4, "t.l1.hills_n1e1s0w0"
  9,  5, "t.l1.hills_n1e1s0w1"
  9,  6, "t.l1.hills_n1e1s1w0"
- 9,  7, "t.l1.hills_n1e1s1w1"
-
-}
+ 9,  7, "t.l1.hills_n1e1s1w1"}
 
 
 [grid_coasts]
@@ -97,6 +94,4 @@
 
 tiles = { "row", "column","tag"
 
-;* previous coordinates now in water.spec
-
-}
+;* previous coordinates now in water.spec}
diff -Nurd -X.diff_ignore freeciv/data/amplio/tiles.spec freeciv/data/amplio/tiles.spec
--- freeciv/data/amplio/tiles.spec	2012-07-17 02:35:52.000000000 +0300
+++ freeciv/data/amplio/tiles.spec	2012-07-17 03:49:42.000000000 +0300
@@ -8,8 +8,7 @@
 
 artists = "
     Tatu Rissanen <tatu.rissanen@hut.fi>
-    Jeff Mallatt <jjm@codewell.com> (miscellaneous)
-"
+    Jeff Mallatt <jjm@codewell.com> (miscellaneous)"
 
 [file]
 gfx = "amplio/tiles"
@@ -24,8 +23,8 @@
 tiles = { "row", "column", "tag"
 ; Unit activity letters:  (note unit icons have just "u.")
 
-  0, 18, "unit.auto_attack",
-         "unit.auto_settler"
+  0, 18, "unit.auto_attack"
+  0, 18, "unit.auto_settler"
   0, 19, "unit.stack"
   1, 17, "unit.loaded"
   1, 18, "unit.connect"
@@ -37,8 +36,8 @@
   2, 16, "unit.pollution"
   2, 17, "unit.road"
   2, 18, "unit.irrigate"
-  2, 19, "unit.fortifying",
-         "unit.fortress"
+  2, 19, "unit.fortifying"
+  2, 19, "unit.fortress"
   3, 16, "unit.airbase"
   3, 17, "unit.pillage"
   3, 18, "unit.fortified"
@@ -149,6 +148,4 @@
   1,  2, "explode.nuke_12"
   2,  0, "explode.nuke_20"
   2,  1, "explode.nuke_21"
-  2,  2, "explode.nuke_22"
-
-}
+  2,  2, "explode.nuke_22"}
diff -Nurd -X.diff_ignore freeciv/data/amplio/units.spec freeciv/data/amplio/units.spec
--- freeciv/data/amplio/units.spec	2012-07-17 02:35:52.000000000 +0300
+++ freeciv/data/amplio/units.spec	2012-07-17 03:49:42.000000000 +0300
@@ -32,8 +32,7 @@
     Vodvakov
     J. W. Bjerk / Eleazar <www.jwbjerk.com>
     qwm
-    FiftyNine
-"
+    FiftyNine"
 
 [file]
 gfx = "amplio/units"
@@ -46,7 +45,6 @@
 dy = 48
 
 tiles = { "row", "column", "tag"
-				; Scenario League tags in brackets
   0,  0, "u.armor"		; [Nemo]
   0,  1, "u.howitzer"		; [Nemo]
   0,  2, "u.battleship"		; [Nemo]
@@ -118,5 +116,4 @@
   3, 8, "unit.vet_9"
 
   3, 11, "unit.lowfuel"
-  3, 11, "unit.tired"
-}
+  3, 11, "unit.tired"}
diff -Nurd -X.diff_ignore freeciv/data/amplio/water.spec freeciv/data/amplio/water.spec
--- freeciv/data/amplio/water.spec	2012-07-17 02:35:52.000000000 +0300
+++ freeciv/data/amplio/water.spec	2012-07-17 03:49:42.000000000 +0300
@@ -11,8 +11,7 @@
     Tim F. Smith <yoohootim@hotmail.com>
     Yautja
     Daniel Speyer
-    Eleazar
-"
+    Eleazar"
 
 [file]
 gfx = "amplio/water"
@@ -71,8 +70,7 @@
  10, 0, "tx.river_outlet_n"
  10, 1, "tx.river_outlet_e"
  10, 2, "tx.river_outlet_s"
- 10, 3, "tx.river_outlet_w"
-}
+ 10, 3, "tx.river_outlet_w"}
 
 
 [grid_coasts]
@@ -193,5 +191,4 @@
  11, 9,  "t.l0.lake_cell_r_s_s_l"
  11, 11, "t.l0.lake_cell_r_l_s_l"
  11, 13, "t.l0.lake_cell_r_s_l_l"
- 11, 15, "t.l0.lake_cell_r_l_l_l"
-}
+ 11, 15, "t.l0.lake_cell_r_l_l_l"}
diff -Nurd -X.diff_ignore freeciv/data/amplio.tilespec freeciv/data/amplio.tilespec
--- freeciv/data/amplio.tilespec	2012-07-17 02:35:37.000000000 +0300
+++ freeciv/data/amplio.tilespec	2012-07-17 03:49:42.000000000 +0300
@@ -94,7 +94,7 @@
 
 
 ; Include color definitions
-*include "misc/colors.tilespec"
+;*include "misc/colors.tilespec"
 
 ; Terrain info - see README.graphics
 
diff -Nurd -X.diff_ignore freeciv/data/misc/buildings-large.spec freeciv/data/misc/buildings-large.spec
--- freeciv/data/misc/buildings-large.spec	2012-07-17 02:35:27.000000000 +0300
+++ freeciv/data/misc/buildings-large.spec	2012-07-17 03:49:42.000000000 +0300
@@ -7,8 +7,8 @@
 [info]
 
 artists = "
-   feler404
-"
+   feler404"
+
 [extra]
 sprites =
 	{	"tag", "file"
diff -Nurd -X.diff_ignore freeciv/data/misc/buildings.spec freeciv/data/misc/buildings.spec
--- freeciv/data/misc/buildings.spec	2012-07-17 02:35:27.000000000 +0300
+++ freeciv/data/misc/buildings.spec	2012-07-17 03:49:42.000000000 +0300
@@ -9,8 +9,7 @@
     Tim F. Smith <yoohootim@hotmail.com>
     Frederic Rodrigo <f.rodrigo@tuxfamily.org>
     Dusan Hukuv <dusan@mirc.cz>
-    Eleazar / J. W. Bjerk <www.jwbjerk.com>
-"
+    Eleazar / J. W. Bjerk <www.jwbjerk.com>"
 
 [file]
 gfx = "misc/buildings"
@@ -103,5 +102,4 @@
   8,  4, "b.statue_of_liberty"
   8,  5, "b.sun_tzus_war_academy"
   8,  6, "b.united_nations"
-  8,  7, "b.womens_suffrage"
-}
+  8,  7, "b.womens_suffrage"}
diff -Nurd -X.diff_ignore freeciv/data/misc/chiefs.spec freeciv/data/misc/chiefs.spec
--- freeciv/data/misc/chiefs.spec	2012-07-17 02:35:27.000000000 +0300
+++ freeciv/data/misc/chiefs.spec	2012-07-17 03:49:42.000000000 +0300
@@ -6,8 +6,7 @@
 [info]
 
 artists = "
-    Falk Hueffner <falk.hueffner@student.uni-tuebingen.de>
-"
+    Falk Hueffner <falk.hueffner@student.uni-tuebingen.de>"
 
 [file]
 gfx = "misc/chiefs"
@@ -20,18 +19,14 @@
 dy = 45
 
 tiles = { "row", "column", "tag"
-
-; Unit hit-point bars: approx percent of hp remaining
-0,  0, "unit.hp_100"
-0,  1, "unit.hp_90"
-0,  2, "unit.hp_80"
-0,  3, "unit.hp_70"
-0,  4, "unit.hp_60"
-0,  5, "unit.hp_50"
-0,  6, "unit.hp_40"
-0,  7, "unit.hp_30"
-0,  8, "unit.hp_20"
-0,  9, "unit.hp_10"
-0, 10, "unit.hp_0"
-
-}
+ 0,  0, "unit.hp_100"
+ 0,  1, "unit.hp_90"
+ 0,  2, "unit.hp_80"
+ 0,  3, "unit.hp_70"
+ 0,  4, "unit.hp_60"
+ 0,  5, "unit.hp_50"
+ 0,  6, "unit.hp_40"
+ 0,  7, "unit.hp_30"
+ 0,  8, "unit.hp_20"
+ 0,  9, "unit.hp_10"
+ 0, 10, "unit.hp_0"}
diff -Nurd -X.diff_ignore freeciv/data/misc/citybar.spec freeciv/data/misc/citybar.spec
--- freeciv/data/misc/citybar.spec	2012-07-17 02:35:27.000000000 +0300
+++ freeciv/data/misc/citybar.spec	2012-07-17 03:49:42.000000000 +0300
@@ -8,8 +8,7 @@
 
 artists = "
      Freim <...>
-     Madeline Book <madeline.book@gmail.com> (citybar.trade)
-"
+     Madeline Book <madeline.book@gmail.com> (citybar.trade)"
 
 [file]
 gfx = "misc/citybar"
@@ -26,9 +25,7 @@
 
   0,  0, "citybar.shields"
   0,  1, "citybar.food"
-  0,  2, "citybar.trade"
-
-}
+  0,  2, "citybar.trade"}
 
 [grid_star]
 
@@ -44,9 +41,7 @@
   0,  1, "citybar.occupancy_0"
   0,  2, "citybar.occupancy_1"
   0,  3, "citybar.occupancy_2"
-  0,  4, "citybar.occupancy_3"
-
-}
+  0,  4, "citybar.occupancy_3"}
 
 [grid_bg]
 x_top_left = 1
@@ -56,7 +51,4 @@
 dy = 152
 
 tiles = { "row", "column", "tag"
-
-  0,  0, "citybar.background"
-
-}
+  0,  0, "citybar.background"}
diff -Nurd -X.diff_ignore freeciv/data/misc/cursors.spec freeciv/data/misc/cursors.spec
--- freeciv/data/misc/cursors.spec	2012-07-17 02:35:27.000000000 +0300
+++ freeciv/data/misc/cursors.spec	2012-07-17 03:49:42.000000000 +0300
@@ -11,9 +11,7 @@
 
     edit_paint cursor take from GIMP
 
-    Hourglass and Add from Bluecurve theme in redhat-artwork
-    
-"
+    Hourglass and Add from Bluecurve theme in redhat-artwork"
 
 [file]
 gfx = "misc/cursors"
@@ -86,5 +84,4 @@
        2, 9, "cursor.wait2", 16, 16
        3, 9, "cursor.wait3", 16, 16
        4, 9, "cursor.wait4", 16, 16
-       5, 9, "cursor.wait5", 16, 16
-}
+       5, 9, "cursor.wait5", 16, 16}
diff -Nurd -X.diff_ignore freeciv/data/misc/editor.spec freeciv/data/misc/editor.spec
--- freeciv/data/misc/editor.spec	2012-07-17 02:35:27.000000000 +0300
+++ freeciv/data/misc/editor.spec	2012-07-17 03:49:42.000000000 +0300
@@ -7,8 +7,7 @@
 [info]
 
 artists = "
-	Pep
-"
+	Pep"
 
 [file]
 gfx = "misc/editor"
@@ -47,8 +46,7 @@
   3,  6, "editor.fill"
   3,  7, "editor.tiles_1"
   3,  8, "editor.tiles_2"
-  3,  9, "editor.terrain_resource_2"
-}
+  3,  9, "editor.terrain_resource_2"}
 
 [grid_mini]
 
@@ -72,8 +70,7 @@
   0, 10, "editor.mini_j" 
   0, 11, "editor.mini_gl"
   0, 12, "editor.mini_erase"
-  0, 13, "editor.mini_freeciv"
-}
+  0, 13, "editor.mini_freeciv"}
 
 [grid_extra]
 
@@ -90,5 +87,4 @@
   0,  3, "editor.type_paint"
   0,  4, "editor.bitmap_to_map"
   0,  5, "editor.copy"
-  0,  6, "editor.paste"
-}
+  0,  6, "editor.paste"}
diff -Nurd -X.diff_ignore freeciv/data/misc/icons.spec freeciv/data/misc/icons.spec
--- freeciv/data/misc/icons.spec	2012-07-17 02:35:27.000000000 +0300
+++ freeciv/data/misc/icons.spec	2012-07-17 03:49:42.000000000 +0300
@@ -7,8 +7,7 @@
 [info]
 
 artists = "
-    James Blewitt (civicon)
-"
+    James Blewitt (civicon)"
 
 [extra]
 sprites = 
diff -Nurd -X.diff_ignore freeciv/data/misc/overlays.spec freeciv/data/misc/overlays.spec
--- freeciv/data/misc/overlays.spec	2012-07-17 02:35:27.000000000 +0300
+++ freeciv/data/misc/overlays.spec	2012-07-17 03:49:42.000000000 +0300
@@ -7,8 +7,7 @@
 [info]
 
 artists = "
-    Jason Dorje Short <jdorje@freeciv.org>
-"
+    Jason Dorje Short <jdorje@freeciv.org>"
 
 [file]
 gfx = "misc/overlays"
@@ -29,5 +28,4 @@
   1, 1, "colors.overlay_3"
 
   0, 2, "mask.worked_tile"
-  1, 2, "mask.unworked_tile"
-}
+  1, 2, "mask.unworked_tile"}
diff -Nurd -X.diff_ignore freeciv/data/misc/shields.spec freeciv/data/misc/shields.spec
--- freeciv/data/misc/shields.spec	2012-07-17 02:35:27.000000000 +0300
+++ freeciv/data/misc/shields.spec	2012-07-17 03:49:42.000000000 +0300
@@ -7,8 +7,7 @@
 [info]
 
 artists = "
-   See flags.spec
-"
+   See flags.spec"
 [extra]
 sprites =
 	{	"tag", "file"
diff -Nurd -X.diff_ignore freeciv/data/misc/small.spec freeciv/data/misc/small.spec
--- freeciv/data/misc/small.spec	2012-07-17 02:35:27.000000000 +0300
+++ freeciv/data/misc/small.spec	2012-07-17 03:49:42.000000000 +0300
@@ -11,8 +11,7 @@
     Jeff Mallatt <jjm@codewell.com> (cooling flakes)
     Davide Pagnin <nightmare@freeciv.it> (angry citizens)
     Enrico Bini <e.bini@sssup.it> (fundamentalism icon)
-    Hogne Håskjold <haskjold@gmail.com> (gold coin)
-"
+    Hogne Håskjold <haskjold@gmail.com> (gold coin)"
 
 [file]
 gfx = "misc/small"
@@ -120,5 +119,4 @@
   1, 9,  "ev.wonderwillbebuilt"
   1, 10, "ev.wonderbuilt"
   1, 11, "s.plus"
-  1, 12, "s.minus"
-}
+  1, 12, "s.minus"}
diff -Nurd -X.diff_ignore freeciv/data/misc/space.spec freeciv/data/misc/space.spec
--- freeciv/data/misc/space.spec	2012-07-17 02:35:27.000000000 +0300
+++ freeciv/data/misc/space.spec	2012-07-17 03:49:42.000000000 +0300
@@ -7,8 +7,7 @@
 [info]
 
 artists = "
-    Alexandre Beraud <a_beraud@lemel.fr>
-"
+    Alexandre Beraud <a_beraud@lemel.fr>"
 
 [file]
 gfx = "misc/space"
@@ -31,5 +30,4 @@
   0,  3, "spaceship.structural"
   0,  4, "spaceship.fuel"
   0,  5, "spaceship.propulsion"
-  0,  6, "spaceship.exhaust"
-}
+  0,  6, "spaceship.exhaust"}
diff -Nurd -X.diff_ignore freeciv/data/misc/techs.spec freeciv/data/misc/techs.spec
--- freeciv/data/misc/techs.spec	2012-07-17 02:35:27.000000000 +0300
+++ freeciv/data/misc/techs.spec	2012-07-17 03:49:42.000000000 +0300
@@ -109,8 +109,7 @@
  PD - Unknown author
  a.communism         a.flight             a.refining
  a.chivalry          a.leadership         a.writing
- a.feudalism         a.mass_production    
-"
+ a.feudalism         a.mass_production    "
 
 
 [file]
@@ -209,5 +208,4 @@
   8, 4, "a.warrior_code"
   8, 5, "a.writing"
   8, 6, "a.environmentalism"
-  8, 7, "a.fundamentalism"
-}
+  8, 7, "a.fundamentalism"}
diff -Nurd -X.diff_ignore freeciv/data/misc/treaty.spec freeciv/data/misc/treaty.spec
--- freeciv/data/misc/treaty.spec	2012-07-17 02:35:27.000000000 +0300
+++ freeciv/data/misc/treaty.spec	2012-07-17 03:49:42.000000000 +0300
@@ -7,8 +7,7 @@
 [info]
 
 artists = "
-    pratheep
-"
+    pratheep"
 
 [file]
 gfx = "misc/treaty"
@@ -25,6 +24,4 @@
 ; Agree/disagree thumbs:
 
   0,  0, "treaty.agree_thumb_up"
-  0,  1, "treaty.disagree_thumb_down"
-
-}
+  0,  1, "treaty.disagree_thumb_down"}
diff -Nurd -X.diff_ignore freeciv/data/misc/wonders-large.spec freeciv/data/misc/wonders-large.spec
--- freeciv/data/misc/wonders-large.spec	2012-07-17 02:35:27.000000000 +0300
+++ freeciv/data/misc/wonders-large.spec	2012-07-17 03:49:42.000000000 +0300
@@ -48,8 +48,7 @@
   b.hoover_dam
   b.manhattan_project
   b.oracle
-  b.statue_of_liberty
-"
+  b.statue_of_liberty"
 
 [extra]
 sprites =
diff -Nurd -X.diff_ignore freeciv/server/console.c freeciv/server/console.c
--- freeciv/server/console.c	2012-07-17 02:35:26.000000000 +0300
+++ freeciv/server/console.c	2012-07-17 03:49:42.000000000 +0300
@@ -60,7 +60,9 @@
     /* Make sure that message is not left to buffers when server dies */
     conn_list_iterate(game.est_connections, pconn) {
       pconn->send_buffer->do_buffer_sends = 0;
+#ifdef USE_COMPRESSION
       pconn->compression.frozen_level = 0;
+#endif
     } conn_list_iterate_end;
 
     notify_conn(NULL, NULL, E_LOG_FATAL, ftc_warning, "%s", message);
diff -Nurd -X.diff_ignore freeciv/server/ruleset.c freeciv/server/ruleset.c
--- freeciv/server/ruleset.c	2012-07-17 03:13:46.000000000 +0300
+++ freeciv/server/ruleset.c	2012-07-17 03:49:42.000000000 +0300
@@ -4011,12 +4011,15 @@
     misc_p.move_bonus[i] = vlevel->move_bonus;
   }
 
+  /* FIXME: removed for webclient. - ANdreas. */
+#if 0
   fc_assert(sizeof(misc_p.global_init_techs)
             == sizeof(game.server.rgame.global_init_techs));
   fc_assert(ARRAY_SIZE(misc_p.global_init_techs)
             == ARRAY_SIZE(game.server.rgame.global_init_techs));
   memcpy(misc_p.global_init_techs, game.server.rgame.global_init_techs,
          sizeof(misc_p.global_init_techs));
+#endif
 
   misc_p.default_specialist = DEFAULT_SPECIALIST;
 
diff -Nurd -X.diff_ignore freeciv/server/settings.c freeciv/server/settings.c
--- freeciv/server/settings.c	2012-07-17 02:35:26.000000000 +0300
+++ freeciv/server/settings.c	2012-07-17 03:49:42.000000000 +0300
@@ -3825,7 +3825,7 @@
     setting.id = setting_number(pset);
     sz_strlcpy(setting.name, setting_name(pset));
     sz_strlcpy(setting.short_help, setting_short_help(pset));
-    sz_strlcpy(setting.extra_help, setting_extra_help(pset));
+    /* sz_strlcpy(setting.extra_help, setting_extra_help(pset)); */
     setting.category = pset->scategory;
 
     send_packet_server_setting_const(pconn, &setting);
