diff -Nurd -X.diff_ignore freeciv/client/packhand.c freeciv/client/packhand.c
--- freeciv/client/packhand.c	2012-05-21 01:53:19.000000000 +0300
+++ freeciv/client/packhand.c	2012-05-21 03:06:43.000000000 +0300
@@ -2924,7 +2924,7 @@
   pl->is_playable = p->is_playable;
   pl->barb_type = p->barbarian_type;
 
-  memcpy(pl->init_techs, p->init_techs, sizeof(pl->init_techs));
+  //memcpy(pl->init_techs, p->init_techs, sizeof(pl->init_techs)); FIXME: removed for webclient - Andreas.
   memcpy(pl->init_buildings, p->init_buildings, 
          sizeof(pl->init_buildings));
   memcpy(pl->init_units, p->init_units, 
diff -Nurd -X.diff_ignore freeciv/common/connection.c freeciv/common/connection.c
--- freeciv/common/connection.c	2012-05-21 01:28:25.000000000 +0300
+++ freeciv/common/connection.c	2012-05-21 03:06:43.000000000 +0300
@@ -350,7 +350,8 @@
 {
   if (pc && pc->used) {
     pc->statistics.bytes_send += len;
-    if(pc->send_buffer->do_buffer_sends > 0) {
+    /* FI*/
+    /*if(pc->send_buffer->do_buffer_sends > 0) {
       flush_connection_send_buffer_packets(pc);
       if (!add_connection_data(pc, data, len)) {
         log_error("cut connection %s due to huge send buffer (1)",
@@ -358,7 +359,7 @@
       }
       flush_connection_send_buffer_packets(pc);
     }
-    else {
+    else {*/
       flush_connection_send_buffer_all(pc);
       if (!add_connection_data(pc, data, len)) {
         log_error("cut connection %s due to huge send buffer (2)",
@@ -366,7 +367,7 @@
       }
       flush_connection_send_buffer_all(pc);
     }
-  }
+  //}
 }
 
 /**************************************************************************
diff -Nurd -X.diff_ignore freeciv/common/connection.h freeciv/common/connection.h
--- freeciv/common/connection.h	2012-05-21 01:28:25.000000000 +0300
+++ freeciv/common/connection.h	2012-05-21 03:06:43.000000000 +0300
@@ -22,7 +22,7 @@
 #include <sys/time.h>
 #endif
 
-#define USE_COMPRESSION
+//#define USE_COMPRESSION   /* webclient doesn't like compression*/
 
 /**************************************************************************
   The connection struct and related stuff.
diff -Nurd -X.diff_ignore freeciv/common/featured_text.c freeciv/common/featured_text.c
--- freeciv/common/featured_text.c	2012-05-21 03:01:59.000000000 +0300
+++ freeciv/common/featured_text.c	2012-05-21 03:08:26.000000000 +0300
@@ -33,8 +33,8 @@
 
 #include "featured_text.h"
 
-#define SEQ_START '['
-#define SEQ_STOP ']'
+#define SEQ_START '<'
+#define SEQ_STOP '>'
 #define SEQ_END '/'
 
 #define MAX_LEN_STR 32
@@ -136,7 +136,7 @@
   case TTT_UNDERLINE:
     return "u";
   case TTT_COLOR:
-    return "c";
+    return "font";
   case TTT_LINK:
     return "l";
   };
@@ -233,7 +233,7 @@
     {
       if (!find_option(sequence, "foreground", ptag->color.foreground,
                        sizeof(ptag->color.foreground))
-          && !find_option(sequence, "fg", ptag->color.foreground,
+          && !find_option(sequence, "color", ptag->color.foreground,
                           sizeof(ptag->color.foreground))) {
         ptag->color.foreground[0] = '\0';
       }
@@ -477,7 +477,7 @@
                                text_tag_type_short_name(ptag->type));
 
       if (ptag->color.foreground[0] != '\0') {
-        ret += fc_snprintf(buf + ret, len - ret, " fg=\"%s\"",
+        ret += fc_snprintf(buf + ret, len - ret, " color=\"%s\"",
                            ptag->color.foreground);
       }
       if (ptag->color.background[0] != '\0') {
diff -Nurd -X.diff_ignore freeciv/common/generate_packets.py freeciv/common/generate_packets.py
--- freeciv/common/generate_packets.py	2012-05-21 01:28:26.000000000 +0300
+++ freeciv/common/generate_packets.py	2012-05-21 03:06:43.000000000 +0300
@@ -27,7 +27,7 @@
 
 ### The following parameters CHANGE the protocol. You have been warned.
 fold_bool_into_header=1
-disable_delta=0
+disable_delta=1
 
 ################# END OF PARAMETERS ####################
 
diff -Nurd -X.diff_ignore freeciv/common/packets.def freeciv/common/packets.def
--- freeciv/common/packets.def	2012-05-21 03:06:36.000000000 +0300
+++ freeciv/common/packets.def	2012-05-21 03:10:06.000000000 +0300
@@ -159,7 +159,7 @@
 
 # typedefs for numbers
 type UINT8		= uint8(int)
-type UINT16		= uint16(int)
+type UINT16		= uint32(int)
 type UINT32		= uint32(int)
 type SINT8		= sint8(int)
 type SINT16		= sint16(int)
@@ -173,15 +173,16 @@
 type STRING		= string(char)
 type BIT_STRING		= bit_string(char)
 type WORKLIST		= worklist(struct worklist)
-type TECH_LIST		= tech_list(int)
+# The TECH_LIST has been removed for the web client.  -Andreas
+#type TECH_LIST		= tech_list(int)
 type REQUIREMENT	= requirement(struct requirement)
 
 # typedefs for enums
 type CLAUSE		= uint8(enum clause_type)
 type ACTIVITY		= uint8(enum unit_activity)
-type EVENT		= sint16(enum event_type)
+type EVENT		= sint32(enum event_type)
 type KNOWN		= uint8(enum known_type)
-type SPECIAL		= uint16(enum tile_special_type)
+type SPECIAL		= uint32(enum tile_special_type)
 type DIPLOMAT_ACTION	= uint8(enum diplomat_actions)
 type CMDLEVEL		= uint8(enum cmdlevel_id)
 type PLACE_TYPE		= uint8(enum spaceship_place_type)
@@ -223,7 +224,7 @@
 type TECH     		= UINT8
 type TILE		= SINT16
 type UNIT_TYPE		= uint8(Unit_type_id)
-type NATION		= sint16(Nation_type_id)
+type NATION		    = UINT32
 type GOVERNMENT		= UINT8
 type CONNECTION		= SINT16
 type TEAM		= UINT8
@@ -1127,7 +1128,7 @@
 PACKET_RULESET_GAME = 141; sc, lsend
   UINT8 default_specialist;
 
-  TECH_LIST global_init_techs[MAX_NUM_TECH_LIST];
+  #TECH_LIST global_init_techs[MAX_NUM_TECH_LIST]; - Removed for web client. - Andreas
   UINT8 work_veteran_chance[MAX_VET_LEVELS];
   UINT8 veteran_chance[MAX_VET_LEVELS];
 end
@@ -1210,7 +1211,7 @@
   STRING legend[MAX_LEN_MSG];
 
   UINT8 city_style;
-  TECH_LIST init_techs[MAX_NUM_TECH_LIST];
+  #TECH_LIST init_techs[MAX_NUM_TECH_LIST]; Removed for web client. - Andreas
   UNIT_TYPE init_units[MAX_NUM_UNIT_LIST];
   IMPROVEMENT init_buildings[MAX_NUM_BUILDING_LIST];
   GOVERNMENT init_government;
@@ -1381,7 +1382,7 @@
   UINT16 id; key
   STRING name[MAX_LEN_NAME];
   STRING short_help[MAX_LEN_PACKET];
-  STRING extra_help[MAX_LEN_PACKET];
+  # STRING extra_help[MAX_LEN_PACKET];
   UINT8 category;
 end
 
diff -Nurd -X.diff_ignore freeciv/data/amplio/ancientcities.spec freeciv/data/amplio/ancientcities.spec
--- freeciv/data/amplio/ancientcities.spec	2012-05-21 01:28:15.000000000 +0300
+++ freeciv/data/amplio/ancientcities.spec	2012-05-21 03:06:43.000000000 +0300
@@ -23,8 +23,7 @@
     Polynesian style by CapTVK
     Celtic style by Erwan, adapted to 96x48 by CapTVK
     Roman style by CapTVK
-    City walls by Hogne Håskjold
-"
+    City walls by Hogne Håskjold"
 
 [file]
 gfx = "amplio/ancientcities"
@@ -108,6 +107,4 @@
  4,  6, "city.babylonian_wall_4"
  4,  7, "city.babylonian_wall_8"
  4,  8, "city.babylonian_wall_12"
- 4,  9, "city.babylonian_wall_16"
-
-}
+ 4,  9, "city.babylonian_wall_16"}
diff -Nurd -X.diff_ignore freeciv/data/amplio/buildings.spec freeciv/data/amplio/buildings.spec
--- freeciv/data/amplio/buildings.spec	2012-05-21 01:28:15.000000000 +0300
+++ freeciv/data/amplio/buildings.spec	2012-05-21 03:06:43.000000000 +0300
@@ -7,8 +7,8 @@
 [info]
 
 artists = "
-   feler404
-"
+   feler404"
+
 [extra]
 sprites =
 	{	"tag", "file"
diff -Nurd -X.diff_ignore freeciv/data/amplio/explosions.spec freeciv/data/amplio/explosions.spec
--- freeciv/data/amplio/explosions.spec	2012-05-21 01:28:15.000000000 +0300
+++ freeciv/data/amplio/explosions.spec	2012-05-21 03:06:43.000000000 +0300
@@ -10,8 +10,7 @@
     Explosions from xgalaga <http://rumsey.org/xgal.html>
     Credits:
     Joe Rumsey <ogre@atomic.com>
-    Mike McGrath <mcgrath@epx.cis.umn.edu>
-"
+    Mike McGrath <mcgrath@epx.cis.umn.edu>"
 
 [file]
 gfx = "amplio/explosions"
@@ -28,5 +27,4 @@
   1,  0, "explode.unit_1"
   2,  0, "explode.unit_2"
   3,  0, "explode.unit_3"
-  4,  0, "explode.unit_4"
-}
+  4,  0, "explode.unit_4"}
diff -Nurd -X.diff_ignore freeciv/data/amplio/fog.spec freeciv/data/amplio/fog.spec
--- freeciv/data/amplio/fog.spec	2012-05-21 01:28:15.000000000 +0300
+++ freeciv/data/amplio/fog.spec	2012-05-21 03:06:43.000000000 +0300
@@ -3,8 +3,7 @@
 options = "+spec3"
 
 artists = "
-    Brendon Oram (Yautja)
-"
+    Brendon Oram (Yautja)"
 
 [file]
 gfx = "amplio/fog"
@@ -98,5 +97,4 @@
   8,  5, "t.fog_u_u_k_u"
   8,  6, "t.fog_u_u_u_f"
   8,  7, "t.fog_u_u_u_k"
-  8,  8, "t.fog_u_u_u_u"
-}
+  8,  8, "t.fog_u_u_u_u"}
diff -Nurd -X.diff_ignore freeciv/data/amplio/grid.spec freeciv/data/amplio/grid.spec
--- freeciv/data/amplio/grid.spec	2012-05-21 01:28:15.000000000 +0300
+++ freeciv/data/amplio/grid.spec	2012-05-21 03:06:43.000000000 +0300
@@ -9,8 +9,7 @@
 artists = "
     Jason Dorje Short <jdorje@freeciv.org>
     Hogne Håskjold    <hogne@freeciv.org>
-    Eleazar / J. W. Bjerk <www.jwbjerk.com>
-"
+    Eleazar / J. W. Bjerk <www.jwbjerk.com>"
 
 [file]
 gfx = "amplio/grid"
@@ -39,5 +38,4 @@
   2, 0, "grid.borders.n"
   2, 1, "grid.borders.s"
   2, 2, "grid.borders.w"
-  2, 3, "grid.borders.e"
-}
+  2, 3, "grid.borders.e"}
diff -Nurd -X.diff_ignore freeciv/data/amplio/icons.spec freeciv/data/amplio/icons.spec
--- freeciv/data/amplio/icons.spec	2012-05-21 01:28:15.000000000 +0300
+++ freeciv/data/amplio/icons.spec	2012-05-21 03:06:43.000000000 +0300
@@ -9,8 +9,7 @@
 artists = "
     Hogne Håskjold <hogne@freeciv.org>
     civicon: Made from the globe in the intro picture
-    cityicon: City gfx by smiley
-"
+    cityicon: City gfx by smiley"
 
 [extra]
 sprites = 
diff -Nurd -X.diff_ignore freeciv/data/amplio/medievalcities.spec freeciv/data/amplio/medievalcities.spec
--- freeciv/data/amplio/medievalcities.spec	2012-05-21 01:28:15.000000000 +0300
+++ freeciv/data/amplio/medievalcities.spec	2012-05-21 03:06:43.000000000 +0300
@@ -21,8 +21,7 @@
 artists = "
     Medieval by CapTVK
     Steam Age by Smiley, www.firstcultural.com
-    City walls by Hogne Håskjold
-"
+    City walls by Hogne Håskjold"
 
 [file]
 gfx = "amplio/medievalcities"
@@ -66,6 +65,4 @@
  1,  6, "city.industrial_wall_4"
  1,  7, "city.industrial_wall_8"
  1,  8, "city.industrial_wall_12"
- 1,  9, "city.industrial_wall_16" 
-
-}
+ 1,  9, "city.industrial_wall_16" }
diff -Nurd -X.diff_ignore freeciv/data/amplio/moderncities.spec freeciv/data/amplio/moderncities.spec
--- freeciv/data/amplio/moderncities.spec	2012-05-21 01:28:15.000000000 +0300
+++ freeciv/data/amplio/moderncities.spec	2012-05-21 03:06:43.000000000 +0300
@@ -23,8 +23,7 @@
     Modern, Post Modern and Electric Age by Smiley, www.firstcultural.com
     City walls by Hogne Håskjold
     Buoy by Eleazar
-    Ruins by Vincent Croisier
-"
+    Ruins by Vincent Croisier"
 
 [file]
 gfx = "amplio/moderncities"
@@ -94,6 +93,4 @@
  4,  6, "city.postmodern_wall_4"
  4,  7, "city.postmodern_wall_8"
  4,  8, "city.postmodern_wall_12"
- 4,  9, "city.postmodern_wall_16" 
-
-}
+ 4,  9, "city.postmodern_wall_16"}
diff -Nurd -X.diff_ignore freeciv/data/amplio/nuke.spec freeciv/data/amplio/nuke.spec
--- freeciv/data/amplio/nuke.spec	2012-05-21 01:28:15.000000000 +0300
+++ freeciv/data/amplio/nuke.spec	2012-05-21 03:06:43.000000000 +0300
@@ -7,8 +7,7 @@
 
 artists = "
     Tatu Rissanen <tatu.rissanen@hut.fi>
-    Jeff Mallatt <jjm@codewell.com> (miscellaneous)
-"
+    Jeff Mallatt <jjm@codewell.com> (miscellaneous)"
 
 [file]
 gfx = "amplio/nuke"
@@ -22,5 +21,4 @@
 pixel_border = 0
 
 tiles = { "row", "column", "tag"
- 0, 0, "explode.nuke"
-}
+ 0, 0, "explode.nuke"}
diff -Nurd -X.diff_ignore freeciv/data/amplio/ocean.spec freeciv/data/amplio/ocean.spec
--- freeciv/data/amplio/ocean.spec	2012-05-21 01:28:15.000000000 +0300
+++ freeciv/data/amplio/ocean.spec	2012-05-21 03:06:43.000000000 +0300
@@ -10,8 +10,7 @@
     Yautja
     CapTVK
     Eleazar
-    William Allen Simpson
-"
+    William Allen Simpson"
 
 [file]
 gfx = "amplio/ocean"
@@ -107,5 +106,4 @@
   0, 3, "t.l0.cellgroup_l_l_l_s"
   0, 2, "t.l0.cellgroup_d_l_l_l"
   0, 1, "t.l0.cellgroup_s_l_l_l"
-  0, 0, "t.l0.cellgroup_l_l_l_l"
-}
+  0, 0, "t.l0.cellgroup_l_l_l_l"}
diff -Nurd -X.diff_ignore freeciv/data/amplio/select.spec freeciv/data/amplio/select.spec
--- freeciv/data/amplio/select.spec	2012-05-21 01:28:15.000000000 +0300
+++ freeciv/data/amplio/select.spec	2012-05-21 03:06:43.000000000 +0300
@@ -8,8 +8,7 @@
 artists = "
     Jason Dorje Short <jdorje@freeciv.org>
     Eleazar / J. W. Bjerk <www.jwbjerk.com>
-    qwm
-"
+    qwm"
 
 [file]
 gfx = "amplio/select-alpha"
@@ -26,5 +25,4 @@
   0, 0, "unit.select0"
   0, 1, "unit.select1"
   0, 2, "unit.select2"
-  0, 3, "unit.select3"
-}
+  0, 3, "unit.select3"}
diff -Nurd -X.diff_ignore freeciv/data/amplio/terrain1.spec freeciv/data/amplio/terrain1.spec
--- freeciv/data/amplio/terrain1.spec	2012-05-21 01:28:15.000000000 +0300
+++ freeciv/data/amplio/terrain1.spec	2012-05-21 03:06:43.000000000 +0300
@@ -10,8 +10,7 @@
     Hogne Håskjold
     Daniel Speyer
     Yautja
-    CapTVK
-"
+    CapTVK"
 
 [file]
 gfx = "amplio/terrain1"
@@ -127,5 +126,4 @@
   7,  0, "t.blend.arctic" ;ice over neighbors
  15,  3, "t.blend.coast"
  15,  4, "user.attention"
- 15,  5, "tx.fog"
-}
+ 15,  5, "tx.fog"}
diff -Nurd -X.diff_ignore freeciv/data/amplio/terrain2.spec freeciv/data/amplio/terrain2.spec
--- freeciv/data/amplio/terrain2.spec	2012-05-21 01:28:15.000000000 +0300
+++ freeciv/data/amplio/terrain2.spec	2012-05-21 03:06:43.000000000 +0300
@@ -11,8 +11,7 @@
     Tim F. Smith <yoohootim@hotmail.com>
     Yautja
     Daniel Speyer
-    Eleazar
-"
+    Eleazar"
 
 [file]
 gfx = "amplio/terrain2"
@@ -82,9 +81,7 @@
  9,  4, "t.l1.hills_n1e1s0w0"
  9,  5, "t.l1.hills_n1e1s0w1"
  9,  6, "t.l1.hills_n1e1s1w0"
- 9,  7, "t.l1.hills_n1e1s1w1"
-
-}
+ 9,  7, "t.l1.hills_n1e1s1w1"}
 
 
 [grid_coasts]
@@ -97,6 +94,4 @@
 
 tiles = { "row", "column","tag"
 
-;* previous coordinates now in water.spec
-
-}
+;* previous coordinates now in water.spec}
diff -Nurd -X.diff_ignore freeciv/data/amplio/tiles.spec freeciv/data/amplio/tiles.spec
--- freeciv/data/amplio/tiles.spec	2012-05-21 01:28:15.000000000 +0300
+++ freeciv/data/amplio/tiles.spec	2012-05-21 03:06:43.000000000 +0300
@@ -8,8 +8,7 @@
 
 artists = "
     Tatu Rissanen <tatu.rissanen@hut.fi>
-    Jeff Mallatt <jjm@codewell.com> (miscellaneous)
-"
+    Jeff Mallatt <jjm@codewell.com> (miscellaneous)"
 
 [file]
 gfx = "amplio/tiles"
@@ -24,8 +23,8 @@
 tiles = { "row", "column", "tag"
 ; Unit activity letters:  (note unit icons have just "u.")
 
-  0, 18, "unit.auto_attack",
-         "unit.auto_settler"
+  0, 18, "unit.auto_attack"
+  0, 18, "unit.auto_settler"
   0, 19, "unit.stack"
   1, 17, "unit.loaded"
   1, 18, "unit.connect"
@@ -37,8 +36,8 @@
   2, 16, "unit.pollution"
   2, 17, "unit.road"
   2, 18, "unit.irrigate"
-  2, 19, "unit.fortifying",
-         "unit.fortress"
+  2, 19, "unit.fortifying"
+  2, 19, "unit.fortress"
   3, 16, "unit.airbase"
   3, 17, "unit.pillage"
   3, 18, "unit.fortified"
@@ -139,6 +138,4 @@
   1,  2, "explode.nuke_12"
   2,  0, "explode.nuke_20"
   2,  1, "explode.nuke_21"
-  2,  2, "explode.nuke_22"
-
-}
+  2,  2, "explode.nuke_22"}
diff -Nurd -X.diff_ignore freeciv/data/amplio/units.spec freeciv/data/amplio/units.spec
--- freeciv/data/amplio/units.spec	2012-05-21 01:28:15.000000000 +0300
+++ freeciv/data/amplio/units.spec	2012-05-21 03:06:43.000000000 +0300
@@ -32,8 +32,7 @@
     Vodvakov
     J. W. Bjerk / Eleazar <www.jwbjerk.com>
     qwm
-    FiftyNine
-"
+    FiftyNine"
 
 [file]
 gfx = "amplio/units"
@@ -46,7 +45,6 @@
 dy = 48
 
 tiles = { "row", "column", "tag"
-				; Scenario League tags in brackets
   0,  0, "u.armor"		; [Nemo]
   0,  1, "u.howitzer"		; [Nemo]
   0,  2, "u.battleship"		; [Nemo]
@@ -118,5 +116,4 @@
   3, 8, "unit.vet_9"
 
   3, 11, "unit.lowfuel"
-  3, 11, "unit.tired"
-}
+  3, 11, "unit.tired"}
diff -Nurd -X.diff_ignore freeciv/data/amplio/water.spec freeciv/data/amplio/water.spec
--- freeciv/data/amplio/water.spec	2012-05-21 01:28:15.000000000 +0300
+++ freeciv/data/amplio/water.spec	2012-05-21 03:06:43.000000000 +0300
@@ -11,8 +11,7 @@
     Tim F. Smith <yoohootim@hotmail.com>
     Yautja
     Daniel Speyer
-    Eleazar
-"
+    Eleazar"
 
 [file]
 gfx = "amplio/water"
@@ -71,8 +70,7 @@
  10, 0, "tx.river_outlet_n"
  10, 1, "tx.river_outlet_e"
  10, 2, "tx.river_outlet_s"
- 10, 3, "tx.river_outlet_w"
-}
+ 10, 3, "tx.river_outlet_w"}
 
 
 [grid_coasts]
@@ -193,5 +191,4 @@
  11, 9,  "t.l0.lake_cell_r_s_s_l"
  11, 11, "t.l0.lake_cell_r_l_s_l"
  11, 13, "t.l0.lake_cell_r_s_l_l"
- 11, 15, "t.l0.lake_cell_r_l_l_l"
-}
+ 11, 15, "t.l0.lake_cell_r_l_l_l"}
diff -Nurd -X.diff_ignore freeciv/data/amplio/wonders.spec freeciv/data/amplio/wonders.spec
--- freeciv/data/amplio/wonders.spec	2012-05-21 01:28:15.000000000 +0300
+++ freeciv/data/amplio/wonders.spec	2012-05-21 03:06:43.000000000 +0300
@@ -48,8 +48,7 @@
   b.hoover_dam
   b.manhattan_project
   b.oracle
-  b.statue_of_liberty
-"
+  b.statue_of_liberty"
 
 [extra]
 sprites =
diff -Nurd -X.diff_ignore freeciv/data/amplio.tilespec freeciv/data/amplio.tilespec
--- freeciv/data/amplio.tilespec	2012-05-21 01:28:16.000000000 +0300
+++ freeciv/data/amplio.tilespec	2012-05-21 03:06:43.000000000 +0300
@@ -92,7 +92,7 @@
 
 
 ; Include color definitions
-*include "misc/colors.tilespec"
+;*include "misc/colors.tilespec"
 
 ; Terrain info - see README.graphics
 
diff -Nurd -X.diff_ignore freeciv/data/misc/buildings.spec freeciv/data/misc/buildings.spec
--- freeciv/data/misc/buildings.spec	2012-05-21 01:28:15.000000000 +0300
+++ freeciv/data/misc/buildings.spec	2012-05-21 03:06:43.000000000 +0300
@@ -9,8 +9,7 @@
     Tim F. Smith <yoohootim@hotmail.com>
     Frederic Rodrigo <f.rodrigo@tuxfamily.org>
     Dusan Hukuv <dusan@mirc.cz>
-    Eleazar / J. W. Bjerk <www.jwbjerk.com>
-"
+    Eleazar / J. W. Bjerk <www.jwbjerk.com>"
 
 [file]
 gfx = "misc/buildings"
@@ -103,5 +102,4 @@
   8,  4, "b.statue_of_liberty"
   8,  5, "b.sun_tzus_war_academy"
   8,  6, "b.united_nations"
-  8,  7, "b.womens_suffrage"
-}
+  8,  7, "b.womens_suffrage"}
diff -Nurd -X.diff_ignore freeciv/data/misc/chiefs.spec freeciv/data/misc/chiefs.spec
--- freeciv/data/misc/chiefs.spec	2012-05-21 01:28:15.000000000 +0300
+++ freeciv/data/misc/chiefs.spec	2012-05-21 03:06:43.000000000 +0300
@@ -6,8 +6,7 @@
 [info]
 
 artists = "
-    Falk Hueffner <falk.hueffner@student.uni-tuebingen.de>
-"
+    Falk Hueffner <falk.hueffner@student.uni-tuebingen.de>"
 
 [file]
 gfx = "misc/chiefs"
@@ -20,18 +19,14 @@
 dy = 45
 
 tiles = { "row", "column", "tag"
-
-; Unit hit-point bars: approx percent of hp remaining
-0,  0, "unit.hp_100"
-0,  1, "unit.hp_90"
-0,  2, "unit.hp_80"
-0,  3, "unit.hp_70"
-0,  4, "unit.hp_60"
-0,  5, "unit.hp_50"
-0,  6, "unit.hp_40"
-0,  7, "unit.hp_30"
-0,  8, "unit.hp_20"
-0,  9, "unit.hp_10"
-0, 10, "unit.hp_0"
-
-}
+ 0,  0, "unit.hp_100"
+ 0,  1, "unit.hp_90"
+ 0,  2, "unit.hp_80"
+ 0,  3, "unit.hp_70"
+ 0,  4, "unit.hp_60"
+ 0,  5, "unit.hp_50"
+ 0,  6, "unit.hp_40"
+ 0,  7, "unit.hp_30"
+ 0,  8, "unit.hp_20"
+ 0,  9, "unit.hp_10"
+ 0, 10, "unit.hp_0"}
diff -Nurd -X.diff_ignore freeciv/data/misc/citybar.spec freeciv/data/misc/citybar.spec
--- freeciv/data/misc/citybar.spec	2012-05-21 01:28:15.000000000 +0300
+++ freeciv/data/misc/citybar.spec	2012-05-21 03:06:43.000000000 +0300
@@ -8,8 +8,7 @@
 
 artists = "
      Freim <...>
-     Madeline Book <madeline.book@gmail.com> (citybar.trade)
-"
+     Madeline Book <madeline.book@gmail.com> (citybar.trade)"
 
 [file]
 gfx = "misc/citybar"
@@ -26,9 +25,7 @@
 
   0,  0, "citybar.shields"
   0,  1, "citybar.food"
-  0,  2, "citybar.trade"
-
-}
+  0,  2, "citybar.trade"}
 
 [grid_star]
 
@@ -44,9 +41,7 @@
   0,  1, "citybar.occupancy_0"
   0,  2, "citybar.occupancy_1"
   0,  3, "citybar.occupancy_2"
-  0,  4, "citybar.occupancy_3"
-
-}
+  0,  4, "citybar.occupancy_3"}
 
 [grid_bg]
 x_top_left = 1
@@ -56,7 +51,4 @@
 dy = 152
 
 tiles = { "row", "column", "tag"
-
-  0,  0, "citybar.background"
-
-}
+  0,  0, "citybar.background"}
diff -Nurd -X.diff_ignore freeciv/data/misc/cursors.spec freeciv/data/misc/cursors.spec
--- freeciv/data/misc/cursors.spec	2012-05-21 01:28:15.000000000 +0300
+++ freeciv/data/misc/cursors.spec	2012-05-21 03:06:43.000000000 +0300
@@ -11,9 +11,7 @@
 
     edit_paint cursor take from GIMP
 
-    Hourglass and Add from Bluecurve theme in redhat-artwork
-    
-"
+    Hourglass and Add from Bluecurve theme in redhat-artwork"
 
 [file]
 gfx = "misc/cursors"
@@ -86,5 +84,4 @@
        2, 9, "cursor.wait2", 16, 16
        3, 9, "cursor.wait3", 16, 16
        4, 9, "cursor.wait4", 16, 16
-       5, 9, "cursor.wait5", 16, 16
-}
+       5, 9, "cursor.wait5", 16, 16}
diff -Nurd -X.diff_ignore freeciv/data/misc/editor.spec freeciv/data/misc/editor.spec
--- freeciv/data/misc/editor.spec	2012-05-21 01:28:15.000000000 +0300
+++ freeciv/data/misc/editor.spec	2012-05-21 03:06:43.000000000 +0300
@@ -7,8 +7,7 @@
 [info]
 
 artists = "
-	Pep
-"
+	Pep"
 
 [file]
 gfx = "misc/editor"
@@ -47,8 +46,7 @@
   3,  6, "editor.fill"
   3,  7, "editor.tiles_1"
   3,  8, "editor.tiles_2"
-  3,  9, "editor.terrain_resource_2"
-}
+  3,  9, "editor.terrain_resource_2"}
 
 [grid_mini]
 
@@ -72,8 +70,7 @@
   0, 10, "editor.mini_j" 
   0, 11, "editor.mini_gl"
   0, 12, "editor.mini_erase"
-  0, 13, "editor.mini_freeciv"
-}
+  0, 13, "editor.mini_freeciv"}
 
 [grid_extra]
 
@@ -90,5 +87,4 @@
   0,  3, "editor.type_paint"
   0,  4, "editor.bitmap_to_map"
   0,  5, "editor.copy"
-  0,  6, "editor.paste"
-}
+  0,  6, "editor.paste"}
diff -Nurd -X.diff_ignore freeciv/data/misc/flags.spec freeciv/data/misc/flags.spec
--- freeciv/data/misc/flags.spec	2012-05-21 01:28:15.000000000 +0300
+++ freeciv/data/misc/flags.spec	2012-05-21 03:06:43.000000000 +0300
@@ -225,9 +225,7 @@
    Viking		Daniel Markstedt
    Wales		Tobias Jakobs
    Westphalia		Daniel Markstedt and Tobias Jakobs
-   Zulu			Daniel Markstedt and Caleb Moore
-
-"
+   Zulu			Daniel Markstedt and Caleb Moore"
 [extra]
 sprites =
 	{	"tag", "file"
diff -Nurd -X.diff_ignore freeciv/data/misc/icons.spec freeciv/data/misc/icons.spec
--- freeciv/data/misc/icons.spec	2012-05-21 01:28:15.000000000 +0300
+++ freeciv/data/misc/icons.spec	2012-05-21 03:06:43.000000000 +0300
@@ -7,8 +7,7 @@
 [info]
 
 artists = "
-    James Blewitt (civicon)
-"
+    James Blewitt (civicon)"
 
 [extra]
 sprites = 
diff -Nurd -X.diff_ignore freeciv/data/misc/overlays.spec freeciv/data/misc/overlays.spec
--- freeciv/data/misc/overlays.spec	2012-05-21 01:28:15.000000000 +0300
+++ freeciv/data/misc/overlays.spec	2012-05-21 03:06:43.000000000 +0300
@@ -7,8 +7,7 @@
 [info]
 
 artists = "
-    Jason Dorje Short <jdorje@freeciv.org>
-"
+    Jason Dorje Short <jdorje@freeciv.org>"
 
 [file]
 gfx = "misc/overlays"
@@ -29,5 +28,4 @@
   1, 1, "colors.overlay_3"
 
   0, 2, "mask.worked_tile"
-  1, 2, "mask.unworked_tile"
-}
+  1, 2, "mask.unworked_tile"}
diff -Nurd -X.diff_ignore freeciv/data/misc/shields.spec freeciv/data/misc/shields.spec
--- freeciv/data/misc/shields.spec	2012-05-21 01:28:15.000000000 +0300
+++ freeciv/data/misc/shields.spec	2012-05-21 03:06:43.000000000 +0300
@@ -7,8 +7,7 @@
 [info]
 
 artists = "
-   See flags.spec
-"
+   See flags.spec"
 [extra]
 sprites =
 	{	"tag", "file"
diff -Nurd -X.diff_ignore freeciv/data/misc/small.spec freeciv/data/misc/small.spec
--- freeciv/data/misc/small.spec	2012-05-21 01:28:15.000000000 +0300
+++ freeciv/data/misc/small.spec	2012-05-21 03:06:43.000000000 +0300
@@ -11,8 +11,7 @@
     Jeff Mallatt <jjm@codewell.com> (cooling flakes)
     Davide Pagnin <nightmare@freeciv.it> (angry citizens)
     Enrico Bini <e.bini@sssup.it> (fundamentalism icon)
-    Hogne Håskjold <haskjold@gmail.com> (gold coin)
-"
+    Hogne Håskjold <haskjold@gmail.com> (gold coin)"
 
 [file]
 gfx = "misc/small"
@@ -120,5 +119,4 @@
   1, 9,  "ev.wonderwillbebuilt"
   1, 10, "ev.wonderbuilt"
   1, 11, "s.plus"
-  1, 12, "s.minus"
-}
+  1, 12, "s.minus"}
diff -Nurd -X.diff_ignore freeciv/data/misc/space.spec freeciv/data/misc/space.spec
--- freeciv/data/misc/space.spec	2012-05-21 01:28:15.000000000 +0300
+++ freeciv/data/misc/space.spec	2012-05-21 03:06:43.000000000 +0300
@@ -7,8 +7,7 @@
 [info]
 
 artists = "
-    Alexandre Beraud <a_beraud@lemel.fr>
-"
+    Alexandre Beraud <a_beraud@lemel.fr>"
 
 [file]
 gfx = "misc/space"
@@ -31,5 +30,4 @@
   0,  3, "spaceship.structural"
   0,  4, "spaceship.fuel"
   0,  5, "spaceship.propulsion"
-  0,  6, "spaceship.exhaust"
-}
+  0,  6, "spaceship.exhaust"}
diff -Nurd -X.diff_ignore freeciv/data/misc/techs.spec freeciv/data/misc/techs.spec
--- freeciv/data/misc/techs.spec	2012-05-21 01:28:15.000000000 +0300
+++ freeciv/data/misc/techs.spec	2012-05-21 03:06:43.000000000 +0300
@@ -109,8 +109,7 @@
  PD - Unknown author
  a.communism         a.flight             a.refining
  a.chivalry          a.leadership         a.writing
- a.feudalism         a.mass_production    
-"
+ a.feudalism         a.mass_production    "
 
 
 [file]
@@ -209,5 +208,4 @@
   8, 4, "a.warrior_code"
   8, 5, "a.writing"
   8, 6, "a.environmentalism"
-  8, 7, "a.fundamentalism"
-}
+  8, 7, "a.fundamentalism"}
diff -Nurd -X.diff_ignore freeciv/data/misc/treaty.spec freeciv/data/misc/treaty.spec
--- freeciv/data/misc/treaty.spec	2012-05-21 01:28:15.000000000 +0300
+++ freeciv/data/misc/treaty.spec	2012-05-21 03:06:43.000000000 +0300
@@ -7,8 +7,7 @@
 [info]
 
 artists = "
-    pratheep
-"
+    pratheep"
 
 [file]
 gfx = "misc/treaty"
@@ -25,6 +24,4 @@
 ; Agree/disagree thumbs:
 
   0,  0, "treaty.agree_thumb_up"
-  0,  1, "treaty.disagree_thumb_down"
-
-}
+  0,  1, "treaty.disagree_thumb_down"}
diff -Nurd -X.diff_ignore freeciv/server/console.c freeciv/server/console.c
--- freeciv/server/console.c	2012-05-21 01:28:24.000000000 +0300
+++ freeciv/server/console.c	2012-05-21 03:06:43.000000000 +0300
@@ -60,7 +60,9 @@
     /* Make sure that message is not left to buffers when server dies */
     conn_list_iterate(game.est_connections, pconn) {
       pconn->send_buffer->do_buffer_sends = 0;
+#ifdef USE_COMPRESSION
       pconn->compression.frozen_level = 0;
+#endif
     } conn_list_iterate_end;
 
     notify_conn(NULL, NULL, E_LOG_FATAL, ftc_warning, "%s", message);
diff -Nurd -X.diff_ignore freeciv/server/ruleset.c freeciv/server/ruleset.c
--- freeciv/server/ruleset.c	2012-05-21 01:53:19.000000000 +0300
+++ freeciv/server/ruleset.c	2012-05-21 03:12:21.000000000 +0300
@@ -3830,10 +3830,12 @@
   } nation_groups_iterate_end;
   lsend_packet_ruleset_nation_groups(dest, &groups_packet);
 
+#if 0
   fc_assert(sizeof(packet.init_techs)
             == FC_MEMBER_SIZEOF(struct nation_type, init_techs));
   fc_assert(ARRAY_SIZE(packet.init_techs)
             == FC_MEMBER_ARRAY_SIZE(struct nation_type, init_techs));
+#endif
 
   nations_iterate(n) {
     packet.id = nation_number(n);
@@ -3850,7 +3852,7 @@
     packet.is_playable = n->is_playable;
     packet.is_available = n->is_available;
     packet.barbarian_type = n->barb_type;
-    memcpy(packet.init_techs, n->init_techs, sizeof(packet.init_techs));
+    //memcpy(packet.init_techs, n->init_techs, sizeof(packet.init_techs));
     memcpy(packet.init_buildings, n->init_buildings, 
            sizeof(packet.init_buildings));
     memcpy(packet.init_units, n->init_units, 
@@ -3914,12 +3916,15 @@
   memcpy(misc_p.veteran_chance, game.veteran_chance, 
          sizeof(game.veteran_chance));
 
+  /* FIXME: removed for webclient. - ANdreas. */
+#if 0
   fc_assert(sizeof(misc_p.global_init_techs)
             == sizeof(game.server.rgame.global_init_techs));
   fc_assert(ARRAY_SIZE(misc_p.global_init_techs)
             == ARRAY_SIZE(game.server.rgame.global_init_techs));
   memcpy(misc_p.global_init_techs, game.server.rgame.global_init_techs,
          sizeof(misc_p.global_init_techs));
+#endif
 
   misc_p.default_specialist = DEFAULT_SPECIALIST;
 
diff -Nurd -X.diff_ignore freeciv/server/settings.c freeciv/server/settings.c
--- freeciv/server/settings.c	2012-05-21 01:28:20.000000000 +0300
+++ freeciv/server/settings.c	2012-05-21 03:13:27.000000000 +0300
@@ -2821,7 +2821,7 @@
     setting.id = setting_number(pset);
     sz_strlcpy(setting.name, setting_name(pset));
     sz_strlcpy(setting.short_help, setting_short_help(pset));
-    sz_strlcpy(setting.extra_help, setting_extra_help(pset));
+    /* sz_strlcpy(setting.extra_help, setting_extra_help(pset)); */
     setting.category = pset->scategory;
 
     send_packet_server_setting_const(pconn, &setting);
