diff -Nurd -X.diff_ignore freeciv/common/packets.def freeciv/common/packets.def
--- freeciv/common/packets.def	2014-11-01 12:31:53.278633440 +0200
+++ freeciv/common/packets.def	2014-11-01 12:34:12.482345375 +0200
@@ -189,17 +189,17 @@
 type UFLOAT10x3         = ufloat100(float)
 type UFLOAT             = ufloat10000(float)
 type UINT8              = uint8(int)
-type UINT16             = uint16(int)
+type UINT16             = uint32(int)
 type UINT32             = uint32(int)
 
 # typedefs for arrays/structs
 type MEMORY             = memory(unsigned char)
 type REQUIREMENT        = requirement(struct requirement)
 type STRING             = string(char)
-type TECH_LIST          = tech_list(int)
-type UNIT_LIST          = unit_list(int)
-type BUILDING_LIST      = building_list(int)
-type WORKLIST           = worklist(struct worklist)
+# type TECH_LIST          = tech_list(int)
+# type UNIT_LIST          = unit_list(int)
+# type BUILDING_LIST      = building_list(int)
+type WORKLIST           = string(char)
 
 # typedefs for enums
 type ACTIVITY           = uint8(enum unit_activity)
@@ -214,7 +214,7 @@
 type DIPLOMACY_MODE     = uint8(enum diplomacy_mode)
 type DIRECTION          = sint8(enum direction8)
 type EFFECT_TYPE        = uint8(enum effect_type)
-type EVENT              = sint16(enum event_type)
+type EVENT              = sint32(enum event_type)
 type GUI_TYPE           = uint8(enum gui_type)
 type IMPR_GENUS         = uint8(enum impr_genus_id)
 type KNOWN              = uint8(enum known_type)
@@ -274,7 +274,7 @@
 type GOVERNMENT         = UINT8
 type IMPROVEMENT        = uint8(Impr_type_id)
 type MULTIPLIER         = uint8(Multiplier_type_id)
-type NATION             = sint16(Nation_type_id)
+type NATION             = UINT32
 type PLAYER             = UINT8
 type RESEARCH           = UINT8
 type RESOURCE           = uint8(Resource_type_id)
@@ -287,14 +287,14 @@
 type UNIT_TYPE          = uint8(Unit_type_id)
 
 # other typedefs
-type CITY_COORD         = UINT8
+type CITY_COORD         = UINT32
 type GOLD               = UINT32
-type HP                 = UINT8
-type PERCENT            = UINT8
+type HP                 = UINT32
+type PERCENT            = UINT32
 type PHASE              = SINT16
 type TURN               = SINT16
-type XYSIZE             = UINT16
-type YEAR               = SINT16
+type XYSIZE             = UINT32
+type YEAR               = SINT32
 type CITIZENS           = UINT8
 
 /****************************************************
@@ -416,7 +416,7 @@
 
   TERRAIN terrain;
   RESOURCE resource;
-  BV_EXTRAS extras;
+  STRING extras[MAX_EXTRA_TYPES];
   STRING spec_sprite[MAX_LEN_NAME];
   STRING label[MAX_LEN_NAME];
 end
@@ -428,7 +428,8 @@
 # The variables in the packet are listed in alphabetical order.
 PACKET_GAME_INFO = 16; sc
   UINT8 add_to_size_limit;
-  UINT8 aifill;
+  UINT32 aifill;
+  UINT32 mapsize;
   PERSISTENT_READY persistent_ready;
   AIRLIFTING_STYLE airlifting_style;
   UINT8 angrycitizen;
@@ -533,6 +534,7 @@
   STRING calendar_fragment_name[MAX_CALENDAR_FRAGMENTS][MAX_LEN_NAME];
   BOOL civil_war_enabled;
   BOOL paradrop_to_transport;
+  STRING meta_message[MAX_LEN_NAME];
 end
 
 PACKET_MAP_INFO = 17; sc, lsend
@@ -535,7 +535,7 @@ end
 # This cannot have is-info set. Sending the same value a second time after a
 # while has passed means a completely reset timeout.
 PACKET_TIMEOUT_INFO = 244; sc
-  SFLOAT10x3 seconds_to_phasedone;
+  UINT32 seconds_to_phasedone;
 end
 
 PACKET_MAP_INFO = 17; sc, lsend
@@ -611,6 +613,9 @@
   SINT16 citizen_base[O_LAST];
   SINT16 usage[O_LAST];
   UINT16 food_stock, shield_stock;
+  UINT16 granary_size;
+  TURN granary_turns;
+  UINT16 buy_gold_cost;
 
   UINT16 trade[MAX_TRADE_ROUTES];
   UINT8 trade_value[MAX_TRADE_ROUTES];
@@ -637,11 +642,19 @@
   UINT8 walls;
   SINT8 city_image;
 
-  WORKLIST worklist;
+  WORKLIST worklist[MAX_LEN_WORKLIST_WEB];
 
-  BV_IMPRS improvements;
-  BV_CITY_OPTIONS city_options;
+  STRING can_build_unit[MAX_LEN_MSG];
+  STRING can_build_improvement[MAX_LEN_MSG];
+  STRING improvements[MAX_LEN_MSG];
+
+  STRING food_output[MAX_LEN_MSG];
+  STRING shield_output[MAX_LEN_MSG];
+  STRING trade_output[MAX_LEN_MSG];
+  #BV_CITY_OPTIONS city_options;
+  BOOL disbandable_city;
   STRING name[MAX_LEN_NAME];
+  BOOL unhappy;
 end
 
 PACKET_CITY_SHORT_INFO = 32; sc, lsend, is-game-info, cancel(PACKET_CITY_INFO)
@@ -660,7 +673,7 @@
 
   SINT8 city_image;
 
-  BV_IMPRS improvements;
+  STRING improvements[MAX_LEN_MSG];
   STRING name[MAX_LEN_NAME];
 end
 
@@ -681,7 +694,7 @@
 
 PACKET_CITY_WORKLIST = 36; cs, dsend
   CITY city_id;
-  WORKLIST worklist;
+  #WORKLIST worklist[MAX_LEN_MSG];
 end
 
 PACKET_CITY_MAKE_SPECIALIST = 37; cs, dsend
@@ -782,14 +795,16 @@
   UINT8 ai_skill_level;
   UINT8 barbarian_type;
   BV_PLAYER gives_shared_vision;
-  UINT16 culture;
-  SINT16 love[MAX_NUM_PLAYER_SLOTS];
+  UINT32 culture;
+  UINT32 love[MAX_NUM_PLAYER_SLOTS];
+  UINT32 current_research_cost;
 
   UINT8 color_red;
   UINT8 color_green;
   UINT8 color_blue;
 
   CITYSPE wonders[B_LAST]; diff
+  UINT32  net_income;
   
   UINT32 multiplier[MAX_MULTIPLIERS_COUNT];
 end
@@ -821,7 +836,7 @@
   UINT32 diplstate_id; key
   PLAYER plr1;
   PLAYER plr2;
-  UINT8 type;
+  UINT32 ds_type;
   UINT16 turns_left;
   UINT8 has_reason_to_cancel;
   UINT16 contact_turns_left;
@@ -908,12 +925,12 @@
 
   UINT8 hp, activity;
   EXTRA activity_tgt;
-  UNIT transported_by; /* Only valid if transported is set. */
+#  UNIT transported_by; /* Only valid if transported is set. */
 
-  /* in packet only, not in unit struct */
-  UINT8 packet_use;	/* see enum unit_info_use */
-  CITY info_city_id;	/* for UNIT_INFO_CITY_SUPPORTED
-  			   and UNIT_INFO_CITY_PRESENT uses */
+#  /* in packet only, not in unit struct */
+#  UINT8 packet_use;	/* see enum unit_info_use */
+#  CITY info_city_id;	/* for UNIT_INFO_CITY_SUPPORTED
+#  			   and UNIT_INFO_CITY_PRESENT uses */
 end
 
 PACKET_UNIT_COMBAT_INFO = 65; sc, lsend
@@ -1020,8 +1037,8 @@
 PACKET_UNIT_DO_ACTION = 84; cs, dsend
   UNIT actor_id;
   SINT32 target_id;   # city_id, unit_id or tile_id
-  SINT16 value;
+  UINT32 value;
   STRING name[MAX_LEN_NAME];
   GEN_ACTION action_type;
 end
 
@@ -1083,25 +1100,25 @@
 
 PACKET_DIPLOMACY_CREATE_CLAUSE_REQ = 99; cs, dsend
   PLAYER counterpart, giver;
-  CLAUSE type;
+  CLAUSE clause_type;
   UINT32 value;
 end
 
 PACKET_DIPLOMACY_CREATE_CLAUSE = 100; sc, dsend, lsend
   PLAYER counterpart, giver;
-  CLAUSE type;
+  CLAUSE clause_type;
   UINT32 value;
 end
 
 PACKET_DIPLOMACY_REMOVE_CLAUSE_REQ = 101; cs, dsend
   PLAYER counterpart, giver;
-  CLAUSE type;
+  CLAUSE clause_type;
   UINT32 value;
 end
 
 PACKET_DIPLOMACY_REMOVE_CLAUSE = 102; sc, dsend, lsend
   PLAYER counterpart, giver;
-  CLAUSE type;
+  CLAUSE clause_type;
   UINT32 value;
 end
 
@@ -1129,7 +1146,7 @@
 end
 
 PACKET_REPORT_REQ = 111; cs, handle-per-conn, dsend
-  REPORT_TYPE type;
+  REPORT_TYPE report_type;
 end
 
 /************** Connection packets **********************/
@@ -1154,9 +1171,9 @@
 
 # Information about the ping times of the connections.
 PACKET_CONN_PING_INFO = 116; sc, lsend
-  UINT8 connections;
+  UINT32 connections;
   CONNECTION conn_id[MAX_NUM_CONNECTIONS:connections];
-  SFLOAT10x7 ping_time[MAX_NUM_CONNECTIONS:connections];
+  UINT32 ping_time[MAX_NUM_CONNECTIONS:connections];
 end
 
 # For compatibility with older versions, this number cannot be changed.
@@ -1320,8 +1337,9 @@
 PACKET_RULESET_GAME = 141; sc, lsend
   UINT8 default_specialist;
 
-  TECH_LIST global_init_techs[MAX_NUM_TECH_LIST];
-  BUILDING_LIST global_init_buildings[MAX_NUM_BUILDING_LIST];
+  # These are removed from freeciv-web
+  # TECH_LIST global_init_techs[MAX_NUM_TECH_LIST];
+  # BUILDING_LIST global_init_buildings[MAX_NUM_BUILDING_LIST];
 
   UINT8 veteran_levels;
   STRING veteran_name[MAX_VET_LEVELS:veteran_levels][MAX_LEN_NAME];
@@ -1443,9 +1461,9 @@
   UINT8 groups[MAX_NUM_NATION_GROUPS:ngroups];
 
   GOVERNMENT init_government_id;
-  TECH_LIST init_techs[MAX_NUM_TECH_LIST];
-  UNIT_LIST init_units[MAX_NUM_UNIT_LIST];
-  BUILDING_LIST init_buildings[MAX_NUM_BUILDING_LIST];
+  # TECH_LIST init_techs[MAX_NUM_TECH_LIST];
+  # UNIT_LIST init_units[MAX_NUM_UNIT_LIST];
+  # BUILDING_LIST init_buildings[MAX_NUM_BUILDING_LIST];
 end
 
 # Separate from PACKET_RULESET_NATION since it can change in pregame
@@ -1726,7 +1744,7 @@
   UINT16 id; key
   STRING name[MAX_LEN_NAME];
   STRING short_help[MAX_LEN_PACKET]; /* untranslated */
-  STRING extra_help[MAX_LEN_PACKET]; /* untranslated */
+  # STRING extra_help[MAX_LEN_PACKET]; /* untranslated */
   UINT8 category;
 end
 
@@ -2075,3 +2093,43 @@
   SINT32 tag;
   SINT32 id;
 end
+
+/*************** Webclient specific packets ****************/
+/* Use range 256:511 for these                             */
+# used for showing GOTO path in web client.
+PACKET_GOTO_PATH_REQ = 256; cs, dsend
+  UNIT unit_id;
+  TILE goal;
+end
+
+PACKET_GOTO_PATH = 257; sc, dsend
+  UNIT unit_id;
+  UINT16 length;
+  DIRECTION dir[MAX_LEN_ROUTE:length];
+  TILE dest;
+  UINT32 turns;
+end
+
+PACKET_INFO_TEXT_REQ = 258; cs, dsend
+  TILE loc;
+  UNIT visible_unit;
+end
+
+PACKET_INFO_TEXT_MESSAGE = 259; sc, dsend
+  STRING message[MAX_LEN_MSG];
+end
+
+PACKET_CITY_DISBANDABLE_REQ = 260; cs, dsend
+  CITY city_id;
+end
+
+PACKET_CITY_WORKLIST_ADD = 261; cs, dsend
+  CITY city_id;
+  UINT32 kind;
+  UINT32 value;
+end
+
+PACKET_CITY_WORKLIST_REMOVE = 262; cs, dsend
+  CITY city_id;
+  UINT32 index;
+end
