diff -Nurd -X.diff_ignore freeciv/utility/shared.c freeciv/utility/shared.c
--- freeciv/utility/shared.c	2015-09-14 18:08:54.439634087 +0300
+++ freeciv/utility/shared.c	2015-09-20 02:19:30.101118011 +0300
@@ -298,7 +298,7 @@
 
   fc_assert_ret_val(NULL != str, -1);
 
-  for (token = 0; token <= num_tokens && *str != '\0'; token++) {
+  for (token = 0; token < num_tokens && *str != '\0'; token++) {
     size_t len, padlength = 0;
 
     /* skip leading delimiters */
