diff -Nurd -X.diff_ignore freeciv/server/unittools.c freeciv/server/unittools.c
--- freeciv/server/unittools.c	2013-05-20 10:53:02.583786921 +0300
+++ freeciv/server/unittools.c	2013-05-24 01:51:23.118889188 +0300
@@ -2320,7 +2320,7 @@
 
   package_unit(punit, &info);
   package_short_unit(punit, &sinfo, UNIT_INFO_IDENTITY, FALSE, FALSE);
-            
+
   conn_list_iterate(dest, pconn) {
     struct player *pplayer = pconn->playing;
 
@@ -2375,11 +2375,12 @@
   conn_list_iterate(dest, pconn) {
     struct player *pplayer = pconn->playing;
 
-    if (can_player_see_unit_at(pplayer, punit, old_tile)
+    if (!conn_is_global_observer(pconn)
+        && can_player_see_unit_at(pplayer, punit, old_tile)
         && !can_player_see_unit_at(pplayer, punit,
                                    unit_tile(punit))) {
       unit_goes_out_of_sight(pplayer, punit);
-    } 
+    }
   } conn_list_iterate_end;
 }
 
