diff -Nurd -Xfreeciv/scripts/diff_ignore freeciv/client/packhand.c freeciv/client/packhand.c
--- freeciv/client/packhand.c	2021-11-03 18:44:06.413154450 +0200
+++ freeciv/client/packhand.c	2021-11-03 18:55:58.745277222 +0200
@@ -5402,3 +5402,13 @@
 {
   play_single_track(tag);
 }
+
+/************************************************************************//**
+  This is a packet that only the web-client needs. The regular client has no
+  use for it.
+  TODO: Do not generate code calling this in the C-client.
+****************************************************************************/
+void handle_goto_path(int unit_id, int length, const enum direction8 *dir,
+                      int dest, int turns)
+{
+}
diff -Nurd -Xfreeciv/scripts/diff_ignore freeciv/common/networking/packets.def freeciv/common/networking/packets.def
--- freeciv/common/networking/packets.def	2021-11-03 18:44:06.437154354 +0200
+++ freeciv/common/networking/packets.def	2021-11-03 18:52:00.194468972 +0200
@@ -2362,3 +2362,19 @@
 
   BV_ACTIONS utype_actions;
 end
+
+# used for showing GOTO path in web client.
+# TODO: Rename to contain '_WEB_' in the name.
+PACKET_GOTO_PATH_REQ = 287; cs, dsend
+  UNIT unit_id;
+  TILE goal;
+end
+
+# TODO: Rename to contain '_WEB_' in the name.
+PACKET_GOTO_PATH = 288; sc, dsend
+  UNIT unit_id;
+  UINT16 length;
+  DIRECTION dir[MAX_LEN_ROUTE:length];
+  TILE dest;
+  UINT32 turns;
+end
diff -Nurd -Xfreeciv/scripts/diff_ignore freeciv/server/unithand.c freeciv/server/unithand.c
--- freeciv/server/unithand.c	2021-11-03 18:44:06.681153381 +0200
+++ freeciv/server/unithand.c	2021-11-03 18:45:35.916620113 +0200
@@ -45,6 +45,9 @@
 #include "unit.h"
 #include "unitlist.h"
 
+/* common/aicore */
+#include "pf_tools.h"
+
 /* common/scriptcore */
 #include "luascript_types.h"
 
@@ -4087,6 +4090,88 @@
 }
 
 /**********************************************************************//**
+  This function handles GOTO path requests from the client.
+**************************************************************************/
+void handle_goto_path_req(struct player *pplayer, int unit_id, int goal)
+{
+  struct unit *punit = player_unit_by_number(pplayer, unit_id);
+  struct tile *ptile = index_to_tile(&(wld.map), goal);
+  struct pf_parameter parameter;
+  struct pf_map *pfm;
+  struct pf_path *path;
+  struct tile *old_tile;
+  int i = 0;
+  struct packet_goto_path p;
+
+  if (NULL == punit) {
+    /* Shouldn't happen */
+    log_error("handle_unit_move()"
+              " invalid unit %d",
+              unit_id);
+    return;
+  }
+
+  if (NULL == ptile) {
+    /* Shouldn't happen */
+    log_error("handle_unit_move()"
+              " invalid %s (%d) tile (%d,%d)",
+              unit_rule_name(punit),
+              unit_id,
+              TILE_XY(ptile));
+    return;
+  }
+
+  if (!is_player_phase(unit_owner(punit), game.info.phase)) {
+    /* Client is out of sync, ignore */
+    log_verbose("handle_unit_move()"
+                " invalid %s (%d) %s != phase %d",
+                unit_rule_name(punit),
+                unit_id,
+                nation_rule_name(nation_of_unit(punit)),
+                game.info.phase);
+    return;
+  }
+
+  p.unit_id = punit->id;
+  p.dest = tile_index(ptile);
+
+  /* Use path-finding to find a goto path. */
+  pft_fill_unit_parameter(&parameter, punit);
+  pfm = pf_map_new(&parameter);
+  path = pf_map_path(pfm, ptile);
+  pf_map_destroy(pfm);
+
+  if (path) {
+    int total_mc = 0;
+
+    p.length = path->length - 1;
+
+    old_tile = path->positions[0].tile;
+
+    for (i = 0; i < path->length - 1; i++) {
+      struct tile *new_tile = path->positions[i + 1].tile;
+      int dir;
+
+      total_mc += path->positions[1].total_MC;
+      if (same_pos(new_tile, old_tile)) {
+        dir = -1;
+      } else {
+        dir = get_direction_for_step(&(wld.map), old_tile, new_tile);
+      }
+      old_tile = new_tile;
+      p.dir[i] = dir;
+
+    }
+    pf_path_destroy(path);
+    p.turns = total_mc / unit_move_rate(punit);
+    send_packet_goto_path(pplayer->current_conn, &p);
+
+  } else {
+    return;
+  }
+}
+
+/**********************************************************************//**
   Handle change in unit activity.
 **************************************************************************/
 static void handle_unit_change_activity_real(struct player *pplayer,
