diff -Nurd -X.diff_ignore freeciv/server/unithand.c freeciv/server/unithand.c
--- freeciv/server/unithand.c	2012-05-21 14:12:04.000000000 +0300
+++ freeciv/server/unithand.c	2012-05-21 14:20:42.000000000 +0300
@@ -824,6 +824,15 @@
 
     old_tile = path->positions[0].tile;
 
+    /* Remove city spot reservations for AI settlers on city founding
+     * mission, before goto_tile reset. */
+    if (punit->server.adv->role != AIUNIT_NONE) {
+      ai_unit_new_role(punit, AIUNIT_NONE, NULL);
+    }
+
+    punit->ai_controlled = FALSE;
+    punit->goto_tile = NULL;
+
     free_unit_orders(punit);
     /* If we waited on a tile, reset punit->done_moving */
     punit->done_moving = (punit->moves_left <= 0);
