diff -Nurd -X.diff_ignore freeciv/server/unithand.c freeciv/server/unithand.c
--- freeciv/server/unithand.c	2012-05-19 04:40:15.000000000 +0300
+++ freeciv/server/unithand.c	2012-05-19 04:42:58.000000000 +0300
@@ -855,9 +855,6 @@
     pf_path_destroy(path);
     return ;
   } else {
-    notify_player(pplayer, punit->tile, E_BAD_COMMAND,
-                  ftc_server, NULL,
-                  _("The unit can't go there."));
     return ;
   }
 
