diff -Nurd -X.diff_ignore freeciv/server/unittools.c freeciv/server/unittools.c
--- freeciv/server/unittools.c	2012-05-16 21:02:54.000000000 +0300
+++ freeciv/server/unittools.c	2012-05-16 21:48:07.000000000 +0300
@@ -3222,7 +3222,8 @@
 	return TRUE;
       }
 
-      if (!last_order
+      /* FIXME: the web client want gotos to result in attack.
+       if (!last_order
 	  && maybe_cancel_goto_due_to_enemy(punit, dst_tile)) {
 	cancel_orders(punit, "  orders canceled because of enemy");
 	notify_player(pplayer, punit->tile, E_UNIT_ORDERS,
@@ -3231,7 +3232,7 @@
                         "are units in the way."),
                       unit_link(punit));
 	return TRUE;
-      }
+      }*/
 
       freelog(LOG_DEBUG, "  moving to %d,%d",
 	      dst_tile->x, dst_tile->y);
