diff -Nurd -X.diff_ignore freeciv/server/unittools.c freeciv/server/unittools.c
--- freeciv/server/unittools.c	2012-05-20 02:59:26.000000000 +0300
+++ freeciv/server/unittools.c	2012-05-20 03:00:34.000000000 +0300
@@ -3227,7 +3227,9 @@
         return TRUE;
       }
 
-      if (!last_order
+      /* FIXME: the web client want gotos to result in attack. */
+#if 0
+       if (!last_order
           && maybe_cancel_goto_due_to_enemy(punit, dst_tile)) {
         cancel_orders(punit, "  orders canceled because of enemy");
         notify_player(pplayer, unit_tile(punit), E_UNIT_ORDERS, ftc_server,
@@ -3236,6 +3238,7 @@
                       unit_link(punit));
         return TRUE;
       }
+#endif /* web client */
 
       freelog(LOG_DEBUG, "  moving to %d,%d",
 	      dst_tile->x, dst_tile->y);
