diff -Nurd -X.diff_ignore freeciv/server/unithand.c freeciv/server/unithand.c
--- freeciv/server/unithand.c	2012-05-20 04:28:48.000000000 +0300
+++ freeciv/server/unithand.c	2012-05-20 04:34:09.000000000 +0300
@@ -61,6 +61,8 @@
 #include "techtools.h"
 #include "unittools.h"
 
+#include "pf_tools.h"
+
 #include "unithand.h"
 
 static void city_add_unit(struct player *pplayer, struct unit *punit);
@@ -765,6 +767,11 @@
 {
   struct unit *punit = player_find_unit_by_id(pplayer, unit_id);
   struct tile *ptile = index_to_tile(tile);
+  struct pf_parameter parameter;
+  struct pf_map *pfm;
+  struct pf_path *path;
+  struct tile *old_tile;
+  int i = 0;
 
   if (NULL == punit) {
     /* Probably died or bribed. */
@@ -779,15 +786,6 @@
     return;
   }
 
-  if (!is_tiles_adjacent(unit_tile(punit), ptile)) {
-    /* Client is out of sync, ignore */
-    log_verbose("handle_unit_move() invalid %s (%d) move "
-                "from (%d, %d) to (%d, %d).",
-                unit_rule_name(punit), unit_id,
-                TILE_XY(unit_tile(punit)), TILE_XY(ptile));
-    return;
-  }
-
   if (!is_player_phase(unit_owner(punit), game.info.phase)) {
     /* Client is out of sync, ignore */
     log_verbose("handle_unit_move() invalid %s (%d) %s != phase %d",
@@ -803,7 +801,56 @@
     unit_activity_handling(punit, ACTIVITY_IDLE);
   }
 
-  (void) unit_move_handling(punit, ptile, FALSE, FALSE);
+  /* Use path-finding to find a goto path. */
+  pft_fill_unit_parameter(&parameter, punit);
+  pfm = pf_map_new(&parameter);
+  path = pf_map_get_path(pfm, ptile);
+  pf_map_destroy(pfm);
+
+  if (path) {
+    //send_goto_path(punit, path, NULL);
+
+    old_tile = path->positions[0].tile;
+
+    free_unit_orders(punit);
+    /* If we waited on a tile, reset punit->done_moving */
+    punit->done_moving = (punit->moves_left <= 0);
+    punit->has_orders = TRUE;
+    punit->orders.length = path->length;
+    punit->orders.index = 0;
+    punit->orders.repeat = FALSE;
+    punit->orders.vigilant = FALSE;
+    punit->orders.list
+      = fc_malloc(path->length * sizeof(*(punit->orders.list)));
+    for (i = 0; i < path->length - 1; i++) {
+      struct tile *new_tile = path->positions[i + 1].tile;
+
+      if (same_pos(new_tile, old_tile)) {
+        punit->orders.list[i].order = ORDER_FULL_MP;
+        punit->orders.list[i].dir = -1;
+      } else {
+        punit->orders.list[i].order = ORDER_MOVE;
+        punit->orders.list[i].dir = get_direction_for_step(old_tile, new_tile);
+        punit->orders.list[i].activity = ACTIVITY_LAST;
+      }
+      old_tile = new_tile;
+
+    }
+
+   punit->goto_tile = ptile;
+
+   if (!is_player_phase(unit_owner(punit), game.info.phase)
+      || execute_orders(punit)) {
+    /* Looks like the unit survived. */
+     send_unit_info(NULL, punit);
+    }
+
+    pf_path_destroy(path);
+    return ;
+  } else {
+    return ;
+  }
+
 }
 
 /**************************************************************************
