diff -Nurd -X.diff_ignore freeciv/client/packhand.c freeciv/client/packhand.c
--- freeciv/client/packhand.c	2012-07-16 00:07:35.000000000 +0300
+++ freeciv/client/packhand.c	2012-07-16 00:16:17.000000000 +0300
@@ -2753,10 +2753,12 @@
 
   gov->item_number = p->id;
 
+#if 0
   for (j = 0; j < p->reqs_count; j++) {
     requirement_vector_append(&gov->reqs, p->reqs[j]);
   }
   fc_assert(gov->reqs.size == p->reqs_count);
+#endif
 
   names_set(&gov->name, p->name, p->rule_name);
   sz_strlcpy(gov->graphic_str, p->graphic_str);
diff -Nurd -X.diff_ignore freeciv/common/packets.def freeciv/common/packets.def
--- freeciv/common/packets.def	2012-07-16 00:11:03.000000000 +0300
+++ freeciv/common/packets.def	2012-07-16 00:15:29.000000000 +0300
@@ -1190,8 +1190,8 @@
 PACKET_RULESET_GOVERNMENT = 145; sc, lsend
   GOVERNMENT id;
 
-  UINT8 reqs_count;
-  REQUIREMENT reqs[MAX_NUM_REQS:reqs_count];
+  #UINT8 reqs_count;
+  #REQUIREMENT reqs[MAX_NUM_REQS:reqs_count];
 
   STRING name[MAX_LEN_NAME];
   STRING rule_name[MAX_LEN_NAME];
diff -Nurd -X.diff_ignore freeciv/server/ruleset.c freeciv/server/ruleset.c
--- freeciv/server/ruleset.c	2012-07-15 21:52:14.000000000 +0300
+++ freeciv/server/ruleset.c	2012-07-16 00:15:29.000000000 +0300
@@ -3787,10 +3787,10 @@
     gov.id = government_number(g);
 
     j = 0;
-    requirement_vector_iterate(&g->reqs, preq) {
+    /*requirement_vector_iterate(&g->reqs, preq) {
       gov.reqs[j++] = *preq;
     } requirement_vector_iterate_end;
-    gov.reqs_count = j;
+    gov.reqs_count = j;*/
 
     sz_strlcpy(gov.name, untranslated_name(&g->name));
     sz_strlcpy(gov.rule_name, rule_name(&g->name));
