diff -Nurd -X.diff_ignore freeciv/common/actions.c freeciv/common/actions.c
--- freeciv/common/actions.c	2016-08-05 22:56:33.622668063 +0300
+++ freeciv/common/actions.c	2016-08-10 23:10:23.700333774 +0300
@@ -1569,7 +1569,7 @@
   case ACTION_HOME_CITY:
     /* Reason: can't change to what is. */
     /* Info leak: The player knows his unit's current home city. */
-    if (homecity->id == target_city->id) {
+    if (homecity != NULL && homecity->id == target_city->id) {
       /* This is already the unit's home city. */
       return TRI_NO;
     }
