Index: freeciv/server/savegame2.c
===================================================================
--- freeciv/server/savegame2.c	(revision 25862)
+++ freeciv/server/savegame2.c	(working copy)
@@ -3850,6 +3850,8 @@
   research->got_tech
     = secfile_lookup_bool_default(loading->file, FALSE,
                                   "player%d.research.got_tech", plrno);
+  research->researching_cost 
+    = research_total_bulbs_required(research, research->researching, FALSE);
 
   string = secfile_lookup_str(loading->file, "player%d.research.done", plrno);
   sg_failure_ret(string != NULL, "%s", secfile_error());
