diff -Nurd -X.diff_ignore freeciv/common/networking/packets.def freeciv/common/networking/packets.def
--- freeciv/common/networking/packets.def	2016-02-24 02:14:06.343479857 +0200
+++ freeciv/common/networking/packets.def	2016-02-24 02:20:02.270903922 +0200
@@ -1347,8 +1347,8 @@
   UINT8 default_specialist;
 
   # These are removed from freeciv-web
-  TECH_LIST global_init_techs[MAX_NUM_TECH_LIST];
-  BUILDING_LIST global_init_buildings[MAX_NUM_BUILDING_LIST];
+  # TECH_LIST global_init_techs[MAX_NUM_TECH_LIST];
+  # BUILDING_LIST global_init_buildings[MAX_NUM_BUILDING_LIST];
 
   UINT8 veteran_levels;
   STRING veteran_name[MAX_VET_LEVELS:veteran_levels][MAX_LEN_NAME];
@@ -1475,9 +1475,9 @@
   UINT8 groups[MAX_NUM_NATION_GROUPS:ngroups];
 
   GOVERNMENT init_government_id;
-  TECH_LIST init_techs[MAX_NUM_TECH_LIST];
-  UNIT_LIST init_units[MAX_NUM_UNIT_LIST];
-  BUILDING_LIST init_buildings[MAX_NUM_BUILDING_LIST];
+  # TECH_LIST init_techs[MAX_NUM_TECH_LIST];
+  # UNIT_LIST init_units[MAX_NUM_UNIT_LIST];
+  # BUILDING_LIST init_buildings[MAX_NUM_BUILDING_LIST];
 end
 
 # Separate from PACKET_RULESET_NATION since it can change in pregame
diff -Nurd -X.diff_ignore freeciv/server/ruleset.c freeciv/server/ruleset.c
--- freeciv/server/ruleset.c	2016-02-24 02:18:56.395009804 +0200
+++ freeciv/server/ruleset.c	2016-02-24 02:19:01.371001797 +0200
@@ -6786,6 +6786,8 @@
 
     packet.init_government_id = n->init_government
       ? government_number(n->init_government) : government_count();
+
+#if 0
     fc_assert(ARRAY_SIZE(packet.init_techs) == ARRAY_SIZE(n->init_techs));
     for (i = 0; i < MAX_NUM_TECH_LIST; i++) {
       packet.init_techs[i] = n->init_techs[i];
@@ -6801,6 +6803,7 @@
       /* Impr_type_id to int */
       packet.init_buildings[i] = n->init_buildings[i];
     }
+#endif
 
     lsend_packet_ruleset_nation(dest, &packet);
   } nations_iterate_end;
