diff -Nurd -X.diff_ignore freeciv/utility/genlist.c freeciv/utility/genlist.c
--- freeciv/utility/genlist.c	2015-09-30 20:13:07.659352223 +0300
+++ freeciv/utility/genlist.c	2016-02-22 02:28:04.254811182 +0200
@@ -25,13 +25,6 @@
 
 #include "genlist.h"
 
-/* A single element of a genlist, storing the pointer to user
- * data, and pointers to the next and previous elements: */
-struct genlist_link {
-  struct genlist_link *next, *prev;
-  void *dataptr;
-};
-
 /****************************************************************************
   Create a new empty genlist.
 ****************************************************************************/
@@ -699,27 +692,3 @@
 {
   fc_release_mutex(&pgenlist->mutex);
 }
-
-/****************************************************************************
-  Returns the pointer of this link.
-****************************************************************************/
-void *genlist_link_data(const struct genlist_link *plink)
-{
-  return (NULL != plink ? plink->dataptr : NULL);
-}
-
-/****************************************************************************
-  Returns the previous link.
-****************************************************************************/
-struct genlist_link *genlist_link_prev(const struct genlist_link *plink)
-{
-  return plink->prev;
-}
-
-/****************************************************************************
-  Returns the next link.
-****************************************************************************/
-struct genlist_link *genlist_link_next(const struct genlist_link *plink)
-{
-  return plink->next;
-}
diff -Nurd -X.diff_ignore freeciv/utility/genlist.h freeciv/utility/genlist.h
--- freeciv/utility/genlist.h	2015-09-30 20:13:07.659352223 +0300
+++ freeciv/utility/genlist.h	2016-02-22 02:33:37.370126830 +0200
@@ -124,12 +124,40 @@
 void genlist_allocate_mutex(struct genlist *pgenlist);
 void genlist_release_mutex(struct genlist *pgenlist);
 
-void *genlist_link_data(const struct genlist_link *plink);
-struct genlist_link *genlist_link_prev(const struct genlist_link *plink)
-                     fc__warn_unused_result;
-struct genlist_link *genlist_link_next(const struct genlist_link *plink)
-                     fc__warn_unused_result;
 
+/* A single element of a genlist, storing the pointer to user
+ * data, and pointers to the next and previous elements: */
+struct genlist_link {
+  struct genlist_link *next, *prev;
+  void *dataptr;
+};
+
+/****************************************************************************
+  Returns the pointer of this link.
+****************************************************************************/
+static inline void *genlist_link_data(const struct genlist_link *plink)
+{
+  return (NULL != plink ? plink->dataptr : NULL);
+}
+
+/****************************************************************************
+  Returns the previous link.
+****************************************************************************/
+fc__warn_unused_result
+static inline struct genlist_link *genlist_link_prev(const struct genlist_link *plink)
+{
+  return plink->prev;
+}
+
+/****************************************************************************
+  Returns the next link.
+****************************************************************************/
+fc__warn_unused_result
+static inline struct genlist_link *genlist_link_next(const struct genlist_link *plink)
+{
+  return plink->next;
+}
+  
 #ifdef __cplusplus
 }
 #endif /* __cplusplus */
