diff -Nurd -X.diff_ignore freeciv/data/default/terrain.ruleset freeciv/data/default/terrain.ruleset
--- freeciv/data/default/terrain.ruleset	2012-05-16 21:02:54.000000000 +0300
+++ freeciv/data/default/terrain.ruleset	2012-05-16 21:35:15.000000000 +0300
@@ -32,7 +32,7 @@
 
 ; Bodies of water up to this size are considered freshwater lakes, if there is
 ; also terrain type(s) with flag "FreshWater" in the ruleset.
-lake_max_size = 14
+lake_max_size = 0 
 
 ; There is no resources in the middle of the oceans, only near coast
 ocean_resources = 0
