diff -Nurd freeciv/server/stdinhand.c freeciv/server/stdinhand.c
--- freeciv/server/stdinhand.c	2023-11-18 19:23:37.229547468 +0200
+++ freeciv/server/stdinhand.c	2023-11-18 19:23:44.417570250 +0200
@@ -3982,6 +3982,9 @@
     } players_iterate_end;
   } achievements_iterate_end;
 
+  /* Do not translate, as freeciv-web relies on exact string */
+  cmd_reply(CMD_DEFAULT, caller, C_OK, "Load complete");
+
   return TRUE;
 }
 
