diff --git a/server/settings.c b/server/settings.c
index 455d2e2..3335023 100644
--- a/server/settings.c
+++ b/server/settings.c
@@ -1050,6 +1050,10 @@ static bool timeout_callback(int value, struct connection *caller,
     return FALSE;
   }
 
+  if ((srvarg.port == 6003 || srvarg.port == 6004) && S_S_RUNNING == server_state()) {
+    return FALSE;  // can't change timeout on LongTurn game.
+  }
+   
   if (value == -1 && game.server.unitwaittime != 0) {
     /* autogame only with 'unitwaittime' = 0 */
     settings_snprintf(reject_msg, reject_msg_len,
diff --git a/server/stdinhand.c b/server/stdinhand.c
index a19a6ba..03f64d2 100644
--- a/server/stdinhand.c
+++ b/server/stdinhand.c
@@ -5785,6 +5785,10 @@ bool start_command(struct connection *caller, bool check, bool notify)
       }
     } players_iterate_end;
 
+    if ((srvarg.port == 6003 || srvarg.port == 6004) && game.server.min_players < 10) {
+      game.server.min_players = 10;
+    }
+
     /* check min_players.
      * Allow continuing of savegames where some of the original
      * players have died */
