diff --git a/server/connecthand.c b/server/connecthand.c
index d74f178..d253eb4 100644
--- a/server/connecthand.c
+++ b/server/connecthand.c
@@ -539,6 +539,7 @@ void send_conn_info_remove(struct conn_list *src, struct conn_list *dest)
 **************************************************************************/
 struct player *find_uncontrolled_player(void)
 {
+  if (srvarg.port == 6003) return NULL;
   players_iterate(played) {
     if (!played->is_connected && !played->was_created) {
       return played;
diff --git a/server/plrhand.c b/server/plrhand.c
index 14a9df2..2917c7b 100644
--- a/server/plrhand.c
+++ b/server/plrhand.c
@@ -1663,6 +1663,8 @@ void server_remove_player(struct player *pplayer)
 {
   const struct player_slot *pslot;
 
+  if (srvarg.port == 6003 && NULL != pplayer->nation) return;
+
   fc_assert_ret(NULL != pplayer);
 
   /* save player slot */
@@ -2429,7 +2431,7 @@ void fit_nationset_to_players(void)
 ****************************************************************************/
 void reset_all_start_commands(bool plrchange)
 {
-  if (S_S_INITIAL != server_state()) {
+  if (S_S_INITIAL != server_state() || srvarg.port == 6003) {
     return;
   }
   players_iterate(pplayer) {
diff --git a/server/settings.c b/server/settings.c
index 455d2e2..4b32a50 100644
--- a/server/settings.c
+++ b/server/settings.c
@@ -617,6 +617,8 @@ static void scorelog_action(const struct setting *pset)
 *************************************************************************/
 static void aifill_action(const struct setting *pset)
 {
+  if (srvarg.port == 6003) *pset->integer.pvalue = 0;
+
   const char *msg = aifill(*pset->integer.pvalue);
   if (msg) {
     log_normal(_("Warning: aifill not met: %s."), msg);
@@ -1050,6 +1052,10 @@ static bool timeout_callback(int value, struct connection *caller,
     return FALSE;
   }
 
+  if (srvarg.port == 6003 && S_S_RUNNING == server_state()) {
+    return FALSE;  // can't change timeout on LongTurn game.
+  }
+   
   if (value == -1 && game.server.unitwaittime != 0) {
     /* autogame only with 'unitwaittime' = 0 */
     settings_snprintf(reject_msg, reject_msg_len,
diff --git a/server/stdinhand.c b/server/stdinhand.c
index a19a6ba..c27309f 100644
--- a/server/stdinhand.c
+++ b/server/stdinhand.c
@@ -3807,6 +3807,8 @@ static bool set_rulesetdir(struct connection *caller, char *str, bool check,
   char filename[512];
   const char *pfilename;
 
+  if (srvarg.port == 6003) str = "longturn\0";
+
   if (NULL == str || '\0' == str[0]) {
     cmd_reply(CMD_RULESETDIR, caller, C_SYNTAX,
               _("You must provide a ruleset name. Use \"/show ruleset\" to "
@@ -5785,6 +5787,10 @@ bool start_command(struct connection *caller, bool check, bool notify)
       }
     } players_iterate_end;
 
+    if (srvarg.port == 6003 && game.server.min_players < 150) {
+      game.server.min_players = 150; 
+    }
+
     /* check min_players.
      * Allow continuing of savegames where some of the original
      * players have died */
