diff --git a/freeciv/common/packets.def b/freeciv/common/packets.def
index ff5f4a6..7fd7584 100644
--- a/freeciv/common/packets.def
+++ b/freeciv/common/packets.def
@@ -372,7 +372,9 @@ PACKET_GAME_INFO=15; sc
   UINT32 tech;
   UINT32 skill_level;
   
-  UINT8 aifill;
+  UINT32 aifill;
+  
+  UINT32 mapsize;
 
   BOOL is_new_game;   # TRUE only in pregame for "new" (not loaded) games
   BOOL is_edit_mode;  # If set, editing is allowed
diff --git a/freeciv/common/packets_gen.c b/freeciv/common/packets_gen.c
index 4f54f0b..2f97c4b 100644
--- a/freeciv/common/packets_gen.c
+++ b/freeciv/common/packets_gen.c
@@ -2558,9 +2558,15 @@ static struct packet_game_info *receive_packet_game_info_100(struct connection *
   {
     int readin;
   
-    dio_get_uint8(&din, &readin);
+    dio_get_uint32(&din, &readin);
     real_packet->aifill = readin;
   }
+  {
+    int readin;
+  
+    dio_get_uint32(&din, &readin);
+    real_packet->mapsize = readin;
+  }
   dio_get_bool8(&din, &real_packet->is_new_game);
   dio_get_bool8(&din, &real_packet->is_edit_mode);
   {
@@ -3198,7 +3204,8 @@ static int send_packet_game_info_100(struct connection *pc, const struct packet_
   dio_put_uint32(&dout, real_packet->gold);
   dio_put_uint32(&dout, real_packet->tech);
   dio_put_uint32(&dout, real_packet->skill_level);
-  dio_put_uint8(&dout, real_packet->aifill);
+  dio_put_uint32(&dout, real_packet->aifill);
+  dio_put_uint32(&dout, real_packet->mapsize);
   dio_put_bool8(&dout, real_packet->is_new_game);
   dio_put_bool8(&dout, real_packet->is_edit_mode);
   dio_put_uint32(&dout, (int)(real_packet->seconds_to_phasedone * 10000));
diff --git a/freeciv/common/packets_gen.h b/freeciv/common/packets_gen.h
index 6392844..dcb92ac 100644
--- a/freeciv/common/packets_gen.h
+++ b/freeciv/common/packets_gen.h
@@ -103,6 +103,7 @@ struct packet_game_info {
   int tech;
   int skill_level;
   int aifill;
+  int mapsize;
   bool is_new_game;
   bool is_edit_mode;
   float seconds_to_phasedone;
diff --git a/freeciv/server/gamehand.c b/freeciv/server/gamehand.c
index baee26b..b6d8c66 100644
--- a/freeciv/server/gamehand.c
+++ b/freeciv/server/gamehand.c
@@ -369,6 +369,8 @@ void send_game_info(struct conn_list *dest)
     ginfo.seconds_to_phasedone = -1.0;
   }
 
+  ginfo.mapsize = map.server.size;
+
   conn_list_iterate(dest, pconn) {
     send_packet_game_info(pconn, &ginfo);
   }
