diff -Nurd -X.diff_ignore freeciv/common/packets.def freeciv/common/packets.def
--- freeciv/common/packets.def	2012-05-16 21:55:12.000000000 +0300
+++ freeciv/common/packets.def	2012-05-16 21:56:32.000000000 +0300
@@ -371,7 +371,9 @@
   UINT32 tech;
   UINT32 skill_level;
   
-  UINT8 aifill;
+  UINT32 aifill;
+  
+  UINT32 mapsize;
 
   BOOL is_new_game;   # TRUE only in pregame for "new" (not loaded) games
   BOOL is_edit_mode;  # If set, editing is allowed
diff -Nurd -X.diff_ignore freeciv/server/gamehand.c freeciv/server/gamehand.c
--- freeciv/server/gamehand.c	2012-05-16 20:56:43.000000000 +0300
+++ freeciv/server/gamehand.c	2012-05-16 21:56:36.000000000 +0300
@@ -369,6 +369,8 @@
     ginfo.seconds_to_phasedone = -1.0;
   }
 
+  ginfo.mapsize = map.server.size;
+
   conn_list_iterate(dest, pconn) {
     send_packet_game_info(pconn, &ginfo);
   }
