diff -Nurd -X.diff_ignore freeciv/common/map.h freeciv/common/map.h
--- freeciv/common/map.h	2012-05-21 12:41:30.000000000 +0300
+++ freeciv/common/map.h	2012-05-21 14:47:57.000000000 +0300
@@ -488,9 +488,9 @@
 /* Size of the map in thousands of tiles.  Setting the maximal size over
  * than 30 is dangerous, because some parts of the code (e.g. path finding)
  * assume the tile index is of type (signed short int). */
-#define MAP_DEFAULT_SIZE         4
+#define MAP_DEFAULT_SIZE         3
 #define MAP_MIN_SIZE             1
-#define MAP_MAX_SIZE             128
+#define MAP_MAX_SIZE             15
 
 #define MAP_DEFAULT_TILESPERPLAYER      100
 #define MAP_MIN_TILESPERPLAYER            1
