diff --git a/common/fc_types.h b/common/fc_types.h
index 8ff8e70..6157b1d 100644
--- a/common/fc_types.h
+++ b/common/fc_types.h
@@ -29,7 +29,7 @@ extern "C" {
  * Nothing in this file should require anything else from the common/
  * directory! */
 
-#define MAX_NUM_PLAYER_SLOTS 160 /* Used in the network protocol. */
+#define MAX_NUM_PLAYER_SLOTS 256 /* Used in the network protocol. */
                                  /* Must be divisable by 32 or iterations
                                   * in savegame2.c needs to be changed */
 #define MAX_NUM_BARBARIANS   10  /* 3, but slots reserved for future use. */
