diff --git a/common/fc_types.h b/common/fc_types.h
index 8ff8e70..47ec371 100644
--- a/common/fc_types.h
+++ b/common/fc_types.h
@@ -29,10 +29,10 @@ extern "C" {
  * Nothing in this file should require anything else from the common/
  * directory! */
 
-#define MAX_NUM_PLAYER_SLOTS 160 /* Used in the network protocol. */
+#define MAX_NUM_PLAYER_SLOTS 256 /* Used in the network protocol. */
                                  /* Must be divisable by 32 or iterations
                                   * in savegame2.c needs to be changed */
-#define MAX_NUM_BARBARIANS   10  /* 3, but slots reserved for future use. */
+#define MAX_NUM_BARBARIANS   6  /* 3, but slots reserved for future use. */
 #define MAX_NUM_PLAYERS      MAX_NUM_PLAYER_SLOTS - MAX_NUM_BARBARIANS
 /* Used in the network protocol. */
 #define MAX_NUM_CONNECTIONS (2 * (MAX_NUM_PLAYER_SLOTS))
