diff -Nurd freeciv/meson.build freeciv/meson.build
--- freeciv/meson.build	2023-05-31 16:37:02.790730613 +0300
+++ freeciv/meson.build	2023-05-31 16:37:27.654994470 +0300
@@ -3833,7 +3833,8 @@
   'sandbox',
   'civ1',
   'civ2',
-  'stub'
+  'stub',
+  'webperimental'
   ]
 
 ruleset_files = [
@@ -3861,6 +3862,8 @@
   'data/sandbox.modpack',
   'data/civ1.modpack',
   'data/civ2.modpack',
+  'data/webperimental.serv',
+  'data/webperimental.modpack',
   install_dir : join_paths(get_option('datadir'), 'freeciv')
   )
 
@@ -3895,6 +3898,11 @@
   )
 
 install_data(
+  'data/webperimental/README.webperimental',
+  install_dir : join_paths(get_option('datadir'), 'freeciv/webperimental')
+  )
+
+install_data(
   'data/alien/nation/adventurers.ruleset',
   'data/alien/nation/galacticsound.ruleset',
   'data/alien/nation/jw.ruleset',
