diff -Nurd freeciv/meson.build freeciv/meson.build
--- freeciv/meson.build	2024-03-24 09:39:05.128314187 +0200
+++ freeciv/meson.build	2024-03-24 09:40:19.164840153 +0200
@@ -4356,7 +4356,8 @@
   'civ1',
   'civ2',
   'granularity',
-  'stub'
+  'stub',
+  'webperimental'
   ]
 
 ruleset_files = [
@@ -4385,6 +4386,7 @@
   'data/civ1.modpack',
   'data/civ2.modpack',
   'data/granularity.modpack',
+  'data/webperimental.modpack',
   install_dir : join_paths(get_option('datadir'), 'freeciv')
   )
 
@@ -4424,6 +4426,11 @@
   )
 
 install_data(
+  'data/webperimental/README.webperimental',
+  install_dir : join_paths(get_option('datadir'), 'freeciv/webperimental')
+  )
+
+install_data(
   'data/alien/nation/adventurers.ruleset',
   'data/alien/nation/galacticsound.ruleset',
   'data/alien/nation/jw.ruleset',
