diff -Nurd freeciv/meson.build freeciv/meson.build
--- freeciv/meson.build	2023-01-03 23:59:50.121401041 +0200
+++ freeciv/meson.build	2023-01-04 00:00:07.345555922 +0200
@@ -3547,7 +3547,8 @@
   'sandbox',
   'civ1',
   'civ2',
-  'stub'
+  'stub',
+  'webperimental'
   ]
 
 ruleset_files = [
@@ -3574,6 +3575,8 @@
   'data/sandbox.modpack',
   'data/civ1.modpack',
   'data/civ2.modpack',
+  'data/webperimental.serv',
+  'data/webperimental.modpack',
   install_dir : join_paths(get_option('datadir'), 'freeciv')
   )
 
@@ -3608,6 +3611,11 @@
   )
 
 install_data(
+  'data/webperimental/README.webperimental',
+  install_dir : join_paths(get_option('datadir'), 'freeciv/webperimental')
+  )
+
+install_data(
   'data/alien/nation/adventurers.ruleset',
   'data/alien/nation/galacticsound.ruleset',
   'data/alien/nation/jw.ruleset',
