diff --git a/meson.build b/meson.build
index 06ad423591..0db6cc3329 100644
--- a/meson.build
+++ b/meson.build
@@ -3079,7 +3079,8 @@ rulesets = [
   'sandbox',
   'civ1',
   'civ2',
-  'stub'
+  'stub',
+  'webperimental'
   ]
 
 ruleset_files = [
@@ -3105,6 +3106,8 @@ install_data(
   'data/sandbox.modpack',
   'data/civ1.modpack',
   'data/civ2.modpack',
+  'data/webperimental.serv',
+  'data/webperimental.modpack',
   install_dir : join_paths(get_option('datadir'), 'freeciv')
   )
 
@@ -3142,6 +3145,11 @@ install_data(
   install_dir : join_paths(get_option('datadir'), 'freeciv/alien')
   )
 
+install_data(
+  'data/webperimental/README.webperimental',
+  install_dir : join_paths(get_option('datadir'), 'freeciv/webperimental')
+  )
+
 install_data(
   'data/alien/nation/adventurers.ruleset',
   'data/alien/nation/galacticsound.ruleset',
