diff -Nurd freeciv/meson.build freeciv/meson.build
--- freeciv/meson.build	2023-07-08 22:29:27.412954972 +0300
+++ freeciv/meson.build	2023-07-08 22:29:38.109087948 +0300
@@ -3913,7 +3913,8 @@
   'sandbox',
   'civ1',
   'civ2',
-  'stub'
+  'stub',
+  'webperimental'
   ]
 
 ruleset_files = [
@@ -3941,6 +3942,8 @@
   'data/sandbox.modpack',
   'data/civ1.modpack',
   'data/civ2.modpack',
+  'data/webperimental.serv',
+  'data/webperimental.modpack',
   install_dir : join_paths(get_option('datadir'), 'freeciv')
   )
 
@@ -3975,6 +3978,11 @@
   )
 
 install_data(
+  'data/webperimental/README.webperimental',
+  install_dir : join_paths(get_option('datadir'), 'freeciv/webperimental')
+  )
+
+install_data(
   'data/alien/nation/adventurers.ruleset',
   'data/alien/nation/galacticsound.ruleset',
   'data/alien/nation/jw.ruleset',
