diff --git a/gen_headers/meson_freeciv_config.h.in b/gen_headers/meson_freeciv_config.h.in
index b81ca74e4c..a902128bf4 100644
--- a/gen_headers/meson_freeciv_config.h.in
+++ b/gen_headers/meson_freeciv_config.h.in
@@ -45,7 +45,7 @@
 /* Location for freeciv to store its information */
 #define FREECIV_STORAGE_DIR "@FREECIV_STORAGE_DIR@"
 
-#define FREECIV_META_URL "http://meta.freeciv.org/metaserver.php"
+#define FREECIV_META_URL "http://localhost:8080/meta/metaserver.php"
 
 /* IPv6 Support built in */
 #define FREECIV_IPV6_SUPPORT
diff --git a/meson.build b/meson.build
index 22839d88ca..4a823e4c31 100644
--- a/meson.build
+++ b/meson.build
@@ -42,7 +42,7 @@ else
 endif
 
 if host_system == 'windows'
-  pub_conf_data.set('FREECIV_STORAGE_DIR', '~\\\\.freeciv')
+  pub_conf_data.set('FREECIV_STORAGE_DIR', '~\\\\.freeciv-web')
   priv_conf_data.set('LOCALEDIR',
                      join_paths('.', get_option('datadir'), 'locale'))
 
@@ -54,9 +54,9 @@ if host_system == 'windows'
     '".;data/scenarios;@FREECIV_STORAGE_DIR@/@DATASUBDIR@/scenarios;@FREECIV_STORAGE_DIR@/scenarios;@DATADIR@/freeciv/scenarios"')
 else
   if host_system == 'haiku'
-    pub_conf_data.set('FREECIV_STORAGE_DIR', '~/config/settings/freeciv')
+    pub_conf_data.set('FREECIV_STORAGE_DIR', '~/config/settings/freeciv-web')
   else
-    pub_conf_data.set('FREECIV_STORAGE_DIR', '~/.freeciv')
+    pub_conf_data.set('FREECIV_STORAGE_DIR', '~/.freeciv-web')
   endif
   priv_conf_data.set('LOCALEDIR',
                      join_paths(get_option('prefix'),
