diff -Nurd freeciv/meson.build freeciv/meson.build
--- freeciv/meson.build	2023-11-14 20:54:52.761945043 +0200
+++ freeciv/meson.build	2023-11-14 20:55:10.361982266 +0200
@@ -4322,7 +4322,8 @@
   'sandbox',
   'civ1',
   'civ2',
-  'stub'
+  'stub',
+  'webperimental'
   ]
 
 ruleset_files = [
@@ -4350,6 +4351,8 @@
   'data/sandbox.modpack',
   'data/civ1.modpack',
   'data/civ2.modpack',
+  'data/webperimental.serv',
+  'data/webperimental.modpack',
   install_dir : join_paths(get_option('datadir'), 'freeciv')
   )
 
@@ -4384,6 +4387,11 @@
   )
 
 install_data(
+  'data/webperimental/README.webperimental',
+  install_dir : join_paths(get_option('datadir'), 'freeciv/webperimental')
+  )
+
+install_data(
   'data/alien/nation/adventurers.ruleset',
   'data/alien/nation/galacticsound.ruleset',
   'data/alien/nation/jw.ruleset',
