diff -Nurd freeciv/meson.build freeciv/meson.build
--- freeciv/meson.build	2023-09-18 14:09:39.566957086 +0300
+++ freeciv/meson.build	2023-09-18 14:09:48.663044888 +0300
@@ -4158,7 +4158,8 @@
   'sandbox',
   'civ1',
   'civ2',
-  'stub'
+  'stub',
+  'webperimental'
   ]
 
 ruleset_files = [
@@ -4186,6 +4187,8 @@
   'data/sandbox.modpack',
   'data/civ1.modpack',
   'data/civ2.modpack',
+  'data/webperimental.serv',
+  'data/webperimental.modpack',
   install_dir : join_paths(get_option('datadir'), 'freeciv')
   )
 
@@ -4220,6 +4223,11 @@
   )
 
 install_data(
+  'data/webperimental/README.webperimental',
+  install_dir : join_paths(get_option('datadir'), 'freeciv/webperimental')
+  )
+
+install_data(
   'data/alien/nation/adventurers.ruleset',
   'data/alien/nation/galacticsound.ruleset',
   'data/alien/nation/jw.ruleset',
