diff -Nurd -X.diff_ignore freeciv/common/game.h freeciv/common/game.h
--- freeciv/common/game.h	2012-10-06 08:58:33.248269092 +0300
+++ freeciv/common/game.h	2012-10-06 09:12:32.140250686 +0300
@@ -220,6 +220,7 @@
       struct {
         bool user_message_set;
         char user_message[256];
+	char type[20];
       } meta_info;
 
       int first_timeout;
diff -Nurd -X.diff_ignore freeciv/server/civserver.c freeciv/server/civserver.c
--- freeciv/server/civserver.c	2012-10-06 08:58:46.812268794 +0300
+++ freeciv/server/civserver.c	2012-10-06 09:13:27.676249466 +0300
@@ -248,6 +248,8 @@
       srvarg.bind_addr = option; /* Never freed. */
     } else if ((option = get_option_malloc("--Bind-meta", argv, &inx, argc))) {
       srvarg.bind_meta_addr = option; /* Never freed. */
+    } else if ((option = get_option_malloc("--type", argv, &inx, argc))) {
+      sz_strlcpy(game.server.meta_info.type, option);
     } else if ((option = get_option_malloc("--read", argv, &inx, argc)))
       srvarg.script_filename = option; /* Never freed. */
     else if ((option = get_option_malloc("--quitidle", argv, &inx, argc))) {
@@ -417,6 +419,8 @@
                 /* TRANS: "Ranklog" is exactly what user must type, do not translate. */
                 _("Ranklog FILE"),
                 _("Use FILE as ranking logfile"));
+    cmdhelp_add(help, "t", _("type TYPE"),
+                _("Set the server metaserver type to TYPE"));
 #ifdef AI_MODULES
     cmdhelp_add(help, "L",
                 /* TRANS: "LoadAI" is exactly what user must type, do not translate. */
diff -Nurd -X.diff_ignore freeciv/server/meta.c freeciv/server/meta.c
--- freeciv/server/meta.c	2012-10-06 08:58:46.832268794 +0300
+++ freeciv/server/meta.c	2012-10-06 09:14:15.700248414 +0300
@@ -113,6 +113,15 @@
 }
 
 /*************************************************************************
+ the server metaserver type
+*************************************************************************/
+const char *get_meta_type_string(void)
+{
+  return game.server.meta_info.type;
+}
+
+
+/*************************************************************************
  The metaserver message set by user
 *************************************************************************/
 const char *get_user_meta_message_string(void)
@@ -278,6 +287,8 @@
   if (flag == META_GOODBYE) {
     netfile_add_form_int(post, "bye", 1);
   } else {
+    netfile_add_form_str(post, "type",
+                         get_meta_type_string());
     netfile_add_form_str(post, "version", VERSION_STRING);
     netfile_add_form_str(post, "patches",
                          get_meta_patches_string());
@@ -316,6 +327,10 @@
                              fc_url_encode(plr->nation != NO_NATION_SELECTED 
                                            ? nation_plural_for_player(plr)
                                            : "none"));
+        netfile_add_form_str(post, "pln[]",
+                             fc_url_encode(plr->nation != NO_NATION_SELECTED 
+                                           ? nation_of_player(plr)->flag_graphic_str
+                                           : "none"));
         netfile_add_form_str(post, "plh[]",
                              pconn ? pconn->addr : "");
 
diff -Nurd -X.diff_ignore freeciv/server/meta.h freeciv/server/meta.h
--- freeciv/server/meta.h	2012-10-06 08:58:46.832268794 +0300
+++ freeciv/server/meta.h	2012-10-06 09:14:32.748248040 +0300
@@ -32,6 +32,7 @@
 const char *get_meta_patches_string(void);
 const char *get_meta_message_string(void);
 const char *get_user_meta_message_string(void);
+const char *get_meta_type_string(void);
 
 void maybe_automatic_meta_message(const char *automatic);
 
