diff -Nurd -X.diff_ignore freeciv/common/game.h freeciv/common/game.h
--- freeciv/common/game.h	2014-02-14 01:23:04.189582544 +0200
+++ freeciv/common/game.h	2014-02-14 02:04:09.823823234 +0200
@@ -229,6 +229,7 @@
       struct {
         bool user_message_set;
         char user_message[256];
+	char type[20];
       } meta_info;
 
       struct {
diff -Nurd -X.diff_ignore freeciv/server/civserver.c freeciv/server/civserver.c
--- freeciv/server/civserver.c	2014-02-14 01:23:05.241580158 +0200
+++ freeciv/server/civserver.c	2014-02-14 02:04:09.839823199 +0200
@@ -249,6 +249,8 @@
       srvarg.bind_addr = option; /* Never freed. */
     } else if ((option = get_option_malloc("--Bind-meta", argv, &inx, argc))) {
       srvarg.bind_meta_addr = option; /* Never freed. */
+    } else if ((option = get_option_malloc("--type", argv, &inx, argc))) {
+      sz_strlcpy(game.server.meta_info.type, option);
     } else if ((option = get_option_malloc("--read", argv, &inx, argc)))
       srvarg.script_filename = option; /* Never freed. */
     else if ((option = get_option_malloc("--quitidle", argv, &inx, argc))) {
@@ -418,6 +420,8 @@
                 /* TRANS: "Ranklog" is exactly what user must type, do not translate. */
                 _("Ranklog FILE"),
                 _("Use FILE as ranking logfile"));
+    cmdhelp_add(help, "t", _("type TYPE"),
+                _("Set the server metaserver type to TYPE"));
 #ifdef AI_MODULES
     cmdhelp_add(help, "L",
                 /* TRANS: "LoadAI" is exactly what user must type, do not translate. */
diff -Nurd -X.diff_ignore freeciv/server/meta.c freeciv/server/meta.c
--- freeciv/server/meta.c	2014-02-14 01:23:05.225580194 +0200
+++ freeciv/server/meta.c	2014-02-14 02:04:09.839823199 +0200
@@ -113,6 +113,15 @@
 }
 
 /*************************************************************************
+ the server metaserver type
+*************************************************************************/
+const char *get_meta_type_string(void)
+{
+  return game.server.meta_info.type;
+}
+
+
+/*************************************************************************
  The metaserver message set by user
 *************************************************************************/
 const char *get_user_meta_message_string(void)
@@ -278,6 +287,8 @@
   if (flag == META_GOODBYE) {
     netfile_add_form_int(post, "bye", 1);
   } else {
+    netfile_add_form_str(post, "type",
+                         get_meta_type_string());
     netfile_add_form_str(post, "version", VERSION_STRING);
     netfile_add_form_str(post, "patches",
                          get_meta_patches_string());
diff -Nurd -X.diff_ignore freeciv/server/meta.h freeciv/server/meta.h
--- freeciv/server/meta.h	2014-02-14 02:04:01.231842771 +0200
+++ freeciv/server/meta.h	2014-02-14 02:04:09.839823199 +0200
@@ -32,6 +32,7 @@
 const char *get_meta_patches_string(void);
 const char *get_meta_message_string(void);
 const char *get_user_meta_message_string(void);
+const char *get_meta_type_string(void);
 
 void maybe_automatic_meta_message(const char *automatic);
 
