diff -Nurd -X.diff_ignore freeciv/common/game.h freeciv/common/game.h
--- freeciv/common/game.h	2012-07-17 02:35:58.000000000 +0300
+++ freeciv/common/game.h	2012-07-17 03:17:40.000000000 +0300
@@ -220,6 +220,7 @@
       struct {
         bool user_message_set;
         char user_message[256];
+	char topic[256];
       } meta_info;
     } server;
   };
diff -Nurd -X.diff_ignore freeciv/common/packets.def freeciv/common/packets.def
--- freeciv/common/packets.def	2012-07-17 03:14:25.000000000 +0300
+++ freeciv/common/packets.def	2012-07-17 03:17:40.000000000 +0300
@@ -1295,12 +1295,12 @@
   STRING rule_name[MAX_LEN_NAME];
   STRING graphic_str[MAX_LEN_NAME];
   STRING graphic_alt[MAX_LEN_NAME];
-  UINT8 reqs_count;
+  UINT32 reqs_count;
   REQUIREMENT reqs[MAX_NUM_REQS:reqs_count];
   TECH obsolete_by;
   IMPROVEMENT replaced_by;
-  UINT16 build_cost;
-  UINT8 upkeep, sabotage;
+  UINT32 build_cost;
+  UINT32 upkeep, sabotage;
   BV_IMPR_FLAGS flags;
   STRING soundtag[MAX_LEN_NAME];
   STRING soundtag_alt[MAX_LEN_NAME];
diff -Nurd -X.diff_ignore freeciv/server/civserver.c freeciv/server/civserver.c
--- freeciv/server/civserver.c	2012-07-17 02:35:27.000000000 +0300
+++ freeciv/server/civserver.c	2012-07-17 03:17:40.000000000 +0300
@@ -244,6 +244,9 @@
       free(option);
     } else if ((option = get_option_malloc("--bind", argv, &inx, argc))) {
       srvarg.bind_addr = option; /* Never freed. */
+    } else if ((option = get_option_malloc("--topic", argv, &inx, argc))) {
+      sz_strlcpy(game.server.meta_info.topic, option);
+
     } else if ((option = get_option_malloc("--read", argv, &inx, argc)))
       srvarg.script_filename = option; /* Never freed. */
     else if ((option = get_option_malloc("--quitidle", argv, &inx, argc))) {
@@ -369,6 +372,7 @@
     fc_fprintf(stderr, _("  -r, --read FILE\tRead startup script FILE\n"));
     fc_fprintf(stderr,
 	       _("  -R, --Ranklog FILE\tUse FILE as ranking logfile\n"));
+    fc_fprintf(stderr, _("  -t, --topic \"MESSAGE\"\t\tSet the metaserver topic\n"));
 #ifdef AI_MODULES
     fc_fprintf(stderr,
                _("  -L, --LoadAI MODULE\tLoad ai module MODULE. Can appear multiple times\n"));
diff -Nurd -X.diff_ignore freeciv/server/meta.c freeciv/server/meta.c
--- freeciv/server/meta.c	2012-07-17 02:35:27.000000000 +0300
+++ freeciv/server/meta.c	2012-07-17 03:22:10.000000000 +0300
@@ -112,6 +112,12 @@
   return meta_message;
 }
 
+const char *get_meta_topic_string(void)
+{
+  return game.server.meta_info.topic;
+}
+
+
 /*************************************************************************
  The metaserver message set by user
 *************************************************************************/
@@ -271,6 +277,8 @@
   if (flag == META_GOODBYE) {
     netfile_add_form_int(post, "bye", 1);
   } else {
+    netfile_add_form_str(post, "topic",
+                         get_meta_topic_string());
     netfile_add_form_str(post, "version", VERSION_STRING);
     netfile_add_form_str(post, "patches",
                          get_meta_patches_string());
@@ -309,6 +317,10 @@
                              fc_url_encode(plr->nation != NO_NATION_SELECTED 
                                            ? nation_plural_for_player(plr)
                                            : "none"));
+        netfile_add_form_str(post, "pln[]",
+                             fc_url_encode(plr->nation != NO_NATION_SELECTED 
+                                           ? nation_of_player(plr)->flag_graphic_str
+                                           : "none"));
         netfile_add_form_str(post, "plh[]",
                              pconn ? pconn->addr : "");
 
diff -Nurd -X.diff_ignore freeciv/server/meta.h freeciv/server/meta.h
--- freeciv/server/meta.h	2012-07-17 03:17:37.000000000 +0300
+++ freeciv/server/meta.h	2012-07-17 03:17:40.000000000 +0300
@@ -32,6 +32,7 @@
 const char *get_meta_patches_string(void);
 const char *get_meta_message_string(void);
 const char *get_user_meta_message_string(void);
+const char *get_meta_topic_string(void);
 
 void maybe_automatic_meta_message(const char *automatic);
 
