diff -Nurd -X.diff_ignore freeciv/server/meta.c freeciv/server/meta.c
--- freeciv/server/meta.c	2014-08-27 04:05:37.297555089 +0300
+++ freeciv/server/meta.c	2014-09-07 08:58:37.357886131 +0300
@@ -333,6 +333,10 @@
                              plr->nation != NO_NATION_SELECTED 
                              ? nation_plural_for_player(plr)
                              : "none");
+        netfile_add_form_str(post, "plf[]",
+                             plr->nation != NO_NATION_SELECTED 
+                             ? nation_of_player(plr)->flag_graphic_str
+                             : "none");
         netfile_add_form_str(post, "plh[]",
                              pconn ? pconn->addr : "");
 
