diff -Nurd -X.diff_ignore freeciv/common/game.c freeciv/common/game.c
--- freeciv/common/game.c	2016-02-24 02:40:28.870661317 +0200
+++ freeciv/common/game.c	2016-02-24 03:12:52.866631507 +0200
@@ -387,7 +387,6 @@
     game.server.last_ping         = 0;
     game.server.max_players       = GAME_DEFAULT_MAX_PLAYERS;
     game.server.meta_info.user_message[0] = '\0';
-    game.server.meta_info.user_message_set = FALSE;
     /* Do not clear meta_info.type here as it's already set to correct value */
     game.server.mgr_distance      = GAME_DEFAULT_MGR_DISTANCE;
     game.server.mgr_foodneeded    = GAME_DEFAULT_MGR_FOODNEEDED;
diff -Nurd -X.diff_ignore freeciv/common/game.h freeciv/common/game.h
--- freeciv/common/game.h	2016-02-24 03:04:07.462414447 +0200
+++ freeciv/common/game.h	2016-02-24 03:12:52.866631507 +0200
@@ -242,7 +242,6 @@
       } save_options;
 
       struct {
-        bool user_message_set;
         char user_message[256];
         char type[20];
       } meta_info;
@@ -589,6 +588,8 @@
 
 #define GAME_DEFAULT_AUTOSAVES       (1 << AS_TURN | 1 << AS_GAME_OVER | 1 << AS_QUITIDLE | 1 << AS_INTERRUPT)
 
+#define GAME_DEFAULT_USER_META_MESSAGE ""
+
 #define GAME_DEFAULT_SKILL_LEVEL     AI_LEVEL_EASY
 #define GAME_HARDCODED_DEFAULT_SKILL_LEVEL 3 /* that was 'easy' in old saves */
 #define GAME_OLD_DEFAULT_SKILL_LEVEL 5  /* normal; for oldest save games */
diff -Nurd -X.diff_ignore freeciv/server/meta.c freeciv/server/meta.c
--- freeciv/server/meta.c	2016-02-24 02:40:28.882661265 +0200
+++ freeciv/server/meta.c	2016-02-24 03:12:52.866631507 +0200
@@ -137,7 +137,8 @@
 *************************************************************************/
 const char *get_user_meta_message_string(void)
 {
-  if (game.server.meta_info.user_message_set) {
+  if (game.server.meta_info.user_message
+      && game.server.meta_info.user_message[0] != '\0') {
     return game.server.meta_info.user_message;
   }
 
@@ -190,12 +191,10 @@
 {
   if (string != NULL && string[0] != '\0') {
     sz_strlcpy(game.server.meta_info.user_message, string);
-    game.server.meta_info.user_message_set = TRUE;
     set_meta_message_string(string);
   } else {
     /* Remove user meta message. We will use automatic messages instead */
     game.server.meta_info.user_message[0] = '\0';
-    game.server.meta_info.user_message_set = FALSE;
     set_meta_message_string(default_meta_message_string());    
   }
 }
diff -Nurd -X.diff_ignore freeciv/server/savegame2.c freeciv/server/savegame2.c
--- freeciv/server/savegame2.c	2016-02-24 02:40:28.886661246 +0200
+++ freeciv/server/savegame2.c	2016-02-24 03:12:52.866631507 +0200
@@ -1466,15 +1466,6 @@
                                       default_meta_patches_string(),
                                       "game.meta_patches");
   set_meta_patches_string(string);
-  game.server.meta_info.user_message_set
-    = secfile_lookup_bool_default(loading->file, FALSE,
-                                  "game.meta_usermessage");
-  if (game.server.meta_info.user_message_set) {
-    string = secfile_lookup_str_default(loading->file,
-                                        default_meta_message_string(),
-                                        "game.meta_message");
-    set_user_meta_message_string(string);
-  }
 
   if (0 == strcmp(DEFAULT_META_SERVER_ADDR, srvarg.metaserver_addr)) {
     /* Do not overwrite this if the user requested a specific metaserver
diff -Nurd -X.diff_ignore freeciv/server/savegame3.c freeciv/server/savegame3.c
--- freeciv/server/savegame3.c	2016-02-24 02:48:28.492919533 +0200
+++ freeciv/server/savegame3.c	2016-02-24 03:16:04.958581672 +0200
@@ -1795,15 +1795,6 @@
                                       default_meta_patches_string(),
                                       "game.meta_patches");
   set_meta_patches_string(string);
-  game.server.meta_info.user_message_set
-    = secfile_lookup_bool_default(loading->file, FALSE,
-                                  "game.meta_usermessage");
-  if (game.server.meta_info.user_message_set) {
-    string = secfile_lookup_str_default(loading->file,
-                                        default_meta_message_string(),
-                                        "game.meta_message");
-    set_user_meta_message_string(string);
-  }
 
   if (0 == strcmp(DEFAULT_META_SERVER_ADDR, srvarg.metaserver_addr)) {
     /* Do not overwrite this if the user requested a specific metaserver
@@ -1949,7 +1940,6 @@
 static void sg_save_game(struct savedata *saving)
 {
   int game_version;
-  const char *user_message;
   enum server_states srv_state;
   char global_advances[game.control.num_tech_types + 1];
   int i;
@@ -1973,12 +1963,6 @@
 
   secfile_insert_str(saving->file, get_meta_patches_string(),
                      "game.meta_patches");
-  secfile_insert_bool(saving->file, game.server.meta_info.user_message_set,
-                      "game.meta_usermessage");
-  user_message = get_user_meta_message_string();
-  if (user_message != NULL) {
-    secfile_insert_str(saving->file, user_message, "game.meta_message");
-  }
   secfile_insert_str(saving->file, meta_addr_port(), "game.meta_server");
 
   secfile_insert_str(saving->file, server.game_identifier, "game.id");
diff -Nurd -X.diff_ignore freeciv/server/legacysave.c freeciv/server/legacysave.c
--- freeciv/server/legacysave.c	2016-02-24 02:40:28.886661246 +0200
+++ freeciv/server/legacysave.c	2016-02-24 03:12:52.866631507 +0200
@@ -3274,9 +3274,9 @@
     set_meta_patches_string(secfile_lookup_str_default(file, 
                                                 default_meta_patches_string(),
                                                 "game.metapatches"));
-    game.server.meta_info.user_message_set =
-      secfile_lookup_bool_default(file, FALSE, "game.user_metamessage");
-    if (game.server.meta_info.user_message_set) {
+
+    if (secfile_lookup_bool_default(file, FALSE,
+                                    "game.user_metamessage")) {
       set_user_meta_message_string(secfile_lookup_str_default(file, 
                                                 default_meta_message_string(),
                                                 "game.metamessage"));
diff -Nurd -X.diff_ignore freeciv/server/settings.c freeciv/server/settings.c
--- freeciv/server/settings.c	2016-02-24 03:11:30.778615303 +0200
+++ freeciv/server/settings.c	2016-02-24 03:12:52.866631507 +0200
@@ -31,6 +31,7 @@
 /* server */
 #include "gamehand.h"
 #include "maphand.h"
+#include "meta.h"
 #include "notify.h"
 #include "plrhand.h"
 #include "report.h"
@@ -699,6 +700,23 @@
 }
 
 /*************************************************************************
+  Update metaserver message string from changed user meta server message
+  string.
+*************************************************************************/
+static void metamessage_action(const struct setting *pset)
+{
+  /* Set the metaserver message based on the new meta server user message.
+   * An empty user metaserver message results in an automatic meta message.
+   * A non empty user meta message results in the user meta message. */
+  set_user_meta_message_string(pset->string.value);
+
+  if (is_metaserver_open()) {
+    /* Update the meta server. */
+    send_server_info_to_metaserver(META_INFO);
+  }
+}
+
+/*************************************************************************
   Validation callback functions.
 *************************************************************************/
 
@@ -2676,6 +2694,16 @@
              "\"kick\" command may reconnect. Changing this setting will "
              "affect users kicked in the past."), NULL, NULL,
           GAME_MIN_KICK_TIME, GAME_MAX_KICK_TIME, GAME_DEFAULT_KICK_TIME)
+
+  GEN_STRING("metamessage", game.server.meta_info.user_message,
+             SSET_META, SSET_INTERNAL, SSET_RARE, SSET_SERVER_ONLY,
+             N_("Metaserver info line"),
+             N_("User defined metaserver info line. For most of the time "
+                "a user defined metamessage will be used instead of an "
+                "automatically generated message. "
+                "Set to empty (\"\", not \"empty\") to always use an "
+                "automatically generated meta server message."),
+             NULL, metamessage_action, GAME_DEFAULT_USER_META_MESSAGE)
 };
 
 #undef GEN_BOOL
