diff -Nurd -X.diff_ignore freeciv/common/packets.def freeciv/common/packets.def
--- freeciv/common/packets.def	2013-03-16 14:45:45.973022671 +0200
+++ freeciv/common/packets.def	2013-03-16 14:46:41.241021458 +0200
@@ -1914,3 +1914,13 @@
   SINT32 tag;
   SINT32 id;
 end
+
+/*************** Webclient specific packets ****************/
+PACKET_INFO_TEXT_REQ = 254; cs, dsend
+  TILE loc;
+  UNIT visible_unit;
+end
+
+PACKET_INFO_TEXT_MESSAGE = 255; sc, dsend
+  STRING message[MAX_LEN_MSG];
+end
endif  /* FC__MAPHAND_H */
