diff -Nurd -X.diff_ignore freeciv/common/packets.def freeciv/common/packets.def
--- freeciv/common/packets.def	2012-05-16 22:03:59.000000000 +0300
+++ freeciv/common/packets.def	2012-05-16 22:04:21.000000000 +0300
@@ -1734,3 +1734,14 @@
   SINT32 tag;
   SINT32 id;
 end
+
+/*************** Webclient specific packets ****************/
+PACKET_INFO_TEXT_REQ=186;cs,dsend
+  COORD x, y;
+  UNIT visible_unit; 
+end
+
+PACKET_INFO_TEXT_MESSAGE=187;sc,dsend
+  STRING message[MAX_LEN_MSG];
+end
+
diff -Nurd -X.diff_ignore freeciv/server/maphand.c freeciv/server/maphand.c
--- freeciv/server/maphand.c	2012-05-16 20:56:43.000000000 +0300
+++ freeciv/server/maphand.c	2012-05-16 22:04:29.000000000 +0300
@@ -36,6 +36,7 @@
 #include "unit.h"
 #include "unitlist.h"
 #include "vision.h"
+#include "astring.h"
 
 /* generator */
 #include "utilities.h"
@@ -1753,3 +1754,403 @@
     }
   }
 }
+
+/****************************************************************************
+  Return a (static) string with a tile's food/prod/trade
+****************************************************************************/
+static const char *get_tile_output_text(const struct tile *ptile, 
+		                        struct player *pplayer)
+{
+  static struct astring str = ASTRING_INIT;
+  int i;
+  char output_text[O_LAST][16];
+
+  for (i = 0; i < O_LAST; i++) {
+    int before_penalty = 0;
+    int x = city_tile_output(NULL, ptile, FALSE, i);
+
+    if (NULL != pplayer) {
+      before_penalty = get_player_output_bonus(pplayer,
+                                               get_output_type(i),
+                                               EFT_OUTPUT_PENALTY_TILE);
+    }
+
+    if (before_penalty > 0 && x > before_penalty) {
+      my_snprintf(output_text[i], sizeof(output_text[i]), "%d(-1)", x);
+    } else {
+      my_snprintf(output_text[i], sizeof(output_text[i]), "%d", x);
+    }
+  }
+  
+  astr_clear(&str);
+  astr_add_line(&str, "%s/%s/%s",
+                output_text[O_FOOD],
+		output_text[O_SHIELD],
+		output_text[O_TRADE]);
+
+  return str.str;
+}
+
+
+
+/****************************************************************************
+  Creates the activity progress text for the given tile.
+
+  This should only be used inside popup_info_text and should eventually be
+  made static.
+****************************************************************************/
+static const char *concat_tile_activity_text(struct tile *ptile)
+{
+  int activity_total[ACTIVITY_LAST];
+  int activity_units[ACTIVITY_LAST];
+  int base_total[MAX_BASE_TYPES];
+  int base_units[MAX_BASE_TYPES];
+  int num_activities = 0;
+  int remains, turns, i;
+  static struct astring str = ASTRING_INIT;
+
+  astr_clear(&str);
+
+  memset(activity_total, 0, sizeof(activity_total));
+  memset(activity_units, 0, sizeof(activity_units));
+  memset(base_total, 0, sizeof(base_total));
+  memset(base_units, 0, sizeof(base_units));
+
+  unit_list_iterate(ptile->units, punit) {
+    if (punit->activity == ACTIVITY_BASE) {
+      base_total[punit->activity_base] += punit->activity_count;
+      base_total[punit->activity_base] += get_activity_rate_this_turn(punit);
+      base_units[punit->activity_base] += get_activity_rate(punit);
+    } else {
+      activity_total[punit->activity] += punit->activity_count;
+      activity_total[punit->activity] += get_activity_rate_this_turn(punit);
+      activity_units[punit->activity] += get_activity_rate(punit);
+    }
+  } unit_list_iterate_end;
+
+  for (i = 0; i < ACTIVITY_LAST; i++) {
+    if (i == ACTIVITY_BASE) {
+      base_type_iterate(bp) {
+        Base_type_id b = base_index(bp);
+	if (base_units[b] > 0) {
+	  remains = tile_activity_base_time(ptile, b) - base_total[b];
+	  if (remains > 0) {
+	    turns = 1 + (remains + base_units[b] - 1) / base_units[b];
+	  } else {
+	    /* base will be finished this turn */
+	    turns = 1;
+	  }
+	  if (num_activities > 0) {
+	    astr_add(&str, "/");
+	  }
+	  astr_add(&str, "%s(%d)", base_name_translation(bp), turns);
+	  num_activities++;
+	}
+      } base_type_iterate_end;
+    } else if (is_build_or_clean_activity(i) && activity_units[i] > 0) {
+      if (num_activities > 0) {
+	astr_add(&str, "/");
+      }
+      remains = tile_activity_time(i, ptile) - activity_total[i];
+      if (remains > 0) {
+	turns = 1 + (remains + activity_units[i] - 1) / activity_units[i];
+      } else {
+	/* activity will be finished this turn */
+	turns = 1;
+      }
+      astr_add(&str, "%s(%d)", get_activity_text(i), turns);
+      num_activities++;
+    }
+  }
+
+  return str.str;
+}
+
+
+/****************************************************************************
+  Fill the buffer with the player's nation name (in adjective form) and
+  optionally add the player's team name.
+****************************************************************************/
+static inline void get_full_nation(char *buf, int buflen,
+                                   const struct player *pplayer)
+{
+  if (!buf || buflen < 1) {
+    return;
+  }
+
+  if (!pplayer) {
+    buf[0] = '\0';
+    return;
+  }
+
+  if (pplayer->team) {
+    /* TRANS: "<nation adjective>, team <team name>" */
+    my_snprintf(buf, buflen, _("%s, team %s"),
+                nation_adjective_for_player(pplayer),
+                team_name_translation(pplayer->team));
+  } else {
+    mystrlcpy(buf, nation_adjective_for_player(pplayer), buflen);
+  }
+}
+
+
+/****************************************************************************
+  For AIs, fill the buffer with their player name prefixed with "AI". For
+  humans, just fill it with their username.
+****************************************************************************/
+static inline void get_full_username(char *buf, int buflen,
+                                     const struct player *pplayer)
+{
+  if (!buf || buflen < 1) {
+    return;
+  }
+
+  if (!pplayer) {
+    buf[0] = '\0';
+    return;
+  }
+
+  if (pplayer->ai_data.control) {
+    /* TRANS: "AI <player name>" */
+    my_snprintf(buf, buflen, _("AI %s"), pplayer->name);
+  } else {
+    mystrlcpy(buf, pplayer->username, buflen);
+  }
+}
+
+
+/****************************************************************************
+  Text to popup on a middle-click in the mapview.
+  - copied from /client/text.c
+****************************************************************************/
+static const char *popup_info_text(struct tile *ptile, struct player *pplayer,
+  	 	                   struct unit *punit)
+{
+  const char *activity_text;
+  struct city *pcity = tile_city(ptile);
+  const char *diplo_nation_plural_adjectives[DS_LAST] =
+    {Q_("?nation:Neutral"), Q_("?nation:Hostile"),
+     Q_("?nation:Neutral"),
+     Q_("?nation:Peaceful"), Q_("?nation:Friendly"), 
+     Q_("?nation:Mysterious"), Q_("?nation:Friendly(team)")};
+  const char *diplo_city_adjectives[DS_LAST] =
+    {Q_("?city:Neutral"), Q_("?city:Hostile"),
+     Q_("?nation:Neutral"),
+     Q_("?city:Peaceful"), Q_("?city:Friendly"), Q_("?city:Mysterious"),
+     Q_("?city:Friendly(team)")};
+  int infracount;
+  bv_special infra;
+  static struct astring str = ASTRING_INIT;
+  char username[MAX_LEN_NAME + 32];
+  char nation[2 * MAX_LEN_NAME + 32];
+
+  astr_clear(&str);
+  astr_add_line(&str, _("Location: (%d, %d)"),
+                ptile->x, ptile->y);
+
+  if (tile_get_known(ptile, pplayer) == TILE_UNKNOWN) {
+    astr_add(&str, _("Unknown"));
+    return str.str;
+  }
+  astr_add_line(&str, _("Terrain: %s"),  tile_get_info_text(ptile, 0));
+  astr_add_line(&str, _("Food/Prod/Trade: %s"),
+		get_tile_output_text(ptile, pplayer));
+  if (tile_has_special(ptile, S_HUT)) {
+    astr_add_line(&str, _("Minor Tribe Village"));
+  }
+  if (game.info.borders > 0 && !pcity) {
+    struct player *owner = tile_owner(ptile);
+
+    get_full_username(username, sizeof(username), owner);
+    get_full_nation(nation, sizeof(nation), owner);
+
+    if (NULL != pplayer && owner == pplayer) {
+      astr_add_line(&str, _("Our territory"));
+    } else if (NULL != owner && NULL == pplayer) {
+      /* TRANS: "Territory of <username> (<nation + team>)" */
+      astr_add_line(&str, _("Territory of %s (%s)"), username, nation);
+    } else if (NULL != owner) {
+      struct player_diplstate *ds = pplayer->diplstates;
+
+      if (ds[player_index(owner)].type == DS_CEASEFIRE) {
+	int turns = ds[player_index(owner)].turns_left;
+
+        /* TRANS: "Territory of <username> (<nation + team>)
+         * (<number> turn cease-fire)" */
+        astr_add_line(&str,
+                      PL_("Territory of %s (%s) (%d turn cease-fire)",
+                          "Territory of %s (%s) (%d turn cease-fire)",
+                          turns),
+                      username, nation, turns);
+      } else {
+	int type = ds[player_index(owner)].type;
+
+        /* TRANS: "Territory of <username>
+         * (<nation + team> | <diplomatic state>)" */
+        astr_add_line(&str, _("Territory of %s (%s | %s)"),
+                      username, nation,
+                      diplo_nation_plural_adjectives[type]);
+      }
+    } else {
+      astr_add_line(&str, _("Unclaimed territory"));
+    }
+  }
+  if (pcity) {
+    /* Look at city owner, not tile owner (the two should be the same, if
+     * borders are in use). */
+    struct player *owner = city_owner(pcity);
+    int has_improvements = 0;
+    struct impr_type *prev_impr = NULL;
+
+    get_full_username(username, sizeof(username), owner);
+    get_full_nation(nation, sizeof(nation), owner);
+
+    if (NULL == pplayer || owner == pplayer) {
+      /* TRANS: "City: <city name> | <username> (<nation + team>)" */
+      astr_add_line(&str, _("City: %s | %s (%s)"),
+                    city_name(pcity), username, nation);
+    } else {
+      struct player_diplstate *ds = pplayer->diplstates;
+
+      if (ds[player_index(owner)].type == DS_CEASEFIRE) {
+	int turns = ds[player_index(owner)].turns_left;
+
+        /* TRANS:  "City: <city name> | <username>
+         * (<nation + team>, <number> turn cease-fire)" */
+        astr_add_line(&str, PL_("City: %s | %s (%s, %d turn cease-fire)",
+                                "City: %s | %s (%s, %d turn cease-fire)",
+                                turns),
+                      city_name(pcity), username, nation, turns);
+      } else {
+        /* TRANS: "City: <city name> | <username>
+         * (<nation + team>, <diplomatic state>)" */
+        astr_add_line(&str, _("City: %s | %s (%s, %s)"),
+                      city_name(pcity), username, nation,
+                      diplo_city_adjectives[ds[player_index(owner)].type]);
+      }
+    }
+    improvement_iterate(pimprove) {
+      if (is_improvement_visible(pimprove)
+          && city_has_building(pcity, pimprove)) {
+        if (has_improvements++ > 1) {
+          /* TRANS: continue list, in case comma is not the separator of choice. */
+          astr_add(&str, Q_("?clistmore:, %s"),
+                   improvement_name_translation(prev_impr));
+        } else if (has_improvements == 1) {
+          astr_add_line(&str, "  ");
+          /* TRANS: previous lines gave other information about the city. */
+          astr_add(&str, Q_("?clistbegin: with %s"),
+                   improvement_name_translation(pimprove));
+        }
+        prev_impr = pimprove;
+      }
+    } improvement_iterate_end;
+
+    if (NULL != prev_impr) {
+      if (has_improvements > 2) {
+        /* TRANS: This appears with two or more previous entries in the list */
+        astr_add(&str, Q_("?clistlast:, and %s"),
+		 improvement_name_translation(prev_impr));
+      } else if (has_improvements > 1) {
+        /* TRANS: This appears with only one previous entry in the list */
+        astr_add(&str, Q_("?clistlast: and %s"),
+		 improvement_name_translation(prev_impr));
+      }
+      astr_add(&str, "%s", Q_("?clistend:."));
+    }
+
+  }
+  infra = get_tile_infrastructure_set(ptile, &infracount);
+  if (infracount == 0) {
+    base_type_iterate(pbase) {
+      if (BV_ISSET(ptile->bases, base_index(pbase))) {
+        infracount = 1;
+        break;
+      }
+    } base_type_iterate_end;
+  }
+  if (infracount > 0) {
+    astr_add_line(&str, _("Infrastructure: %s"),
+		  get_infrastructure_text(ptile->special, ptile->bases));
+  }
+  activity_text = concat_tile_activity_text(ptile);
+  if (strlen(activity_text) > 0) {
+    astr_add_line(&str, _("Activity: %s"), activity_text);
+  }
+  if (punit && !pcity) {
+    struct player *owner = unit_owner(punit);
+    struct unit_type *ptype = unit_type(punit);
+
+    get_full_username(username, sizeof(username), owner);
+    get_full_nation(nation, sizeof(nation), owner);
+
+    if (NULL == pplayer || owner == pplayer) {
+      struct city *pcity = player_find_city_by_id(owner, punit->homecity);
+
+      if (pcity) {
+        /* TRANS: "Unit: <unit type> | <username>
+         * (<nation + team>, <homecity>)" */
+        astr_add_line(&str, _("Unit: %s | %s (%s, %s)"),
+                      utype_name_translation(ptype), username,
+                      nation, city_name(pcity));
+      } else {
+        /* TRANS: "Unit: <unit type> | <username> (<nation + team>)" */
+        astr_add_line(&str, _("Unit: %s | %s (%s)"),
+                      utype_name_translation(ptype), username, nation);
+      }
+    } else if (NULL != owner) {
+      struct player_diplstate *ds = pplayer->diplstates;
+
+      if (ds[player_index(owner)].type == DS_CEASEFIRE) {
+	int turns = ds[player_index(owner)].turns_left;
+
+        /* TRANS:  "Unit: <unit type> | <username> (<nation + team>,
+         * <number> turn cease-fire)" */
+        astr_add_line(&str, PL_("Unit: %s | %s (%s, %d turn cease-fire)",
+                                "Unit: %s | %s (%s, %d turn cease-fire)",
+                                turns),
+                      utype_name_translation(ptype),
+                      username, nation, turns);
+      } else {
+        /* TRANS: "Unit: <unit type> | <username> (<nation + team>,
+         * <diplomatic state>)" */
+        astr_add_line(&str, _("Unit: %s | %s (%s, %s)"),
+                      utype_name_translation(ptype), username, nation,
+                      diplo_city_adjectives[ds[player_index(owner)].type]);
+      }
+    }
+
+    /* TRANS: A is attack power, D is defense power, FP is firepower,
+     * HP is hitpoints (current and max). */
+    astr_add_line(&str, _("A:%d D:%d FP:%d HP:%d/%d (%s)"),
+		  ptype->attack_strength, 
+		  ptype->defense_strength, ptype->firepower, punit->hp, 
+		  ptype->hp,
+		  _(ptype->veteran[punit->veteran].name));
+
+    if ((NULL == pplayer || owner == pplayer)
+	&& unit_list_size(ptile->units) >= 2) {
+      /* TRANS: "5 more" units on this tile */
+      astr_add(&str, _("  (%d more)"), unit_list_size(ptile->units) - 1);
+    }
+  } 
+  return str.str;
+}
+
+
+
+/****************************************************************************
+...
+****************************************************************************/
+void handle_info_text_req(struct player *pplayer, int x, int y, int visible_unit_id)
+{
+  struct tile *ptile = map_pos_to_tile(x, y);
+  struct unit *punit = (visible_unit_id == 0) ? NULL : game_find_unit_by_number(visible_unit_id);
+  const char *info_text = popup_info_text(ptile, pplayer, punit);
+
+  conn_list_iterate(pplayer->connections, pconn) {
+    dsend_packet_info_text_message(pconn, info_text);
+  } conn_list_iterate_end;
+
+}
+
diff -Nurd -X.diff_ignore freeciv/server/maphand.h freeciv/server/maphand.h
--- freeciv/server/maphand.h	2012-05-16 20:56:43.000000000 +0300
+++ freeciv/server/maphand.h	2012-05-16 22:04:29.000000000 +0300
@@ -121,5 +121,7 @@
 
 void create_base(struct tile *ptile, struct base_type *pbase,
                  struct player *pplayer);
+void handle_info_text_req(struct player *pplayer, int x, int y, int visible_unit_id);
+
 
 #endif  /* FC__MAPHAND_H */
