diff -Nurd freeciv/common/networking/packets.def freeciv/common/networking/packets.def
--- freeciv/common/networking/packets.def	2022-08-16 07:48:01.863025037 +0300
+++ freeciv/common/networking/packets.def	2022-08-16 07:38:08.965080331 +0300
@@ -760,6 +760,10 @@
   WORKLIST worklist;
 
   BV_IMPRS improvements;
+
+  STRING food_output[MAX_LEN_MSG];
+  STRING shield_output[MAX_LEN_MSG];
+  STRING trade_output[MAX_LEN_MSG];
   BV_CITY_OPTIONS city_options;
   ESTRING name[MAX_LEN_CITYNAME];
 
diff -Nurd freeciv/server/citytools.c freeciv/server/citytools.c
--- freeciv/server/citytools.c	2022-08-16 07:47:15.467188383 +0300
+++ freeciv/server/citytools.c	2022-08-16 07:42:50.552114473 +0300
@@ -2627,6 +2627,11 @@
   if (web_packet != NULL) {
     char can_build_impr_buf[MAX_NUM_ITEMS + 1];
     char can_build_unit_buf[MAX_NUM_ITEMS + 1];
+    char food_output_buf[MAX_NUM_ITEMS + 1];
+    char shield_output_buf[MAX_NUM_ITEMS + 1];
+    char trade_output_buf[MAX_NUM_ITEMS + 1];
+    struct tile *pcenter = city_tile(pcity);
+    int c = 0;
 
     web_packet->id = pcity->id;
 
@@ -2648,6 +2653,29 @@
     } unit_type_iterate_end;
     can_build_unit_buf[utype_count()] = '\0';
     sz_strlcpy(web_packet->can_build_unit, can_build_unit_buf);
+
+    city_tile_iterate(city_map_radius_sq_get(pcity), pcenter, ptile) {
+      char f[2];
+      char s[2];
+      char t[2];
+
+      fc_snprintf(f, sizeof(f), "%d", city_tile_output_now(pcity, ptile, O_FOOD));
+      fc_snprintf(s, sizeof(s), "%d", city_tile_output_now(pcity, ptile, O_SHIELD));
+      fc_snprintf(t, sizeof(t), "%d", city_tile_output_now(pcity, ptile, O_TRADE));
+      food_output_buf[c] = f[0];
+      shield_output_buf[c] = s[0];
+      trade_output_buf[c] = t[0];
+
+      c += 1;
+
+    } city_tile_iterate_end;
+
+    food_output_buf[c] = '\0';
+    shield_output_buf[c] = '\0';
+    trade_output_buf[c] = '\0';
+    sz_strlcpy(packet->food_output, food_output_buf);
+    sz_strlcpy(packet->shield_output, shield_output_buf);
+    sz_strlcpy(packet->trade_output, trade_output_buf);
   }
 #endif /* FREECIV_WEB */
 }
