diff -Nurd freeciv/common/networking/packets.def freeciv/common/networking/packets.def
--- freeciv/common/networking/packets.def	2022-11-05 01:33:56.941573783 +0200
+++ freeciv/common/networking/packets.def	2022-11-05 01:34:08.593678176 +0200
@@ -2385,6 +2385,10 @@
 
   BV_UTYPES can_build_unit;
   BV_IMPRS can_build_improvement;
+
+  STRING food_output[MAX_CITY_TILES + 1];
+  STRING shield_output[MAX_CITY_TILES + 1];
+  STRING trade_output[MAX_CITY_TILES + 1];
 end
 
 PACKET_WEB_CMA_SET = 257; cs, handle-via-fields
diff -Nurd freeciv/server/citytools.c freeciv/server/citytools.c
--- freeciv/server/citytools.c	2022-11-05 01:33:57.309577081 +0200
+++ freeciv/server/citytools.c	2022-11-05 01:34:08.593678176 +0200
@@ -2631,6 +2631,9 @@
 
 #ifdef FREECIV_WEB
   if (web_packet != NULL) {
+    struct tile *pcenter = city_tile(pcity);
+    int c = 0;
+
     BV_CLR_ALL(web_packet->can_build_unit);
     BV_CLR_ALL(web_packet->can_build_improvement);
 
@@ -2658,6 +2661,25 @@
         BV_SET(web_packet->can_build_unit, utype_index(punittype));
       }
     } unit_type_iterate_end;
+
+    city_tile_iterate(city_map_radius_sq_get(pcity), pcenter, ptile) {
+      char f[2];
+      char s[2];
+      char t[2];
+
+      fc_snprintf(f, sizeof(f), "%d", city_tile_output_now(pcity, ptile, O_FOOD));
+      fc_snprintf(s, sizeof(s), "%d", city_tile_output_now(pcity, ptile, O_SHIELD));
+      fc_snprintf(t, sizeof(t), "%d", city_tile_output_now(pcity, ptile, O_TRADE));
+      web_packet->food_output[c] = f[0];
+      web_packet->shield_output[c] = s[0];
+      web_packet->trade_output[c] = t[0];
+
+      c++;
+    } city_tile_iterate_end;
+
+    web_packet->food_output[c] = '\0';
+    web_packet->shield_output[c] = '\0';
+    web_packet->trade_output[c] = '\0';
   }
 #endif /* FREECIV_WEB */
 }
