diff -Nurd -X.diff_ignore freeciv/utility/registry.c freeciv/utility/registry.c
--- freeciv/utility/registry.c	2012-05-19 01:03:30.000000000 +0300
+++ freeciv/utility/registry.c	2012-05-19 02:34:08.000000000 +0300
@@ -928,9 +928,7 @@
   va_end(ap);
 
   if(!(pentry=section_file_lookup_internal(my_section_file, buf))) {
-    freelog(LOG_FATAL, "sectionfile %s doesn't contain a '%s' entry",
-	    secfile_filename(my_section_file), buf);
-    exit(EXIT_FAILURE);
+    return NULL;
   }
 
   if(!pentry->svalue) {
