From d9345c8803d16ffee4601ed68d6303b78f350c0f Mon Sep 17 00:00:00 2001
From: Sveinung Kvilhaugsvik <sveinung84@users.sourceforge.net>
Date: Sun, 10 Jan 2016 20:20:54 +0100
Subject: [PATCH 3/3] Don't crash on missing trade partner.

---
 common/city.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/common/city.c b/common/city.c
index d24eb50..c216c4e 100644
--- a/common/city.c
+++ b/common/city.c
@@ -2780,7 +2780,7 @@ inline void set_city_production(struct city *pcity)
     struct city *tcity = game_city_by_number(proute->partner);
     bool can_trade;
 
-    fc_assert(tcity != NULL);
+    fc_assert_action(tcity != NULL, continue);
 
     can_trade = can_cities_trade(pcity, tcity);
 
-- 
2.6.4

