diff -Nurd -X.diff_ignore freeciv/common/connection.h freeciv/common/connection.h
--- freeciv/common/connection.h	2015-03-16 07:03:22.369089846 +0200
+++ freeciv/common/connection.h	2015-03-16 07:03:56.053089107 +0200
@@ -53,7 +53,7 @@
 struct timer_list;
 
 /* Used in the network protocol. */
-#define MAX_LEN_PACKET   4096
+#define MAX_LEN_PACKET    16384
 #define MAX_LEN_CAPSTR    512
 #define MAX_LEN_PASSWORD  512 /* do not change this under any circumstances */
 
diff -Nurd -X.diff_ignore freeciv/common/generate_packets.py freeciv/common/generate_packets.py
--- freeciv/common/generate_packets.py	2015-03-16 07:03:22.401089845 +0200
+++ freeciv/common/generate_packets.py	2015-03-16 07:07:53.705083893 +0200
@@ -250,7 +250,7 @@
         if self.is_array==1:
             tmp="real_packet->%(name)s[i] = %(name)s[i]"%self.__dict__
             return '''  {
-    int i;
+    int i; /* this is for put. */
 
     for (i = 0; i < %(array_size_u) s; i++) {
       %(tmp)s;
@@ -364,25 +364,20 @@
         if (self.dataio_type in arr_types and self.is_array==1) or \
            (self.dataio_type not in arr_types and self.is_array==0):
             return "  DIO_PUT(%(dataio_type)s, &dout, \"%(name)s\", real_packet->%(name)s);"%self.__dict__
-        if self.is_struct:
-            if self.is_array==2:
-                c="dio_put_%(dataio_type)s_raw(&dout, &real_packet->%(name)s[i][j]);"%self.__dict__
-            else:
-                c="dio_put_%(dataio_type)s_raw(&dout, &real_packet->%(name)s[i]);"%self.__dict__
         elif self.dataio_type=="string":
-            c="dio_put_%(dataio_type)s_raw(&dout, real_packet->%(name)s[i]);"%self.__dict__
+            c="/* dio_put_%(dataio_type)s_raw(&dout, \"%(name)s\", real_packet->%(name)s[i]); */"%self.__dict__
             array_size_u=self.array_size1_u
+            self.__dict__["array_size_u"] = array_size_u
 
         elif self.struct_type=="float":
-            if self.is_array==2:
-                c="  dio_put_%(dataio_type)s_raw(&dout, real_packet->%(name)s[i][j], %(float_factor)d);"%self.__dict__
-            else:
-                c="  dio_put_%(dataio_type)s_raw(&dout, real_packet->%(name)s[i], %(float_factor)d);"%self.__dict__
+            c="/*  dio_put_%(dataio_type)s_raw(&dout, \"%(name)s\", real_packet->%(name)s[i][j], %(float_factor)d); */"%self.__dict__
+        elif self.dataio_type=="requirement":
+            c="  DIO_PUT(%(dataio_type)s, &dout, \"%(name)s\", (struct requirement *)real_packet->%(name)s, %(array_size_u)s);"%self.__dict__
         else:
             if self.is_array==2:
-                c="dio_put_%(dataio_type)s_raw(&dout, real_packet->%(name)s[i][j]);"%self.__dict__
+                c="/* dio_put_%(dataio_type)s_raw(&dout, \"%(name)s\", real_packet->%(name)s[i][j]);"%self.__dict__
             else:
-                c="dio_put_%(dataio_type)s_raw(&dout, real_packet->%(name)s[i]);"%self.__dict__
+                c="dio_put_array_%(dataio_type)s_json(&dout, \"%(name)s\", (void *)real_packet->%(name)s, %(array_size_u)s);"%self.__dict__
 
         if not self.diff:
             if self.is_array==2 and self.dataio_type!="string":
@@ -392,27 +387,18 @@
 
       for (i = 0; i < %(array_size1_u)s; i++) {
         for (j = 0; j < %(array_size2_u)s; j++) {
-#ifndef FREECIV_JSON_CONNECTION
           %(c)s
-#endif /* FREECIV_JSON_CONNECTION */
         }
       }
     } '''%self.get_dict(vars())
             else:
                 return '''
     {
-      int i;
-
-      for (i = 0; i < %(array_size_u)s; i++) {
-#ifndef FREECIV_JSON_CONNECTION
         %(c)s
-#endif /* FREECIV_JSON_CONNECTION */
-      }
     } '''%self.get_dict(vars())
         else:
             return '''
     {
-#ifndef FREECIV_JSON_CONNECTION
       int i;
 
       fc_assert(%(array_size_u)s < 255);
@@ -424,7 +410,6 @@
         }
       }
       dio_put_uint8_raw(&dout, 255);
-#endif /* FREECIV_JSON_CONNECTION */
     } '''%self.get_dict(vars())
 
     # Returns a code fragement which will get the field if the
@@ -549,10 +534,10 @@
             else:
                 extra=""
             if self.dataio_type=="memory":
-                return '''%(extra)s
-  if (!DIO_GET(%(dataio_type)s, &din, \"%(name)s\", real_packet->%(name)s, %(array_size_u)s)){
+                return ''' /* not supported yet. %(extra)s
+  if (!dio_get_%(dataio_type)s(pc->json_packet, \"%(name)s\", real_packet->%(name)s, %(array_size_u)s)){
     RECEIVE_PACKET_FIELD_ERROR(%(name)s);
-  }'''%self.get_dict(vars())
+  } */'''%self.get_dict(vars())
             elif self.is_array==2 and self.dataio_type!="string":
                 return '''
 {
@@ -937,7 +922,7 @@
   struct %(packet_name)s *old;
   struct genhash **hash = pc->phs.received + %(type)s;
 '''
-            delta_body1="\n  DIO_BV_GET(&din, \"fields\", fields);\n"
+            delta_body1="\n  DIO_BV_GET(pc->json_packet, \"fields\", fields);\n"
             body1=""
             for field in self.key_fields:
                 body1=body1+prefix("  ",field.get_get())+"\n"
diff -Nurd -X.diff_ignore freeciv/server/handchat.c freeciv/server/handchat.c
--- freeciv/server/handchat.c	2015-03-16 07:03:22.481089843 +0200
+++ freeciv/server/handchat.c	2015-03-16 07:03:56.053089107 +0200
@@ -19,7 +19,10 @@
 #include <stdio.h>
 #include <string.h>
 
+#include <glib.h>
+
 /* utility */
+#include "fciconv.h"
 #include "fcintl.h"
 #include "log.h"
 #include "shared.h"
@@ -319,14 +322,17 @@
 void handle_chat_msg_req(struct connection *pconn, const char *message)
 {
   char real_message[MAX_LEN_MSG], *cp;
+  char* unesc_message;
   bool double_colon;
 
-  sz_strlcpy(real_message, message);
+  unesc_message = g_uri_unescape_string(message, NULL);
+  sz_strlcpy(real_message, unesc_message);
+  g_free(unesc_message);
 
   /* This loop to prevent players from sending multiple lines which can
    * be abused */
   for (cp = real_message; *cp != '\0'; cp++) {
-    if (*cp == '\n' || *cp == '\r') {
+    if (*cp == '\n' || *cp == '\r' || *cp == '<' || *cp == '>' || *cp == '"' || *cp == '\'') {
       *cp = '\0';
       break;
     }
diff -Nurd -X.diff_ignore freeciv/server/plrhand.c freeciv/server/plrhand.c
--- freeciv/server/plrhand.c	2015-03-16 07:03:22.493089843 +0200
+++ freeciv/server/plrhand.c	2015-03-16 07:03:56.053089107 +0200
@@ -1243,12 +1243,12 @@
       || (receiver
           && player_diplstate_get(receiver, plr1)->contact_turns_left > 0)
       || (receiver && receiver == plr2)) {
-    packet_ds->type                 = ds->type;
+    packet_ds->ds_type              = ds->type;
     packet_ds->turns_left           = ds->turns_left;
     packet_ds->has_reason_to_cancel = ds->has_reason_to_cancel;
     packet_ds->contact_turns_left   = ds->contact_turns_left;
   } else {
-    packet_ds->type                 = DS_WAR;
+    packet_ds->ds_type              = DS_WAR;
     packet_ds->turns_left           = 0;
     packet_ds->has_reason_to_cancel = 0;
     packet_ds->contact_turns_left   = 0;
diff -Nurd -X.diff_ignore freeciv/server/ruleset.c freeciv/server/ruleset.c
--- freeciv/server/ruleset.c	2015-03-16 07:03:22.417089845 +0200
+++ freeciv/server/ruleset.c	2015-03-16 07:03:56.057089107 +0200
@@ -6380,6 +6380,8 @@
     packet.ngroups = i;
 
     packet.init_government_id = government_number(n->init_government);
+
+#if 0
     fc_assert(ARRAY_SIZE(packet.init_techs) == ARRAY_SIZE(n->init_techs));
     for (i = 0; i < MAX_NUM_TECH_LIST; i++) {
       packet.init_techs[i] = n->init_techs[i];
@@ -6395,6 +6397,7 @@
       /* Impr_type_id to int */
       packet.init_buildings[i] = n->init_buildings[i];
     }
+#endif
 
     lsend_packet_ruleset_nation(dest, &packet);
   } nations_iterate_end;
