diff -Nurd -X.diff_ignore freeciv/server/sernet.c freeciv/server/sernet.c
--- freeciv/server/sernet.c	2012-05-21 12:41:29.000000000 +0300
+++ freeciv/server/sernet.c	2012-05-23 00:46:33.000000000 +0300
@@ -497,7 +497,7 @@
       if (connections && conn_list_size(game.est_connections) == 0) {
 	if (last_noplayers != 0) {
 	  if (time(NULL) > last_noplayers + srvarg.quitidle) {
-	    save_game_auto("Lost all connections", "quitidle");
+            /* save_game_auto("Lost all connections", "quitidle"); */
             log_normal(_("Restarting for lack of players."));
             set_meta_message_string("restarting for lack of players");
 	    (void) send_server_info_to_metaserver(META_INFO);
