Index: server/sernet.c
===================================================================
--- server/sernet.c	(revisjon 30)
+++ server/sernet.c	(arbeidskopi)
@@ -501,7 +501,7 @@
       if (connections && conn_list_size(game.est_connections) == 0) {
 	if (last_noplayers != 0) {
 	  if (time(NULL) > last_noplayers + srvarg.quitidle) {
-	    save_game_auto("Lost all connections", "quitidle");
+	    //save_game_auto("Lost all connections", "quitidle");
 	    set_meta_message_string(N_("restarting for lack of players"));
 	    freelog(LOG_NORMAL, "%s", Q_(get_meta_message_string()));
 	    (void) send_server_info_to_metaserver(META_INFO);
