diff -Nurd -X.diff_ignore freeciv/data/classic/game.ruleset freeciv/data/classic/game.ruleset
--- freeciv/data/classic/game.ruleset	2012-08-08 00:18:48.000000000 +0300
+++ freeciv/data/classic/game.ruleset	2012-08-08 00:55:09.000000000 +0300
@@ -307,3 +307,8 @@
 ;       "int_set", 123, FALSE
 ;       "str_set", "test", FALSE
 ;     }
+
+set =
+  { "name", "value", "lock"
+    "autosaves", "", TRUE
+  }
