diff -Nurd -X.diff_ignore freeciv/data/default/game.ruleset freeciv/data/default/game.ruleset
--- freeciv/data/default/game.ruleset	2012-07-17 02:35:27.000000000 +0300
+++ freeciv/data/default/game.ruleset	2012-07-17 02:57:36.000000000 +0300
@@ -259,3 +259,8 @@
 ;       "int_set", 123, FALSE
 ;       "str_set", "test", FALSE
 ;     }
+
+set =
+  { "name", "value", "lock"
+    "autosaves", "", TRUE
+  }
