From 018430031c3459e633d094dc2ab8f09dbfc9ee6a Mon Sep 17 00:00:00 2001
From: Sveinung Kvilhaugsvik <sveinung84@users.sourceforge.net>
Date: Wed, 29 Jul 2015 00:55:21 +0200
Subject: [PATCH] handle_unit_orders() log errors caught by the order
 validation.

---
 server/unithand.c | 31 +++++++++++++++++++++++++++++++
 1 file changed, 31 insertions(+)

diff --git a/server/unithand.c b/server/unithand.c
index f45c7e8..05aa230 100644
--- a/server/unithand.c
+++ b/server/unithand.c
@@ -3414,6 +3414,11 @@ void handle_unit_orders(struct player *pplayer,
     case ORDER_MOVE:
     case ORDER_ACTION_MOVE:
       if (!is_valid_dir(packet->dir[i])) {
+        log_error("handle_unit_orders() %d isn't a valid move direction. "
+                  "Sent in order number %d from %s to unit number %d.",
+                  packet->dir[i], i,
+                  player_name(pplayer), packet->unit_id);
+
 	return;
       }
       break;
@@ -3432,16 +3437,32 @@ void handle_unit_orders(struct player *pplayer,
       case ACTIVITY_SENTRY:
         if (i != length - 1) {
           /* Only allowed as the last order. */
+          log_error("handle_unit_orders() activity %d is only allowed in "
+                    "the last order. "
+                    "Sent in order number %d from %s to unit number %d.",
+                    packet->activity[i], i,
+                    player_name(pplayer), packet->unit_id);
+
           return;
         }
         break;
       case ACTIVITY_BASE:
         if (!is_extra_caused_by(extra_by_number(packet->target[i]), EC_BASE)) {
+          log_error("handle_unit_orders() %s isn't a base. "
+                    "Sent in order number %d from %s to unit number %d.",
+                    extra_rule_name(extra_by_number(packet->target[i])), i,
+                    player_name(pplayer), packet->unit_id);
+
           return;
         }
         break;
       case ACTIVITY_GEN_ROAD:
         if (!is_extra_caused_by(extra_by_number(packet->target[i]), EC_ROAD)) {
+          log_error("handle_unit_orders() %s isn't a road. "
+                    "Sent in order number %d from %s to unit number %d.",
+                    extra_rule_name(extra_by_number(packet->target[i])), i,
+                    player_name(pplayer), packet->unit_id);
+
           return;
         }
         break;
@@ -3460,6 +3481,11 @@ void handle_unit_orders(struct player *pplayer,
       case ACTIVITY_PATROL_UNUSED:
       case ACTIVITY_LAST:
       case ACTIVITY_UNKNOWN:
+        log_error("handle_unit_orders() unsupported activity %d. "
+                  "Sent in order number %d from %s to unit number %d.",
+                  packet->activity[i], i,
+                  player_name(pplayer), packet->unit_id);
+
         return;
       }
 
@@ -3467,6 +3493,11 @@ void handle_unit_orders(struct player *pplayer,
           && unit_activity_needs_target_from_client(packet->activity[i])) {
         /* The orders system can't do server side target assignment for
          * this activity. */
+        log_error("handle_unit_orders() can't assign target for %d. "
+                  "Sent in order number %d from %s to unit number %d.",
+                  packet->activity[i], i,
+                  player_name(pplayer), packet->unit_id);
+
         return;
       }
 
-- 
2.1.4

