diff -Nurd -X.diff_ignore freeciv/server/unittools.c freeciv/server/unittools.c
--- freeciv/server/unittools.c	2012-05-16 22:08:49.000000000 +0300
+++ freeciv/server/unittools.c	2012-05-16 22:09:00.000000000 +0300
@@ -3254,10 +3254,10 @@
       if (!res && punit->moves_left > 0) {
 	/* Movement failed (ZOC, etc.) */
 	cancel_orders(punit, "  attempt to move failed.");
-	notify_player(pplayer, punit->tile, E_UNIT_ORDERS,
+	/*notify_player(pplayer, punit->tile, E_UNIT_ORDERS,
                       FTC_SERVER_INFO, NULL,
                       _("Orders for %s aborted because of failed move."),
-                      unit_link(punit));
+                      unit_link(punit));*/
 	return TRUE;
       }
 
