diff -Nurd -X.diff_ignore freeciv/common/packets.def freeciv/common/packets.def
--- freeciv/common/packets.def	2014-02-08 18:30:52.537057466 +0200
+++ freeciv/common/packets.def	2014-02-08 18:44:24.854939992 +0200
@@ -1373,13 +1373,13 @@
   STRING rule_name[MAX_LEN_NAME];
   STRING graphic_str[MAX_LEN_NAME];
   STRING graphic_alt[MAX_LEN_NAME];
-  UINT8 reqs_count;
+  UINT32 reqs_count;
   REQUIREMENT reqs[MAX_NUM_REQS:reqs_count];
   UINT8 obs_count;
   REQUIREMENT obs_reqs[MAX_NUM_REQS:obs_count];
   IMPROVEMENT replaced_by;
-  UINT16 build_cost;
-  UINT8 upkeep, sabotage;
+  UINT32 build_cost;
+  UINT32 upkeep, sabotage;
   BV_IMPR_FLAGS flags;
   STRING soundtag[MAX_LEN_NAME];
   STRING soundtag_alt[MAX_LEN_NAME];
