Index: freeciv/common/packets.def
===================================================================
--- freeciv/common/packets.def	(revision 23003)
+++ freeciv/common/packets.def	(working copy)
@@ -506,6 +507,7 @@
   BOOL year_0_hack;
   BOOL civil_war_enabled;
   BOOL paradrop_to_transport;
+  STRING meta_message[MAX_LEN_NAME];
 end
 
 PACKET_MAP_INFO = 17; sc, lsend
@@ -1375,13 +1375,13 @@
   STRING rule_name[MAX_LEN_NAME];
   STRING graphic_str[MAX_LEN_NAME];
   STRING graphic_alt[MAX_LEN_NAME];
-  UINT8 reqs_count;
+  UINT32 reqs_count;
   REQUIREMENT reqs[MAX_NUM_REQS:reqs_count];
   UINT8 obs_count;
   REQUIREMENT obs_reqs[MAX_NUM_REQS:obs_count];
   IMPROVEMENT replaced_by;
-  UINT16 build_cost;
-  UINT8 upkeep, sabotage;
+  UINT32 build_cost;
+  UINT32 upkeep, sabotage;
   BV_IMPR_FLAGS flags;
   STRING soundtag[MAX_LEN_NAME];
   STRING soundtag_alt[MAX_LEN_NAME];
