diff -Nurd -X.diff_ignore freeciv/server/unittools.c freeciv/server/unittools.c
--- freeciv/server/unittools.c	2014-08-13 19:24:28.262961196 +0300
+++ freeciv/server/unittools.c	2014-08-15 23:32:38.026650601 +0300
@@ -2759,6 +2759,7 @@
     }
 
     if (NULL != plrtile->site
+        && plrtile->owner != NULL
         && pplayers_non_attack(pplayer, plrtile->owner)) {
       notify_player(pplayer, ptile, E_BAD_COMMAND, ftc_server,
                     _("Cannot attack unless you declare war first."));
