diff -Nurd -X.diff_ignore freeciv/common/packets_json.c freeciv/common/packets_json.c
--- freeciv/common/packets_json.c	2015-02-19 21:36:24.107331510 +0200
+++ freeciv/common/packets_json.c	2015-02-19 23:30:25.207181415 +0200
@@ -201,6 +201,8 @@
 #endif
   void *data;
   void *(*receive_handler)(struct connection *);
+  json_error_t error;
+  json_t *pint;
 
   if (!pc->used) {
     return NULL;		/* connection was closed, stop reading */
@@ -323,7 +325,6 @@
   }
 
   /* Parse JSON packet. */
-  json_error_t error;
 
   pc->json_packet = json_loadb((char*)pc->buffer->data + 4, whole_packet_len, 0, &error);
 
@@ -335,7 +336,7 @@
     return NULL;
   }
 
-  json_t *pint = json_object_get(pc->json_packet, "type");
+  pint = json_object_get(pc->json_packet, "pid");
 
   if (!pint) {
     log_error("ERROR: Unable to get packet type.");
