diff -Nurd -X.diff_ignore freeciv/utility/ioz.c freeciv/utility/ioz.c
--- freeciv/utility/ioz.c	2015-09-24 03:29:52.706101859 +0300
+++ freeciv/utility/ioz.c	2015-10-27 23:39:42.807298300 +0200
@@ -70,7 +70,7 @@
 
 #ifdef FREECIV_HAVE_LIBLZMA
 
-#define PLAIN_FILE_BUF_SIZE (250*1024)     /* 250kb */
+#define PLAIN_FILE_BUF_SIZE (1024*1024)    /* 1024kb */
 #define XZ_DECODER_TEST_SIZE (4*1024)      /* 4kb */
 
 /* In my tests 7Mb proved to be not enough and with 10Mb decompression
