diff -Nurd freeciv/common/city.h freeciv/common/city.h
--- freeciv/common/city.h	2018-03-05 17:14:08.620868766 +0200
+++ freeciv/common/city.h	2018-03-05 17:14:29.248740773 +0200
@@ -302,6 +302,7 @@
 
 struct adv_city; /* defined in ./server/advisors/infracache.h */
 
+#pragma pack(push, 1)
 struct city {
   char name[MAX_LEN_CITYNAME];
   struct tile *tile; /* May be NULL, should check! */
@@ -449,6 +450,7 @@
     } client;
   };
 };
+#pragma pack(pop)
 
 struct citystyle {
   struct name_translation name;
diff -Nurd freeciv/common/fc_types.h freeciv/common/fc_types.h
--- freeciv/common/fc_types.h	2018-03-05 17:14:08.620868766 +0200
+++ freeciv/common/fc_types.h	2018-03-05 17:15:49.704398376 +0200
@@ -65,7 +65,7 @@
 
 /* Changing these will probably break network compatability. */
 #define MAX_LEN_NAME       48
-#define MAX_LEN_CITYNAME   80
+#define MAX_LEN_CITYNAME   50
 #define MAX_LEN_MAP_LABEL  64
 #define MAX_LEN_DEMOGRAPHY 16
 #define MAX_LEN_ALLOW_TAKE 16
Binary files freeciv/.git/index and freeciv/.git/index differ
