diff -Nurd -X.diff_ignore freeciv/server/savegame2.c freeciv/server/savegame2.c
--- freeciv/server/savegame2.c	2016-02-24 04:41:27.639083775 +0200
+++ freeciv/server/savegame2.c	2016-02-24 04:53:11.557909440 +0200
@@ -4097,29 +4097,6 @@
     punit->upkeep[o] = utype_upkeep_cost(unit_type_get(punit), plr, o);
   } output_type_iterate_end;
 
-  punit->action_decision_want
-      = secfile_lookup_enum_default(loading->file,
-                                    ACT_DEC_NOTHING, action_decision,
-                                    "%s.action_decision_want", unitstr);
-
-  if (punit->action_decision_want != ACT_DEC_NOTHING) {
-    /* Load the tile to act against. */
-    int adwt_x, adwt_y;
-
-    if (secfile_lookup_int(loading->file, &adwt_x,
-                           "%s.action_decision_tile_x", unitstr)
-        && secfile_lookup_int(loading->file, &adwt_y,
-                              "%s.action_decision_tile_y", unitstr)) {
-      punit->action_decision_tile = native_pos_to_tile(adwt_x, adwt_y);
-    } else {
-      punit->action_decision_want = ACT_DEC_NOTHING;
-      punit->action_decision_tile = NULL;
-      log_sg("Bad action_decision_tile for unit %d", punit->id);
-    }
-  } else {
-    punit->action_decision_tile = NULL;
-  }
-
   /* load the unit orders */
   {
     int len = secfile_lookup_int_default(loading->file, 0,
diff -Nurd -X.diff_ignore freeciv/server/savegame3.c freeciv/server/savegame3.c
--- freeciv/server/savegame3.c	2016-02-24 04:41:27.639083775 +0200
+++ freeciv/server/savegame3.c	2016-02-24 04:53:11.557909440 +0200
@@ -5279,29 +5279,6 @@
     punit->upkeep[o] = utype_upkeep_cost(unit_type_get(punit), plr, o);
   } output_type_iterate_end;
 
-  punit->action_decision_want
-      = secfile_lookup_enum_default(loading->file,
-                                    ACT_DEC_NOTHING, action_decision,
-                                    "%s.action_decision_want", unitstr);
-
-  if (punit->action_decision_want != ACT_DEC_NOTHING) {
-    /* Load the tile to act against. */
-    int adwt_x, adwt_y;
-
-    if (secfile_lookup_int(loading->file, &adwt_x,
-                           "%s.action_decision_tile_x", unitstr)
-        && secfile_lookup_int(loading->file, &adwt_y,
-                              "%s.action_decision_tile_y", unitstr)) {
-      punit->action_decision_tile = native_pos_to_tile(adwt_x, adwt_y);
-    } else {
-      punit->action_decision_want = ACT_DEC_NOTHING;
-      punit->action_decision_tile = NULL;
-      log_sg("Bad action_decision_tile for unit %d", punit->id);
-    }
-  } else {
-    punit->action_decision_tile = NULL;
-  }
-
   /* load the unit orders */
   {
     int len = secfile_lookup_int_default(loading->file, 0,
@@ -5642,26 +5619,6 @@
                                      ? unit_transport_get(punit)->id : -1,
                        "%s.transported_by", buf);
 
-    secfile_insert_enum(saving->file, punit->action_decision_want,
-                        action_decision, "%s.action_decision_want", buf);
-
-    /* Stored as tile rather than direction to make sure the target tile is
-     * sane. */
-    if (punit->action_decision_tile) {
-      index_to_native_pos(&nat_x, &nat_y,
-                          tile_index(punit->action_decision_tile));
-      secfile_insert_int(saving->file, nat_x,
-                         "%s.action_decision_tile_x", buf);
-      secfile_insert_int(saving->file, nat_y,
-                         "%s.action_decision_tile_y", buf);
-    } else {
-      /* Dummy values to get tabular format. */
-      secfile_insert_int(saving->file, -1,
-                         "%s.action_decision_tile_x", buf);
-      secfile_insert_int(saving->file, -1,
-                         "%s.action_decision_tile_y", buf);
-    }
-
     if (punit->has_orders) {
       int len = punit->orders.length;
       char orders_buf[len + 1], dir_buf[len + 1];
