diff -Nurd -X.diff_ignore freeciv/common/aicore/pf_tools.c freeciv/common/aicore/pf_tools.c
--- freeciv/common/aicore/pf_tools.c	2014-08-31 17:28:38.891428325 +0300
+++ freeciv/common/aicore/pf_tools.c	2014-09-06 03:31:41.331103747 +0300
@@ -691,19 +691,11 @@
                                  struct tile *pstart_tile,
                                  struct player *powner)
 {
-  int veteran_level = get_unittype_bonus(powner, pstart_tile, punittype,
-                                         EFT_VETERAN_BUILD);
-
-  if (veteran_level >= utype_veteran_levels(punittype)) {
-    veteran_level = utype_veteran_levels(punittype) - 1;
-  }
-
   pft_fill_default_parameter(parameter, punittype);
 
   parameter->start_tile = pstart_tile;
   parameter->moves_left_initially = punittype->move_rate;
-  parameter->move_rate = utype_move_rate(punittype, pstart_tile, powner,
-                                         veteran_level, punittype->hp);
+  parameter->move_rate = punittype->move_rate;
   if (utype_fuel(punittype)) {
     parameter->fuel_left_initially = utype_fuel(punittype);
     parameter->fuel = utype_fuel(punittype);
