diff --git a/data/amplio2/extra_units.spec b/data/amplio2/extra_units.spec
index 97457e7063..8890179983 100644
--- a/data/amplio2/extra_units.spec
+++ b/data/amplio2/extra_units.spec
@@ -26,105 +26,27 @@ dy = 48
 pixel_border = 1
 
 tiles = { "row", "column", "tag"
-  0,  0, "u.gladiators_Idle"
-  0,  1, "u.arbalests_Idle"
-  0,  2, "u.halberdiers_Idle"
+
   0,  3, "u.samurai_Idle"
-  0,  4, "u.militia_Idle"
-  0,  5, "u.line_infantry_Idle"
-  0,  6, "u.grenadier_Idle"
-  0,  7, "u.fusiliers_Idle"
-  0,  8, "u.musket_militia_Idle"
-  0,  9, "u.marksman_Idle"
-
-  1,  0, "u.stormtroopers_Idle"
-  1,  1, "u.commando_Idle"
-  1,  7, "u.motorized_infantry_Idle"
-  1,  8, "u.tachanka_Idle"
-  1,  9, "u.mobile_fanatics_Idle"
-
-  2,  0, "u.dromedari_Idle"
-  2,  1, "u.mounted_archers_Idle"
-  2,  2, "u.mounted_samurai_Idle"
-  2,  3, "u.uhlan_Idle"
-  2,  4, "u.mounted_militia_Idle"
-  2,  5, "u.mounted_marksman_Idle"
-
-  3,  0, "u.early_light_tank_Idle", "u.mark_iv_Idle"
-  3,  1, "u.early_tank_Idle", "u.whippet_Idle"
-  3,  2, "u.early_heavy_tank_Idle", "u.renault_ft_Idle"
-  3,  3, "u.tank_Idle", "u.sherman_Idle"
-  3,  4, "u.heavy_tank_Idle", "u.leopard_Idle"
-  3,  5, "u.reactive_armor_Idle", "u.abrahams_Idle"
-  3,  6, "u.armored_car_Idle", "u.rolls_royce_armored_car_Idle"
-  3,  7, "u.armored_recon_vehicle_Idle", "u.rooikat_Idle"
-
-  4,  0, "u.trebuchet_Idle"
-  4,  1, "u.cannon_alternative_Idle"
-  4,  2, "u.longrange_howitzer_Idle", "u.panzerhaubitze2000_Idle"
-  4,  3, "u.ballista_Idle"
-  4,  4, "u.siege_tower_Idle"
-  4,  5, "u.battering_ram_Idle"
-
-  5,  0, "u.biplane_Idle"
-  5,  1, "u.gliderplane_Idle", "u.gotha_Idle"
-  5,  2, "u.ekranoplan_Idle"
-  5,  3, "u.divebomber_Idle"
-  5,  4, "u.transportplane_Idle", "u.c47_Idle"
-  5,  5, "u.jetfighter_Idle", "u.mig_Idle"
-  5,  6, "u.jetbomber_Idle"
-  5,  7, "u.supersonic_fighter_Idle"
-  5,  8, "u.transportjet_Idle"
-  5,  9, "u.strikeplane_Idle", "u.a10_warthog_Idle"
-  5, 10, "u.strikejet_Idle", "u.f4_Idle"
+
   5, 12, "u.balloon_Idle"
   5, 13, "u.zeppelin_Idle"
 
-  6,  0, "u.light_helicopter_Idle", "u.alouette_iii_Idle"
-  6,  1, "u.transport_helicopter_Idle", "u.chinook_Idle"
-  6,  2, "u.attack_transport_helicopter_Idle", "u.mi24_Idle"
   6,  4, "u.drone_Idle"
-  6,  6, "u.v2_rocket_Idle"
-
-  7,  0, "u.quinquireme_Idle"
-  7,  1, "u.longboat_Idle"
-  7,  2, "u.junk_Idle"
-  7,  3, "u.cog_Idle"
-  7,  4, "u.galleon_alternative_Idle"
-  7,  5, "u.frigate_alternative_Idle"
-  7,  6, "u.flagship_Idle"
-  7,  7, "u.paddle_steamer_Idle"
-  7,  8, "u.patrol_boat_Idle"
+
   7,  9, "u.feluca_Idle"
-  7, 10, "u.container_ship_Idle"
 
   8,  0, "u.dredger_Idle"
-  8,  1, "u.icebreaker_Idle"
-  8,  2, "u.minesweeper_Idle"
+  8,  1, "u.longboat_Idle"
+
   8,  4, "u.rowboat_Idle"
-  8,  6, "u.early_submarine_Idle"
 
-  9,  0, "u.slave_Idle"
   9,  1, "u.emissary_Idle"
-  9,  2, "u.barbarian_leader_alternative_Idle"
-  9,  3, "u.early_explorer_Idle"
+
   9,  4, "u.operative_Idle"
-  9,  5, "u.armoured_train_Idle"
-  9,  6, "u.rail_artillery_Idle"
 
- 10,  0, "u.wolves_Idle"
- 10,  1, "u.panther_Idle"
- 10,  2, "u.tiger_Idle"
  10,  3, "u.lion_Idle"
- 10,  4, "u.bear_Idle"
- 10,  5, "u.snake_Idle"
- 10,  6, "u.alligator_Idle"
- 10,  7, "u.orangutan_Idle"
- 10,  8, "u.hippopotamus_Idle"
- 10,  9, "u.rhinoceros_Idle"
- 10, 10, "u.elephant_Idle"
- 10, 11, "u.polar_bear_Idle"
- 10, 12, "u.squid_Idle"
+
  10, 13, "u.storm_Idle"
- 10, 14, "u.mammoth_Idle"
+
 }
