diff -Nurd -X.diff_ignore freeciv/common/game.c freeciv/common/game.c
--- freeciv/common/game.c	2012-05-21 12:41:30.000000000 +0300
+++ freeciv/common/game.c	2012-05-23 00:47:36.000000000 +0300
@@ -724,6 +724,13 @@
     CF_INT_SEQ('Y', game.info.year),
     CF_END
   };
+  time_t     now;
+  struct tm  *ts;
+  char buff[80];
+
+  now = time(0); 
+  ts = localtime(&now);
+  strftime(buff, sizeof(buff), "%Y-%m-%d-%H-%M-%S", ts);
 
   fc_vsnprintcf(buf, buflen, format, sequences, -1);
 
@@ -732,8 +739,8 @@
      * printf information. */
     char savename[512];
 
-    fc_snprintf(savename, sizeof(savename), "%s-T%%04T-Y%%05Y-%%R",
-                format);
+    fc_snprintf(savename, sizeof(savename), "%s-T%%04T-Y%%05Y-%%R-%s",
+                format, buff);
     fc_vsnprintcf(buf, buflen, savename, sequences, -1);
   }
 
diff -Nurd -X.diff_ignore freeciv/configure.ac freeciv/configure.ac
--- freeciv/configure.ac	2012-05-21 13:37:46.000000000 +0300
+++ freeciv/configure.ac	2012-05-23 00:47:36.000000000 +0300
@@ -844,7 +844,7 @@
   DEFAULT_SCENARIO_PATH=".;data/scenario;~/.freeciv/scenario;$datadir/freeciv/scenario"
 else
   DEFAULT_DATA_PATH=".:data:~/.freeciv/$DATASUBDIR:$datadir/freeciv"
-  DEFAULT_SAVE_PATH=".:~/.freeciv/saves"
+  DEFAULT_SAVE_PATH=".:~/.freeciv/saves:/mnt/savegames"
   DEFAULT_SCENARIO_PATH=".:data/scenario:~/.freeciv/scenarios:$datadir/freeciv/scenario"
 fi
 CPPFLAGS="$CPPFLAGS -DDEFAULT_DATA_PATH=\"\\\"$DEFAULT_DATA_PATH\\\"\""
diff -Nurd -X.diff_ignore freeciv/server/commands.c freeciv/server/commands.c
--- freeciv/server/commands.c	2012-05-21 12:41:29.000000000 +0300
+++ freeciv/server/commands.c	2012-05-23 00:47:36.000000000 +0300
@@ -436,7 +436,7 @@
       "all cities and units etc. Use with care!"),
    VCF_NONE, 50
   },
-  {"save",	ALLOW_ADMIN,
+  {"save",	ALLOW_CTRL,
    /* TRANS: translate text between <> only */
    N_("save\n"
       "save <file-name>"),
diff -Nurd -X.diff_ignore freeciv/server/meta.h freeciv/server/meta.h
--- freeciv/server/meta.h	2012-05-23 00:39:46.000000000 +0300
+++ freeciv/server/meta.h	2012-05-23 00:47:36.000000000 +0300
@@ -17,7 +17,7 @@
 
 #define DEFAULT_META_SERVER_NO_SEND  TRUE
 #define DEFAULT_META_SERVER_ADDR     META_URL /* config.h */
-#define METASERVER_REFRESH_INTERVAL   (60)
+#define METASERVER_REFRESH_INTERVAL   (20)
 #define METASERVER_MIN_UPDATE_INTERVAL 7   /* not too short, not too long */
 
 enum meta_flag {
diff -Nurd -X.diff_ignore freeciv/server/savegame.c freeciv/server/savegame.c
--- freeciv/server/savegame.c	2012-05-21 12:41:29.000000000 +0300
+++ freeciv/server/savegame.c	2012-05-23 00:47:36.000000000 +0300
@@ -423,8 +423,12 @@
     break;
   }
 
+  /* FIXME: Should not get here! */
+#if 0
   fc_assert(FALSE);
   return '?';
+#endif
+  return 'a';
 }
 
 /****************************************************************************
diff -Nurd -X.diff_ignore freeciv/server/score.c freeciv/server/score.c
--- freeciv/server/score.c	2012-05-21 12:41:29.000000000 +0300
+++ freeciv/server/score.c	2012-05-23 00:47:36.000000000 +0300
@@ -414,13 +414,15 @@
   static const int watercol[3] = {0,0,255}; /* blue */
   static const int landcol[3] =  {0,0,0};   /* black */
 
+#if 0
   if (!srvarg.save_ppm) {
     return;
   }
+#endif
 
   /* put this file in the same place we put savegames */
   fc_snprintf(filename, sizeof(filename),
-              "%s%+05d.int.ppm", game.server.save_name, game.info.year);
+              "overviewmap-%04d.ppm", srvarg.port);
 
   /* Ensure the saves directory exists. */
   make_dir(srvarg.saves_pathname);
diff -Nurd -X.diff_ignore freeciv/server/srv_main.c freeciv/server/srv_main.c
--- freeciv/server/srv_main.c	2012-05-21 14:11:05.000000000 +0300
+++ freeciv/server/srv_main.c	2012-05-23 00:47:36.000000000 +0300
@@ -1011,18 +1011,19 @@
 Note that if !HAVE_LIBZ, then game.server.save_compress_level should never
 become non-zero, so no need to check HAVE_LIBZ explicitly here as well.
 **************************************************************************/
-void save_game(char *orig_filename, const char *save_reason, bool scenario)
+void save_game(char *username, const char *save_reason, bool scenario)
 {
   char filepath[600];
+  char genname[600];
   char *dot, *filename;
   struct section_file *file;
   struct timer *timer_cpu, *timer_user;
 
-  if (!orig_filename) {
-    filepath[0] = '\0';
-    filename = filepath;
+  if (!username) {
+    con_write(C_FAIL, _("Failed saving game. Missing filename."));
+    return;
   } else {
-    sz_strlcpy(filepath, orig_filename);
+    sz_strlcpy(filepath, username);
     if ((filename = strrchr(filepath, '/'))) {
       filename++;
     } else {
@@ -1042,12 +1043,9 @@
     }
   }
 
-  /* If orig_filename is NULL or empty, use a generated default name. */
-  if (filename[0] == '\0'){
-    /* manual save */
-    generate_save_name(game.server.save_name, filename,
-                       sizeof(filepath) + filepath - filename, "manual");
-  }
+  /* manual save */
+  generate_save_name(game.server.save_name, filepath,
+                     sizeof(filepath), "manual");
 
   timer_cpu = new_timer_start(TIMER_CPU, TIMER_ACTIVE);
   timer_user = new_timer_start(TIMER_USER, TIMER_ACTIVE);
@@ -1091,12 +1089,12 @@
 
     if (!scenario) {
       /* Ensure the saves directory exists. */
-      make_dir(srvarg.saves_pathname);
+/*      make_dir(srvarg.saves_pathname); */
 
       sz_strlcpy(tmpname, srvarg.saves_pathname);
     } else {
       /* Make sure scenario directory exist */
-      make_dir(srvarg.scenarios_pathname);
+/*      make_dir(srvarg.scenarios_pathname); */
 
       sz_strlcpy(tmpname, srvarg.scenarios_pathname);
     }
@@ -1104,7 +1102,14 @@
     if (tmpname[0] != '\0') {
       sz_strlcat(tmpname, "/");
     }
+    sz_strlcat(tmpname, username);
+    sz_strlcat(tmpname, "/");
+
+    /* Ensure the saves directory exists. */
+    make_dir(tmpname);
+
     sz_strlcat(tmpname, filepath);
+
     sz_strlcpy(filepath, tmpname);
   }
 
@@ -2070,6 +2075,8 @@
 	save_counter++;
       }
 
+      save_ppm();
+
       log_debug("sniffingpackets");
       check_for_full_turn_done(); /* HACK: don't wait during AI phases */
       while (server_sniff_all_input() == S_E_OTHERWISE) {
diff -Nurd -X.diff_ignore freeciv/server/stdinhand.c freeciv/server/stdinhand.c
--- freeciv/server/stdinhand.c	2012-05-21 12:41:29.000000000 +0300
+++ freeciv/server/stdinhand.c	2012-05-23 00:47:36.000000000 +0300
@@ -677,11 +677,6 @@
 **************************************************************************/
 static bool save_command(struct connection *caller, char *arg, bool check)
 {
-  if (is_restricted(caller)) {
-    cmd_reply(CMD_SAVE, caller, C_FAIL,
-              _("You cannot save games manually on this server."));
-    return FALSE;
-  }
   if (!check) {
     save_game(arg, "User request", FALSE);
   }
@@ -3546,15 +3541,17 @@
   {
     /* it is a normal savegame or maybe a scenario */
     char testfile[MAX_LEN_PATH];
-    const struct strvec *pathes[] = {
-      get_save_dirs(), get_scenario_dirs(), NULL
-    };
+    char usrdir[MAX_LEN_PATH];
+
+    fc_snprintf(usrdir, sizeof(usrdir), "%s/", 
+		caller->username);
+
+    const char *pathes[] = { "", usrdir, "scenario/", NULL };
     const char *exts[] = {
       "sav", "gz", "bz2", "sav.gz", "sav.bz2", NULL
     };
-    const char **ext, *found = NULL;
-    const struct strvec **path;
-
+    const char **path, **ext, *found = NULL;
+ 
     for (path = pathes; !found && *path; path++) {
       for (ext = exts; !found && *ext; ext++) {
         fc_snprintf(testfile, sizeof(testfile), "%s.%s", filename, *ext);
@@ -3868,6 +3865,11 @@
 
   level = command_level(command_by_number(cmd));
 
+  /* hack for web client: savegames are saved with username as proposed filename. */
+  if (cmd == CMD_SAVE && caller) {
+    sz_strlcpy(allargs, caller->username);
+  }
+
   if (conn_can_vote(caller, NULL) && level == ALLOW_CTRL
       && conn_get_access(caller) == ALLOW_BASIC && !check) {
     struct vote *vote;
