Index: freeciv/server/commands.c
===================================================================
--- freeciv/server/commands.c	(revision 22812)
+++ freeciv/server/commands.c	(working copy)
@@ -533,7 +533,7 @@
       "all cities and units etc. Use with care!"), NULL,
    CMD_ECHO_ALL, VCF_NONE, 50
   },
-  {"save",	ALLOW_ADMIN,
+  {"save",	ALLOW_CTRL,
    /* TRANS: translate text between <> only */
    N_("save\n"
       "save <file-name>"),
Index: freeciv/server/srv_main.c
===================================================================
--- freeciv/server/srv_main.c	(revision 22812)
+++ freeciv/server/srv_main.c	(working copy)
@@ -1152,7 +1152,7 @@
 Note that if !HAVE_LIBZ, then game.server.save_compress_level should never
 become non-zero, so no need to check HAVE_LIBZ explicitly here as well.
 **************************************************************************/
-void save_game(const char *orig_filename, const char *save_reason,
+void save_game(const char *username, const char *save_reason,
                bool scenario)
 {
   char filepath[600];
@@ -1160,11 +1160,11 @@
   struct section_file *file;
   struct timer *timer_cpu, *timer_user;
 
-  if (!orig_filename) {
-    filepath[0] = '\0';
-    filename = filepath;
+  if (!username) {
+    con_write(C_FAIL, _("Failed saving game. Missing filename."));
+    return;
   } else {
-    sz_strlcpy(filepath, orig_filename);
+    sz_strlcpy(filepath, username);
     if ((filename = strrchr(filepath, '/'))) {
       filename++;
     } else {
@@ -1197,13 +1197,6 @@
     }
   }
 
-  /* If orig_filename is NULL or empty, use a generated default name. */
-  if (filename[0] == '\0'){
-    /* manual save */
-    generate_save_name(game.server.save_name, filename,
-                       sizeof(filepath) + filepath - filename, "manual");
-  }
-
   timer_cpu = timer_new(TIMER_CPU, TIMER_ACTIVE);
   timer_start(timer_cpu);
   timer_user = timer_new(TIMER_USER, TIMER_ACTIVE);
Index: freeciv/server/stdinhand.c
===================================================================
--- freeciv/server/stdinhand.c	(revision 22812)
+++ freeciv/server/stdinhand.c	(working copy)
@@ -731,11 +731,6 @@
 **************************************************************************/
 static bool save_command(struct connection *caller, char *arg, bool check)
 {
-  if (is_restricted(caller)) {
-    cmd_reply(CMD_SAVE, caller, C_FAIL,
-              _("You cannot save games manually on this server."));
-    return FALSE;
-  }
   if (!check) {
     save_game(arg, "User request", FALSE);
   }
@@ -4184,6 +4179,13 @@
 
   level = command_level(command_by_number(cmd));
 
+  /* hack for web client: savegames are saved with username as proposed filename. */
+  if (cmd == CMD_SAVE && caller) {
+    sz_strlcpy(allargs, caller->username);
+  }
+
+
+
   if (conn_can_vote(caller, NULL) && level == ALLOW_CTRL
       && conn_get_access(caller) == ALLOW_BASIC && !check) {
     struct vote *vote;
Index: freeciv/data/scenarios/tutorial.sav
===================================================================
--- freeciv/data/scenarios/tutorial.sav	(revision 22812)
+++ freeciv/data/scenarios/tutorial.sav	(working copy)
@@ -435,7 +435,6 @@
 
 set = {"name", "value", "gamestart"
 "aifill", 2, 2
-"topology", "ISO|WRAPX", "ISO|WRAPX"
 "mapsize", "FULLSIZE", "FULLSIZE"
 "size", 1, 1
 }
