diff --git a/data/scenarios/british-isles-85x80-v2.80.sav b/data/scenarios/british-isles-85x80-v2.80.sav
index ba23ad4..03687f1 100644
--- a/data/scenarios/british-isles-85x80-v2.80.sav
+++ b/data/scenarios/british-isles-85x80-v2.80.sav
@@ -15,3 +15,3 @@ reason="Scenario"
 revision="2.5.99-alpha"
-rulesetdir="classic"
+rulesetdir="fcweb"
 improvement_size=68
diff --git a/data/scenarios/earth-160x90-v2.sav b/data/scenarios/earth-160x90-v2.sav
index 81cba3d..26d750c 100644
--- a/data/scenarios/earth-160x90-v2.sav
+++ b/data/scenarios/earth-160x90-v2.sav
@@ -15,3 +15,3 @@ reason="Scenario"
 revision="2.5.99-alpha"
-rulesetdir="classic"
+rulesetdir="fcweb"
 improvement_size=68
diff --git a/data/scenarios/earth-80x50-v3.sav b/data/scenarios/earth-80x50-v3.sav
index 63c0909..af04c08 100644
--- a/data/scenarios/earth-80x50-v3.sav
+++ b/data/scenarios/earth-80x50-v3.sav
@@ -15,3 +15,3 @@ reason="Scenario"
 revision="2.5.99-alpha"
-rulesetdir="classic"
+rulesetdir="fcweb"
 improvement_size=68
diff --git a/data/scenarios/europe-200x100-v2.sav b/data/scenarios/europe-200x100-v2.sav
index 324da72..db9b8c0 100644
--- a/data/scenarios/europe-200x100-v2.sav
+++ b/data/scenarios/europe-200x100-v2.sav
@@ -12,3 +12,3 @@ version=20
 reason="Scenario"
-rulesetdir="classic"
+rulesetdir="fcweb"
 improvement_size=68
diff --git a/data/scenarios/france-140x90-v2.sav b/data/scenarios/france-140x90-v2.sav
index 6a288ae..f16ba66 100644
--- a/data/scenarios/france-140x90-v2.sav
+++ b/data/scenarios/france-140x90-v2.sav
@@ -15,3 +15,3 @@ reason="Scenario"
 revision="2.5.99-alpha"
-rulesetdir="classic"
+rulesetdir="fcweb"
 improvement_size=68
diff --git a/data/scenarios/hagworld-120x60-v1.2.sav b/data/scenarios/hagworld-120x60-v1.2.sav
index c602baf..40088b8 100644
--- a/data/scenarios/hagworld-120x60-v1.2.sav
+++ b/data/scenarios/hagworld-120x60-v1.2.sav
@@ -15,3 +15,3 @@ reason="Scenario"
 revision="2.5.99-alpha"
-rulesetdir="classic"
+rulesetdir="fcweb"
 improvement_size=68
diff --git a/data/scenarios/iberian-peninsula-136x100-v1.0.sav b/data/scenarios/iberian-peninsula-136x100-v1.0.sav
index cde96ee..121da94 100644
--- a/data/scenarios/iberian-peninsula-136x100-v1.0.sav
+++ b/data/scenarios/iberian-peninsula-136x100-v1.0.sav
@@ -15,3 +15,3 @@ reason="Scenario"
 revision="2.5.99-alpha"
-rulesetdir="classic"
+rulesetdir="fcweb"
 improvement_size=68
diff --git a/data/scenarios/italy-100x100-v1.5.sav b/data/scenarios/italy-100x100-v1.5.sav
index a2ad586..059ba36 100644
--- a/data/scenarios/italy-100x100-v1.5.sav
+++ b/data/scenarios/italy-100x100-v1.5.sav
@@ -15,3 +15,3 @@ reason="Scenario"
 revision="2.5.99-alpha"
-rulesetdir="classic"
+rulesetdir="fcweb"
 improvement_size=68
diff --git a/data/scenarios/japan-88x100-v1.3.sav b/data/scenarios/japan-88x100-v1.3.sav
index 24602c7..6de5394 100644
--- a/data/scenarios/japan-88x100-v1.3.sav
+++ b/data/scenarios/japan-88x100-v1.3.sav
@@ -15,3 +15,3 @@ reason="Scenario"
 revision="2.5.99-alpha"
-rulesetdir="classic"
+rulesetdir="fcweb"
 improvement_size=68
diff --git a/data/scenarios/north_america_116x100-v1.2.sav b/data/scenarios/north_america_116x100-v1.2.sav
index f937f35..88c3bf5 100644
--- a/data/scenarios/north_america_116x100-v1.2.sav
+++ b/data/scenarios/north_america_116x100-v1.2.sav
@@ -15,3 +15,3 @@ reason="Scenario"
 revision="2.5.99-alpha"
-rulesetdir="classic"
+rulesetdir="fcweb"
 improvement_size=68
diff --git a/data/scenarios/tutorial.sav b/data/scenarios/tutorial.sav
index 09e4614..c13385a 100644
--- a/data/scenarios/tutorial.sav
+++ b/data/scenarios/tutorial.sav
@@ -15,3 +15,3 @@ reason="Scenario"
 revision="2.5.99-alpha"
-rulesetdir="civ2civ3"
+rulesetdir="fcweb"
 improvement_size=68
@@ -313,3 +313,2 @@ set={"name","value","gamestart"
 "size",1,1
-"topology","WRAPX|ISO","WRAPX|ISO"
 "victories","",""
