diff -Nurd freeciv/common/game.h freeciv/common/game.h
--- freeciv/common/game.h	2018-03-05 17:21:24.098920332 +0200
+++ freeciv/common/game.h	2018-03-05 17:21:46.850809317 +0200
@@ -214,7 +214,7 @@
       char save_name[MAX_LEN_NAME];
       bool scorelog;
       enum scorelog_level scoreloglevel;
-      char scorefile[MAX_LEN_NAME];
+      char scorefile[100];
       int scoreturn;    /* next make_history_report() */
       int seed_setting;
       int seed;
diff -Nurd freeciv/server/report.c freeciv/server/report.c
--- freeciv/server/report.c	2018-03-05 17:21:25.186915047 +0200
+++ freeciv/server/report.c	2018-03-05 17:26:45.817270125 +0200
@@ -581,6 +581,7 @@
   return result;
 }
 
+#if 0
 /**********************************************************************//**
   Number of city building units.
 **************************************************************************/
@@ -599,6 +600,7 @@
 
   return result;
 }
+#endif
 
 /**********************************************************************//**
   Wonder score
@@ -616,6 +618,7 @@
   return pplayer->score.techout;
 }
 
+#if 0
 /**********************************************************************//**
   Literacy score calculated one way. See also get_literacy() to see
   alternative way.
@@ -624,6 +627,7 @@
 {
   return pplayer->score.literacy;
 }
+#endif
 
 /**********************************************************************//**
   Spaceship score
@@ -665,6 +669,7 @@
   return pplayer->economic.gold;
 }
 
+#if 0
 /**********************************************************************//**
   Tax rate
 **************************************************************************/
@@ -764,6 +769,7 @@
 
   return result;
 }
+#endif
 
 /**********************************************************************//**
   Total score
@@ -1270,6 +1276,10 @@
     return;
   }
 
+  fc_snprintf(game.server.scorefile, 100, 
+              "/var/lib/tomcat9/webapps/data/scorelogs/score-%d.log",
+              srvarg.port);
+
   score_log = fc_calloc(1, sizeof(*score_log));
   score_log->fp = NULL;
   score_log->last_turn = -1;
@@ -1329,41 +1339,25 @@
     char *name;
     int (*get_value) (const struct player *);
   } score_tags[] = {
+    {"score",           get_total_score}, /* New 2.1.10 tag end here. */
     {"pop",             get_pop},
     {"bnp",             get_economics},
     {"mfg",             get_production},
     {"cities",          get_cities},
     {"techs",           get_techs},
     {"munits",          get_munits},
-    {"settlers",        get_settlers},  /* "original" tags end here */
 
     {"wonders",         get_wonders},
     {"techout",         get_techout},
     {"landarea",        get_landarea},
     {"settledarea",     get_settledarea},
-    {"pollution",       get_pollution},
-    {"literacy",        get_literacy2},
-    {"spaceship",       get_spaceship}, /* new 1.8.2 tags end here */
 
     {"gold",            get_gold},
-    {"taxrate",         get_taxrate},
-    {"scirate",         get_scirate},
-    {"luxrate",         get_luxrate},
-    {"riots",           get_riots},
-    {"happypop",        get_happypop},
-    {"contentpop",      get_contentpop},
-    {"unhappypop",      get_unhappypop},
-    {"specialists",     get_specialists},
-    {"gov",             get_gov},
-    {"corruption",      get_corruption}, /* new 1.11.5 tags end here */
-
-    {"score",           get_total_score}, /* New 2.1.10 tag end here. */
 
     {"unitsbuilt",      get_units_built}, /* New tags since 2.3.0. */
     {"unitskilled",     get_units_killed},
     {"unitslost",       get_units_lost},
 
-    {"culture",         get_culture}      /* New tag in 2.6.0. */
   };
 
   if (!game.server.scorelog) {
