diff -Nurd freeciv/server/report.c freeciv/server/report.c
--- freeciv/server/report.c	2023-11-05 23:18:20.627736421 +0200
+++ freeciv/server/report.c	2023-11-05 23:18:33.095790052 +0200
@@ -709,6 +709,7 @@
   return result;
 }
 
+#if 0
 /**********************************************************************//**
   Number of city building units.
 **************************************************************************/
@@ -727,6 +728,7 @@
 
   return result;
 }
+#endif
 
 /**********************************************************************//**
   Wonder score
@@ -744,6 +746,7 @@
   return pplayer->score.techout;
 }
 
+#if 0
 /**********************************************************************//**
   Literacy score calculated one way. See also get_literacy() to see
   alternative way.
@@ -752,6 +755,7 @@
 {
   return pplayer->score.literacy;
 }
+#endif
 
 /**********************************************************************//**
   Spaceship score
@@ -801,6 +805,7 @@
   return pplayer->economic.gold;
 }
 
+#if 0
 /**********************************************************************//**
   Tax rate
 **************************************************************************/
@@ -900,6 +905,7 @@
 
   return result;
 }
+#endif
 
 /**********************************************************************//**
   Total score
@@ -1415,6 +1421,10 @@
     return;
   }
 
+  fc_snprintf(game.server.scorefile, 100, 
+              "/var/lib/tomcat10/webapps/data/scorelogs/score-%d.log",
+              srvarg.port);
+
   score_log = fc_calloc(1, sizeof(*score_log));
   score_log->fp = nullptr;
   score_log->last_turn = -1;
@@ -1474,42 +1484,26 @@
     char *name;
     int (*get_value) (const struct player *);
   } score_tags[] = {
+    {"score",           get_total_score}, /* 2.1.10 tag actually */
+
     {"pop",             get_pop},
     {"bnp",             get_economics},
     {"mfg",             get_production},
     {"cities",          get_cities},
     {"techs",           get_techs},
     {"munits",          get_munits},
-    {"settlers",        get_settlers},  /* "original" tags end here */
 
     {"wonders",         get_wonders},
     {"techout",         get_techout},
     {"landarea",        get_landarea},
-    {"settledarea",     get_settledarea},
-    {"pollution",       get_pollution},
-    {"literacy",        get_literacy2},
-    {"spaceship",       get_spaceship}, /* new 1.8.2 tags end here */
-
-    {"gold",            get_gold},
-    {"taxrate",         get_taxrate},
-    {"scirate",         get_scirate},
-    {"luxrate",         get_luxrate},
-    {"riots",           get_riots},
-    {"happypop",        get_happypop},
-    {"contentpop",      get_contentpop},
-    {"unhappypop",      get_unhappypop},
-    {"specialists",     get_specialists},
-    {"gov",             get_gov},
-    {"corruption",      get_corruption}, /* new 1.11.5 tags end here */
+    {"settledarea",     get_settledarea}, /* new 1.8.2 tags end here */
 
-    {"score",           get_total_score}, /* New 2.1.10 tag end here. */
+    {"gold",            get_gold},        /* new 1.11.5 tags end here */
 
     {"unitsbuilt",      get_units_built}, /* New tags since 2.3.0. */
     {"unitskilled",     get_units_killed},
     {"unitslost",       get_units_lost},
 
-    {"culture",         get_culture},     /* New tag in 2.6.0. */
-
     {"unitsused",       get_units_used}   /* New tag in 3.2.0. */
   };
 
