diff -Nurd -X.diff_ignore freeciv/common/game.h freeciv/common/game.h
--- freeciv/common/game.h	2016-05-02 13:03:18.357230715 +0300
+++ freeciv/common/game.h	2016-05-02 13:03:40.033167249 +0300
@@ -209,7 +209,7 @@
       char save_name[MAX_LEN_NAME];
       bool scorelog;
       enum scorelog_level scoreloglevel;
-      char scorefile[MAX_LEN_NAME];
+      char scorefile[100];
       int scoreturn;    /* next make_history_report() */
       int seed;
 
diff -Nurd -X.diff_ignore freeciv/server/report.c freeciv/server/report.c
--- freeciv/server/report.c	2016-05-02 13:03:18.413230550 +0300
+++ freeciv/server/report.c	2016-05-02 13:04:38.892996083 +0300
@@ -599,6 +599,7 @@
   return result;
 }
 
+#if 0
 /****************************************************************************
   Number of city building units.
 ****************************************************************************/
@@ -617,6 +618,7 @@
 
   return result;
 }
+#endif
 
 /****************************************************************************
   Wonder scpre
@@ -634,6 +636,7 @@
   return pplayer->score.techout;
 }
 
+#if 0
 /****************************************************************************
   Literacy score calculated one way. See also get_literacy() to see
   alternative way.
@@ -642,6 +645,7 @@
 {
   return pplayer->score.literacy;
 }
+#endif
 
 /****************************************************************************
   Spaceship score
@@ -683,6 +687,7 @@
   return pplayer->economic.gold;
 }
 
+#if 0
 /****************************************************************************
   Tax rate
 ****************************************************************************/
@@ -782,6 +787,7 @@
 
   return result;
 }
+#endif
 
 /****************************************************************************
   Total score
@@ -1288,6 +1294,10 @@
     return;
   }
 
+  fc_snprintf(game.server.scorefile, 100, 
+              "/var/lib/tomcat8/webapps/data/scorelogs/score-%d.log",
+              srvarg.port);
+
   score_log = fc_calloc(1, sizeof(*score_log));
   score_log->fp = NULL;
   score_log->last_turn = -1;
@@ -1347,41 +1357,25 @@
     char *name;
     int (*get_value) (const struct player *);
   } score_tags[] = {
+    {"score",           get_total_score}, /* New 2.1.10 tag end here. */
     {"pop",             get_pop},
     {"bnp",             get_economics},
     {"mfg",             get_production},
     {"cities",          get_cities},
     {"techs",           get_techs},
     {"munits",          get_munits},
-    {"settlers",        get_settlers},  /* "original" tags end here */
 
     {"wonders",         get_wonders},
     {"techout",         get_techout},
     {"landarea",        get_landarea},
     {"settledarea",     get_settledarea},
-    {"pollution",       get_pollution},
-    {"literacy",        get_literacy2},
-    {"spaceship",       get_spaceship}, /* new 1.8.2 tags end here */
 
     {"gold",            get_gold},
-    {"taxrate",         get_taxrate},
-    {"scirate",         get_scirate},
-    {"luxrate",         get_luxrate},
-    {"riots",           get_riots},
-    {"happypop",        get_happypop},
-    {"contentpop",      get_contentpop},
-    {"unhappypop",      get_unhappypop},
-    {"specialists",     get_specialists},
-    {"gov",             get_gov},
-    {"corruption",      get_corruption}, /* new 1.11.5 tags end here */
-
-    {"score",           get_total_score}, /* New 2.1.10 tag end here. */
 
     {"unitsbuilt",      get_units_built}, /* New tags since 2.3.0. */
     {"unitskilled",     get_units_killed},
     {"unitslost",       get_units_lost},
 
-    {"culture",         get_culture}      /* New tag in 2.6.0. */
   };
 
   if (!game.server.scorelog) {
