diff -Nurd -X.diff_ignore freeciv/server/settings.c freeciv/server/settings.c
--- freeciv/server/settings.c	2016-02-23 07:21:02.858745169 +0200
+++ freeciv/server/settings.c	2016-05-02 13:52:21.925148433 +0300
@@ -817,6 +817,7 @@
 /****************************************************************************
   Verify the name for the score log file.
 ****************************************************************************/
+#ifndef FREECIV_WEB
 static bool scorefile_validate(const char *value, struct connection *caller,
                                char *reject_msg, size_t reject_msg_len)
 {
@@ -828,6 +829,7 @@
 
   return TRUE;
 }
+#endif /* !FREECIV_WEB */
 
 /*************************************************************************
   Verify that a given demography string is valid. See
@@ -2805,6 +2807,7 @@
               "or only from human players."), NULL, NULL, NULL,
            scoreloglevel_name, GAME_DEFAULT_SCORELOGLEVEL)
 
+#ifndef FREECIV_WEB
   GEN_STRING("scorefile", game.server.scorefile,
              SSET_META, SSET_INTERNAL, SSET_SITUATIONAL,
              ALLOW_HACK, ALLOW_HACK,
@@ -2813,6 +2816,7 @@
              N_("The default name for the score log file is "
               "'freeciv-score.log'."),
              scorefile_validate, NULL, GAME_DEFAULT_SCOREFILE)
+#endif /* !FREECIV_WEB */
 
   GEN_INT("maxconnectionsperhost", game.server.maxconnectionsperhost,
           SSET_RULES_FLEXIBLE, SSET_NETWORK, SSET_RARE,
