diff -Nurd freeciv/server/auth.c freeciv/server/auth.c
--- freeciv/server/auth.c	2023-09-03 05:29:42.695378863 +0300
+++ freeciv/server/auth.c	2023-09-03 05:30:58.580116657 +0300
@@ -74,6 +74,18 @@
 {
   char tmpname[MAX_LEN_NAME] = "\0";
 
+#ifdef FREECIV_WEB
+  if (srvarg.server_password_enabled) {
+    char buffer[MAX_LEN_MSG];
+
+    fc_snprintf(buffer, sizeof(buffer), _("Enter password for this private game:"));
+    dsend_packet_authentication_req(pconn, AUTH_LOGIN_FIRST, buffer);
+    pconn->server.auth_settime = time(nullptr);
+
+    return TRUE;
+  }
+#endif /* FREECIV_WEB */
+
   /* Assign the client a unique guest name/Reject if guests aren't allowed */
   if (is_guest_name(username)) {
     if (srvarg.auth_allow_guests) {
@@ -157,6 +169,32 @@
 {
   char msg[MAX_LEN_MSG];
 
+#ifdef FREECIV_WEB
+  if (srvarg.server_password_enabled) {
+    if (pconn->server.status == AS_ESTABLISHED) {
+      notify_conn(nullptr, nullptr, E_SETTING, ftc_server,
+                  _("Server password already set."));
+    } else if (!fc_strncmp(srvarg.server_password, password, MAX_LEN_PASSWORD)) {
+      establish_new_connection(pconn);
+    } else {
+      char buffer[MAX_LEN_MSG];
+
+      fc_snprintf(buffer, sizeof(buffer),
+                  _("Incorrect password entered. Please try again."));
+      dsend_packet_authentication_req(pconn, AUTH_LOGIN_FIRST, buffer);
+      pconn->server.auth_settime = time(nullptr);
+    }
+
+    return TRUE;
+  } else {
+    srvarg.server_password_enabled = TRUE;
+    sz_strlcpy(srvarg.server_password, password);
+    notify_conn(nullptr, nullptr, E_SETTING, ftc_server, _("Server password set."));
+
+    return TRUE;
+  }
+#endif /* FREECIV_WEB */
+
   if (pconn->server.status == AS_REQUESTING_NEW_PASS) {
 
     /* check if the new password is acceptable */
diff -Nurd freeciv/server/connecthand.c freeciv/server/connecthand.c
--- freeciv/server/connecthand.c	2023-09-03 05:29:42.687378786 +0300
+++ freeciv/server/connecthand.c	2023-09-03 05:29:56.087509001 +0300
@@ -510,7 +510,7 @@
     dsend_packet_connect_msg(pconn, game.server.connectmsg);
   }
 
-  if (srvarg.auth_enabled) {
+  if (srvarg.auth_enabled || srvarg.server_password_enabled) {
     return auth_user(pconn, req->username);
   } else {
     sz_strlcpy(pconn->username, req->username);
diff -Nurd freeciv/server/srv_main.c freeciv/server/srv_main.c
--- freeciv/server/srv_main.c	2023-09-03 05:29:42.687378786 +0300
+++ freeciv/server/srv_main.c	2023-09-03 05:29:56.087509001 +0300
@@ -284,6 +284,7 @@
   srvarg.auth_enabled = FALSE;
   srvarg.auth_allow_guests = FALSE;
   srvarg.auth_allow_newusers = FALSE;
+  srvarg.server_password_enabled = FALSE;
 
   /* Mark as initialized */
   has_been_srv_init = TRUE;
diff -Nurd freeciv/server/srv_main.h freeciv/server/srv_main.h
--- freeciv/server/srv_main.h	2023-09-03 05:29:42.695378863 +0300
+++ freeciv/server/srv_main.h	2023-09-03 05:29:56.087509001 +0300
@@ -57,6 +57,8 @@
   bool auth_enabled;            /* defaults to FALSE */
   bool auth_allow_guests;       /* defaults to FALSE */
   bool auth_allow_newusers;     /* defaults to FALSE */
+  bool server_password_enabled; /* Defaults to FALSE */
+  char server_password[MAX_LEN_PASSWORD]; /* Server password */
   enum announce_type announce;
   int fatal_assertions;         /* default to -1 (disabled). */
 };
