diff -Nurd -X.diff_ignore freeciv/server/unithand.c freeciv/server/unithand.c
--- freeciv/server/unithand.c	2014-09-13 04:35:03.723942718 +0300
+++ freeciv/server/unithand.c	2014-09-13 04:35:34.923870105 +0300
@@ -1214,6 +1214,59 @@
     activity_target = extra_by_number(target_id);
   }
 
+  if (activity_target == NULL) {
+    struct unit *punit = player_unit_by_number(pplayer, unit_id);
+    bool required = TRUE;
+
+    if (punit == NULL) {
+      return;
+    }
+
+    if (activity == ACTIVITY_IRRIGATE) {
+      struct tile *ptile = unit_tile(punit);
+      struct terrain *pterrain = tile_terrain(ptile);
+
+      if (pterrain->irrigation_result != pterrain) {
+        required = FALSE;
+      } else {
+        activity_target = next_extra_for_tile(ptile, EC_IRRIGATION,
+                                              pplayer, punit);
+      }
+    } else if (activity == ACTIVITY_MINE) {
+      struct tile *ptile = unit_tile(punit);
+      struct terrain *pterrain = tile_terrain(ptile);
+
+      if (pterrain->mining_result != pterrain) {
+        required = FALSE;
+      } else {
+        activity_target = next_extra_for_tile(ptile, EC_MINE,
+                                              pplayer, punit);
+      }
+    } else if (activity == ACTIVITY_BASE) {
+      struct tile *ptile = unit_tile(punit);
+      struct base_type *pbase =
+        get_base_by_gui_type(BASE_GUI_FORTRESS, punit, ptile);
+
+      if (pbase != NULL) {
+        activity_target = base_extra_get(pbase);
+      }
+
+    } else if (activity == ACTIVITY_POLLUTION) {
+      activity_target = prev_extra_in_tile(unit_tile(punit), ERM_CLEANPOLLUTION,
+                                           pplayer, punit);
+    } else if (activity == ACTIVITY_FALLOUT) {
+      activity_target = prev_extra_in_tile(unit_tile(punit), ERM_CLEANFALLOUT,
+                                           pplayer, punit);
+    } else {
+      required = FALSE;
+    }
+
+    if (activity_target == NULL && required) {
+      /* Nothing more we can do */
+      return;
+    }
+  }
+
   handle_unit_change_activity_real(pplayer, unit_id, activity, activity_target);
 }
 
