diff -Nurd -X.diff_ignore freeciv/server/spacerace.c freeciv/server/spacerace.c
--- freeciv/server/spacerace.c	2014-09-13 01:05:38.640191068 +0300
+++ freeciv/server/spacerace.c	2014-09-13 01:06:15.752106442 +0300
@@ -136,7 +136,11 @@
     if (!src || pplayer == src) {
       struct packet_spaceship_info info;
       struct player_spaceship *ship = &pplayer->spaceship;
-	  
+
+      if (pplayer->spaceship.state == SSHIP_STARTED) {
+        adv_spaceship_autoplace(pplayer, ship);
+      }
+
       info.player_num = player_number(pplayer);
       info.sship_state = ship->state;
       info.structurals = ship->structurals;
