diff -Nurd -X.diff_ignore freeciv/data/amplio2/ancientcities.spec freeciv/data/amplio2/ancientcities.spec
--- freeciv/data/amplio2/ancientcities.spec	2012-08-30 14:52:34.000000000 +0300
+++ freeciv/data/amplio2/ancientcities.spec	2012-08-30 14:58:50.000000000 +0300
@@ -25,8 +25,7 @@
     Polynesian style by CapTVK
     Celtic style by Erwan, adapted to 96x48 by CapTVK
     Roman style by CapTVK
-    City walls by Hogne Håskjold
-"
+    City walls by Hogne Håskjold"
 
 [file]
 gfx = "amplio2/ancientcities"
@@ -110,6 +109,4 @@
  4,  6, "city.babylonian_wall_1"
  4,  7, "city.babylonian_wall_2"
  4,  8, "city.babylonian_wall_3"
- 4,  9, "city.babylonian_wall_4"
-
-}
+ 4,  9, "city.babylonian_wall_4"}
diff -Nurd -X.diff_ignore freeciv/data/amplio2/explosions.spec freeciv/data/amplio2/explosions.spec
--- freeciv/data/amplio2/explosions.spec	2012-08-30 14:52:34.000000000 +0300
+++ freeciv/data/amplio2/explosions.spec	2012-08-30 14:58:50.000000000 +0300
@@ -10,8 +10,7 @@
     Explosions from xgalaga <http://rumsey.org/xgal.html>
     Credits:
     Joe Rumsey <ogre@atomic.com>
-    Mike McGrath <mcgrath@epx.cis.umn.edu>
-"
+    Mike McGrath <mcgrath@epx.cis.umn.edu>"
 
 [file]
 gfx = "amplio2/explosions"
@@ -28,5 +27,4 @@
   1,  0, "explode.unit_1"
   2,  0, "explode.unit_2"
   3,  0, "explode.unit_3"
-  4,  0, "explode.unit_4"
-}
+  4,  0, "explode.unit_4"}
diff -Nurd -X.diff_ignore freeciv/data/amplio2/fog.spec freeciv/data/amplio2/fog.spec
--- freeciv/data/amplio2/fog.spec	2012-08-30 14:52:34.000000000 +0300
+++ freeciv/data/amplio2/fog.spec	2012-08-30 14:58:50.000000000 +0300
@@ -3,8 +3,7 @@
 options = "+Freeciv-2.3-spec"
 
 artists = "
-    Brendon Oram (Yautja)
-"
+    Brendon Oram (Yautja)"
 
 [file]
 gfx = "amplio2/fog"
@@ -98,5 +97,4 @@
   8,  5, "t.fog_u_u_k_u"
   8,  6, "t.fog_u_u_u_f"
   8,  7, "t.fog_u_u_u_k"
-  8,  8, "t.fog_u_u_u_u"
-}
+  8,  8, "t.fog_u_u_u_u"}
diff -Nurd -X.diff_ignore freeciv/data/amplio2/grid.spec freeciv/data/amplio2/grid.spec
--- freeciv/data/amplio2/grid.spec	2012-08-30 14:52:34.000000000 +0300
+++ freeciv/data/amplio2/grid.spec	2012-08-30 14:58:50.000000000 +0300
@@ -9,8 +9,7 @@
 artists = "
     Jason Dorje Short <jdorje@freeciv.org>
     Hogne Håskjold    <hogne@freeciv.org>
-    Eleazar / J. W. Bjerk <www.jwbjerk.com>
-"
+    Eleazar / J. W. Bjerk <www.jwbjerk.com>"
 
 [file]
 gfx = "amplio2/grid"
@@ -39,5 +38,4 @@
   2, 0, "grid.borders.n"
   2, 1, "grid.borders.s"
   2, 2, "grid.borders.w"
-  2, 3, "grid.borders.e"
-}
+  2, 3, "grid.borders.e"}
diff -Nurd -X.diff_ignore freeciv/data/amplio2/hills.spec freeciv/data/amplio2/hills.spec
--- freeciv/data/amplio2/hills.spec	2012-08-30 14:52:34.000000000 +0300
+++ freeciv/data/amplio2/hills.spec	2012-08-30 14:58:50.000000000 +0300
@@ -4,8 +4,7 @@
 
 [info]
 artists = "
-	 Peter Arbor <peter.arbor@gmail.com>
-"
+	 Peter Arbor <peter.arbor@gmail.com>"
 
 [file]
 gfx = "amplio2/hills"
@@ -37,5 +36,4 @@
  3,  0, "t.l1.hills_n0e1s1w0"
  3,  1, "t.l1.hills_n0e1s1w1"
  3,  2, "t.l1.hills_n1e1s1w0"
- 3,  3, "t.l1.hills_n1e1s1w1"
-}
+ 3,  3, "t.l1.hills_n1e1s1w1" }
diff -Nurd -X.diff_ignore freeciv/data/amplio2/medievalcities.spec freeciv/data/amplio2/medievalcities.spec
--- freeciv/data/amplio2/medievalcities.spec	2012-08-30 14:52:34.000000000 +0300
+++ freeciv/data/amplio2/medievalcities.spec	2012-08-30 14:58:50.000000000 +0300
@@ -23,8 +23,7 @@
 artists = "
     Medieval by CapTVK
     Steam Age by Smiley, www.firstcultural.com
-    City walls by Hogne Håskjold
-"
+    City walls by Hogne Håskjold"
 
 [file]
 gfx = "amplio2/medievalcities"
@@ -68,6 +67,4 @@
  1,  6, "city.industrial_wall_1"
  1,  7, "city.industrial_wall_2"
  1,  8, "city.industrial_wall_3"
- 1,  9, "city.industrial_wall_4" 
-
-}
+ 1,  9, "city.industrial_wall_4"}
diff -Nurd -X.diff_ignore freeciv/data/amplio2/moderncities.spec freeciv/data/amplio2/moderncities.spec
--- freeciv/data/amplio2/moderncities.spec	2012-08-30 14:52:34.000000000 +0300
+++ freeciv/data/amplio2/moderncities.spec	2012-08-30 14:58:50.000000000 +0300
@@ -25,8 +25,7 @@
     Modern, Post Modern and Electric Age by Smiley, www.firstcultural.com
     City walls by Hogne Håskjold
     Buoy by Eleazar
-    Ruins by Vincent Croisier
-"
+    Ruins by Vincent Croisier"
 
 [file]
 gfx = "amplio2/moderncities"
@@ -96,6 +95,4 @@
  4,  6, "city.postmodern_wall_1"
  4,  7, "city.postmodern_wall_2"
  4,  8, "city.postmodern_wall_3"
- 4,  9, "city.postmodern_wall_4" 
-
-}
+ 4,  9, "city.postmodern_wall_4" }
diff -Nurd -X.diff_ignore freeciv/data/amplio2/mountains.spec freeciv/data/amplio2/mountains.spec
--- freeciv/data/amplio2/mountains.spec	2012-08-30 14:52:34.000000000 +0300
+++ freeciv/data/amplio2/mountains.spec	2012-08-30 14:58:50.000000000 +0300
@@ -4,8 +4,7 @@
 
 artists = "
 	 Peter Arbor <peter.arbor@gmail.com>
-         Hogne Håskjold <haskjold@gmail.com>
-"
+         Hogne Håskjold <haskjold@gmail.com>"
 
 [file]
 gfx = "amplio2/mountains"
@@ -37,5 +36,4 @@
  3,  0, "t.l1.mountains_n0e1s1w0"
  3,  1, "t.l1.mountains_n0e1s1w1"
  3,  2, "t.l1.mountains_n1e1s1w0"
- 3,  3, "t.l1.mountains_n1e1s1w1"
-}
+ 3,  3, "t.l1.mountains_n1e1s1w1" }
diff -Nurd -X.diff_ignore freeciv/data/amplio2/ocean.spec freeciv/data/amplio2/ocean.spec
--- freeciv/data/amplio2/ocean.spec	2012-08-30 14:52:34.000000000 +0300
+++ freeciv/data/amplio2/ocean.spec	2012-08-30 14:58:50.000000000 +0300
@@ -10,8 +10,7 @@
     Yautja
     CapTVK
     Eleazar
-    William Allen Simpson
-"
+    William Allen Simpson"
 
 [file]
 gfx = "amplio2/ocean"
@@ -107,5 +106,4 @@
   0, 3, "t.l0.cellgroup_l_l_l_s"
   0, 2, "t.l0.cellgroup_d_l_l_l"
   0, 1, "t.l0.cellgroup_s_l_l_l"
-  0, 0, "t.l0.cellgroup_l_l_l_l"
-}
+  0, 0, "t.l0.cellgroup_l_l_l_l" }
diff -Nurd -X.diff_ignore freeciv/data/amplio2/select.spec freeciv/data/amplio2/select.spec
--- freeciv/data/amplio2/select.spec	2012-08-30 14:52:34.000000000 +0300
+++ freeciv/data/amplio2/select.spec	2012-08-30 14:58:50.000000000 +0300
@@ -8,8 +8,7 @@
 artists = "
     Jason Dorje Short <jdorje@freeciv.org>
     Eleazar / J. W. Bjerk <www.jwbjerk.com>
-    qwm
-"
+    qwm"
 
 [file]
 gfx = "amplio2/select-alpha"
@@ -26,5 +25,4 @@
   0, 0, "unit.select0"
   0, 1, "unit.select1"
   0, 2, "unit.select2"
-  0, 3, "unit.select3"
-}
+  0, 3, "unit.select3" }
diff -Nurd -X.diff_ignore freeciv/data/amplio2/terrain1.spec freeciv/data/amplio2/terrain1.spec
--- freeciv/data/amplio2/terrain1.spec	2012-08-30 14:52:34.000000000 +0300
+++ freeciv/data/amplio2/terrain1.spec	2012-08-30 14:58:50.000000000 +0300
@@ -10,8 +10,7 @@
     Hogne Håskjold
     Daniel Speyer
     Yautja
-    CapTVK
-"
+    CapTVK"
 
 [file]
 gfx = "amplio2/terrain1"
@@ -128,5 +127,4 @@
  15,  3, "t.blend.coast"
  15,  3, "t.blend.lake"
  15,  4, "user.attention"
- 15,  5, "tx.fog"
-}
+ 15,  5, "tx.fog" }
diff -Nurd -X.diff_ignore freeciv/data/amplio2/terrain2.spec freeciv/data/amplio2/terrain2.spec
--- freeciv/data/amplio2/terrain2.spec	2012-08-30 14:52:34.000000000 +0300
+++ freeciv/data/amplio2/terrain2.spec	2012-08-30 14:58:50.000000000 +0300
@@ -9,8 +9,7 @@
     Tim F. Smith <yoohootim@hotmail.com>
     Yautja
     Daniel Speyer
-    Eleazar
-"
+    Eleazar"
 
 [file]
 gfx = "amplio2/terrain2"
@@ -62,9 +61,7 @@
  5,  4, "t.l1.forest_n0e0s1w1"
  5,  5, "t.l1.forest_n1e0s1w1"
  5,  6, "t.l1.forest_n0e1s1w1"
- 5,  7, "t.l1.forest_n1e1s1w1"
-
-}
+ 5,  7, "t.l1.forest_n1e1s1w1" }
 
 
 [grid_coasts]
@@ -76,7 +73,4 @@
 pixel_border = 1
 
 tiles = { "row", "column","tag"
-
-;* previous coordinates now in water.spec
-
-}
+;* previous coordinates now in water.spec }
diff -Nurd -X.diff_ignore freeciv/data/amplio2/tiles.spec freeciv/data/amplio2/tiles.spec
--- freeciv/data/amplio2/tiles.spec	2012-08-30 14:52:34.000000000 +0300
+++ freeciv/data/amplio2/tiles.spec	2012-08-30 14:58:50.000000000 +0300
@@ -8,8 +8,7 @@
 
 artists = "
     Tatu Rissanen <tatu.rissanen@hut.fi>
-    Jeff Mallatt <jjm@codewell.com> (miscellaneous)
-"
+    Jeff Mallatt <jjm@codewell.com> (miscellaneous)"
 
 [file]
 gfx = "amplio2/tiles"
@@ -150,6 +149,4 @@
   1,  2, "explode.nuke_12"
   2,  0, "explode.nuke_20"
   2,  1, "explode.nuke_21"
-  2,  2, "explode.nuke_22"
-
-}
+  2,  2, "explode.nuke_22" }
diff -Nurd -X.diff_ignore freeciv/data/amplio2/units.spec freeciv/data/amplio2/units.spec
--- freeciv/data/amplio2/units.spec	2012-08-30 14:52:34.000000000 +0300
+++ freeciv/data/amplio2/units.spec	2012-08-30 14:59:27.000000000 +0300
@@ -32,8 +32,7 @@
     Vodvakov
     J. W. Bjerk / Eleazar <www.jwbjerk.com>
     qwm
-    FiftyNine
-"
+    FiftyNine"
 
 [file]
 gfx = "amplio2/units"
@@ -103,6 +102,4 @@
   2, 15, "u.leader"		; [GB]
   2, 16, "u.barbarian_leader"	; FiftyNine
   2, 17, "u.refugee"		; Eleazar
-;  3, 15, "u.train"		; Eleazar
-
-}
+;  3, 15, "u.train"		; Eleazar }
diff -Nurd -X.diff_ignore freeciv/data/amplio2/veterancy.spec freeciv/data/amplio2/veterancy.spec
--- freeciv/data/amplio2/veterancy.spec	2012-08-30 14:52:34.000000000 +0300
+++ freeciv/data/amplio2/veterancy.spec	2012-08-30 14:58:50.000000000 +0300
@@ -32,8 +32,7 @@
     Vodvakov
     J. W. Bjerk / Eleazar <www.jwbjerk.com>
     qwm
-    FiftyNine
-"
+    FiftyNine"
 
 [file]
 gfx = "amplio2/veterancy"
@@ -60,5 +59,4 @@
   0, 8, "unit.vet_9"
 
   0, 9, "unit.lowfuel"
-  0, 9, "unit.tired"
-}
+  0, 9, "unit.tired" }
diff -Nurd -X.diff_ignore freeciv/data/amplio2/water.spec freeciv/data/amplio2/water.spec
--- freeciv/data/amplio2/water.spec	2012-08-30 14:52:34.000000000 +0300
+++ freeciv/data/amplio2/water.spec	2012-08-30 14:58:50.000000000 +0300
@@ -11,8 +11,7 @@
     Tim F. Smith <yoohootim@hotmail.com>
     Yautja
     Daniel Speyer
-    Eleazar
-"
+    Eleazar"
 
 [file]
 gfx = "amplio2/water"
@@ -71,8 +70,7 @@
  4,  0, "tx.river_outlet_n"
  4,  1, "tx.river_outlet_e"
  4,  2, "tx.river_outlet_s"
- 4,  3, "tx.river_outlet_w"
-}
+ 4,  3, "tx.river_outlet_w" }
 
 
 [grid_coasts]
@@ -193,5 +191,4 @@
  11, 9,  "t.l0.lake_cell_r_s_s_l"
  11, 11, "t.l0.lake_cell_r_l_s_l"
  11, 13, "t.l0.lake_cell_r_s_l_l"
- 11, 15, "t.l0.lake_cell_r_l_l_l"
-}
+ 11, 15, "t.l0.lake_cell_r_l_l_l" }
diff -Nurd -X.diff_ignore freeciv/data/amplio2.tilespec freeciv/data/amplio2.tilespec
--- freeciv/data/amplio2.tilespec	2012-08-30 14:52:49.000000000 +0300
+++ freeciv/data/amplio2.tilespec	2012-08-30 14:58:50.000000000 +0300
@@ -94,7 +94,7 @@
 
 
 ; Include color definitions
-*include "misc/colors.tilespec"
+;*include "misc/colors.tilespec"
 
 ; Terrain info - see README.graphics
 
diff -Nurd -X.diff_ignore freeciv/data/amplio.tilespec freeciv/data/amplio.tilespec
--- freeciv/data/amplio.tilespec	2012-08-30 14:52:49.000000000 +0300
+++ freeciv/data/amplio.tilespec	2012-08-30 14:58:50.000000000 +0300
@@ -91,7 +91,7 @@
 
 
 ; Include color definitions
-*include "misc/colors.tilespec"
+;*include "misc/colors.tilespec"
 
 ; Terrain info - see README.graphics
 
diff -Nurd -X.diff_ignore freeciv/data/misc/buildings-large.spec freeciv/data/misc/buildings-large.spec
--- freeciv/data/misc/buildings-large.spec	2012-08-30 14:52:35.000000000 +0300
+++ freeciv/data/misc/buildings-large.spec	2012-08-30 14:58:50.000000000 +0300
@@ -7,8 +7,8 @@
 [info]
 
 artists = "
-   feler404
-"
+   feler404"
+
 [extra]
 sprites =
 	{	"tag", "file"
diff -Nurd -X.diff_ignore freeciv/data/misc/buildings.spec freeciv/data/misc/buildings.spec
--- freeciv/data/misc/buildings.spec	2012-08-30 14:52:35.000000000 +0300
+++ freeciv/data/misc/buildings.spec	2012-08-30 14:58:50.000000000 +0300
@@ -9,8 +9,7 @@
     Tim F. Smith <yoohootim@hotmail.com>
     Frederic Rodrigo <f.rodrigo@tuxfamily.org>
     Dusan Hukuv <dusan@mirc.cz>
-    Eleazar / J. W. Bjerk <www.jwbjerk.com>
-"
+    Eleazar / J. W. Bjerk <www.jwbjerk.com>"
 
 [file]
 gfx = "misc/buildings"
@@ -103,5 +102,4 @@
   8,  4, "b.statue_of_liberty"
   8,  5, "b.sun_tzus_war_academy"
   8,  6, "b.united_nations"
-  8,  7, "b.womens_suffrage"
-}
+  8,  7, "b.womens_suffrage"}
diff -Nurd -X.diff_ignore freeciv/data/misc/chiefs.spec freeciv/data/misc/chiefs.spec
--- freeciv/data/misc/chiefs.spec	2012-08-30 14:52:35.000000000 +0300
+++ freeciv/data/misc/chiefs.spec	2012-08-30 14:58:50.000000000 +0300
@@ -6,8 +6,7 @@
 [info]
 
 artists = "
-    Falk Hueffner <falk.hueffner@student.uni-tuebingen.de>
-"
+    Falk Hueffner <falk.hueffner@student.uni-tuebingen.de>"
 
 [file]
 gfx = "misc/chiefs"
@@ -20,18 +19,14 @@
 dy = 45
 
 tiles = { "row", "column", "tag"
-
-; Unit hit-point bars: approx percent of hp remaining
-0,  0, "unit.hp_100"
-0,  1, "unit.hp_90"
-0,  2, "unit.hp_80"
-0,  3, "unit.hp_70"
-0,  4, "unit.hp_60"
-0,  5, "unit.hp_50"
-0,  6, "unit.hp_40"
-0,  7, "unit.hp_30"
-0,  8, "unit.hp_20"
-0,  9, "unit.hp_10"
-0, 10, "unit.hp_0"
-
-}
+ 0,  0, "unit.hp_100"
+ 0,  1, "unit.hp_90"
+ 0,  2, "unit.hp_80"
+ 0,  3, "unit.hp_70"
+ 0,  4, "unit.hp_60"
+ 0,  5, "unit.hp_50"
+ 0,  6, "unit.hp_40"
+ 0,  7, "unit.hp_30"
+ 0,  8, "unit.hp_20"
+ 0,  9, "unit.hp_10"
+ 0, 10, "unit.hp_0"}
diff -Nurd -X.diff_ignore freeciv/data/misc/citybar.spec freeciv/data/misc/citybar.spec
--- freeciv/data/misc/citybar.spec	2012-08-30 14:52:35.000000000 +0300
+++ freeciv/data/misc/citybar.spec	2012-08-30 14:58:50.000000000 +0300
@@ -8,8 +8,7 @@
 
 artists = "
      Freim <...>
-     Madeline Book <madeline.book@gmail.com> (citybar.trade)
-"
+     Madeline Book <madeline.book@gmail.com> (citybar.trade)"
 
 [file]
 gfx = "misc/citybar"
@@ -26,9 +25,7 @@
 
   0,  0, "citybar.shields"
   0,  1, "citybar.food"
-  0,  2, "citybar.trade"
-
-}
+  0,  2, "citybar.trade"}
 
 [grid_star]
 
@@ -44,9 +41,7 @@
   0,  1, "citybar.occupancy_0"
   0,  2, "citybar.occupancy_1"
   0,  3, "citybar.occupancy_2"
-  0,  4, "citybar.occupancy_3"
-
-}
+  0,  4, "citybar.occupancy_3"}
 
 [grid_bg]
 x_top_left = 1
@@ -56,7 +51,4 @@
 dy = 152
 
 tiles = { "row", "column", "tag"
-
-  0,  0, "citybar.background"
-
-}
+  0,  0, "citybar.background"}
diff -Nurd -X.diff_ignore freeciv/data/misc/colors.tilespec freeciv/data/misc/colors.tilespec
--- freeciv/data/misc/colors.tilespec	2012-08-30 14:52:35.000000000 +0300
+++ freeciv/data/misc/colors.tilespec	2012-08-30 14:58:50.000000000 +0300
@@ -2,114 +2,80 @@
 
 ; Mapview
 mapview_unknown = {"r", "g", "b"
-    0,     0,   0
-}
+    0,     0,   0}
 mapview_citytext = {"r", "g", "b"
-    255, 255, 255
-}
+    255, 255, 255}
 mapview_cityblocked = {"r", "g", "b"
-    255,   0,   0
-}
+    255,   0,   0}
 mapview_goto = {"r", "g", "b"
-    0,   255, 200
-}
+    0,   255, 200}
 mapview_selection = {"r", "g", "b"
-    255, 255,   0
-}
+    255, 255,   0}
 mapview_trade_route_line = {"r", "g", "b"
-    0,   255, 127
-}
+    0,   255, 127}
 mapview_trade_routes_all_built = {"r", "g", "b"
-    0,   255,   0
-}
+    0,   255,   0}
 mapview_trade_routes_some_built = {"r", "g", "b"
-    255, 255,   0
-}
+    255, 255,   0}
 mapview_trade_routes_no_built = {"r", "g", "b"
-    255,   0,   0
-}
+    255,   0,   0}
 mapview_city_link = {"r", "g", "b"
-    0,   255,   0
-}
+    0,   255,   0}
 mapview_tile_link = {"r", "g", "b"
-    255,   0,   0
-}
+    255,   0,   0}
 mapview_unit_link = {"r", "g", "b"
-    0,   255, 255
-}
+    0,   255, 255}
 
 ; Spaceship
 spaceship_background = {"r", "g", "b"
-    0,   0,   0
-}
+    0,   0,   0}
 
 ; Overview
 overview_unknown = {"r", "g", "b"
-    0,     0,   0
-}
+    0,     0,   0}
 overview_mycity = {"r", "g", "b"
-    255, 255, 255
-}
+    255, 255, 255}
 overview_alliedcity = {"r", "g", "b"
-    255, 255, 255
-}
+    255, 255, 255}
 overview_enemycity = {"r", "g", "b"
-    0,   255, 200
-}
+    0,   255, 200}
 overview_myunit = {"r", "g", "b"
-    255, 255,   0
-}
+    255, 255,   0}
 overview_alliedunit = {"r", "g", "b"
-    255, 255, 255
-}
+    255, 255, 255}
 overview_enemyunit = {"r", "g", "b"
-    255,   0,   0
-}
+    255,   0,   0}
 overview_ocean = {"r", "g", "b"
 ; Generic ocean, not the same as the Ocean terrain (see "tiles" above)
-    0,     0, 200
-}
+    0,     0, 200}
 overview_ground = {"r", "g", "b"
-    0,   200,   0
-}
+    0,   200,   0}
 overview_viewrect = {"r", "g", "b"
-    255, 255, 255
-}
+    255, 255, 255}
 
 ; Reqtree
 reqtree_known = {"r", "g", "b"
-    255, 255, 255
-}
+    255, 255, 255}
 reqtree_prereqs_known = {"r", "g", "b"
-    210, 180, 140
-}
+    210, 180, 140}
 reqtree_unknown = {"r", "g", "b"
-    190, 150, 110
-}
+    190, 150, 110}
 reqtree_researching = {"r", "g", "b"
-    255, 250, 70
-}
+    255, 250, 70}
 reqtree_goal_prereqs_known = {"r", "g", "b"
-    95,  160, 190
-}
+    95,  160, 190}
 reqtree_goal_unknown = {"r", "g", "b"
-    85,  120, 180
-}
+    85,  120, 180}
 reqtree_unreachable = {"r", "g", "b"
-    225, 150, 110
-}
+    225, 150, 110}
 reqtree_background = {"r", "g", "b"
-    0,   0,   0
-}
+    0,   0,   0}
 reqtree_text = {"r", "g", "b"
-    0,   0,   0
-}
+    0,   0,   0}
 
 reqtree_edge = {"r", "g", "b"
-    127,   127,   127
-}
+    127,   127,   127}
   
 ; Player dialog
 playerdlg_background = {"r", "g", "b"
-    0,   0,   0
-}
+    0,   0,   0}
diff -Nurd -X.diff_ignore freeciv/data/misc/cursors.spec freeciv/data/misc/cursors.spec
--- freeciv/data/misc/cursors.spec	2012-08-30 14:52:35.000000000 +0300
+++ freeciv/data/misc/cursors.spec	2012-08-30 14:58:50.000000000 +0300
@@ -11,9 +11,7 @@
 
     edit_paint cursor take from GIMP
 
-    Hourglass and Add from Bluecurve theme in redhat-artwork
-    
-"
+    Hourglass and Add from Bluecurve theme in redhat-artwork"
 
 [file]
 gfx = "misc/cursors"
@@ -86,5 +84,4 @@
        2, 9, "cursor.wait2", 16, 16
        3, 9, "cursor.wait3", 16, 16
        4, 9, "cursor.wait4", 16, 16
-       5, 9, "cursor.wait5", 16, 16
-}
+       5, 9, "cursor.wait5", 16, 16}
diff -Nurd -X.diff_ignore freeciv/data/misc/editor.spec freeciv/data/misc/editor.spec
--- freeciv/data/misc/editor.spec	2012-08-30 14:52:35.000000000 +0300
+++ freeciv/data/misc/editor.spec	2012-08-30 14:58:50.000000000 +0300
@@ -7,8 +7,7 @@
 [info]
 
 artists = "
-	Pep
-"
+	Pep"
 
 [file]
 gfx = "misc/editor"
@@ -48,8 +47,7 @@
   3,  6, "editor.fill"
   3,  7, "editor.tiles_1"
   3,  8, "editor.tiles_2"
-  3,  9, "editor.terrain_resource_2"
-}
+  3,  9, "editor.terrain_resource_2"}
 
 [grid_mini]
 
@@ -73,8 +71,7 @@
   0, 10, "editor.mini_j" 
   0, 11, "editor.mini_gl"
   0, 12, "editor.mini_erase"
-  0, 13, "editor.mini_freeciv"
-}
+  0, 13, "editor.mini_freeciv"}
 
 [grid_extra]
 
@@ -91,5 +88,4 @@
   0,  3, "editor.type_paint"
   0,  4, "editor.bitmap_to_map"
   0,  5, "editor.copy"
-  0,  6, "editor.paste"
-}
+  0,  6, "editor.paste"}
diff -Nurd -X.diff_ignore freeciv/data/misc/flags.spec freeciv/data/misc/flags.spec
--- freeciv/data/misc/flags.spec	2012-08-30 14:52:35.000000000 +0300
+++ freeciv/data/misc/flags.spec	2012-08-30 14:58:50.000000000 +0300
@@ -561,10 +561,10 @@
    Zapotec		J.M. Maalderink
    Zhuang		Daniel Markstedt
    Zimbabwe		Madden (Wikimedia Commons)
-   Zulu			Daniel Markstedt and Caleb Moore
+   Zulu			Daniel Markstedt and Caleb Moore"
 
-"
 [extra]
+
 sprites =
 	{	"tag", "file"
 		"f.abkhazia", "flags/abkhazia"
diff -Nurd -X.diff_ignore freeciv/data/misc/overlays.spec freeciv/data/misc/overlays.spec
--- freeciv/data/misc/overlays.spec	2012-08-30 14:52:35.000000000 +0300
+++ freeciv/data/misc/overlays.spec	2012-08-30 14:58:50.000000000 +0300
@@ -7,8 +7,7 @@
 [info]
 
 artists = "
-    Jason Dorje Short <jdorje@freeciv.org>
-"
+    Jason Dorje Short <jdorje@freeciv.org>"
 
 [file]
 gfx = "misc/overlays"
@@ -29,5 +28,4 @@
   1, 1, "colors.overlay_3"
 
   0, 2, "mask.worked_tile"
-  1, 2, "mask.unworked_tile"
-}
+  1, 2, "mask.unworked_tile"}
diff -Nurd -X.diff_ignore freeciv/data/misc/shields.spec freeciv/data/misc/shields.spec
--- freeciv/data/misc/shields.spec	2012-08-30 14:52:35.000000000 +0300
+++ freeciv/data/misc/shields.spec	2012-08-30 14:58:50.000000000 +0300
@@ -7,8 +7,7 @@
 [info]
 
 artists = "
-   See flags.spec
-"
+   See flags.spec"
 [extra]
 sprites =
 	{	"tag", "file"
diff -Nurd -X.diff_ignore freeciv/data/misc/small.spec freeciv/data/misc/small.spec
--- freeciv/data/misc/small.spec	2012-08-30 14:52:35.000000000 +0300
+++ freeciv/data/misc/small.spec	2012-08-30 14:58:50.000000000 +0300
@@ -11,8 +11,7 @@
     Jeff Mallatt <jjm@codewell.com> (cooling flakes)
     Davide Pagnin <nightmare@freeciv.it> (angry citizens)
     Enrico Bini <e.bini@sssup.it> (fundamentalism icon)
-    Hogne Håskjold <haskjold@gmail.com> (gold coin)
-"
+    Hogne Håskjold <haskjold@gmail.com> (gold coin)"
 
 [file]
 gfx = "misc/small"
@@ -120,5 +119,4 @@
   1, 9,  "ev.wonderwillbebuilt"
   1, 10, "ev.wonderbuilt"
   1, 11, "s.plus"
-  1, 12, "s.minus"
-}
+  1, 12, "s.minus"}
diff -Nurd -X.diff_ignore freeciv/data/misc/space.spec freeciv/data/misc/space.spec
--- freeciv/data/misc/space.spec	2012-08-30 14:52:35.000000000 +0300
+++ freeciv/data/misc/space.spec	2012-08-30 14:58:50.000000000 +0300
@@ -7,8 +7,7 @@
 [info]
 
 artists = "
-    Alexandre Beraud <a_beraud@lemel.fr>
-"
+    Alexandre Beraud <a_beraud@lemel.fr>"
 
 [file]
 gfx = "misc/space"
@@ -31,5 +30,4 @@
   0,  3, "spaceship.structural"
   0,  4, "spaceship.fuel"
   0,  5, "spaceship.propulsion"
-  0,  6, "spaceship.exhaust"
-}
+  0,  6, "spaceship.exhaust"}
diff -Nurd -X.diff_ignore freeciv/data/misc/techs.spec freeciv/data/misc/techs.spec
--- freeciv/data/misc/techs.spec	2012-08-30 14:52:35.000000000 +0300
+++ freeciv/data/misc/techs.spec	2012-08-30 14:58:50.000000000 +0300
@@ -109,8 +109,7 @@
  PD - Unknown author
  a.communism         a.flight             a.refining
  a.chivalry          a.leadership         a.writing
- a.feudalism         a.mass_production    
-"
+ a.feudalism         a.mass_production    "
 
 
 [file]
@@ -209,5 +208,4 @@
   8, 4, "a.warrior_code"
   8, 5, "a.writing"
   8, 6, "a.environmentalism"
-  8, 7, "a.fundamentalism"
-}
+  8, 7, "a.fundamentalism"}
diff -Nurd -X.diff_ignore freeciv/data/misc/treaty.spec freeciv/data/misc/treaty.spec
--- freeciv/data/misc/treaty.spec	2012-08-30 14:52:35.000000000 +0300
+++ freeciv/data/misc/treaty.spec	2012-08-30 14:58:50.000000000 +0300
@@ -7,8 +7,7 @@
 [info]
 
 artists = "
-    pratheep
-"
+    pratheep"
 
 [file]
 gfx = "misc/treaty"
@@ -25,6 +24,4 @@
 ; Agree/disagree thumbs:
 
   0,  0, "treaty.agree_thumb_up"
-  0,  1, "treaty.disagree_thumb_down"
-
-}
+  0,  1, "treaty.disagree_thumb_down"}
diff -Nurd -X.diff_ignore freeciv/data/misc/wonders-large.spec freeciv/data/misc/wonders-large.spec
--- freeciv/data/misc/wonders-large.spec	2012-08-30 14:52:35.000000000 +0300
+++ freeciv/data/misc/wonders-large.spec	2012-08-30 14:58:50.000000000 +0300
@@ -48,8 +48,7 @@
   b.hoover_dam
   b.manhattan_project
   b.oracle
-  b.statue_of_liberty
-"
+  b.statue_of_liberty"
 
 [extra]
 sprites =
