diff -Nurd -X.diff_ignore freeciv/common/research.c freeciv/common/research.c
--- freeciv/common/research.c	2014-07-25 10:52:30.659308973 +0300
+++ freeciv/common/research.c	2014-07-31 21:46:33.627026326 +0300
@@ -760,7 +760,8 @@
 {
   const struct research *presearch = research_get(pplayer);
   int f = presearch->future_tech, t = presearch->techs_researched;
-  double tech_upkeep, total_research_factor;
+  double tech_upkeep = 0.0;
+  double total_research_factor;
   int members;
 
   if (TECH_UPKEEP_NONE == game.info.tech_upkeep_style) {
@@ -807,7 +808,6 @@
   default:
     fc_assert_msg(FALSE, "Invalid tech_cost_style %d",
                   game.info.tech_cost_style);
-    tech_upkeep = 0.0;
   }
 
   tech_upkeep *= total_research_factor / members;
