diff -Nurd -X.diff_ignore freeciv/common/featured_text.c freeciv/common/featured_text.c
--- freeciv/common/featured_text.c	2012-05-21 12:41:30.000000000 +0300
+++ freeciv/common/featured_text.c	2012-05-23 00:58:46.000000000 +0300
@@ -73,7 +73,7 @@
 
 const struct ft_color ftc_warning       = FT_COLOR("#FF0000",   NULL);
 const struct ft_color ftc_log           = FT_COLOR("#7F7F7F",   NULL);
-const struct ft_color ftc_server        = FT_COLOR("#8B0000",   NULL);
+const struct ft_color ftc_server        = FT_COLOR("#FF0000",   NULL);
 const struct ft_color ftc_client        = FT_COLOR("#EF7F00",   NULL);
 const struct ft_color ftc_editor        = FT_COLOR("#0000FF",   NULL);
 const struct ft_color ftc_command       = FT_COLOR("#006400",   NULL);
@@ -82,7 +82,7 @@
 const struct ft_color ftc_player_lost   = FT_COLOR("#FFFFFF",   "#000000");
 const struct ft_color ftc_game_start    = FT_COLOR("#00FF00",   "#115511");
 
-const struct ft_color ftc_chat_public   = FT_COLOR("#00008B",   NULL);
+const struct ft_color ftc_chat_public   = FT_COLOR("#FFFFFF",   NULL);
 const struct ft_color ftc_chat_ally     = FT_COLOR("#551166",   NULL);
 const struct ft_color ftc_chat_private  = FT_COLOR("#A020F0",   NULL);
 
@@ -1048,10 +1048,9 @@
 {
   static char buf[MAX_LEN_LINK];
 
-  fc_snprintf(buf, sizeof(buf), "%c%s tgt=\"%s\" id=%d name=\"%s\" %c%c",
-              SEQ_START, text_tag_type_short_name(TTT_LINK),
-              text_link_type_name(TLT_CITY), pcity->id,
-              city_name(pcity), SEQ_END, SEQ_STOP);
+  fc_snprintf(buf, sizeof(buf), "<a href=\"#\" onclick=\"show_city_dialog_by_id(%d);\">%s</a>",
+              pcity->id,
+              city_name(pcity));
   return buf;
 }
 
@@ -1098,12 +1097,8 @@
 {
   static char buf[MAX_LEN_LINK];
 
-  /* We use the rule name of the unit, it will be translated in every
-   * local sides in the function text_tag_replace_text(). */
-  fc_snprintf(buf, sizeof(buf), "%c%s tgt=\"%s\" id=%d name=\"%s\" %c%c",
-              SEQ_START, text_tag_type_short_name(TTT_LINK),
-              text_link_type_name(TLT_UNIT), punit->id,
-              unit_rule_name(punit), SEQ_END, SEQ_STOP);
+  fc_snprintf(buf, sizeof(buf), "%s", unit_name_translation(punit));
+
   return buf;
 }
 
diff -Nurd -X.diff_ignore freeciv/server/srv_main.c freeciv/server/srv_main.c
--- freeciv/server/srv_main.c	2012-05-23 00:58:34.000000000 +0300
+++ freeciv/server/srv_main.c	2012-05-23 00:59:29.000000000 +0300
@@ -1915,7 +1915,13 @@
       char leader_name[MAX_LEN_NAME];
 
       pick_random_player_name(nation_of_player(pplayer), leader_name);
-      sz_strlcpy(pplayer->name, leader_name);
+      /* FIXME: in Web client, connection username == player name.*/
+      if (pplayer->ai_controlled == TRUE) {
+        sz_strlcpy(pplayer->name, leader_name);
+      } else {
+        sz_strlcpy(pplayer->name, pplayer->username);
+      }
+
     }
 
     if (check_nation_leader_name(nation_of_player(pplayer), pplayer->name)) {
