diff -Nurd -X.diff_ignore freeciv/common/packets.def freeciv/common/packets.def
--- freeciv/common/packets.def	2013-05-26 23:49:30.059232854 +0300
+++ freeciv/common/packets.def	2013-05-26 23:49:56.083233730 +0300
@@ -400,8 +400,8 @@
   TERRAIN terrain;
   RESOURCE resource;
   BOOL special[S_LAST];
-  BV_BASES bases;
-  BV_ROADS roads;
+  STRING bases[MAX_BASE_TYPES];
+  STRING roads[MAX_ROAD_TYPES];
   STRING spec_sprite[MAX_LEN_NAME];
   STRING label[MAX_LEN_NAME];
 end
