Index: configure.ac
===================================================================
--- a/configure.ac	(revision 35267)
+++ b/configure.ac	(working copy)
@@ -1535,6 +1535,7 @@
     if test "x$sys_tolua_cmd" = "xtest" ; then
       TOLUA="\$(top_builddir)/dependencies/tolua-5.2/src/bin/tolua$EXEEXT"
       feature_systolua_cmd=missing
+      sys_tolua_cmd=false
     else
       AC_MSG_ERROR([tolua command not found from the system])
     fi
