diff -Nurd -X.diff_ignore freeciv/server/unithand.c freeciv/server/unithand.c
--- freeciv/server/unithand.c	2012-05-16 21:53:25.000000000 +0300
+++ freeciv/server/unithand.c	2012-05-16 22:16:45.000000000 +0300
@@ -1416,7 +1416,7 @@
       notify_player(pplayer, punit->tile, E_BAD_COMMAND,
                     FTC_SERVER_INFO, NULL,
                     _("You must declare war on %s first.  Try using "
-                      "players dialog (F3)."),
+                      "players dialog."),
                     victim == NULL
                     ? player_name(city_owner(pcity))
                     : player_name(unit_owner(victim)));
