diff -Nurd -X.diff_ignore freeciv/server/unittools.c freeciv/server/unittools.c
--- freeciv/server/unittools.c	2014-09-13 04:10:46.083198276 +0300
+++ freeciv/server/unittools.c	2014-09-13 04:11:57.915046204 +0300
@@ -2219,8 +2219,8 @@
 			struct packet_unit_short_info *packet,
                         enum unit_info_use packet_use, int info_city_id)
 {
-  packet->packet_use = packet_use;
-  packet->info_city_id = info_city_id;
+  /*  packet->packet_use = packet_use;
+      packet->info_city_id = info_city_id; */
 
   packet->id = punit->id;
   packet->owner = player_number(unit_owner(punit));
@@ -2247,13 +2247,13 @@
    * aren't fully known.  Note that for non-allied players, any transported
    * unit can't be seen at all.  For allied players we have to know if
    * transporters have room in them so that we can load units properly. */
-  if (!unit_transported(punit)) {
+  /*if (!unit_transported(punit)) {
     packet->transported = FALSE;
     packet->transported_by = 0;
   } else {
     packet->transported = TRUE;
     packet->transported_by = unit_transport_get(punit)->id;
-  }
+  }*/
 }
 
 /**************************************************************************
