diff -Nurd freeciv/server/unittools.c freeciv/server/unittools.c
--- freeciv/server/unittools.c	2014-08-30 01:53:26.575475382 +0300
+++ freeciv/server/unittools.c	2014-08-30 01:54:35.811373310 +0300
@@ -2219,8 +2219,8 @@
 			struct packet_unit_short_info *packet,
                         enum unit_info_use packet_use, int info_city_id)
 {
-  packet->packet_use = packet_use;
-  packet->info_city_id = info_city_id;
+  /*  packet->packet_use = packet_use;
+      packet->info_city_id = info_city_id; */
 
   packet->id = punit->id;
   packet->owner = player_number(unit_owner(punit));
@@ -2237,23 +2237,23 @@
     packet->activity = punit->activity;
   }
 
-  if (punit->activity_target == NULL) {
+  /*if (punit->activity_target == NULL) {
     packet->activity_tgt = EXTRA_NONE;
   } else {
     packet->activity_tgt = extra_index(punit->activity_target);
-  }
+  }*/
 
   /* Transported_by information is sent to the client even for units that
    * aren't fully known.  Note that for non-allied players, any transported
    * unit can't be seen at all.  For allied players we have to know if
    * transporters have room in them so that we can load units properly. */
-  if (!unit_transported(punit)) {
+  /*if (!unit_transported(punit)) {
     packet->transported = FALSE;
     packet->transported_by = 0;
   } else {
     packet->transported = TRUE;
     packet->transported_by = unit_transport_get(punit)->id;
-  }
+  }*/
 }
 
 /**************************************************************************
