diff -Nurd -X.diff_ignore freeciv/data/default/units.ruleset freeciv/data/default/units.ruleset
--- freeciv/data/default/units.ruleset	2012-05-16 20:56:42.000000000 +0300
+++ freeciv/data/default/units.ruleset	2012-05-16 22:15:37.000000000 +0300
@@ -324,7 +324,7 @@
 ")
 
 [unit_worker]
-name          = _("?unit:Workers")
+name          = _("Workers")
 class         = "Land"
 tech_req      = "Pottery"
 obsolete_by   = "Engineers"
@@ -1975,7 +1975,7 @@
 ")
 
 [unit_leader]
-name          = _("?unit:Leader")
+name          = _("Leader")
 class         = "Land"
 tech_req      = "None"
 obsolete_by   = "None"
