Index: unithand.c
===================================================================
--- unithand.c	(revisjon 63)
+++ unithand.c	(revisjon 64)
@@ -704,6 +704,7 @@
 {
   enum add_build_city_result res;
   struct unit *punit = player_find_unit_by_id(pplayer, unit_id);
+  char *wname = city_name_suggestion(pplayer, punit->tile);
 
   if (NULL == punit) {
     /* Shouldn't happen */
@@ -716,7 +717,7 @@
   res = test_unit_add_or_build_city(punit);
 
   if (res == AB_BUILD_OK) {
-    city_build(pplayer, punit, name);
+    city_build(pplayer, punit, wname);
   } else if (res == AB_ADD_OK) {
     city_add_unit(pplayer, punit);
   } else {
