Index: data/default/units.ruleset
===================================================================
--- data/default/units.ruleset	(revisjon 30)
+++ data/default/units.ruleset	(arbeidskopi)
@@ -324,7 +324,7 @@
 ")
 
 [unit_worker]
-name          = _("?unit:Workers")
+name          = _("Workers")
 class         = "Land"
 tech_req      = "Pottery"
 obsolete_by   = "Engineers"
@@ -1975,7 +1975,7 @@
 ")
 
 [unit_leader]
-name          = _("?unit:Leader")
+name          = _("Leader")
 class         = "Land"
 tech_req      = "None"
 obsolete_by   = "None"
