diff -Nurd -X.diff_ignore freeciv/server/maphand.c freeciv/server/maphand.c
--- freeciv/server/maphand.c	2015-02-19 02:25:39.636846363 +0200
+++ freeciv/server/maphand.c	2015-03-03 23:54:55.880401500 +0200
@@ -2218,14 +2218,17 @@
 void destroy_extra(struct tile *ptile, struct extra_type *pextra)
 {
   bv_player base_seen;
+  bool virtual = tile_virtual_check(ptile);
 
-  /* Remember what play was able to see the base. */
-  BV_CLR_ALL(base_seen);
-  players_iterate(pplayer) {
-    if (map_is_known_and_seen(ptile, pplayer, V_MAIN)) {
-      BV_SET(base_seen, player_index(pplayer));
-    }
-  } players_iterate_end;
+  /* Remember what players were able to see the base. */
+  if (!virtual) {
+    BV_CLR_ALL(base_seen);
+    players_iterate(pplayer) {
+      if (map_is_known_and_seen(ptile, pplayer, V_MAIN)) {
+        BV_SET(base_seen, player_index(pplayer));
+      }
+    } players_iterate_end;
+  }
 
   if (is_extra_caused_by(pextra, EC_BASE)) {
     struct base_type *pbase = extra_base_get(pextra);
@@ -2250,13 +2253,15 @@
 
   tile_remove_extra(ptile, pextra);
 
-  /* Remove base from vision of players which were able to see the base. */
-  players_iterate(pplayer) {
-    if (BV_ISSET(base_seen, player_index(pplayer))
-        && update_player_tile_knowledge(pplayer, ptile)) {
-      send_tile_info(pplayer->connections, ptile, FALSE);
-    }
-  } players_iterate_end;
+  if (!virtual) {
+    /* Remove base from vision of players which were able to see the base. */
+    players_iterate(pplayer) {
+      if (BV_ISSET(base_seen, player_index(pplayer))
+          && update_player_tile_knowledge(pplayer, ptile)) {
+        send_tile_info(pplayer->connections, ptile, FALSE);
+      }
+    } players_iterate_end;
+  }
 }
 
 /****************************************************************************
