diff --git freeciv/server/plrhand.c freeciv/server/plrhand.c
index 4a7c654..ad373a9 100644
--- freeciv/server/plrhand.c
+++ freeciv/server/plrhand.c
@@ -1012,7 +1012,7 @@ static void package_player_common(struct player *plr,
   packet->ai_skill_level = plr->ai_controlled
                            ? plr->ai_common.skill_level : 0;
   for (i = 0; i < player_slot_count(); i++) {
-    packet->love[i] = plr->ai_common.love[i];
+    packet->love[i] = plr->ai_common.love[i] + MAX_AI_LOVE;
   }
   packet->barbarian_type = plr->ai_common.barbarian_type;
 
