diff --git freeciv/common/packets.def freeciv/common/packets.def
index 41be296..065062e 100644
--- freeciv/common/packets.def
+++ freeciv/common/packets.def
@@ -772,7 +772,7 @@ PACKET_PLAYER_INFO = 51; sc, is-info
   UINT8 ai_skill_level;
   UINT8 barbarian_type;
   BV_PLAYER gives_shared_vision;
-  SINT16 love[MAX_NUM_PLAYER_SLOTS];
+  UINT32 love[MAX_NUM_PLAYER_SLOTS];
 
   UINT8 color_red;
   UINT8 color_green;
diff --git freeciv/server/plrhand.c freeciv/server/plrhand.c
index 4a7c654..ad373a9 100644
--- freeciv/server/plrhand.c
+++ freeciv/server/plrhand.c
@@ -1012,7 +1012,7 @@ static void package_player_common(struct player *plr,
   packet->ai_skill_level = plr->ai_controlled
                            ? plr->ai_common.skill_level : 0;
   for (i = 0; i < player_slot_count(); i++) {
-    packet->love[i] = plr->ai_common.love[i];
+    packet->love[i] = plr->ai_common.love[i] + MAX_AI_LOVE;
   }
   packet->barbarian_type = plr->ai_common.barbarian_type;
 
