diff -Nurd -X.diff_ignore freeciv/common/packets.def freeciv/common/packets.def
--- freeciv/common/packets.def	2012-05-23 01:00:48.000000000 +0300
+++ freeciv/common/packets.def	2012-05-23 01:02:48.000000000 +0300
@@ -673,7 +673,7 @@
   UINT8 barbarian_type;
   BV_PLAYER gives_shared_vision;
   STRING inventions[A_LAST+1];
-  SINT16 love[MAX_NUM_PLAYER_SLOTS];
+  UINT32 love[MAX_NUM_PLAYER_SLOTS];
 
   CITY wonders[B_LAST]; diff
 end
diff -Nurd -X.diff_ignore freeciv/server/plrhand.c freeciv/server/plrhand.c
--- freeciv/server/plrhand.c	2012-05-21 13:56:40.000000000 +0300
+++ freeciv/server/plrhand.c	2012-05-23 01:04:12.000000000 +0300
@@ -835,7 +835,7 @@
   packet->ai_skill_level = plr->ai_controlled
                            ? plr->ai_common.skill_level : 0;
   for (i = 0; i < player_slot_count(); i++) {
-    packet->love[i] = plr->ai_common.love[i];
+    packet->love[i] = plr->ai_common.love[i] + MAX_AI_LOVE;
   }
   packet->barbarian_type = plr->ai_common.barbarian_type;
 
