diff --git a/freeciv/common/packets.def b/freeciv/common/packets.def
index ab841af..ff5f4a6 100644
--- a/freeciv/common/packets.def
+++ b/freeciv/common/packets.def
@@ -704,6 +704,8 @@ PACKET_PLAYER_INFO=39;sc
   BOOL is_alive;
 
   DIPLSTATE diplstates[MAX_NUM_PLAYERS + MAX_NUM_BARBARIANS];
+
+  UINT32 love[MAX_NUM_PLAYERS + MAX_NUM_BARBARIANS];
   
   GOLD gold;
   PERCENT tax, science,luxury;
@@ -724,7 +726,6 @@ PACKET_PLAYER_INFO=39;sc
   UINT8 barbarian_type;
   VISION gives_shared_vision;
   STRING inventions[A_LAST+1];
-  SINT16 love[MAX_NUM_PLAYERS + MAX_NUM_BARBARIANS];
 
   CITY wonders[B_LAST]; diff
 end
diff --git a/freeciv/common/packets_gen.c b/freeciv/common/packets_gen.c
index 8ff7825..4f54f0b 100644
--- a/freeciv/common/packets_gen.c
+++ b/freeciv/common/packets_gen.c
@@ -6118,6 +6118,19 @@ static struct packet_player_info *receive_packet_player_info_100(struct connecti
       dio_get_diplstate(&din, &real_packet->diplstates[i]);
     }
   }
+  
+  {
+    int i;
+  
+    for (i = 0; i < MAX_NUM_PLAYERS + MAX_NUM_BARBARIANS; i++) {
+      {
+    int readin;
+  
+    dio_get_uint32(&din, &readin);
+    real_packet->love[i] = readin;
+  }
+    }
+  }
   {
     int readin;
   
@@ -6221,19 +6234,6 @@ static struct packet_player_info *receive_packet_player_info_100(struct connecti
   {
     int i;
   
-    for (i = 0; i < MAX_NUM_PLAYERS + MAX_NUM_BARBARIANS; i++) {
-      {
-    int readin;
-  
-    dio_get_sint16(&din, &readin);
-    real_packet->love[i] = readin;
-  }
-    }
-  }
-  
-  {
-    int i;
-  
     for (i = 0; i < B_LAST; i++) {
       {
     int readin;
@@ -6283,6 +6283,14 @@ static int send_packet_player_info_100(struct connection *pc, const struct packe
         dio_put_diplstate(&dout, &real_packet->diplstates[i]);
       }
     } 
+
+    {
+      int i;
+
+      for (i = 0; i < MAX_NUM_PLAYERS + MAX_NUM_BARBARIANS; i++) {
+        dio_put_uint32(&dout, real_packet->love[i]);
+      }
+    } 
   dio_put_uint32(&dout, real_packet->gold);
   dio_put_uint32(&dout, real_packet->tax);
   dio_put_uint32(&dout, real_packet->science);
@@ -6306,14 +6314,6 @@ static int send_packet_player_info_100(struct connection *pc, const struct packe
     {
       int i;
 
-      for (i = 0; i < MAX_NUM_PLAYERS + MAX_NUM_BARBARIANS; i++) {
-        dio_put_sint16(&dout, real_packet->love[i]);
-      }
-    } 
-
-    {
-      int i;
-
       for (i = 0; i < B_LAST; i++) {
         dio_put_uint32(&dout, real_packet->wonders[i]);
       }
diff --git a/freeciv/common/packets_gen.h b/freeciv/common/packets_gen.h
index fad53b0..6392844 100644
--- a/freeciv/common/packets_gen.h
+++ b/freeciv/common/packets_gen.h
@@ -408,6 +408,7 @@ struct packet_player_info {
   int nturns_idle;
   bool is_alive;
   struct player_diplstate diplstates[MAX_NUM_PLAYERS + MAX_NUM_BARBARIANS];
+  int love[MAX_NUM_PLAYERS + MAX_NUM_BARBARIANS];
   int gold;
   int tax;
   int science;
@@ -427,7 +428,6 @@ struct packet_player_info {
   int barbarian_type;
   unsigned int gives_shared_vision;
   char inventions[A_LAST+1];
-  int love[MAX_NUM_PLAYERS + MAX_NUM_BARBARIANS];
   int wonders[B_LAST];
 };
 
diff --git a/freeciv/server/plrhand.c b/freeciv/server/plrhand.c
index e07a77a..c298d36 100644
--- a/freeciv/server/plrhand.c
+++ b/freeciv/server/plrhand.c
@@ -754,7 +754,7 @@ static void package_player_common(struct player *plr,
   packet->ai = plr->ai_data.control;
   packet->ai_skill_level = plr->ai_data.control ? plr->ai_data.skill_level : 0;
   for (i = 0; i < MAX_NUM_PLAYERS + MAX_NUM_BARBARIANS; i++) {
-    packet->love[i] = plr->ai_data.love[i];
+    packet->love[i] = plr->ai_data.love[i] + MAX_AI_LOVE;
   }
   packet->barbarian_type = plr->ai_data.barbarian_type;
 
