diff -Nurd -X.diff_ignore freeciv/common/packets.def freeciv/common/packets.def
--- freeciv/common/packets.def	2012-05-16 22:15:19.000000000 +0300
+++ freeciv/common/packets.def	2012-05-16 22:17:49.000000000 +0300
@@ -705,6 +705,8 @@
   BOOL is_alive;
 
   DIPLSTATE diplstates[MAX_NUM_PLAYERS + MAX_NUM_BARBARIANS];
+
+  UINT32 love[MAX_NUM_PLAYERS + MAX_NUM_BARBARIANS];
   
   GOLD gold;
   PERCENT tax, science,luxury;
@@ -725,7 +727,6 @@
   UINT8 barbarian_type;
   VISION gives_shared_vision;
   STRING inventions[A_LAST+1];
-  SINT16 love[MAX_NUM_PLAYERS + MAX_NUM_BARBARIANS];
 
   CITY wonders[B_LAST]; diff
 end
diff -Nurd -X.diff_ignore freeciv/server/plrhand.c freeciv/server/plrhand.c
--- freeciv/server/plrhand.c	2012-05-16 21:19:06.000000000 +0300
+++ freeciv/server/plrhand.c	2012-05-16 22:17:52.000000000 +0300
@@ -754,7 +754,7 @@
   packet->ai = plr->ai_data.control;
   packet->ai_skill_level = plr->ai_data.control ? plr->ai_data.skill_level : 0;
   for (i = 0; i < MAX_NUM_PLAYERS + MAX_NUM_BARBARIANS; i++) {
-    packet->love[i] = plr->ai_data.love[i];
+    packet->love[i] = plr->ai_data.love[i] + MAX_AI_LOVE;
   }
   packet->barbarian_type = plr->ai_data.barbarian_type;
 
