diff -Nurd freeciv/server/maphand.c freeciv/server/maphand.c
--- freeciv/server/maphand.c	2021-03-17 21:56:31.888810886 +0200
+++ freeciv/server/maphand.c	2021-03-17 21:59:28.318532671 +0200
@@ -446,7 +446,7 @@
       conn_list_do_buffer(dest);
     }
 
-    send_tile_info(dest, ptile, FALSE);
+    send_tile_info(dest, ptile, TRUE);
   } whole_map_iterate_end;
 
   conn_list_do_unbuffer(dest);
@@ -582,11 +582,13 @@
       info.extras_owner = MAP_TILE_OWNER_NULL;
       info.worked = IDENTITY_NUMBER_ZERO;
 
-      info.terrain = terrain_count();
+      info.terrain = (NULL != tile_terrain(ptile))
+                      ? terrain_number(tile_terrain(ptile))
+                      : terrain_count();
       info.resource = MAX_EXTRA_TYPES;
       info.placing = -1;
 
-      BV_CLR_ALL(info.extras);
+      info.extras = ptile->extras; // WebGL client needs to know which tiles are rivers on game start.
 
       info.label[0] = '\0';
 
