diff -Nurd freeciv/configure.ac freeciv/configure.ac
--- freeciv/configure.ac	2022-04-01 21:41:37.043188391 +0300
+++ freeciv/configure.ac	2022-04-01 21:43:11.766676994 +0300
@@ -1775,6 +1775,7 @@
 	  data/civ2/Makefile
           data/stub/Makefile
           data/stub/nations/Makefile
+          data/webperimental/Makefile
 	  data/scenarios/Makefile
 	  data/nation/Makefile
           data/ruledit/Makefile
diff -Nurd freeciv/data/Makefile.am freeciv/data/Makefile.am
--- freeciv/data/Makefile.am	2022-04-01 21:41:37.043188391 +0300
+++ freeciv/data/Makefile.am	2022-04-01 21:43:43.942505580 +0300
@@ -79,7 +79,7 @@
 endif
 
 SRV_RE_DATADIRS = default classic civ1 civ2 sandbox multiplayer \
-                  alien civ2civ3 stub nation
+                  alien civ2civ3 stub webperimental nation
 if SERVER
 SERVERDATADIRS = $(SRV_RE_DATADIRS) scenarios
 else
diff -Nurd freeciv/data/webperimental/Makefile.am freeciv/data/webperimental/Makefile.am
--- freeciv/data/webperimental/Makefile.am	1970-01-01 02:00:00.000000000 +0200
+++ freeciv/data/webperimental/Makefile.am	2022-04-01 21:42:23.942933854 +0300
@@ -0,0 +1,21 @@
+## Process this file with automake to produce Makefile.in
+
+## Override automake so that "make install" puts these in proper place:
+pkgdatadir = $(datadir)/$(PACKAGE)/webperimental
+
+pkgdata_DATA =			\
+	buildings.ruleset	\
+	cities.ruleset		\
+	effects.ruleset		\
+	game.ruleset		\
+	governments.ruleset	\
+	nations.ruleset		\
+	script.lua		\
+	parser.lua		\
+	styles.ruleset		\
+	techs.ruleset		\
+	terrain.ruleset		\
+	README.webperimental	\
+	units.ruleset
+
+EXTRA_DIST = $(pkgdata_DATA)
diff -Nurd freeciv/translations/core/POTFILES.in freeciv/translations/core/POTFILES.in
--- freeciv/translations/core/POTFILES.in	2022-04-01 21:41:37.815184179 +0300
+++ freeciv/translations/core/POTFILES.in	2022-04-01 21:42:23.942933854 +0300
@@ -234,6 +234,15 @@
 data/alien/nation/mathclub.ruleset
 data/alien/nation/secretsociety.ruleset
 data/alien/nation/teamgladiators.ruleset
+data/webperimental/buildings.ruleset
+data/webperimental/cities.ruleset
+data/webperimental/game.ruleset
+data/webperimental/governments.ruleset
+data/webperimental/script.lua
+data/webperimental/styles.ruleset
+data/webperimental/techs.ruleset
+data/webperimental/terrain.ruleset
+data/webperimental/units.ruleset
 data/helpdata.txt
 data/multiplayer/buildings.ruleset
 data/multiplayer/cities.ruleset
