diff -Nurd -X.diff_ignore freeciv/common/packets.def freeciv/common/packets.def
--- freeciv/common/packets.def	2016-02-24 02:47:13.117158150 +0200
+++ freeciv/common/packets.def	2016-02-24 02:56:43.539526073 +0200
@@ -203,7 +203,7 @@
 type TECH_LIST          = tech_list(int)
 type UNIT_LIST          = unit_list(int)
 type BUILDING_LIST      = building_list(int)
-type WORKLIST           = worklist(struct worklist)
+type WORKLIST           = string(char)
 # string that is URI encoded in the JSON protocol
 type ESTRING            = estring(char)
 
@@ -660,7 +660,7 @@
   UINT8 walls;
   SINT8 city_image;
 
-  WORKLIST worklist;
+  WORKLIST worklist[MAX_LEN_WORKLIST_WEB];
 
   STRING can_build_unit[MAX_LEN_MSG];
   STRING can_build_improvement[MAX_LEN_MSG];
@@ -719,7 +719,7 @@
 
 PACKET_CITY_WORKLIST = 36; cs, dsend
   CITY city_id;
-  WORKLIST worklist;
+  # WORKLIST worklist;
 end
 
 PACKET_CITY_MAKE_SPECIALIST = 37; cs, dsend
@@ -2178,3 +2178,14 @@
 PACKET_INFO_TEXT_MESSAGE = 259; sc, dsend
   STRING message[MAX_LEN_MSG];
 end
+
+PACKET_CITY_WORKLIST_ADD = 260; cs, dsend
+  CITY city_id;
+  UINT32 kind;
+  UINT32 value;
+end
+
+PACKET_CITY_WORKLIST_REMOVE = 261; cs, dsend
+  CITY city_id;
+  UINT32 index;
+end
diff -Nurd -X.diff_ignore freeciv/common/worklist.h freeciv/common/worklist.h
--- freeciv/common/worklist.h	2016-02-24 02:40:28.870661317 +0200
+++ freeciv/common/worklist.h	2016-02-24 02:56:09.587614879 +0200
@@ -17,13 +17,20 @@
 extern "C" {
 #endif /* __cplusplus */
 
+/* utility */
 #include "registry.h"
 
+/* common */
 #include "fc_types.h"
 
 #define MAX_LEN_WORKLIST 64
 #define MAX_NUM_WORKLISTS 16
 
+#ifdef FREECIV_WEB
+#define MAX_LEN_WORKLIST_ITEM 8
+#define MAX_LEN_WORKLIST_WEB (64 * MAX_LEN_WORKLIST_ITEM)
+#endif /* FREECIV_WEB */
+
 /* a worklist */
 struct worklist {
   int length;
diff -Nurd -X.diff_ignore freeciv/server/cityhand.c freeciv/server/cityhand.c
--- freeciv/server/cityhand.c	2016-02-24 02:40:28.886661246 +0200
+++ freeciv/server/cityhand.c	2016-02-24 02:50:22.124576311 +0200
@@ -374,8 +374,7 @@
 /**************************************************************************
   Handle city worklist update request
 **************************************************************************/
-void handle_city_worklist(struct player *pplayer, int city_id,
-                          const struct worklist *worklist)
+void handle_city_worklist(struct player *pplayer, int city_id)
 {
   struct city *pcity = player_city_by_number(pplayer, city_id);
 
@@ -383,12 +382,44 @@
     return;
   }
 
-  worklist_copy(&pcity->worklist, worklist);
+  // not in use by webclient.
+  //worklist_copy(&pcity->worklist, worklist);
+  //send_city_info(pplayer, pcity);
+}
+
+/**************************************************************************
+  Handle city worklist add request
+**************************************************************************/
+void handle_city_worklist_add(struct player *pplayer, int city_id,
+                              int kind, int value)
+{
+  struct city *pcity = player_city_by_number(pplayer, city_id);
+
+  if (!pcity) {
+    return;
+  }
+  struct universal addme = universal_by_number(kind, value);
+  worklist_append(&pcity->worklist, &addme);
 
   send_city_info(pplayer, pcity);
 }
 
 /**************************************************************************
+  Handle city worklist remove request
+**************************************************************************/
+void handle_city_worklist_remove(struct player *pplayer, int city_id,
+                          int index)
+{
+  struct city *pcity = player_city_by_number(pplayer, city_id);
+
+  if (!pcity) {
+    return;
+  }
+  worklist_remove(&pcity->worklist, index);
+
+  send_city_info(pplayer, pcity);
+}
+/**************************************************************************
   Handle buying request. This function does properly check its input as
   it may come from untrusted source over the network.
 **************************************************************************/
diff -Nurd -X.diff_ignore freeciv/server/citytools.c freeciv/server/citytools.c
--- freeciv/server/citytools.c	2016-02-24 02:40:28.898661194 +0200
+++ freeciv/server/citytools.c	2016-02-24 02:50:22.124576311 +0200
@@ -2299,6 +2299,7 @@
   char food_output_buf[MAX_NUM_ITEMS + 1];
   char shield_output_buf[MAX_NUM_ITEMS + 1];
   char trade_output_buf[MAX_NUM_ITEMS + 1];
+  char worklist_buf[MAX_LEN_WORKLIST_WEB];
   struct tile *pcenter = city_tile(pcity);
   int c = 0;
 
@@ -2431,7 +2432,20 @@
   packet->caravan_shields=pcity->caravan_shields;
   packet->last_turns_shield_surplus = pcity->last_turns_shield_surplus;
 
-  //worklist_copy(&packet->worklist, &pcity->worklist);
+  if (worklist_is_empty(&pcity->worklist)) {
+    sz_strlcpy(packet->worklist, "-");
+  } else {
+    worklist_buf[0] = '\0';
+    int w = 0;
+    for (w = 0; w < worklist_length(&pcity->worklist); w++) {
+      struct universal target;
+      char f[MAX_LEN_WORKLIST_ITEM];
+      worklist_peek_ith(&pcity->worklist, &target, w);
+      fc_snprintf(f, sizeof(f), "%d,%d;", target.kind, universal_number(&target));
+      strcat(worklist_buf, f);
+    }
+    sz_strlcpy(packet->worklist, worklist_buf);
+  }
   packet->diplomat_investigate=dipl_invest;
 
   packet->airlift = pcity->airlift;
