diff -Nurd -X.diff_ignore freeciv/server/srv_main.c freeciv/server/srv_main.c
--- freeciv/server/srv_main.c	2014-10-09 00:02:31.573964752 +0300
+++ freeciv/server/srv_main.c	2014-10-10 23:20:13.285228866 +0300
@@ -900,7 +900,7 @@
         struct tile *ptile = punit->tile;
 
         if (unit_can_be_retired(punit)
-            && fc_rand(100) <= get_unit_bonus(punit, EFT_RETIRE_PCT)) {
+            && fc_rand(100) < get_unit_bonus(punit, EFT_RETIRE_PCT)) {
           notify_player(pplayer, ptile, E_UNIT_LOST_MISC, ftc_server,
                         _("%s retired!"), unit_tile_link(punit));
           wipe_unit(punit, ULR_RETIRED, NULL);
