diff -ru -Xfreeciv-2.2.3/diff_ignore freeciv-2.2.3/common/game.h freeciv-2.2.3-changed//common/game.h
--- freeciv-2.2.3/common/game.h	2010-05-23 23:39:02.000000000 +0200
+++ freeciv-2.2.3-changed//common/game.h	2010-07-17 21:37:22.000000000 +0200
@@ -120,6 +120,7 @@
       struct {
         bool user_message_set;
         char user_message[256];
+	char topic[256];
       } meta_info;
     } server;
   };
@@ -319,7 +320,7 @@
 
 #define GAME_DEFAULT_AUTO_AI_TOGGLE  FALSE
 
-#define GAME_DEFAULT_TIMEOUT         0
+#define GAME_DEFAULT_TIMEOUT         60
 #define GAME_DEFAULT_TIMEOUTINT      0
 #define GAME_DEFAULT_TIMEOUTINTINC   0
 #define GAME_DEFAULT_TIMEOUTINC      0
@@ -327,7 +328,7 @@
 #define GAME_DEFAULT_TIMEOUTADDEMOVE 0
 #define GAME_DEFAULT_TIMEOUTCOUNTER  1
 
-#define GAME_MIN_TIMEOUT             -1
+#define GAME_MIN_TIMEOUT             50
 #define GAME_MAX_TIMEOUT             8639999
 
 #define GAME_DEFAULT_PHASE_MODE 0
Only in freeciv-2.2.3-changed//common: game.lo
Only in freeciv-2.2.3-changed//common: government.lo
Only in freeciv-2.2.3-changed//common: idex.lo
Only in freeciv-2.2.3-changed//common: improvement.lo
Only in freeciv-2.2.3-changed//common: libfreeciv.la
Only in freeciv-2.2.3-changed//common: .libs
diff -ru -Xfreeciv-2.2.3/diff_ignore freeciv-2.2.3/common/map.h freeciv-2.2.3-changed//common/map.h
--- freeciv-2.2.3/common/map.h	2010-05-23 23:39:02.000000000 +0200
+++ freeciv-2.2.3-changed//common/map.h	2010-07-17 21:21:07.000000000 +0200
@@ -473,9 +473,9 @@
 /* Size of the map in thousands of tiles.  Setting the maximal size over
  * than 30 is dangerous, because some parts of the code (e.g. path finding)
  * assume the tile index is of type (signed short int). */
-#define MAP_DEFAULT_SIZE         4
+#define MAP_DEFAULT_SIZE         3
 #define MAP_MIN_SIZE             1
-#define MAP_MAX_SIZE             30
+#define MAP_MAX_SIZE             15
 
 /* This defines the maximum linear size in _map_ coordinates.
  * This must be smaller than 255 because of the way coordinates are sent
Only in freeciv-2.2.3-changed//common: map.lo
Only in freeciv-2.2.3-changed//common: movement.lo
Only in freeciv-2.2.3-changed//common: nation.lo
Only in freeciv-2.2.3-changed//common: packets_gen.lo
Only in freeciv-2.2.3-changed//common: packets.lo
Only in freeciv-2.2.3-changed//common: player.lo
Only in freeciv-2.2.3-changed//common: requirements.lo
Only in freeciv-2.2.3-changed//common: spaceship.lo
Only in freeciv-2.2.3-changed//common: specialist.lo
Only in freeciv-2.2.3-changed//common: team.lo
Only in freeciv-2.2.3-changed//common: tech.lo
Only in freeciv-2.2.3-changed//common: terrain.lo
Only in freeciv-2.2.3-changed//common: tile.lo
Only in freeciv-2.2.3-changed//common: unitlist.lo
Only in freeciv-2.2.3-changed//common: unit.lo
Only in freeciv-2.2.3-changed//common: unittype.lo
Only in freeciv-2.2.3-changed//common: version.lo
Only in freeciv-2.2.3-changed//common: vision.lo
Only in freeciv-2.2.3-changed//common: worklist.lo
Only in freeciv-2.2.3-changed//data: civclient.dsc
Only in freeciv-2.2.3-changed//data: civserver.dsc
Only in freeciv-2.2.3-changed//data: civserver.room
diff -ru -Xfreeciv-2.2.3/diff_ignore freeciv-2.2.3/data/Freeciv freeciv-2.2.3-changed//data/Freeciv
--- freeciv-2.2.3/data/Freeciv	2010-05-23 23:46:17.000000000 +0200
+++ freeciv-2.2.3-changed//data/Freeciv	2010-07-17 22:08:57.000000000 +0200
@@ -2814,6 +2814,9 @@
 Freeciv*outputwindowtext.*.foreground:		   black
 Freeciv*outputwindowtext*wrap:			   word
 Freeciv*outputwindowtext*String: _("\
+Welcome to the Freeciv.net public server! Please also visit http://www.freeciv.net \n\
+for the latest information about this server. Send feedback to:  andrearo@pvv.ntnu.no\n\
+ \n\
 Freeciv is free software and you are welcome to distribute copies of it\n\
 under certain conditions; See the \"Copying\" item on the Help menu.\n\
 Now.. Go give'em hell!")
diff -ru -Xfreeciv-2.2.3/diff_ignore freeciv-2.2.3/data/Freeciv.in freeciv-2.2.3-changed//data/Freeciv.in
--- freeciv-2.2.3/data/Freeciv.in	2010-05-23 23:43:00.000000000 +0200
+++ freeciv-2.2.3-changed//data/Freeciv.in	2010-07-17 22:07:47.000000000 +0200
@@ -2814,6 +2814,9 @@
 Freeciv*outputwindowtext.*.foreground:		   black
 Freeciv*outputwindowtext*wrap:			   word
 Freeciv*outputwindowtext*String: _("\
+Welcome to the Freeciv.net public server! Please also visit http://www.freeciv.net \n\
+for the latest information about this server. Send feedback to:  andrearo@pvv.ntnu.no\n\
+ \n\
 Freeciv is free software and you are welcome to distribute copies of it\n\
 under certain conditions; See the \"Copying\" item on the Help menu.\n\
 Now.. Go give'em hell!")
Only in freeciv-2.2.3-changed//dependencies/tolua-5.1/src/bin: .libs
Only in freeciv-2.2.3-changed/: libtool
Only in freeciv-2.2.3: po
Only in freeciv-2.2.3-changed//server: aiiface.lo
Only in freeciv-2.2.3-changed//server: auth.lo
Only in freeciv-2.2.3-changed//server: barbarian.lo
Only in freeciv-2.2.3-changed//server: cityhand.lo
Only in freeciv-2.2.3-changed//server: citytools.lo
Only in freeciv-2.2.3-changed//server: cityturn.lo
diff -ru -Xfreeciv-2.2.3/diff_ignore freeciv-2.2.3/server/civserver.c freeciv-2.2.3-changed//server/civserver.c
--- freeciv-2.2.3/server/civserver.c	2010-05-23 23:39:14.000000000 +0200
+++ freeciv-2.2.3-changed//server/civserver.c	2010-07-17 21:12:59.000000000 +0200
@@ -281,6 +281,8 @@
         freelog(LOG_ERROR, _("Illegal value \"%s\" for --Announce"), option);
       }
       free(option);
+    } else if ((option = get_option_malloc("--topic", argv, &inx, argc))) {
+      sz_strlcpy(game.server.meta_info.topic, option);
     } else {
       fc_fprintf(stderr, _("Error: unknown option '%s'\n"), argv[inx]);
       showhelp = TRUE;
@@ -343,6 +345,7 @@
     fc_fprintf(stderr, _("  -r, --read FILE\tRead startup script FILE\n"));
     fc_fprintf(stderr,
 	       _("  -R, --Ranklog FILE\tUse FILE as ranking logfile\n"));
+    fc_fprintf(stderr, _("  -t, --topic \"MESSAGE\"\t\tSet the metaserver topic\n"));
     fc_fprintf(stderr, _("  -v, --version\t\tPrint the version number\n"));
     /* TRANS: No full stop after the URL, could cause confusion. */
     fc_fprintf(stderr, _("Report bugs at %s\n"), BUG_URL);
diff -ru -Xfreeciv-2.2.3/diff_ignore freeciv-2.2.3/server/commands.c freeciv-2.2.3-changed//server/commands.c
--- freeciv-2.2.3/server/commands.c	2010-05-23 23:39:14.000000000 +0200
+++ freeciv-2.2.3-changed//server/commands.c	2010-07-17 21:35:06.000000000 +0200
@@ -417,57 +417,7 @@
       "then the game ends."),
    VCF_NONE, 0
   },
-  {"remove",	ALLOW_CTRL,
-   /* TRANS: translate text between <> only */
-   N_("remove <player-name>"),
-   N_("Fully remove player from game."),
-   N_("This *completely* removes a player from the game, including "
-      "all cities and units etc.  Use with care!"),
-   VCF_NONE, 50
-  },
-  {"save",	ALLOW_ADMIN,
-   /* TRANS: translate text between <> only */
-   N_("save\n"
-      "save <file-name>"),
-   N_("Save game to file."),
-   N_("Save the current game to file <file-name>.  If no file-name "
-      "argument is given saves to \"<auto-save name prefix><year>m.sav[.gz]\".\n"
-      "To reload a savegame created by 'save', start the server with "
-      "the command-line argument:\n"
-      "    --file <filename>\n"
-      "and use the 'start' command once players have reconnected."),
-   VCF_NONE, 0
-  },
-  {"load",      ALLOW_CTRL,
-   /* TRANS: translate text between <> only */
-   N_("load\n"
-      "load <file-name>"),
-   N_("Load game from file."),
-   N_("Load a game from <file-name>. Any current data including players, "
-      "rulesets and server options are lost.\n"),
-   VCF_NONE, 50
-  },
-  {"read",	ALLOW_CTRL,
-   /* TRANS: translate text between <> only */
-   N_("read <file-name>"),
-   N_("Process server commands from file."), NULL,
-   VCF_NONE, 50
-  },
-  {"write",	ALLOW_HACK,
-   /* TRANS: translate text between <> only */
-   N_("write <file-name>"),
-   N_("Write current settings as server commands to file."), NULL,
-   VCF_NONE, 0
-  },
-  {"reset",	ALLOW_CTRL,
-   /* no translatable parameters */
-   SYN_ORIG_("reset"),
-   N_("Reset all server settings."),
-   N_("Reset all settings and re-read the server start script, "
-      "if there was one given with the --read server argument. "),
-   VCF_NONE, 50
-  },
-  {"rfcstyle",	ALLOW_HACK,
+   {"rfcstyle",	ALLOW_HACK,
    /* no translatable parameters */
    SYN_ORIG_("rfcstyle"),
    N_("Switch server output between 'RFC-style' and normal style."), NULL,
Only in freeciv-2.2.3-changed//server: commands.lo
diff -ru -Xfreeciv-2.2.3/diff_ignore freeciv-2.2.3/server/connecthand.c freeciv-2.2.3-changed//server/connecthand.c
--- freeciv-2.2.3/server/connecthand.c	2010-05-23 23:39:14.000000000 +0200
+++ freeciv-2.2.3-changed//server/connecthand.c	2010-07-17 22:20:21.000000000 +0200
@@ -148,8 +148,8 @@
   /* introduce the server to the connection */
   if (my_gethostname(hostname, sizeof(hostname)) == 0) {
     notify_conn(dest, NULL, E_CONNECTION, ftc_any,
-                _("Welcome to the %s Server running at %s port %d."),
-                freeciv_name_version(), hostname, srvarg.port);
+                _("Welcome to the Freeciv.net public server! Please also visit http://www.freeciv.net \n\
+for the latest information about this server. Send feedback to:  andrearo@pvv.ntnu.no\n"));
   } else {
     notify_conn(dest, NULL, E_CONNECTION, ftc_any,
                 _("Welcome to the %s Server at port %d."),
Only in freeciv-2.2.3-changed//server: connecthand.lo
Only in freeciv-2.2.3-changed//server: console.lo
Only in freeciv-2.2.3-changed//server: diplhand.lo
Only in freeciv-2.2.3-changed//server: diplomats.lo
Only in freeciv-2.2.3-changed//server: edithand.lo
Only in freeciv-2.2.3-changed//server: freeciv-server
Only in freeciv-2.2.3-changed//server: freeciv-server.desktop
Only in freeciv-2.2.3-changed//server: gamehand.lo
Only in freeciv-2.2.3-changed//server/generator: height_map.lo
Only in freeciv-2.2.3-changed//server/generator: libgenerator.la
Only in freeciv-2.2.3-changed//server/generator: .libs
Only in freeciv-2.2.3-changed//server/generator: mapgen.lo
Only in freeciv-2.2.3-changed//server/generator: mapgen_topology.lo
Only in freeciv-2.2.3-changed//server/generator: startpos.lo
Only in freeciv-2.2.3-changed//server/generator: temperature_map.lo
Only in freeciv-2.2.3-changed//server/generator: utilities.lo
Only in freeciv-2.2.3-changed//server: ggzserver.lo
Only in freeciv-2.2.3-changed//server: gotohand.lo
Only in freeciv-2.2.3-changed//server: handchat.lo
Only in freeciv-2.2.3-changed//server: hand_gen.lo
Only in freeciv-2.2.3-changed//server: libfreeciv-srv.la
Only in freeciv-2.2.3-changed//server: .libs
Only in freeciv-2.2.3-changed//server: maphand.lo
diff -ru -Xfreeciv-2.2.3/diff_ignore freeciv-2.2.3/server/meta.c freeciv-2.2.3-changed//server/meta.c
--- freeciv-2.2.3/server/meta.c	2010-05-23 23:39:14.000000000 +0200
+++ freeciv-2.2.3-changed//server/meta.c	2010-07-17 21:13:26.000000000 +0200
@@ -109,6 +109,11 @@
   return meta_message;
 }
 
+const char *get_meta_topic_string(void)
+{
+  return game.server.meta_info.topic;
+}
+
 /*************************************************************************
  The metaserver message set by user
 *************************************************************************/
@@ -194,6 +199,12 @@
   con_flush();
 
   server_close_meta();
+
+  if (conn_list_size(game.est_connections) == 0
+      && srvarg.quitidle != 0) {
+    set_server_state(S_S_OVER);
+    force_end_of_sniff = TRUE;
+  }
 }
 
 /*************************************************************************
@@ -258,6 +269,8 @@
     mystrlcpy(s, "bye=1&", rest);
     s = end_of_strn(s, &rest);
   } else {
+    const char *topic;
+
     my_snprintf(s, rest, "version=%s&", fc_url_encode(VERSION_STRING));
     s = end_of_strn(s, &rest);
 
@@ -272,6 +285,13 @@
                 fc_url_encode(srvarg.serverid));
     s = end_of_strn(s, &rest);
 
+    topic = get_meta_topic_string();
+    if (topic != NULL && topic[0] != '\0') {
+      my_snprintf(s, rest, "topic=%s&",
+                  fc_url_encode(topic));
+      s = end_of_strn(s, &rest);
+    }
+
     my_snprintf(s, rest, "message=%s&",
                 fc_url_encode(get_meta_message_string()));
     s = end_of_strn(s, &rest);
diff -ru -Xfreeciv-2.2.3/diff_ignore freeciv-2.2.3/server/meta.h freeciv-2.2.3-changed//server/meta.h
--- freeciv-2.2.3/server/meta.h	2010-05-23 23:39:14.000000000 +0200
+++ freeciv-2.2.3-changed//server/meta.h	2010-07-17 21:19:42.000000000 +0200
@@ -17,7 +17,7 @@
 
 #define DEFAULT_META_SERVER_NO_SEND  TRUE
 #define DEFAULT_META_SERVER_ADDR     META_URL /* config.h */
-#define METASERVER_REFRESH_INTERVAL   (3*60)
+#define METASERVER_REFRESH_INTERVAL   (60)
 #define METASERVER_MIN_UPDATE_INTERVAL 7   /* not too short, not too long */
 
 enum meta_flag {
@@ -29,6 +29,8 @@
 const char *default_meta_patches_string(void);
 const char *default_meta_message_string(void);
 
+const char *get_meta_topic_string(void);
+
 const char *get_meta_patches_string(void);
 const char *get_meta_message_string(void);
 const char *get_user_meta_message_string(void);
Only in freeciv-2.2.3-changed//server: meta.lo
Only in freeciv-2.2.3-changed//server: notify.lo
Only in freeciv-2.2.3-changed//server: plrhand.lo
Only in freeciv-2.2.3-changed//server: report.lo
Only in freeciv-2.2.3-changed//server: ruleset.lo
Only in freeciv-2.2.3-changed//server: sanitycheck.lo
Only in freeciv-2.2.3-changed//server: savegame.lo
Only in freeciv-2.2.3-changed//server: score.lo
Only in freeciv-2.2.3-changed//server/scripting: api_actions.lo
Only in freeciv-2.2.3-changed//server/scripting: api_find.lo
Only in freeciv-2.2.3-changed//server/scripting: api_gen.lo
Only in freeciv-2.2.3-changed//server/scripting: api_intl.lo
Only in freeciv-2.2.3-changed//server/scripting: api_methods.lo
Only in freeciv-2.2.3-changed//server/scripting: api_notify.lo
Only in freeciv-2.2.3-changed//server/scripting: api_utilities.lo
Only in freeciv-2.2.3-changed//server/scripting: .libs
Only in freeciv-2.2.3-changed//server/scripting: libscripting.la
Only in freeciv-2.2.3-changed//server/scripting: script.lo
Only in freeciv-2.2.3-changed//server/scripting: script_signal.lo
diff -ru -Xfreeciv-2.2.3/diff_ignore freeciv-2.2.3/server/sernet.c freeciv-2.2.3-changed//server/sernet.c
--- freeciv-2.2.3/server/sernet.c	2010-05-23 23:39:14.000000000 +0200
+++ freeciv-2.2.3-changed//server/sernet.c	2010-07-17 21:18:56.000000000 +0200
@@ -498,7 +498,7 @@
       if (connections && conn_list_size(game.est_connections) == 0) {
 	if (last_noplayers != 0) {
 	  if (time(NULL) > last_noplayers + srvarg.quitidle) {
-	    save_game_auto("Lost all connections", "quitidle");
+	    //save_game_auto("Lost all connections", "quitidle");
 	    set_meta_message_string(N_("restarting for lack of players"));
 	    freelog(LOG_NORMAL, "%s", Q_(get_meta_message_string()));
 	    (void) send_server_info_to_metaserver(META_INFO);
Only in freeciv-2.2.3-changed//server: sernet.lo
Only in freeciv-2.2.3-changed//server: settings.lo
Only in freeciv-2.2.3-changed//server: settlers.lo
Only in freeciv-2.2.3-changed//server: spacerace.lo
diff -ru -Xfreeciv-2.2.3/diff_ignore freeciv-2.2.3/server/srv_main.c freeciv-2.2.3-changed//server/srv_main.c
--- freeciv-2.2.3/server/srv_main.c	2010-05-23 23:39:14.000000000 +0200
+++ freeciv-2.2.3-changed//server/srv_main.c	2010-07-17 21:18:14.000000000 +0200
@@ -1117,8 +1117,8 @@
 
   generate_save_name(game.server.save_name, filename, sizeof(filename),
                      reason_filename);
-  save_game(filename, save_reason, FALSE);
-  save_ppm();
+  //save_game(filename, save_reason, FALSE);
+  //save_ppm();
 }
 
 /**************************************************************************
@@ -2049,7 +2049,7 @@
       if (game.info.phase == 0) {
 	if (save_counter >= game.info.save_nturns && game.info.save_nturns > 0) {
 	  save_counter = 0;
-	  save_game_auto("Autosave", NULL);
+	  //save_game_auto("Autosave", NULL);
 	}
 	save_counter++;
       }
@@ -2209,7 +2209,7 @@
       && conn_list_size(game.est_connections) > 0) {
     /* Save game on game_over, but not when the gameover was caused by
      * the -q parameter. */
-    save_game_auto("Game over", "final");
+    //save_game_auto("Game over", "final");
   }
 }
 
Only in freeciv-2.2.3-changed//server: srv_main.lo
Only in freeciv-2.2.3-changed//server: stdinhand.lo
Only in freeciv-2.2.3-changed//server: techtools.lo
Only in freeciv-2.2.3-changed//server: unithand.lo
Only in freeciv-2.2.3-changed//server: unittools.lo
Only in freeciv-2.2.3-changed//server: voting.lo
Only in freeciv-2.2.3-changed//utility: astring.lo
Only in freeciv-2.2.3-changed//utility: capability.lo
Only in freeciv-2.2.3-changed//utility: distribute.lo
Only in freeciv-2.2.3-changed//utility: fciconv.lo
Only in freeciv-2.2.3-changed//utility: fcintl.lo
Only in freeciv-2.2.3-changed//utility: genlist.lo
Only in freeciv-2.2.3-changed//utility: hash.lo
Only in freeciv-2.2.3-changed//utility: inputfile.lo
Only in freeciv-2.2.3-changed//utility: ioz.lo
Only in freeciv-2.2.3-changed//utility: iterator.lo
Only in freeciv-2.2.3-changed//utility: libcivutility.la
Only in freeciv-2.2.3-changed//utility: .libs
Only in freeciv-2.2.3-changed//utility: log.lo
Only in freeciv-2.2.3-changed//utility: md5.lo
Only in freeciv-2.2.3-changed//utility: mem.lo
Only in freeciv-2.2.3-changed//utility: netintf.lo
Only in freeciv-2.2.3-changed//utility: pqueue.lo
Only in freeciv-2.2.3-changed//utility: rand.lo
Only in freeciv-2.2.3-changed//utility: registry.lo
Only in freeciv-2.2.3-changed//utility: sbuffer.lo
Only in freeciv-2.2.3-changed//utility: shared.lo
Only in freeciv-2.2.3-changed//utility: string_vector.lo
Only in freeciv-2.2.3-changed//utility: support.lo
Only in freeciv-2.2.3-changed//utility: timing.lo
