diff -Nurd -X.diff_ignore freeciv/utility/shared.c freeciv/utility/shared.c
--- freeciv/utility/shared.c	2012-05-16 20:56:42.000000000 +0300
+++ freeciv/utility/shared.c	2012-05-16 21:12:39.000000000 +0300
@@ -439,7 +439,7 @@
 ***************************************************************/
 bool is_ascii_name(const char *name)
 {
-  const char illegal_chars[] = {'|', '%', '"', ',', '*', '<', '>', '\0'};
+  const char illegal_chars[] = {'|', '"', ',', '*', '<', '>', '\0'};
   int i, j;
 
   /* must not be NULL or empty */
