diff --git a/server/handchat.c b/server/handchat.c
index fcbd198..1cdbc3e 100644
--- a/server/handchat.c
+++ b/server/handchat.c
@@ -326,7 +326,7 @@ void handle_chat_msg_req(struct connection *pconn, const char *message)
   /* This loop to prevent players from sending multiple lines which can
    * be abused */
   for (cp = real_message; *cp != '\0'; cp++) {
-    if (*cp == '\n' || *cp == '\r') {
+    if (*cp == '\n' || *cp == '\r' || *cp == '<' || *cp == '>' || *cp == '"' || *cp == '\'') {
       *cp = '\0';
       break;
     }
